{
 "cells": [
  {
   "cell_type": "raw",
   "id": "a2f71401",
   "metadata": {},
   "source": [
    "---\n",
    "output-file: 14-human_genetics.html\n",
    "title: 14-human_genetics\n",
    "\n",
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5e465208",
   "metadata": {},
   "source": [
    "# 14 - human genetics\n",
    "\n",
    "The Human Phenotype Project collects genomic variation data on all its participants. The genomic data together with the Human Phenotype Project  deep-phenotypes allows to investigate the progression of disease, and to explore personalized treatments. We genotype millions of positions by low-pass sequencing combined with imputation using gencove platform technologies. Genotype imputation is a process of statistically inferring unobserved genotypes using known haplotypes in a population. The performance of Gencove genotype imputation is very high ( accuracy > 98% ) (Wasik et al. 2021). \n",
    "\n",
    "The information is stored in a number of statistics parquet files:\n",
    "- `main.parquet`: sample metadata, including QC statistics, paths to PLINK variant files (raw and post-QC), and principal components (PCs).\n",
    "- `variant_qc.parquet`: variant QC statistics.\n",
    "- `relatives/plink_ibs.parquet`: IBS calculated by PLINK for pairs of participants.\n",
    "- `relatives/king_kinship.parquet`: King kinship coefficients for pairs of participants.\n",
    "\n",
    "And a PLINK text file `pca/eigenvec.var` containing the principal component loadings."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "915be9af",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-29T16:15:21.442707Z",
     "iopub.status.busy": "2023-04-29T16:15:21.442263Z",
     "iopub.status.idle": "2023-04-29T16:15:21.708039Z",
     "shell.execute_reply": "2023-04-29T16:15:21.707555Z"
    }
   },
   "outputs": [],
   "source": [
    "#| hide\n",
    "import pandas as pd\n",
    "pd.set_option('display.max_rows', 500)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "e4f14213",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-29T16:15:21.710266Z",
     "iopub.status.busy": "2023-04-29T16:15:21.710087Z",
     "iopub.status.idle": "2023-04-29T16:15:22.584976Z",
     "shell.execute_reply": "2023-04-29T16:15:22.584468Z"
    }
   },
   "outputs": [],
   "source": [
    "from pheno_utils import PhenoLoader"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "bbcc681b",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-29T16:15:22.587076Z",
     "iopub.status.busy": "2023-04-29T16:15:22.586948Z",
     "iopub.status.idle": "2023-04-29T16:15:26.980247Z",
     "shell.execute_reply": "2023-04-29T16:15:26.979800Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "DataLoader for human_genetics with\n",
       "119 fields\n",
       "1 tables: ['main']"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dl = PhenoLoader('human_genetics', age_sex_dataset=None)\n",
    "dl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "67969e14",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-04-29T16:15:26.982087Z",
     "iopub.status.busy": "2023-04-29T16:15:26.981859Z",
     "iopub.status.idle": "2023-04-29T16:15:27.005635Z",
     "shell.execute_reply": "2023-04-29T16:15:27.005217Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>field_string</th>\n",
       "      <th>description_string</th>\n",
       "      <th>parent_dataframe</th>\n",
       "      <th>relative_location</th>\n",
       "      <th>value_type</th>\n",
       "      <th>units</th>\n",
       "      <th>sampling_rate</th>\n",
       "      <th>item_type</th>\n",
       "      <th>array</th>\n",
       "      <th>cohorts</th>\n",
       "      <th>data_type</th>\n",
       "      <th>debut</th>\n",
       "      <th>pandas_dtype</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tabular_field_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>collection_date</th>\n",
       "      <td>Collection date</td>\n",
       "      <td>The date of downloading Gencove results from t...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Time</td>\n",
       "      <td>Time</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>datetime64[ns]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>version</th>\n",
       "      <td>Gencove version</td>\n",
       "      <td>Gencove API version 1 or 2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Categorical (multiple)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genecov_qc_bases</th>\n",
       "      <td>Bases sequenced</td>\n",
       "      <td>number of total bases sequenced. from genecov ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Integer</td>\n",
       "      <td>Count</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genecov_qc_bases_dedup</th>\n",
       "      <td>Deduplicated bases</td>\n",
       "      <td>number of deduplicated bases. from genecov qc ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Integer</td>\n",
       "      <td>Count</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gencove_qc_bases_dedup_mapped</th>\n",
       "      <td>Deduplicated bases aligned</td>\n",
       "      <td>number of deduplicated bases that have aligned...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Integer</td>\n",
       "      <td>Count</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genecov_qc_effective_coverage</th>\n",
       "      <td>Effective coverage</td>\n",
       "      <td>effective coverage. from genecov qc file</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>Precent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genecov_qc_fraction_contamination</th>\n",
       "      <td>DNA contamination</td>\n",
       "      <td>contamination by DNA from another sample of th...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>Precent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genecov_qc_snps</th>\n",
       "      <td>SNPs covered</td>\n",
       "      <td>number of variants in reference panel that are...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Integer</td>\n",
       "      <td>Count</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genecov_qc_format_passed</th>\n",
       "      <td>Passed FASTQ format validity</td>\n",
       "      <td>Indicates proper formatting of the input FASTQ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genecov_qc_r1_eq_r2_passed</th>\n",
       "      <td>Passed paired-end count</td>\n",
       "      <td>Indicates number of bases in R1 file equal to ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genecov_qc_r1_r2_ids_match_passed</th>\n",
       "      <td>Passed paired-end match</td>\n",
       "      <td>Indicates R1 read identifiers match R2 read id...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genecov_qc_bases_passed</th>\n",
       "      <td>Passed Bases sequenced</td>\n",
       "      <td>Indicates number of total bases sequenced grea...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genecov_qc_bases_dedup_passed</th>\n",
       "      <td>Passed Deduplicated bases</td>\n",
       "      <td>Indicates number of deduplicated bases greater...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genecov_qc_bases_dedup_mapped_passed</th>\n",
       "      <td>Passed Deduplicated bases aligned</td>\n",
       "      <td>Indicates number of deduplicated bases aligned...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genecov_qc_effective_coverage_passed</th>\n",
       "      <td>Passed Effective coverage</td>\n",
       "      <td>Indicates effective coverage greater than 0. f...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genecov_qc_fraction_contamination_passed</th>\n",
       "      <td>Passed DNA contamination</td>\n",
       "      <td>Indicates contamination by DNA less than 0.06....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genecov_qc_snps_passed</th>\n",
       "      <td>Passed SNPs covered</td>\n",
       "      <td>Indicates number of variants covered by at lea...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>heterozygosity_proportion</th>\n",
       "      <td>Heterozygosity proportion</td>\n",
       "      <td>Estimated heterozygosity proportion across a s...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>Precent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pass_het_check</th>\n",
       "      <td>Passed Heterozygosity proportion</td>\n",
       "      <td>Indicates heterozygosity proportion is less th...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>bool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_qc_gp_pass_rate</th>\n",
       "      <td>Genotype Probabilities pass rate</td>\n",
       "      <td>Proportion of genotypes with maximum GP greate...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>Precent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_qc_gp_pass_check</th>\n",
       "      <td>Passed Genotype Probabilities</td>\n",
       "      <td>Indicates proportion of GP greater than 0.9 is...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>bool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genetic_sex</th>\n",
       "      <td>DNA sex</td>\n",
       "      <td>Sex as determined by the percentage of read-al...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_x_proportion</th>\n",
       "      <td>Proportion of alignment to chromosome X</td>\n",
       "      <td>percentage of reads aligned to chromosome X</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>Precent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_y_proportion</th>\n",
       "      <td>Proportion of alignment to chromosome Y</td>\n",
       "      <td>percentage of reads aligned to chromosome Y</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>Precent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_qc_gender_match_check</th>\n",
       "      <td>Passed gender check</td>\n",
       "      <td>Indicates inferred_gender and submitted_gender...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>bool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>relatives</th>\n",
       "      <td>Relatives based on IBD estimation</td>\n",
       "      <td>List of second and first degree relatives acco...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>String</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>self_report_ashkenaz_proportion</th>\n",
       "      <td>Self-reported ashkenazi descent proportion</td>\n",
       "      <td>Proportion of great-grandparents with ashkenaz...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>Precent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>self_report_middleeastern_proportion</th>\n",
       "      <td>Self-reported Middle-Eastern descent proportion</td>\n",
       "      <td>Proportion of great-grandparents with Middle-E...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>Precent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>self_report_northafrican_proportion</th>\n",
       "      <td>Self-reported North-African descent proportion</td>\n",
       "      <td>Proportion of great-grandparents with ashkenaz...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>Precent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>self_report_sephardi_proportion</th>\n",
       "      <td>Self-reported Sephardi descent proportion</td>\n",
       "      <td>Proportion of great-grandparents with North-Af...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>Precent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>self_report_yemen_proportion</th>\n",
       "      <td>Self-reported Yemen descent proportion</td>\n",
       "      <td>Proportion of great-grandparents with Yemen de...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>Precent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>self_report_unknownother_proportion</th>\n",
       "      <td>Self-reported unknown or other descent proport...</td>\n",
       "      <td>Proportion of great-grandparents with unknown ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>Precent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc1</th>\n",
       "      <td>Principal component 1</td>\n",
       "      <td>Score for projected principal component 1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc2</th>\n",
       "      <td>Principal component 2</td>\n",
       "      <td>Score for projected principal component 2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc3</th>\n",
       "      <td>Principal component 3</td>\n",
       "      <td>Score for projected principal component 3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc4</th>\n",
       "      <td>Principal component 4</td>\n",
       "      <td>Score for projected principal component 4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc5</th>\n",
       "      <td>Principal component 5</td>\n",
       "      <td>Score for projected principal component 5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc6</th>\n",
       "      <td>Principal component 6</td>\n",
       "      <td>Score for projected principal component 6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc7</th>\n",
       "      <td>Principal component 7</td>\n",
       "      <td>Score for projected principal component 7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc8</th>\n",
       "      <td>Principal component 8</td>\n",
       "      <td>Score for projected principal component 8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc9</th>\n",
       "      <td>Principal component 9</td>\n",
       "      <td>Score for projected principal component 9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc10</th>\n",
       "      <td>Principal component 10</td>\n",
       "      <td>Score for projected principal component 10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc11</th>\n",
       "      <td>Principal component 11</td>\n",
       "      <td>Score for projected principal component 11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc12</th>\n",
       "      <td>Principal component 12</td>\n",
       "      <td>Score for projected principal component 12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc13</th>\n",
       "      <td>Principal component 13</td>\n",
       "      <td>Score for projected principal component 13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc14</th>\n",
       "      <td>Principal component 14</td>\n",
       "      <td>Score for projected principal component 14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc15</th>\n",
       "      <td>Principal component 15</td>\n",
       "      <td>Score for projected principal component 15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc16</th>\n",
       "      <td>Principal component 16</td>\n",
       "      <td>Score for projected principal component 16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc17</th>\n",
       "      <td>Principal component 17</td>\n",
       "      <td>Score for projected principal component 17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc18</th>\n",
       "      <td>Principal component 18</td>\n",
       "      <td>Score for projected principal component 18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc19</th>\n",
       "      <td>Principal component 19</td>\n",
       "      <td>Score for projected principal component 19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc20</th>\n",
       "      <td>Principal component 20</td>\n",
       "      <td>Score for projected principal component 20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>float64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>used_in_pca_calculation</th>\n",
       "      <td>Included in principal component analysis</td>\n",
       "      <td>Indicates samples was in the input for princip...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>bool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bed_chr_1</th>\n",
       "      <td>PLINK BED file for post QC genotype data chr1</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Compressed binary file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_fam_chr_1</th>\n",
       "      <td>PLINK FAM file accompanying QC .bed chr1</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bim_chr_1</th>\n",
       "      <td>PLINK BIM file for the post QC genotype data chr1</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bed_chr_2</th>\n",
       "      <td>PLINK BED file for post QC genotype data chr2</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Compressed binary file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_fam_chr_2</th>\n",
       "      <td>PLINK FAM file accompanying QC .bed chr2</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bim_chr_2</th>\n",
       "      <td>PLINK BIM file for the post QC genotype data chr2</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bed_chr_3</th>\n",
       "      <td>PLINK BED file for post QC genotype data chr3</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Compressed binary file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_fam_chr_3</th>\n",
       "      <td>PLINK FAM file accompanying QC .bed chr3</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bim_chr_3</th>\n",
       "      <td>PLINK BIM file for the post QC genotype data chr3</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bed_chr_4</th>\n",
       "      <td>PLINK BED file for post QC genotype data chr4</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Compressed binary file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_fam_chr_4</th>\n",
       "      <td>PLINK FAM file accompanying QC .bed chr4</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bim_chr_4</th>\n",
       "      <td>PLINK BIM file for the post QC genotype data chr4</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bed_chr_5</th>\n",
       "      <td>PLINK BED file for post QC genotype data chr5</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Compressed binary file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_fam_chr_5</th>\n",
       "      <td>PLINK FAM file accompanying QC .bed chr5</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bim_chr_5</th>\n",
       "      <td>PLINK BIM file for the post QC genotype data chr5</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bed_chr_6</th>\n",
       "      <td>PLINK BED file for post QC genotype data chr6</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Compressed binary file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_fam_chr_6</th>\n",
       "      <td>PLINK FAM file accompanying QC .bed chr6</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bim_chr_6</th>\n",
       "      <td>PLINK BIM file for the post QC genotype data chr6</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bed_chr_7</th>\n",
       "      <td>PLINK BED file for post QC genotype data chr7</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Compressed binary file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_fam_chr_7</th>\n",
       "      <td>PLINK FAM file accompanying QC .bed chr7</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bim_chr_7</th>\n",
       "      <td>PLINK BIM file for the post QC genotype data chr7</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bed_chr_8</th>\n",
       "      <td>PLINK BED file for post QC genotype data chr8</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Compressed binary file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_fam_chr_8</th>\n",
       "      <td>PLINK FAM file accompanying QC .bed chr8</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bim_chr_8</th>\n",
       "      <td>PLINK BIM file for the post QC genotype data chr8</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bed_chr_9</th>\n",
       "      <td>PLINK BED file for post QC genotype data chr9</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Compressed binary file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_fam_chr_9</th>\n",
       "      <td>PLINK FAM file accompanying QC .bed chr9</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bim_chr_9</th>\n",
       "      <td>PLINK BIM file for the post QC genotype data chr9</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bed_chr_10</th>\n",
       "      <td>PLINK BED file for post QC genotype data chr10</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Compressed binary file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_fam_chr_10</th>\n",
       "      <td>PLINK FAM file accompanying QC .bed chr10</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bim_chr_10</th>\n",
       "      <td>PLINK BIM file for the post QC genotype data c...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bed_chr_11</th>\n",
       "      <td>PLINK BED file for post QC genotype data chr11</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Compressed binary file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_fam_chr_11</th>\n",
       "      <td>PLINK FAM file accompanying QC .bed chr11</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bim_chr_11</th>\n",
       "      <td>PLINK BIM file for the post QC genotype data c...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bed_chr_12</th>\n",
       "      <td>PLINK BED file for post QC genotype data chr12</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Compressed binary file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_fam_chr_12</th>\n",
       "      <td>PLINK FAM file accompanying QC .bed chr12</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bim_chr_12</th>\n",
       "      <td>PLINK BIM file for the post QC genotype data c...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bed_chr_13</th>\n",
       "      <td>PLINK BED file for post QC genotype data chr13</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Compressed binary file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_fam_chr_13</th>\n",
       "      <td>PLINK FAM file accompanying QC .bed chr13</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bim_chr_13</th>\n",
       "      <td>PLINK BIM file for the post QC genotype data c...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bed_chr_14</th>\n",
       "      <td>PLINK BED file for post QC genotype data chr14</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Compressed binary file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_fam_chr_14</th>\n",
       "      <td>PLINK FAM file accompanying QC .bed chr14</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bim_chr_14</th>\n",
       "      <td>PLINK BIM file for the post QC genotype data c...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bed_chr_15</th>\n",
       "      <td>PLINK BED file for post QC genotype data chr15</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Compressed binary file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_fam_chr_15</th>\n",
       "      <td>PLINK FAM file accompanying QC .bed chr15</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bim_chr_15</th>\n",
       "      <td>PLINK BIM file for the post QC genotype data c...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bed_chr_16</th>\n",
       "      <td>PLINK BED file for post QC genotype data chr16</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Compressed binary file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_fam_chr_16</th>\n",
       "      <td>PLINK FAM file accompanying QC .bed chr16</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bim_chr_16</th>\n",
       "      <td>PLINK BIM file for the post QC genotype data c...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bed_chr_17</th>\n",
       "      <td>PLINK BED file for post QC genotype data chr17</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Compressed binary file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_fam_chr_17</th>\n",
       "      <td>PLINK FAM file accompanying QC .bed chr17</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bim_chr_17</th>\n",
       "      <td>PLINK BIM file for the post QC genotype data c...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bed_chr_18</th>\n",
       "      <td>PLINK BED file for post QC genotype data chr18</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Compressed binary file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_fam_chr_18</th>\n",
       "      <td>PLINK FAM file accompanying QC .bed chr18</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bim_chr_18</th>\n",
       "      <td>PLINK BIM file for the post QC genotype data c...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bed_chr_19</th>\n",
       "      <td>PLINK BED file for post QC genotype data chr19</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Compressed binary file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_fam_chr_19</th>\n",
       "      <td>PLINK FAM file accompanying QC .bed chr19</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bim_chr_19</th>\n",
       "      <td>PLINK BIM file for the post QC genotype data c...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bed_chr_20</th>\n",
       "      <td>PLINK BED file for post QC genotype data chr20</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Compressed binary file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_fam_chr_20</th>\n",
       "      <td>PLINK FAM file accompanying QC .bed chr20</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bim_chr_20</th>\n",
       "      <td>PLINK BIM file for the post QC genotype data c...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bed_chr_21</th>\n",
       "      <td>PLINK BED file for post QC genotype data chr21</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Compressed binary file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_fam_chr_21</th>\n",
       "      <td>PLINK FAM file accompanying QC .bed chr21</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bim_chr_21</th>\n",
       "      <td>PLINK BIM file for the post QC genotype data c...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bed_chr_22</th>\n",
       "      <td>PLINK BED file for post QC genotype data chr22</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Compressed binary file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_fam_chr_22</th>\n",
       "      <td>PLINK FAM file accompanying QC .bed chr22</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>postqc_bim_chr_22</th>\n",
       "      <td>PLINK BIM file for the post QC genotype data c...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>main.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10K</td>\n",
       "      <td>Text file</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                               field_string  \\\n",
       "tabular_field_name                                                                            \n",
       "collection_date                                                             Collection date   \n",
       "version                                                                    Gencove version    \n",
       "genecov_qc_bases                                                           Bases sequenced    \n",
       "genecov_qc_bases_dedup                                                   Deduplicated bases   \n",
       "gencove_qc_bases_dedup_mapped                                    Deduplicated bases aligned   \n",
       "genecov_qc_effective_coverage                                            Effective coverage   \n",
       "genecov_qc_fraction_contamination                                         DNA contamination   \n",
       "genecov_qc_snps                                                                SNPs covered   \n",
       "genecov_qc_format_passed                                       Passed FASTQ format validity   \n",
       "genecov_qc_r1_eq_r2_passed                                         Passed paired-end count    \n",
       "genecov_qc_r1_r2_ids_match_passed                                   Passed paired-end match   \n",
       "genecov_qc_bases_passed                                             Passed Bases sequenced    \n",
       "genecov_qc_bases_dedup_passed                                     Passed Deduplicated bases   \n",
       "genecov_qc_bases_dedup_mapped_passed                      Passed Deduplicated bases aligned   \n",
       "genecov_qc_effective_coverage_passed                              Passed Effective coverage   \n",
       "genecov_qc_fraction_contamination_passed                           Passed DNA contamination   \n",
       "genecov_qc_snps_passed                                                  Passed SNPs covered   \n",
       "heterozygosity_proportion                                         Heterozygosity proportion   \n",
       "pass_het_check                                             Passed Heterozygosity proportion   \n",
       "sample_qc_gp_pass_rate                                    Genotype Probabilities pass rate    \n",
       "sample_qc_gp_pass_check                                       Passed Genotype Probabilities   \n",
       "genetic_sex                                                                         DNA sex   \n",
       "chr_x_proportion                                    Proportion of alignment to chromosome X   \n",
       "chr_y_proportion                                    Proportion of alignment to chromosome Y   \n",
       "sample_qc_gender_match_check                                            Passed gender check   \n",
       "relatives                                                 Relatives based on IBD estimation   \n",
       "self_report_ashkenaz_proportion                 Self-reported ashkenazi descent proportion    \n",
       "self_report_middleeastern_proportion       Self-reported Middle-Eastern descent proportion    \n",
       "self_report_northafrican_proportion         Self-reported North-African descent proportion    \n",
       "self_report_sephardi_proportion                  Self-reported Sephardi descent proportion    \n",
       "self_report_yemen_proportion                        Self-reported Yemen descent proportion    \n",
       "self_report_unknownother_proportion       Self-reported unknown or other descent proport...   \n",
       "pc1                                                                   Principal component 1   \n",
       "pc2                                                                   Principal component 2   \n",
       "pc3                                                                   Principal component 3   \n",
       "pc4                                                                   Principal component 4   \n",
       "pc5                                                                   Principal component 5   \n",
       "pc6                                                                   Principal component 6   \n",
       "pc7                                                                   Principal component 7   \n",
       "pc8                                                                   Principal component 8   \n",
       "pc9                                                                   Principal component 9   \n",
       "pc10                                                                 Principal component 10   \n",
       "pc11                                                                 Principal component 11   \n",
       "pc12                                                                 Principal component 12   \n",
       "pc13                                                                 Principal component 13   \n",
       "pc14                                                                 Principal component 14   \n",
       "pc15                                                                 Principal component 15   \n",
       "pc16                                                                 Principal component 16   \n",
       "pc17                                                                 Principal component 17   \n",
       "pc18                                                                 Principal component 18   \n",
       "pc19                                                                 Principal component 19   \n",
       "pc20                                                                 Principal component 20   \n",
       "used_in_pca_calculation                            Included in principal component analysis   \n",
       "postqc_bed_chr_1                              PLINK BED file for post QC genotype data chr1   \n",
       "postqc_fam_chr_1                                   PLINK FAM file accompanying QC .bed chr1   \n",
       "postqc_bim_chr_1                          PLINK BIM file for the post QC genotype data chr1   \n",
       "postqc_bed_chr_2                              PLINK BED file for post QC genotype data chr2   \n",
       "postqc_fam_chr_2                                   PLINK FAM file accompanying QC .bed chr2   \n",
       "postqc_bim_chr_2                          PLINK BIM file for the post QC genotype data chr2   \n",
       "postqc_bed_chr_3                              PLINK BED file for post QC genotype data chr3   \n",
       "postqc_fam_chr_3                                   PLINK FAM file accompanying QC .bed chr3   \n",
       "postqc_bim_chr_3                          PLINK BIM file for the post QC genotype data chr3   \n",
       "postqc_bed_chr_4                              PLINK BED file for post QC genotype data chr4   \n",
       "postqc_fam_chr_4                                   PLINK FAM file accompanying QC .bed chr4   \n",
       "postqc_bim_chr_4                          PLINK BIM file for the post QC genotype data chr4   \n",
       "postqc_bed_chr_5                              PLINK BED file for post QC genotype data chr5   \n",
       "postqc_fam_chr_5                                   PLINK FAM file accompanying QC .bed chr5   \n",
       "postqc_bim_chr_5                          PLINK BIM file for the post QC genotype data chr5   \n",
       "postqc_bed_chr_6                              PLINK BED file for post QC genotype data chr6   \n",
       "postqc_fam_chr_6                                   PLINK FAM file accompanying QC .bed chr6   \n",
       "postqc_bim_chr_6                          PLINK BIM file for the post QC genotype data chr6   \n",
       "postqc_bed_chr_7                              PLINK BED file for post QC genotype data chr7   \n",
       "postqc_fam_chr_7                                   PLINK FAM file accompanying QC .bed chr7   \n",
       "postqc_bim_chr_7                          PLINK BIM file for the post QC genotype data chr7   \n",
       "postqc_bed_chr_8                              PLINK BED file for post QC genotype data chr8   \n",
       "postqc_fam_chr_8                                   PLINK FAM file accompanying QC .bed chr8   \n",
       "postqc_bim_chr_8                          PLINK BIM file for the post QC genotype data chr8   \n",
       "postqc_bed_chr_9                              PLINK BED file for post QC genotype data chr9   \n",
       "postqc_fam_chr_9                                   PLINK FAM file accompanying QC .bed chr9   \n",
       "postqc_bim_chr_9                          PLINK BIM file for the post QC genotype data chr9   \n",
       "postqc_bed_chr_10                            PLINK BED file for post QC genotype data chr10   \n",
       "postqc_fam_chr_10                                 PLINK FAM file accompanying QC .bed chr10   \n",
       "postqc_bim_chr_10                         PLINK BIM file for the post QC genotype data c...   \n",
       "postqc_bed_chr_11                            PLINK BED file for post QC genotype data chr11   \n",
       "postqc_fam_chr_11                                 PLINK FAM file accompanying QC .bed chr11   \n",
       "postqc_bim_chr_11                         PLINK BIM file for the post QC genotype data c...   \n",
       "postqc_bed_chr_12                            PLINK BED file for post QC genotype data chr12   \n",
       "postqc_fam_chr_12                                 PLINK FAM file accompanying QC .bed chr12   \n",
       "postqc_bim_chr_12                         PLINK BIM file for the post QC genotype data c...   \n",
       "postqc_bed_chr_13                            PLINK BED file for post QC genotype data chr13   \n",
       "postqc_fam_chr_13                                 PLINK FAM file accompanying QC .bed chr13   \n",
       "postqc_bim_chr_13                         PLINK BIM file for the post QC genotype data c...   \n",
       "postqc_bed_chr_14                            PLINK BED file for post QC genotype data chr14   \n",
       "postqc_fam_chr_14                                 PLINK FAM file accompanying QC .bed chr14   \n",
       "postqc_bim_chr_14                         PLINK BIM file for the post QC genotype data c...   \n",
       "postqc_bed_chr_15                            PLINK BED file for post QC genotype data chr15   \n",
       "postqc_fam_chr_15                                 PLINK FAM file accompanying QC .bed chr15   \n",
       "postqc_bim_chr_15                         PLINK BIM file for the post QC genotype data c...   \n",
       "postqc_bed_chr_16                            PLINK BED file for post QC genotype data chr16   \n",
       "postqc_fam_chr_16                                 PLINK FAM file accompanying QC .bed chr16   \n",
       "postqc_bim_chr_16                         PLINK BIM file for the post QC genotype data c...   \n",
       "postqc_bed_chr_17                            PLINK BED file for post QC genotype data chr17   \n",
       "postqc_fam_chr_17                                 PLINK FAM file accompanying QC .bed chr17   \n",
       "postqc_bim_chr_17                         PLINK BIM file for the post QC genotype data c...   \n",
       "postqc_bed_chr_18                            PLINK BED file for post QC genotype data chr18   \n",
       "postqc_fam_chr_18                                 PLINK FAM file accompanying QC .bed chr18   \n",
       "postqc_bim_chr_18                         PLINK BIM file for the post QC genotype data c...   \n",
       "postqc_bed_chr_19                            PLINK BED file for post QC genotype data chr19   \n",
       "postqc_fam_chr_19                                 PLINK FAM file accompanying QC .bed chr19   \n",
       "postqc_bim_chr_19                         PLINK BIM file for the post QC genotype data c...   \n",
       "postqc_bed_chr_20                            PLINK BED file for post QC genotype data chr20   \n",
       "postqc_fam_chr_20                                 PLINK FAM file accompanying QC .bed chr20   \n",
       "postqc_bim_chr_20                         PLINK BIM file for the post QC genotype data c...   \n",
       "postqc_bed_chr_21                            PLINK BED file for post QC genotype data chr21   \n",
       "postqc_fam_chr_21                                 PLINK FAM file accompanying QC .bed chr21   \n",
       "postqc_bim_chr_21                         PLINK BIM file for the post QC genotype data c...   \n",
       "postqc_bed_chr_22                            PLINK BED file for post QC genotype data chr22   \n",
       "postqc_fam_chr_22                                 PLINK FAM file accompanying QC .bed chr22   \n",
       "postqc_bim_chr_22                         PLINK BIM file for the post QC genotype data c...   \n",
       "\n",
       "                                                                         description_string  \\\n",
       "tabular_field_name                                                                            \n",
       "collection_date                           The date of downloading Gencove results from t...   \n",
       "version                                                          Gencove API version 1 or 2   \n",
       "genecov_qc_bases                          number of total bases sequenced. from genecov ...   \n",
       "genecov_qc_bases_dedup                    number of deduplicated bases. from genecov qc ...   \n",
       "gencove_qc_bases_dedup_mapped             number of deduplicated bases that have aligned...   \n",
       "genecov_qc_effective_coverage                      effective coverage. from genecov qc file   \n",
       "genecov_qc_fraction_contamination         contamination by DNA from another sample of th...   \n",
       "genecov_qc_snps                           number of variants in reference panel that are...   \n",
       "genecov_qc_format_passed                  Indicates proper formatting of the input FASTQ...   \n",
       "genecov_qc_r1_eq_r2_passed                Indicates number of bases in R1 file equal to ...   \n",
       "genecov_qc_r1_r2_ids_match_passed         Indicates R1 read identifiers match R2 read id...   \n",
       "genecov_qc_bases_passed                   Indicates number of total bases sequenced grea...   \n",
       "genecov_qc_bases_dedup_passed             Indicates number of deduplicated bases greater...   \n",
       "genecov_qc_bases_dedup_mapped_passed      Indicates number of deduplicated bases aligned...   \n",
       "genecov_qc_effective_coverage_passed      Indicates effective coverage greater than 0. f...   \n",
       "genecov_qc_fraction_contamination_passed  Indicates contamination by DNA less than 0.06....   \n",
       "genecov_qc_snps_passed                    Indicates number of variants covered by at lea...   \n",
       "heterozygosity_proportion                 Estimated heterozygosity proportion across a s...   \n",
       "pass_het_check                            Indicates heterozygosity proportion is less th...   \n",
       "sample_qc_gp_pass_rate                    Proportion of genotypes with maximum GP greate...   \n",
       "sample_qc_gp_pass_check                   Indicates proportion of GP greater than 0.9 is...   \n",
       "genetic_sex                               Sex as determined by the percentage of read-al...   \n",
       "chr_x_proportion                                percentage of reads aligned to chromosome X   \n",
       "chr_y_proportion                                percentage of reads aligned to chromosome Y   \n",
       "sample_qc_gender_match_check              Indicates inferred_gender and submitted_gender...   \n",
       "relatives                                 List of second and first degree relatives acco...   \n",
       "self_report_ashkenaz_proportion           Proportion of great-grandparents with ashkenaz...   \n",
       "self_report_middleeastern_proportion      Proportion of great-grandparents with Middle-E...   \n",
       "self_report_northafrican_proportion       Proportion of great-grandparents with ashkenaz...   \n",
       "self_report_sephardi_proportion           Proportion of great-grandparents with North-Af...   \n",
       "self_report_yemen_proportion              Proportion of great-grandparents with Yemen de...   \n",
       "self_report_unknownother_proportion       Proportion of great-grandparents with unknown ...   \n",
       "pc1                                               Score for projected principal component 1   \n",
       "pc2                                               Score for projected principal component 2   \n",
       "pc3                                               Score for projected principal component 3   \n",
       "pc4                                               Score for projected principal component 4   \n",
       "pc5                                               Score for projected principal component 5   \n",
       "pc6                                               Score for projected principal component 6   \n",
       "pc7                                               Score for projected principal component 7   \n",
       "pc8                                               Score for projected principal component 8   \n",
       "pc9                                               Score for projected principal component 9   \n",
       "pc10                                             Score for projected principal component 10   \n",
       "pc11                                             Score for projected principal component 11   \n",
       "pc12                                             Score for projected principal component 12   \n",
       "pc13                                             Score for projected principal component 13   \n",
       "pc14                                             Score for projected principal component 14   \n",
       "pc15                                             Score for projected principal component 15   \n",
       "pc16                                             Score for projected principal component 16   \n",
       "pc17                                             Score for projected principal component 17   \n",
       "pc18                                             Score for projected principal component 18   \n",
       "pc19                                             Score for projected principal component 19   \n",
       "pc20                                             Score for projected principal component 20   \n",
       "used_in_pca_calculation                   Indicates samples was in the input for princip...   \n",
       "postqc_bed_chr_1                          Raw genotype calls (per chromosome) in binary ...   \n",
       "postqc_fam_chr_1                          PLINK sample information file accompanying a ....   \n",
       "postqc_bim_chr_1                          PLINK variant information file (per chromosome...   \n",
       "postqc_bed_chr_2                          Raw genotype calls (per chromosome) in binary ...   \n",
       "postqc_fam_chr_2                          PLINK sample information file accompanying a ....   \n",
       "postqc_bim_chr_2                          PLINK variant information file (per chromosome...   \n",
       "postqc_bed_chr_3                          Raw genotype calls (per chromosome) in binary ...   \n",
       "postqc_fam_chr_3                          PLINK sample information file accompanying a ....   \n",
       "postqc_bim_chr_3                          PLINK variant information file (per chromosome...   \n",
       "postqc_bed_chr_4                          Raw genotype calls (per chromosome) in binary ...   \n",
       "postqc_fam_chr_4                          PLINK sample information file accompanying a ....   \n",
       "postqc_bim_chr_4                          PLINK variant information file (per chromosome...   \n",
       "postqc_bed_chr_5                          Raw genotype calls (per chromosome) in binary ...   \n",
       "postqc_fam_chr_5                          PLINK sample information file accompanying a ....   \n",
       "postqc_bim_chr_5                          PLINK variant information file (per chromosome...   \n",
       "postqc_bed_chr_6                          Raw genotype calls (per chromosome) in binary ...   \n",
       "postqc_fam_chr_6                          PLINK sample information file accompanying a ....   \n",
       "postqc_bim_chr_6                          PLINK variant information file (per chromosome...   \n",
       "postqc_bed_chr_7                          Raw genotype calls (per chromosome) in binary ...   \n",
       "postqc_fam_chr_7                          PLINK sample information file accompanying a ....   \n",
       "postqc_bim_chr_7                          PLINK variant information file (per chromosome...   \n",
       "postqc_bed_chr_8                          Raw genotype calls (per chromosome) in binary ...   \n",
       "postqc_fam_chr_8                          PLINK sample information file accompanying a ....   \n",
       "postqc_bim_chr_8                          PLINK variant information file (per chromosome...   \n",
       "postqc_bed_chr_9                          Raw genotype calls (per chromosome) in binary ...   \n",
       "postqc_fam_chr_9                          PLINK sample information file accompanying a ....   \n",
       "postqc_bim_chr_9                          PLINK variant information file (per chromosome...   \n",
       "postqc_bed_chr_10                         Raw genotype calls (per chromosome) in binary ...   \n",
       "postqc_fam_chr_10                         PLINK sample information file accompanying a ....   \n",
       "postqc_bim_chr_10                         PLINK variant information file (per chromosome...   \n",
       "postqc_bed_chr_11                         Raw genotype calls (per chromosome) in binary ...   \n",
       "postqc_fam_chr_11                         PLINK sample information file accompanying a ....   \n",
       "postqc_bim_chr_11                         PLINK variant information file (per chromosome...   \n",
       "postqc_bed_chr_12                         Raw genotype calls (per chromosome) in binary ...   \n",
       "postqc_fam_chr_12                         PLINK sample information file accompanying a ....   \n",
       "postqc_bim_chr_12                         PLINK variant information file (per chromosome...   \n",
       "postqc_bed_chr_13                         Raw genotype calls (per chromosome) in binary ...   \n",
       "postqc_fam_chr_13                         PLINK sample information file accompanying a ....   \n",
       "postqc_bim_chr_13                         PLINK variant information file (per chromosome...   \n",
       "postqc_bed_chr_14                         Raw genotype calls (per chromosome) in binary ...   \n",
       "postqc_fam_chr_14                         PLINK sample information file accompanying a ....   \n",
       "postqc_bim_chr_14                         PLINK variant information file (per chromosome...   \n",
       "postqc_bed_chr_15                         Raw genotype calls (per chromosome) in binary ...   \n",
       "postqc_fam_chr_15                         PLINK sample information file accompanying a ....   \n",
       "postqc_bim_chr_15                         PLINK variant information file (per chromosome...   \n",
       "postqc_bed_chr_16                         Raw genotype calls (per chromosome) in binary ...   \n",
       "postqc_fam_chr_16                         PLINK sample information file accompanying a ....   \n",
       "postqc_bim_chr_16                         PLINK variant information file (per chromosome...   \n",
       "postqc_bed_chr_17                         Raw genotype calls (per chromosome) in binary ...   \n",
       "postqc_fam_chr_17                         PLINK sample information file accompanying a ....   \n",
       "postqc_bim_chr_17                         PLINK variant information file (per chromosome...   \n",
       "postqc_bed_chr_18                         Raw genotype calls (per chromosome) in binary ...   \n",
       "postqc_fam_chr_18                         PLINK sample information file accompanying a ....   \n",
       "postqc_bim_chr_18                         PLINK variant information file (per chromosome...   \n",
       "postqc_bed_chr_19                         Raw genotype calls (per chromosome) in binary ...   \n",
       "postqc_fam_chr_19                         PLINK sample information file accompanying a ....   \n",
       "postqc_bim_chr_19                         PLINK variant information file (per chromosome...   \n",
       "postqc_bed_chr_20                         Raw genotype calls (per chromosome) in binary ...   \n",
       "postqc_fam_chr_20                         PLINK sample information file accompanying a ....   \n",
       "postqc_bim_chr_20                         PLINK variant information file (per chromosome...   \n",
       "postqc_bed_chr_21                         Raw genotype calls (per chromosome) in binary ...   \n",
       "postqc_fam_chr_21                         PLINK sample information file accompanying a ....   \n",
       "postqc_bim_chr_21                         PLINK variant information file (per chromosome...   \n",
       "postqc_bed_chr_22                         Raw genotype calls (per chromosome) in binary ...   \n",
       "postqc_fam_chr_22                         PLINK sample information file accompanying a ....   \n",
       "postqc_bim_chr_22                         PLINK variant information file (per chromosome...   \n",
       "\n",
       "                                          parent_dataframe relative_location  \\\n",
       "tabular_field_name                                                             \n",
       "collection_date                                        NaN      main.parquet   \n",
       "version                                                NaN      main.parquet   \n",
       "genecov_qc_bases                                       NaN      main.parquet   \n",
       "genecov_qc_bases_dedup                                 NaN      main.parquet   \n",
       "gencove_qc_bases_dedup_mapped                          NaN      main.parquet   \n",
       "genecov_qc_effective_coverage                          NaN      main.parquet   \n",
       "genecov_qc_fraction_contamination                      NaN      main.parquet   \n",
       "genecov_qc_snps                                        NaN      main.parquet   \n",
       "genecov_qc_format_passed                               NaN      main.parquet   \n",
       "genecov_qc_r1_eq_r2_passed                             NaN      main.parquet   \n",
       "genecov_qc_r1_r2_ids_match_passed                      NaN      main.parquet   \n",
       "genecov_qc_bases_passed                                NaN      main.parquet   \n",
       "genecov_qc_bases_dedup_passed                          NaN      main.parquet   \n",
       "genecov_qc_bases_dedup_mapped_passed                   NaN      main.parquet   \n",
       "genecov_qc_effective_coverage_passed                   NaN      main.parquet   \n",
       "genecov_qc_fraction_contamination_passed               NaN      main.parquet   \n",
       "genecov_qc_snps_passed                                 NaN      main.parquet   \n",
       "heterozygosity_proportion                              NaN      main.parquet   \n",
       "pass_het_check                                         NaN      main.parquet   \n",
       "sample_qc_gp_pass_rate                                 NaN      main.parquet   \n",
       "sample_qc_gp_pass_check                                NaN      main.parquet   \n",
       "genetic_sex                                            NaN      main.parquet   \n",
       "chr_x_proportion                                       NaN      main.parquet   \n",
       "chr_y_proportion                                       NaN      main.parquet   \n",
       "sample_qc_gender_match_check                           NaN      main.parquet   \n",
       "relatives                                              NaN      main.parquet   \n",
       "self_report_ashkenaz_proportion                        NaN      main.parquet   \n",
       "self_report_middleeastern_proportion                   NaN      main.parquet   \n",
       "self_report_northafrican_proportion                    NaN      main.parquet   \n",
       "self_report_sephardi_proportion                        NaN      main.parquet   \n",
       "self_report_yemen_proportion                           NaN      main.parquet   \n",
       "self_report_unknownother_proportion                    NaN      main.parquet   \n",
       "pc1                                                    NaN      main.parquet   \n",
       "pc2                                                    NaN      main.parquet   \n",
       "pc3                                                    NaN      main.parquet   \n",
       "pc4                                                    NaN      main.parquet   \n",
       "pc5                                                    NaN      main.parquet   \n",
       "pc6                                                    NaN      main.parquet   \n",
       "pc7                                                    NaN      main.parquet   \n",
       "pc8                                                    NaN      main.parquet   \n",
       "pc9                                                    NaN      main.parquet   \n",
       "pc10                                                   NaN      main.parquet   \n",
       "pc11                                                   NaN      main.parquet   \n",
       "pc12                                                   NaN      main.parquet   \n",
       "pc13                                                   NaN      main.parquet   \n",
       "pc14                                                   NaN      main.parquet   \n",
       "pc15                                                   NaN      main.parquet   \n",
       "pc16                                                   NaN      main.parquet   \n",
       "pc17                                                   NaN      main.parquet   \n",
       "pc18                                                   NaN      main.parquet   \n",
       "pc19                                                   NaN      main.parquet   \n",
       "pc20                                                   NaN      main.parquet   \n",
       "used_in_pca_calculation                                NaN      main.parquet   \n",
       "postqc_bed_chr_1                                       NaN      main.parquet   \n",
       "postqc_fam_chr_1                                       NaN      main.parquet   \n",
       "postqc_bim_chr_1                                       NaN      main.parquet   \n",
       "postqc_bed_chr_2                                       NaN      main.parquet   \n",
       "postqc_fam_chr_2                                       NaN      main.parquet   \n",
       "postqc_bim_chr_2                                       NaN      main.parquet   \n",
       "postqc_bed_chr_3                                       NaN      main.parquet   \n",
       "postqc_fam_chr_3                                       NaN      main.parquet   \n",
       "postqc_bim_chr_3                                       NaN      main.parquet   \n",
       "postqc_bed_chr_4                                       NaN      main.parquet   \n",
       "postqc_fam_chr_4                                       NaN      main.parquet   \n",
       "postqc_bim_chr_4                                       NaN      main.parquet   \n",
       "postqc_bed_chr_5                                       NaN      main.parquet   \n",
       "postqc_fam_chr_5                                       NaN      main.parquet   \n",
       "postqc_bim_chr_5                                       NaN      main.parquet   \n",
       "postqc_bed_chr_6                                       NaN      main.parquet   \n",
       "postqc_fam_chr_6                                       NaN      main.parquet   \n",
       "postqc_bim_chr_6                                       NaN      main.parquet   \n",
       "postqc_bed_chr_7                                       NaN      main.parquet   \n",
       "postqc_fam_chr_7                                       NaN      main.parquet   \n",
       "postqc_bim_chr_7                                       NaN      main.parquet   \n",
       "postqc_bed_chr_8                                       NaN      main.parquet   \n",
       "postqc_fam_chr_8                                       NaN      main.parquet   \n",
       "postqc_bim_chr_8                                       NaN      main.parquet   \n",
       "postqc_bed_chr_9                                       NaN      main.parquet   \n",
       "postqc_fam_chr_9                                       NaN      main.parquet   \n",
       "postqc_bim_chr_9                                       NaN      main.parquet   \n",
       "postqc_bed_chr_10                                      NaN      main.parquet   \n",
       "postqc_fam_chr_10                                      NaN      main.parquet   \n",
       "postqc_bim_chr_10                                      NaN      main.parquet   \n",
       "postqc_bed_chr_11                                      NaN      main.parquet   \n",
       "postqc_fam_chr_11                                      NaN      main.parquet   \n",
       "postqc_bim_chr_11                                      NaN      main.parquet   \n",
       "postqc_bed_chr_12                                      NaN      main.parquet   \n",
       "postqc_fam_chr_12                                      NaN      main.parquet   \n",
       "postqc_bim_chr_12                                      NaN      main.parquet   \n",
       "postqc_bed_chr_13                                      NaN      main.parquet   \n",
       "postqc_fam_chr_13                                      NaN      main.parquet   \n",
       "postqc_bim_chr_13                                      NaN      main.parquet   \n",
       "postqc_bed_chr_14                                      NaN      main.parquet   \n",
       "postqc_fam_chr_14                                      NaN      main.parquet   \n",
       "postqc_bim_chr_14                                      NaN      main.parquet   \n",
       "postqc_bed_chr_15                                      NaN      main.parquet   \n",
       "postqc_fam_chr_15                                      NaN      main.parquet   \n",
       "postqc_bim_chr_15                                      NaN      main.parquet   \n",
       "postqc_bed_chr_16                                      NaN      main.parquet   \n",
       "postqc_fam_chr_16                                      NaN      main.parquet   \n",
       "postqc_bim_chr_16                                      NaN      main.parquet   \n",
       "postqc_bed_chr_17                                      NaN      main.parquet   \n",
       "postqc_fam_chr_17                                      NaN      main.parquet   \n",
       "postqc_bim_chr_17                                      NaN      main.parquet   \n",
       "postqc_bed_chr_18                                      NaN      main.parquet   \n",
       "postqc_fam_chr_18                                      NaN      main.parquet   \n",
       "postqc_bim_chr_18                                      NaN      main.parquet   \n",
       "postqc_bed_chr_19                                      NaN      main.parquet   \n",
       "postqc_fam_chr_19                                      NaN      main.parquet   \n",
       "postqc_bim_chr_19                                      NaN      main.parquet   \n",
       "postqc_bed_chr_20                                      NaN      main.parquet   \n",
       "postqc_fam_chr_20                                      NaN      main.parquet   \n",
       "postqc_bim_chr_20                                      NaN      main.parquet   \n",
       "postqc_bed_chr_21                                      NaN      main.parquet   \n",
       "postqc_fam_chr_21                                      NaN      main.parquet   \n",
       "postqc_bim_chr_21                                      NaN      main.parquet   \n",
       "postqc_bed_chr_22                                      NaN      main.parquet   \n",
       "postqc_fam_chr_22                                      NaN      main.parquet   \n",
       "postqc_bim_chr_22                                      NaN      main.parquet   \n",
       "\n",
       "                                                      value_type    units  \\\n",
       "tabular_field_name                                                          \n",
       "collection_date                                             Time     Time   \n",
       "version                                   Categorical (multiple)      NaN   \n",
       "genecov_qc_bases                                         Integer    Count   \n",
       "genecov_qc_bases_dedup                                   Integer    Count   \n",
       "gencove_qc_bases_dedup_mapped                            Integer    Count   \n",
       "genecov_qc_effective_coverage                         Continuous  Precent   \n",
       "genecov_qc_fraction_contamination                     Continuous  Precent   \n",
       "genecov_qc_snps                                          Integer    Count   \n",
       "genecov_qc_format_passed                    Categorical (single)  Boolean   \n",
       "genecov_qc_r1_eq_r2_passed                  Categorical (single)  Boolean   \n",
       "genecov_qc_r1_r2_ids_match_passed           Categorical (single)  Boolean   \n",
       "genecov_qc_bases_passed                     Categorical (single)  Boolean   \n",
       "genecov_qc_bases_dedup_passed               Categorical (single)  Boolean   \n",
       "genecov_qc_bases_dedup_mapped_passed        Categorical (single)  Boolean   \n",
       "genecov_qc_effective_coverage_passed        Categorical (single)  Boolean   \n",
       "genecov_qc_fraction_contamination_passed    Categorical (single)  Boolean   \n",
       "genecov_qc_snps_passed                      Categorical (single)  Boolean   \n",
       "heterozygosity_proportion                             Continuous  Precent   \n",
       "pass_het_check                              Categorical (single)  Boolean   \n",
       "sample_qc_gp_pass_rate                                Continuous  Precent   \n",
       "sample_qc_gp_pass_check                     Categorical (single)  Boolean   \n",
       "genetic_sex                                 Categorical (single)  Boolean   \n",
       "chr_x_proportion                                      Continuous  Precent   \n",
       "chr_y_proportion                                      Continuous  Precent   \n",
       "sample_qc_gender_match_check                Categorical (single)  Boolean   \n",
       "relatives                                                 String     Text   \n",
       "self_report_ashkenaz_proportion                       Continuous  Precent   \n",
       "self_report_middleeastern_proportion                  Continuous  Precent   \n",
       "self_report_northafrican_proportion                   Continuous  Precent   \n",
       "self_report_sephardi_proportion                       Continuous  Precent   \n",
       "self_report_yemen_proportion                          Continuous  Precent   \n",
       "self_report_unknownother_proportion                   Continuous  Precent   \n",
       "pc1                                                   Continuous     None   \n",
       "pc2                                                   Continuous     None   \n",
       "pc3                                                   Continuous     None   \n",
       "pc4                                                   Continuous     None   \n",
       "pc5                                                   Continuous     None   \n",
       "pc6                                                   Continuous     None   \n",
       "pc7                                                   Continuous     None   \n",
       "pc8                                                   Continuous     None   \n",
       "pc9                                                   Continuous     None   \n",
       "pc10                                                  Continuous     None   \n",
       "pc11                                                  Continuous     None   \n",
       "pc12                                                  Continuous     None   \n",
       "pc13                                                  Continuous     None   \n",
       "pc14                                                  Continuous     None   \n",
       "pc15                                                  Continuous     None   \n",
       "pc16                                                  Continuous     None   \n",
       "pc17                                                  Continuous     None   \n",
       "pc18                                                  Continuous     None   \n",
       "pc19                                                  Continuous     None   \n",
       "pc20                                                  Continuous     None   \n",
       "used_in_pca_calculation                     Categorical (single)  Boolean   \n",
       "postqc_bed_chr_1                                             NaN      NaN   \n",
       "postqc_fam_chr_1                                             NaN      NaN   \n",
       "postqc_bim_chr_1                                             NaN      NaN   \n",
       "postqc_bed_chr_2                                             NaN      NaN   \n",
       "postqc_fam_chr_2                                             NaN      NaN   \n",
       "postqc_bim_chr_2                                             NaN      NaN   \n",
       "postqc_bed_chr_3                                             NaN      NaN   \n",
       "postqc_fam_chr_3                                             NaN      NaN   \n",
       "postqc_bim_chr_3                                             NaN      NaN   \n",
       "postqc_bed_chr_4                                             NaN      NaN   \n",
       "postqc_fam_chr_4                                             NaN      NaN   \n",
       "postqc_bim_chr_4                                             NaN      NaN   \n",
       "postqc_bed_chr_5                                             NaN      NaN   \n",
       "postqc_fam_chr_5                                             NaN      NaN   \n",
       "postqc_bim_chr_5                                             NaN      NaN   \n",
       "postqc_bed_chr_6                                             NaN      NaN   \n",
       "postqc_fam_chr_6                                             NaN      NaN   \n",
       "postqc_bim_chr_6                                             NaN      NaN   \n",
       "postqc_bed_chr_7                                             NaN      NaN   \n",
       "postqc_fam_chr_7                                             NaN      NaN   \n",
       "postqc_bim_chr_7                                             NaN      NaN   \n",
       "postqc_bed_chr_8                                             NaN      NaN   \n",
       "postqc_fam_chr_8                                             NaN      NaN   \n",
       "postqc_bim_chr_8                                             NaN      NaN   \n",
       "postqc_bed_chr_9                                             NaN      NaN   \n",
       "postqc_fam_chr_9                                             NaN      NaN   \n",
       "postqc_bim_chr_9                                             NaN      NaN   \n",
       "postqc_bed_chr_10                                            NaN      NaN   \n",
       "postqc_fam_chr_10                                            NaN      NaN   \n",
       "postqc_bim_chr_10                                            NaN      NaN   \n",
       "postqc_bed_chr_11                                            NaN      NaN   \n",
       "postqc_fam_chr_11                                            NaN      NaN   \n",
       "postqc_bim_chr_11                                            NaN      NaN   \n",
       "postqc_bed_chr_12                                            NaN      NaN   \n",
       "postqc_fam_chr_12                                            NaN      NaN   \n",
       "postqc_bim_chr_12                                            NaN      NaN   \n",
       "postqc_bed_chr_13                                            NaN      NaN   \n",
       "postqc_fam_chr_13                                            NaN      NaN   \n",
       "postqc_bim_chr_13                                            NaN      NaN   \n",
       "postqc_bed_chr_14                                            NaN      NaN   \n",
       "postqc_fam_chr_14                                            NaN      NaN   \n",
       "postqc_bim_chr_14                                            NaN      NaN   \n",
       "postqc_bed_chr_15                                            NaN      NaN   \n",
       "postqc_fam_chr_15                                            NaN      NaN   \n",
       "postqc_bim_chr_15                                            NaN      NaN   \n",
       "postqc_bed_chr_16                                            NaN      NaN   \n",
       "postqc_fam_chr_16                                            NaN      NaN   \n",
       "postqc_bim_chr_16                                            NaN      NaN   \n",
       "postqc_bed_chr_17                                            NaN      NaN   \n",
       "postqc_fam_chr_17                                            NaN      NaN   \n",
       "postqc_bim_chr_17                                            NaN      NaN   \n",
       "postqc_bed_chr_18                                            NaN      NaN   \n",
       "postqc_fam_chr_18                                            NaN      NaN   \n",
       "postqc_bim_chr_18                                            NaN      NaN   \n",
       "postqc_bed_chr_19                                            NaN      NaN   \n",
       "postqc_fam_chr_19                                            NaN      NaN   \n",
       "postqc_bim_chr_19                                            NaN      NaN   \n",
       "postqc_bed_chr_20                                            NaN      NaN   \n",
       "postqc_fam_chr_20                                            NaN      NaN   \n",
       "postqc_bim_chr_20                                            NaN      NaN   \n",
       "postqc_bed_chr_21                                            NaN      NaN   \n",
       "postqc_fam_chr_21                                            NaN      NaN   \n",
       "postqc_bim_chr_21                                            NaN      NaN   \n",
       "postqc_bed_chr_22                                            NaN      NaN   \n",
       "postqc_fam_chr_22                                            NaN      NaN   \n",
       "postqc_bim_chr_22                                            NaN      NaN   \n",
       "\n",
       "                                          sampling_rate item_type    array  \\\n",
       "tabular_field_name                                                           \n",
       "collection_date                                     NaN      Data  Single    \n",
       "version                                             NaN      Data  Single    \n",
       "genecov_qc_bases                                    NaN      Data  Single    \n",
       "genecov_qc_bases_dedup                              NaN      Data  Single    \n",
       "gencove_qc_bases_dedup_mapped                       NaN      Data  Single    \n",
       "genecov_qc_effective_coverage                       NaN      Data  Single    \n",
       "genecov_qc_fraction_contamination                   NaN      Data  Single    \n",
       "genecov_qc_snps                                     NaN      Data  Single    \n",
       "genecov_qc_format_passed                            NaN      Data  Single    \n",
       "genecov_qc_r1_eq_r2_passed                          NaN      Data  Single    \n",
       "genecov_qc_r1_r2_ids_match_passed                   NaN      Data  Single    \n",
       "genecov_qc_bases_passed                             NaN      Data  Single    \n",
       "genecov_qc_bases_dedup_passed                       NaN      Data  Single    \n",
       "genecov_qc_bases_dedup_mapped_passed                NaN      Data  Single    \n",
       "genecov_qc_effective_coverage_passed                NaN      Data  Single    \n",
       "genecov_qc_fraction_contamination_passed            NaN      Data  Single    \n",
       "genecov_qc_snps_passed                              NaN      Data  Single    \n",
       "heterozygosity_proportion                           NaN      Data  Single    \n",
       "pass_het_check                                      NaN      Data  Single    \n",
       "sample_qc_gp_pass_rate                              NaN      Data  Single    \n",
       "sample_qc_gp_pass_check                             NaN      Data  Single    \n",
       "genetic_sex                                         NaN      Data  Single    \n",
       "chr_x_proportion                                    NaN      Data  Single    \n",
       "chr_y_proportion                                    NaN      Data  Single    \n",
       "sample_qc_gender_match_check                        NaN      Data  Single    \n",
       "relatives                                           NaN      Data  Single    \n",
       "self_report_ashkenaz_proportion                     NaN      Data  Single    \n",
       "self_report_middleeastern_proportion                NaN      Data  Single    \n",
       "self_report_northafrican_proportion                 NaN      Data  Single    \n",
       "self_report_sephardi_proportion                     NaN      Data  Single    \n",
       "self_report_yemen_proportion                        NaN      Data  Single    \n",
       "self_report_unknownother_proportion                 NaN      Data  Single    \n",
       "pc1                                                 NaN      Data  Single    \n",
       "pc2                                                 NaN      Data  Single    \n",
       "pc3                                                 NaN      Data  Single    \n",
       "pc4                                                 NaN      Data  Single    \n",
       "pc5                                                 NaN      Data  Single    \n",
       "pc6                                                 NaN      Data  Single    \n",
       "pc7                                                 NaN      Data  Single    \n",
       "pc8                                                 NaN      Data  Single    \n",
       "pc9                                                 NaN      Data  Single    \n",
       "pc10                                                NaN      Data  Single    \n",
       "pc11                                                NaN      Data  Single    \n",
       "pc12                                                NaN      Data  Single    \n",
       "pc13                                                NaN      Data  Single    \n",
       "pc14                                                NaN      Data  Single    \n",
       "pc15                                                NaN      Data  Single    \n",
       "pc16                                                NaN      Data  Single    \n",
       "pc17                                                NaN      Data  Single    \n",
       "pc18                                                NaN      Data  Single    \n",
       "pc19                                                NaN      Data  Single    \n",
       "pc20                                                NaN      Data  Single    \n",
       "used_in_pca_calculation                             NaN      Data  Single    \n",
       "postqc_bed_chr_1                                    NaN      Bulk  Single    \n",
       "postqc_fam_chr_1                                    NaN      Bulk  Single    \n",
       "postqc_bim_chr_1                                    NaN      Bulk  Single    \n",
       "postqc_bed_chr_2                                    NaN      Bulk  Single    \n",
       "postqc_fam_chr_2                                    NaN      Bulk  Single    \n",
       "postqc_bim_chr_2                                    NaN      Bulk  Single    \n",
       "postqc_bed_chr_3                                    NaN      Bulk  Single    \n",
       "postqc_fam_chr_3                                    NaN      Bulk  Single    \n",
       "postqc_bim_chr_3                                    NaN      Bulk  Single    \n",
       "postqc_bed_chr_4                                    NaN      Bulk  Single    \n",
       "postqc_fam_chr_4                                    NaN      Bulk  Single    \n",
       "postqc_bim_chr_4                                    NaN      Bulk  Single    \n",
       "postqc_bed_chr_5                                    NaN      Bulk  Single    \n",
       "postqc_fam_chr_5                                    NaN      Bulk  Single    \n",
       "postqc_bim_chr_5                                    NaN      Bulk  Single    \n",
       "postqc_bed_chr_6                                    NaN      Bulk  Single    \n",
       "postqc_fam_chr_6                                    NaN      Bulk  Single    \n",
       "postqc_bim_chr_6                                    NaN      Bulk  Single    \n",
       "postqc_bed_chr_7                                    NaN      Bulk  Single    \n",
       "postqc_fam_chr_7                                    NaN      Bulk  Single    \n",
       "postqc_bim_chr_7                                    NaN      Bulk  Single    \n",
       "postqc_bed_chr_8                                    NaN      Bulk  Single    \n",
       "postqc_fam_chr_8                                    NaN      Bulk  Single    \n",
       "postqc_bim_chr_8                                    NaN      Bulk  Single    \n",
       "postqc_bed_chr_9                                    NaN      Bulk  Single    \n",
       "postqc_fam_chr_9                                    NaN      Bulk  Single    \n",
       "postqc_bim_chr_9                                    NaN      Bulk  Single    \n",
       "postqc_bed_chr_10                                   NaN      Bulk  Single    \n",
       "postqc_fam_chr_10                                   NaN      Bulk  Single    \n",
       "postqc_bim_chr_10                                   NaN      Bulk  Single    \n",
       "postqc_bed_chr_11                                   NaN      Bulk  Single    \n",
       "postqc_fam_chr_11                                   NaN      Bulk  Single    \n",
       "postqc_bim_chr_11                                   NaN      Bulk  Single    \n",
       "postqc_bed_chr_12                                   NaN      Bulk  Single    \n",
       "postqc_fam_chr_12                                   NaN      Bulk  Single    \n",
       "postqc_bim_chr_12                                   NaN      Bulk  Single    \n",
       "postqc_bed_chr_13                                   NaN      Bulk  Single    \n",
       "postqc_fam_chr_13                                   NaN      Bulk  Single    \n",
       "postqc_bim_chr_13                                   NaN      Bulk  Single    \n",
       "postqc_bed_chr_14                                   NaN      Bulk  Single    \n",
       "postqc_fam_chr_14                                   NaN      Bulk  Single    \n",
       "postqc_bim_chr_14                                   NaN      Bulk  Single    \n",
       "postqc_bed_chr_15                                   NaN      Bulk  Single    \n",
       "postqc_fam_chr_15                                   NaN      Bulk  Single    \n",
       "postqc_bim_chr_15                                   NaN      Bulk  Single    \n",
       "postqc_bed_chr_16                                   NaN      Bulk  Single    \n",
       "postqc_fam_chr_16                                   NaN      Bulk  Single    \n",
       "postqc_bim_chr_16                                   NaN      Bulk  Single    \n",
       "postqc_bed_chr_17                                   NaN      Bulk  Single    \n",
       "postqc_fam_chr_17                                   NaN      Bulk  Single    \n",
       "postqc_bim_chr_17                                   NaN      Bulk  Single    \n",
       "postqc_bed_chr_18                                   NaN      Bulk  Single    \n",
       "postqc_fam_chr_18                                   NaN      Bulk  Single    \n",
       "postqc_bim_chr_18                                   NaN      Bulk  Single    \n",
       "postqc_bed_chr_19                                   NaN      Bulk  Single    \n",
       "postqc_fam_chr_19                                   NaN      Bulk  Single    \n",
       "postqc_bim_chr_19                                   NaN      Bulk  Single    \n",
       "postqc_bed_chr_20                                   NaN      Bulk  Single    \n",
       "postqc_fam_chr_20                                   NaN      Bulk  Single    \n",
       "postqc_bim_chr_20                                   NaN      Bulk  Single    \n",
       "postqc_bed_chr_21                                   NaN      Bulk  Single    \n",
       "postqc_fam_chr_21                                   NaN      Bulk  Single    \n",
       "postqc_bim_chr_21                                   NaN      Bulk  Single    \n",
       "postqc_bed_chr_22                                   NaN      Bulk  Single    \n",
       "postqc_fam_chr_22                                   NaN      Bulk  Single    \n",
       "postqc_bim_chr_22                                   NaN      Bulk  Single    \n",
       "\n",
       "                                         cohorts               data_type  \\\n",
       "tabular_field_name                                                         \n",
       "collection_date                              10K                 Tabular   \n",
       "version                                      10K                 Tabular   \n",
       "genecov_qc_bases                             10K                 Tabular   \n",
       "genecov_qc_bases_dedup                       10K                 Tabular   \n",
       "gencove_qc_bases_dedup_mapped                10K                 Tabular   \n",
       "genecov_qc_effective_coverage                10K                 Tabular   \n",
       "genecov_qc_fraction_contamination            10K                 Tabular   \n",
       "genecov_qc_snps                              10K                 Tabular   \n",
       "genecov_qc_format_passed                     10K                 Tabular   \n",
       "genecov_qc_r1_eq_r2_passed                   10K                 Tabular   \n",
       "genecov_qc_r1_r2_ids_match_passed            10K                 Tabular   \n",
       "genecov_qc_bases_passed                      10K                 Tabular   \n",
       "genecov_qc_bases_dedup_passed                10K                 Tabular   \n",
       "genecov_qc_bases_dedup_mapped_passed         10K                 Tabular   \n",
       "genecov_qc_effective_coverage_passed         10K                 Tabular   \n",
       "genecov_qc_fraction_contamination_passed     10K                 Tabular   \n",
       "genecov_qc_snps_passed                       10K                 Tabular   \n",
       "heterozygosity_proportion                    10K                 Tabular   \n",
       "pass_het_check                               10K                 Tabular   \n",
       "sample_qc_gp_pass_rate                       10K                 Tabular   \n",
       "sample_qc_gp_pass_check                      10K                 Tabular   \n",
       "genetic_sex                                  10K                 Tabular   \n",
       "chr_x_proportion                             10K                 Tabular   \n",
       "chr_y_proportion                             10K                 Tabular   \n",
       "sample_qc_gender_match_check                 10K                 Tabular   \n",
       "relatives                                    10K                 Tabular   \n",
       "self_report_ashkenaz_proportion              10K                 Tabular   \n",
       "self_report_middleeastern_proportion         10K                 Tabular   \n",
       "self_report_northafrican_proportion          10K                 Tabular   \n",
       "self_report_sephardi_proportion              10K                 Tabular   \n",
       "self_report_yemen_proportion                 10K                 Tabular   \n",
       "self_report_unknownother_proportion          10K                 Tabular   \n",
       "pc1                                          10K                 Tabular   \n",
       "pc2                                          10K                 Tabular   \n",
       "pc3                                          10K                 Tabular   \n",
       "pc4                                          10K                 Tabular   \n",
       "pc5                                          10K                 Tabular   \n",
       "pc6                                          10K                 Tabular   \n",
       "pc7                                          10K                 Tabular   \n",
       "pc8                                          10K                 Tabular   \n",
       "pc9                                          10K                 Tabular   \n",
       "pc10                                         10K                 Tabular   \n",
       "pc11                                         10K                 Tabular   \n",
       "pc12                                         10K                 Tabular   \n",
       "pc13                                         10K                 Tabular   \n",
       "pc14                                         10K                 Tabular   \n",
       "pc15                                         10K                 Tabular   \n",
       "pc16                                         10K                 Tabular   \n",
       "pc17                                         10K                 Tabular   \n",
       "pc18                                         10K                 Tabular   \n",
       "pc19                                         10K                 Tabular   \n",
       "pc20                                         10K                 Tabular   \n",
       "used_in_pca_calculation                      10K                 Tabular   \n",
       "postqc_bed_chr_1                             10K  Compressed binary file   \n",
       "postqc_fam_chr_1                             10K               Text file   \n",
       "postqc_bim_chr_1                             10K               Text file   \n",
       "postqc_bed_chr_2                             10K  Compressed binary file   \n",
       "postqc_fam_chr_2                             10K               Text file   \n",
       "postqc_bim_chr_2                             10K               Text file   \n",
       "postqc_bed_chr_3                             10K  Compressed binary file   \n",
       "postqc_fam_chr_3                             10K               Text file   \n",
       "postqc_bim_chr_3                             10K               Text file   \n",
       "postqc_bed_chr_4                             10K  Compressed binary file   \n",
       "postqc_fam_chr_4                             10K               Text file   \n",
       "postqc_bim_chr_4                             10K               Text file   \n",
       "postqc_bed_chr_5                             10K  Compressed binary file   \n",
       "postqc_fam_chr_5                             10K               Text file   \n",
       "postqc_bim_chr_5                             10K               Text file   \n",
       "postqc_bed_chr_6                             10K  Compressed binary file   \n",
       "postqc_fam_chr_6                             10K               Text file   \n",
       "postqc_bim_chr_6                             10K               Text file   \n",
       "postqc_bed_chr_7                             10K  Compressed binary file   \n",
       "postqc_fam_chr_7                             10K               Text file   \n",
       "postqc_bim_chr_7                             10K               Text file   \n",
       "postqc_bed_chr_8                             10K  Compressed binary file   \n",
       "postqc_fam_chr_8                             10K               Text file   \n",
       "postqc_bim_chr_8                             10K               Text file   \n",
       "postqc_bed_chr_9                             10K  Compressed binary file   \n",
       "postqc_fam_chr_9                             10K               Text file   \n",
       "postqc_bim_chr_9                             10K               Text file   \n",
       "postqc_bed_chr_10                            10K  Compressed binary file   \n",
       "postqc_fam_chr_10                            10K               Text file   \n",
       "postqc_bim_chr_10                            10K               Text file   \n",
       "postqc_bed_chr_11                            10K  Compressed binary file   \n",
       "postqc_fam_chr_11                            10K               Text file   \n",
       "postqc_bim_chr_11                            10K               Text file   \n",
       "postqc_bed_chr_12                            10K  Compressed binary file   \n",
       "postqc_fam_chr_12                            10K               Text file   \n",
       "postqc_bim_chr_12                            10K               Text file   \n",
       "postqc_bed_chr_13                            10K  Compressed binary file   \n",
       "postqc_fam_chr_13                            10K               Text file   \n",
       "postqc_bim_chr_13                            10K               Text file   \n",
       "postqc_bed_chr_14                            10K  Compressed binary file   \n",
       "postqc_fam_chr_14                            10K               Text file   \n",
       "postqc_bim_chr_14                            10K               Text file   \n",
       "postqc_bed_chr_15                            10K  Compressed binary file   \n",
       "postqc_fam_chr_15                            10K               Text file   \n",
       "postqc_bim_chr_15                            10K               Text file   \n",
       "postqc_bed_chr_16                            10K  Compressed binary file   \n",
       "postqc_fam_chr_16                            10K               Text file   \n",
       "postqc_bim_chr_16                            10K               Text file   \n",
       "postqc_bed_chr_17                            10K  Compressed binary file   \n",
       "postqc_fam_chr_17                            10K               Text file   \n",
       "postqc_bim_chr_17                            10K               Text file   \n",
       "postqc_bed_chr_18                            10K  Compressed binary file   \n",
       "postqc_fam_chr_18                            10K               Text file   \n",
       "postqc_bim_chr_18                            10K               Text file   \n",
       "postqc_bed_chr_19                            10K  Compressed binary file   \n",
       "postqc_fam_chr_19                            10K               Text file   \n",
       "postqc_bim_chr_19                            10K               Text file   \n",
       "postqc_bed_chr_20                            10K  Compressed binary file   \n",
       "postqc_fam_chr_20                            10K               Text file   \n",
       "postqc_bim_chr_20                            10K               Text file   \n",
       "postqc_bed_chr_21                            10K  Compressed binary file   \n",
       "postqc_fam_chr_21                            10K               Text file   \n",
       "postqc_bim_chr_21                            10K               Text file   \n",
       "postqc_bed_chr_22                            10K  Compressed binary file   \n",
       "postqc_fam_chr_22                            10K               Text file   \n",
       "postqc_bim_chr_22                            10K               Text file   \n",
       "\n",
       "                                               debut    pandas_dtype  \n",
       "tabular_field_name                                                    \n",
       "collection_date                           2019-03-11  datetime64[ns]  \n",
       "version                                   2019-03-11          object  \n",
       "genecov_qc_bases                          2019-03-11         float64  \n",
       "genecov_qc_bases_dedup                    2019-03-11         float64  \n",
       "gencove_qc_bases_dedup_mapped             2019-03-11         float64  \n",
       "genecov_qc_effective_coverage             2019-03-11         float64  \n",
       "genecov_qc_fraction_contamination         2019-03-11         float64  \n",
       "genecov_qc_snps                           2019-03-11         float64  \n",
       "genecov_qc_format_passed                  2019-03-11          object  \n",
       "genecov_qc_r1_eq_r2_passed                2019-03-11          object  \n",
       "genecov_qc_r1_r2_ids_match_passed         2019-03-11          object  \n",
       "genecov_qc_bases_passed                   2019-03-11          object  \n",
       "genecov_qc_bases_dedup_passed             2019-03-11          object  \n",
       "genecov_qc_bases_dedup_mapped_passed      2019-03-11          object  \n",
       "genecov_qc_effective_coverage_passed      2019-03-11          object  \n",
       "genecov_qc_fraction_contamination_passed  2019-03-11          object  \n",
       "genecov_qc_snps_passed                    2019-03-11          object  \n",
       "heterozygosity_proportion                 2019-03-11         float64  \n",
       "pass_het_check                            2019-03-11            bool  \n",
       "sample_qc_gp_pass_rate                    2019-03-11         float64  \n",
       "sample_qc_gp_pass_check                   2019-03-11            bool  \n",
       "genetic_sex                               2019-03-11         float64  \n",
       "chr_x_proportion                          2019-03-11         float64  \n",
       "chr_y_proportion                          2019-03-11         float64  \n",
       "sample_qc_gender_match_check              2019-03-11            bool  \n",
       "relatives                                 2019-03-11          object  \n",
       "self_report_ashkenaz_proportion           2019-03-11         float64  \n",
       "self_report_middleeastern_proportion      2019-03-11         float64  \n",
       "self_report_northafrican_proportion       2019-03-11         float64  \n",
       "self_report_sephardi_proportion           2019-03-11         float64  \n",
       "self_report_yemen_proportion              2019-03-11         float64  \n",
       "self_report_unknownother_proportion       2019-03-11         float64  \n",
       "pc1                                       2019-03-11         float64  \n",
       "pc2                                       2019-03-11         float64  \n",
       "pc3                                       2019-03-11         float64  \n",
       "pc4                                       2019-03-11         float64  \n",
       "pc5                                       2019-03-11         float64  \n",
       "pc6                                       2019-03-11         float64  \n",
       "pc7                                       2019-03-11         float64  \n",
       "pc8                                       2019-03-11         float64  \n",
       "pc9                                       2019-03-11         float64  \n",
       "pc10                                      2019-03-11         float64  \n",
       "pc11                                      2019-03-11         float64  \n",
       "pc12                                      2019-03-11         float64  \n",
       "pc13                                      2019-03-11         float64  \n",
       "pc14                                      2019-03-11         float64  \n",
       "pc15                                      2019-03-11         float64  \n",
       "pc16                                      2019-03-11         float64  \n",
       "pc17                                      2019-03-11         float64  \n",
       "pc18                                      2019-03-11         float64  \n",
       "pc19                                      2019-03-11         float64  \n",
       "pc20                                      2019-03-11         float64  \n",
       "used_in_pca_calculation                   2019-03-11            bool  \n",
       "postqc_bed_chr_1                          2019-03-11          object  \n",
       "postqc_fam_chr_1                          2019-03-11          object  \n",
       "postqc_bim_chr_1                          2019-03-11          object  \n",
       "postqc_bed_chr_2                          2019-03-11          object  \n",
       "postqc_fam_chr_2                          2019-03-11          object  \n",
       "postqc_bim_chr_2                          2019-03-11          object  \n",
       "postqc_bed_chr_3                          2019-03-11          object  \n",
       "postqc_fam_chr_3                          2019-03-11          object  \n",
       "postqc_bim_chr_3                          2019-03-11          object  \n",
       "postqc_bed_chr_4                          2019-03-11          object  \n",
       "postqc_fam_chr_4                          2019-03-11          object  \n",
       "postqc_bim_chr_4                          2019-03-11          object  \n",
       "postqc_bed_chr_5                          2019-03-11          object  \n",
       "postqc_fam_chr_5                          2019-03-11          object  \n",
       "postqc_bim_chr_5                          2019-03-11          object  \n",
       "postqc_bed_chr_6                          2019-03-11          object  \n",
       "postqc_fam_chr_6                          2019-03-11          object  \n",
       "postqc_bim_chr_6                          2019-03-11          object  \n",
       "postqc_bed_chr_7                          2019-03-11          object  \n",
       "postqc_fam_chr_7                          2019-03-11          object  \n",
       "postqc_bim_chr_7                          2019-03-11          object  \n",
       "postqc_bed_chr_8                          2019-03-11          object  \n",
       "postqc_fam_chr_8                          2019-03-11          object  \n",
       "postqc_bim_chr_8                          2019-03-11          object  \n",
       "postqc_bed_chr_9                          2019-03-11          object  \n",
       "postqc_fam_chr_9                          2019-03-11          object  \n",
       "postqc_bim_chr_9                          2019-03-11          object  \n",
       "postqc_bed_chr_10                         2019-03-11          object  \n",
       "postqc_fam_chr_10                         2019-03-11          object  \n",
       "postqc_bim_chr_10                         2019-03-11          object  \n",
       "postqc_bed_chr_11                         2019-03-11          object  \n",
       "postqc_fam_chr_11                         2019-03-11          object  \n",
       "postqc_bim_chr_11                         2019-03-11          object  \n",
       "postqc_bed_chr_12                         2019-03-11          object  \n",
       "postqc_fam_chr_12                         2019-03-11          object  \n",
       "postqc_bim_chr_12                         2019-03-11          object  \n",
       "postqc_bed_chr_13                         2019-03-11          object  \n",
       "postqc_fam_chr_13                         2019-03-11          object  \n",
       "postqc_bim_chr_13                         2019-03-11          object  \n",
       "postqc_bed_chr_14                         2019-03-11          object  \n",
       "postqc_fam_chr_14                         2019-03-11          object  \n",
       "postqc_bim_chr_14                         2019-03-11          object  \n",
       "postqc_bed_chr_15                         2019-03-11          object  \n",
       "postqc_fam_chr_15                         2019-03-11          object  \n",
       "postqc_bim_chr_15                         2019-03-11          object  \n",
       "postqc_bed_chr_16                         2019-03-11          object  \n",
       "postqc_fam_chr_16                         2019-03-11          object  \n",
       "postqc_bim_chr_16                         2019-03-11          object  \n",
       "postqc_bed_chr_17                         2019-03-11          object  \n",
       "postqc_fam_chr_17                         2019-03-11          object  \n",
       "postqc_bim_chr_17                         2019-03-11          object  \n",
       "postqc_bed_chr_18                         2019-03-11          object  \n",
       "postqc_fam_chr_18                         2019-03-11          object  \n",
       "postqc_bim_chr_18                         2019-03-11          object  \n",
       "postqc_bed_chr_19                         2019-03-11          object  \n",
       "postqc_fam_chr_19                         2019-03-11          object  \n",
       "postqc_bim_chr_19                         2019-03-11          object  \n",
       "postqc_bed_chr_20                         2019-03-11          object  \n",
       "postqc_fam_chr_20                         2019-03-11          object  \n",
       "postqc_bim_chr_20                         2019-03-11          object  \n",
       "postqc_bed_chr_21                         2019-03-11          object  \n",
       "postqc_fam_chr_21                         2019-03-11          object  \n",
       "postqc_bim_chr_21                         2019-03-11          object  \n",
       "postqc_bed_chr_22                         2019-03-11          object  \n",
       "postqc_fam_chr_22                         2019-03-11          object  \n",
       "postqc_bim_chr_22                         2019-03-11          object  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dl.dict"
   ]
  }
 ],
 "metadata": {
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
