{
 "cells": [
  {
   "cell_type": "raw",
   "id": "5d779971",
   "metadata": {},
   "source": [
    "---\n",
    "output-file: 014-human_genetics.html\n",
    "title: 014-human_genetics\n",
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7533508e",
   "metadata": {},
   "source": [
    "# Human genetics \n",
    "\n",
    "### Description \n",
    "\n",
    "The Human Phenotype Project collects genomic variation data on all its participants. The genomic data together with the Human Phenotype Project  deep-phenotypes allows to investigate the progression of disease, and to explore personalized treatments. We genotype millions of positions by low-pass sequencing combined with imputation using gencove platform technologies. Genotype imputation is a process of statistically inferring unobserved genotypes using known haplotypes in a population. The performance of Gencove genotype imputation is very high ( accuracy > 98% ) ([Wasik et al. 2021](https://link.springer.com/article/10.1186/s12864-021-07508-2?utm_campaign=2022.01%20Publications&utm_source=LP%20for%20Pharmacogenomics)). \n",
    "\n",
    "### Introduction\n",
    "\n",
    "Genomic variation refers to DNA sequence differences between individuals. Some genetic variants can influence biological function (such as a mutation that causes a genetic disease), while others have no known effects. The genomic data together with the Human Phenotype Project rich variety of phenotypic data enables investigation of how genetics impact the progression of disease, and exploration of personalized prevention strategies and  treatments. \n",
    "\n",
    "The Human Phenotype Project uses low-pass sequencing combined with [imputation](https://en.wikipedia.org/wiki/Imputation_(genetics)) to obtain genomic variation, using [genocve](https://gencove.com) platform technologies. We do not target specific regions by sequencing, but rather sample the entire genome with low average depth of coverage, and impute the sequence data to the haplotype reference panel which includes a large set of known genetic variants. Briefly, the model approach is to identify haplotypes shared between a study sample and a reference panel based on the study sample observed genotypes. Next, the unobserved genotypes are extrapolated according to the large reference data as illustrated in Figure 1. Imputation is performed for single nucleotide polymorphisms (SNPs) and short insertion–deletion variants (indels), structural variation are currently not included. The overall accuracy of gencove imputed genotypes is 98.22% (Wasik et al. 2021). We also validated the accuracy of the imputation using two hundred Human Phenotype Project participants from several ancestries which had both low-pass sequencing done and Illumina genotyping array data. We obtained high concordance rates, greater than 96%.\n",
    "\n",
    "![study sample](study_sample.png)\n",
    "\n",
    "Figure 1: taken from [Li et al. 2013](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2925172/).\n",
    "Panel A illustrates the observed data which consists of genotypes at a modest number of genetic markers in each sample being studied and of detailed information on genotypes (or haplotypes) for a reference sample. Panel B illustrates the process of identifying regions of chromosomes shared between a study sample and individuals in the reference panel. In Panel C, observed genotypes and haplotype sharing information have been combined to fill in a series of unobserved genotypes in the study sample.*\n",
    "\n",
    "### Measurement protocol \n",
    "<!-- long measurment protocol for the data browser -->\n",
    "#### Genotyping Procedure\n",
    "DNA is collected from Human Phenotype Project participants using a buccal swab during their baseline visit. DNA is collected only once as germline DNA is considered constant. The buccal swabs are shipped to Gencove laboratories. The entire sequencing and genotyping workflow is performed by Gencove technologies. The Detailed information about the Gencove algorithms and quality control protocols are available in Wasik et al. 2021,  we briefly describe the key steps here.\n",
    "\n",
    "Gencove’s workflow contains three steps: DNA extraction and library preparation, low-pass whole genome sequencing, and imputation analysis. Sequencing is performed at Neogen GeneSeek Laboratories using Illumina Novaseq at low-pass, targeting coverage levels of 0.5× and 1×. Whole-genome coverage is defined as the number of mapped DNA bases divided by the size of the genome (∼3 billion base pairs). We perform paired-end sequencing at length of 150-bp resulting in two FASTQ files (R1 and R2). Sequencing at paired-end mode significantly improves alignment to the genome. Next, the FASTQ files are aligned to the hs37-1kg reference genome, resulting in a BAM (Binary Alignment Map) file. In the sequence alignment procedure the DNA reads are mapped (in-pair) to the human genome reference sequence by similarity between the sequences. Lastly, loimpute-v0.1.0, Gencove’s imputation procedure, is used over the aligned reads. Imputation is done for the autosomal chromosomes (1-22) using The 1000 Genomes phase 3 haplotypes (1KGP3) as a reference panel, resulting in a VCF (Variant Call Format) file. The VCF contains the most likely genotype (GT) at 37,559,140 positions and their corresponding genotype probabilities (GP). Finally, Gencove employs quality control procedures to exclude samples with poor sequencing and genotyping results.\n",
    "\n",
    "### Data availability \n",
    "<!-- for the example notebooks -->\n",
    "The information is stored in a number of statistics parquet files:<br>\n",
    "- `human_genetics.parquet`: sample metadata, including QC statistics, paths to PLINK variant files (raw and post-QC), and principal components (PCs).<br>\n",
    "- `variants_qc.parquet`: variant QC statistics.<br>\n",
    "- `relationship/relationship_ibs.txt`: IBS calculated by PLINK for pairs of participants.<br>\n",
    "- `relationship/relationship_king.tsv`: King kinship coefficients for pairs of participants.<br>\n",
    "- `pca/pca.parquet`: a PLINK file containing  principal components.<br>\n",
    "- `pca/pca_loadings.tsv`: a PLINK file containing principal component loadings calculated.<br>\n",
    "\n",
    "```mermaid\n",
    "graph LR;\n",
    "    D[DNA extraction] -->|sequencing| F(FASTQ) -->|alignment| B(BAM) -->|imputation| V(VCF)\n",
    "    H[REFERENCE HAPLOTYPES] -->|imputation| V -->|\"sample<br>aggregation\"| R(BED<br>raw) -->|QC| Q(BED<br>post-qc)\n",
    "    Q --> PCA(PCA)\n",
    "    Q --> IBS(PLINK IBS<br>relationships)\n",
    "    Q --> KING(KING<br>relationships)\n",
    "```\n",
    "\n",
    "### Summary of available data \n",
    "<!-- for the data browser -->\n",
    "#### Raw DNA data\n",
    "Pheno provides per each sample all of the raw sequencing and genotyping workflow data, including FASTQ, BAM, VCF, and QC files.\n",
    "\n",
    "- FASTQ files are the sequencing machine outputs. They are text files that store the DNA sequences and their corresponding quality scores. We perform the sequencing at paired-end mode, meaning that each DNA molecule is sequenced from both sides, therefore there are two FASTQ files (R1 and R2).\n",
    "- Binary Alignment Map (BAM) files are a binary and indexed representation of sequence alignment results. The BAM file stores the position of the DNA reads in the reference genome and the corresponding mapping quality. Indexing is provided in the bam.bai file, and allows to retrieve alignments that overlap a specific location quickly.\n",
    "- Variant Call Format (VCF) file is a text file used for storing genetic variations. VCF consists of a header and a table. The header provides metadata describing the table fields and their values. The FILTER field holds a flag indicating whether the variant has failed or successfully passed filters. The raw genotype data includes all 37M positions regardless of their FILTER flag status. Genotypes are flagged as ‘LOWCONF’ at the FILTER field if their maximum GP is less than 0.9, and otherwise flagged as ‘PASS’ quality check. The following FORMAT fields are provided in the VCF - GT:RC:AC:GP:DS ([VCF specification](https://samtools.github.io/hts-specs/VCFv4.3.pdf)). Please note that the data is unphased.\n",
    "- PLINK binary biallelic genotype file - In addition to the single sample VCFs we also provide the raw genotype data of all samples jointly in PLINK bed format. Because of the large number of variants, we split the data to chromosomes  (22 PLINK bed/bim/bam files). PLINK bed file is a compacted binary representation of genotypes at bi-allelic variants. It is accompanied by .bim and .fam files, which are the metadata files for variants and samples, respectively ([PLINK specification](https://www.cog-genomics.org/plink/1.9/formats#bed)).\n",
    "- Gencove QC file - Gencove performs quality control to exclude samples with poor sequencing and genotyping results. Gencove’s quality control matrices are included in the sample metadata.\n",
    "\n",
    "The raw genotype data includes 34,580,532 SNPs and 2,978,606 short indels, 37,494,604 of the variants are bi-alleleic and 30,990 are multi-allelic. Multi-allelic variants are represented in variant normalization format, meaning each alternate allele is listed in a separate line, in both the VCFs and PLINK.\n",
    "\n",
    "Notably, we do not provide the raw data of samples who failed Gencove QC or for which we identify a possibility of DNA contamination or of mislabelling (i.e. DNA samples are associated with the wrong individual), as explained in section ‘Sample evaluation’. \n",
    "\n",
    "#### Post-genotyping QC data\n",
    "- We provide the QC genotyped data in PLINK format as explained above. The post-genotyping QC procedure resulted in genotypes for ~99.98% of total samples, and 5,770,633 variants that passed variant QC (~15.36% of the total variants). \n",
    "- In addition to the genotype data we provide auxiliary tabulated datasets such as : variants and sample QC metrics, principal component projection, and familial relationships estimates.\n",
    "\n",
    "![availbale data](genetics_data.png)\n",
    "\n",
    "### Relevant links\n",
    "\n",
    "* Pheno Knowledgebase: https://knowledgebase.pheno.ai/datasets/014-human_genetics.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d54e32bf",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2024-04-02T11:45:48.273065Z",
     "iopub.status.busy": "2024-04-02T11:45:48.272875Z",
     "iopub.status.idle": "2024-04-02T11:45:48.556510Z",
     "shell.execute_reply": "2024-04-02T11:45:48.555936Z"
    }
   },
   "outputs": [],
   "source": [
    "#| echo: false\n",
    "import pandas as pd\n",
    "pd.set_option(\"display.max_rows\", 500)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "e4e8d45d",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2024-04-02T11:45:48.559058Z",
     "iopub.status.busy": "2024-04-02T11:45:48.558776Z",
     "iopub.status.idle": "2024-04-02T11:45:49.837380Z",
     "shell.execute_reply": "2024-04-02T11:45:49.836123Z"
    }
   },
   "outputs": [],
   "source": [
    "from pheno_utils import PhenoLoader"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c888255f",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2024-04-02T11:45:49.839785Z",
     "iopub.status.busy": "2024-04-02T11:45:49.839462Z",
     "iopub.status.idle": "2024-04-02T11:45:50.841929Z",
     "shell.execute_reply": "2024-04-02T11:45:50.840701Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "PhenoLoader for human_genetics with\n",
       "189 fields\n",
       "3 tables: ['human_genetics', 'pca', 'age_sex']"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pl = PhenoLoader('human_genetics')\n",
    "pl"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f7060e75",
   "metadata": {},
   "source": [
    "# Data dictionary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "7f82b1ad",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2024-04-02T11:45:50.844207Z",
     "iopub.status.busy": "2024-04-02T11:45:50.843992Z",
     "iopub.status.idle": "2024-04-02T11:45:50.883726Z",
     "shell.execute_reply": "2024-04-02T11:45:50.883264Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>folder_id</th>\n",
       "      <th>feature_set</th>\n",
       "      <th>field_string</th>\n",
       "      <th>description_string</th>\n",
       "      <th>field_type</th>\n",
       "      <th>bulk_dictionary</th>\n",
       "      <th>bulk_file_extension</th>\n",
       "      <th>relative_location</th>\n",
       "      <th>data_coding</th>\n",
       "      <th>stability</th>\n",
       "      <th>units</th>\n",
       "      <th>sampling_rate</th>\n",
       "      <th>strata</th>\n",
       "      <th>sexed</th>\n",
       "      <th>array</th>\n",
       "      <th>completed</th>\n",
       "      <th>debut</th>\n",
       "      <th>transformation</th>\n",
       "      <th>list_of_tags</th>\n",
       "      <th>pandas_dtype</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tabular_field_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>collection_date</th>\n",
       "      <td>14</td>\n",
       "      <td>human_genetics</td>\n",
       "      <td>Collection date</td>\n",
       "      <td>The date of downloading Gencove results from t...</td>\n",
       "      <td>Date</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Time</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Auxiliary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>datetime64[ns]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>version</th>\n",
       "      <td>14</td>\n",
       "      <td>human_genetics</td>\n",
       "      <td>Gencove version</td>\n",
       "      <td>Gencove API version 1 or 2</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Auxiliary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gencove_fastq_r1</th>\n",
       "      <td>14</td>\n",
       "      <td>fastq_bam</td>\n",
       "      <td>R1_FASTQ</td>\n",
       "      <td>Per sample FASTQ file, a text file that contai...</td>\n",
       "      <td>Fastq file (individual)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R1.fastq.gz</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gencove_fastq_r2</th>\n",
       "      <td>14</td>\n",
       "      <td>fastq_bam</td>\n",
       "      <td>R2_FASTQ</td>\n",
       "      <td>Per sample FASTQ file, a text file that contai...</td>\n",
       "      <td>Fastq file (individual)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R2.fastq.gz</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gencove_bam</th>\n",
       "      <td>14</td>\n",
       "      <td>fastq_bam</td>\n",
       "      <td>BAM</td>\n",
       "      <td>Per sample Binary Alignment Map (BAM) file for...</td>\n",
       "      <td>Bulk file (individual)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gencove_bai</th>\n",
       "      <td>14</td>\n",
       "      <td>fastq_bam</td>\n",
       "      <td>BAM indices</td>\n",
       "      <td>Per sample indices for BAM file</td>\n",
       "      <td>Bulk file (individual)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bam.bai</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gencove_vcf</th>\n",
       "      <td>14</td>\n",
       "      <td>fastq_bam</td>\n",
       "      <td>Imputation Variant Call Format (VCF)</td>\n",
       "      <td>Per sample VCF, a text file storing imputed ge...</td>\n",
       "      <td>Bulk file (individual)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>vcf.gz</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genecov_qc_bases</th>\n",
       "      <td>14</td>\n",
       "      <td>human_genetics</td>\n",
       "      <td>Bases sequenced</td>\n",
       "      <td>number of total bases sequenced. from genecov ...</td>\n",
       "      <td>Integer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Count</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Auxiliary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>int</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genecov_qc_bases_dedup</th>\n",
       "      <td>14</td>\n",
       "      <td>human_genetics</td>\n",
       "      <td>Deduplicated bases</td>\n",
       "      <td>number of deduplicated bases. from genecov qc ...</td>\n",
       "      <td>Integer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Count</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Auxiliary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>int</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gencove_qc_bases_dedup_mapped</th>\n",
       "      <td>14</td>\n",
       "      <td>human_genetics</td>\n",
       "      <td>Deduplicated bases aligned</td>\n",
       "      <td>number of deduplicated bases that have aligned...</td>\n",
       "      <td>Integer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Count</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Auxiliary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>int</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genecov_qc_effective_coverage</th>\n",
       "      <td>14</td>\n",
       "      <td>human_genetics</td>\n",
       "      <td>Effective coverage</td>\n",
       "      <td>effective coverage. from genecov qc file</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Precent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Auxiliary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>float</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genecov_qc_fraction_contamination</th>\n",
       "      <td>14</td>\n",
       "      <td>human_genetics</td>\n",
       "      <td>DNA contamination</td>\n",
       "      <td>contamination by DNA from another sample of th...</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Precent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Auxiliary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>float</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genecov_qc_snps</th>\n",
       "      <td>14</td>\n",
       "      <td>human_genetics</td>\n",
       "      <td>SNPs covered</td>\n",
       "      <td>number of variants in reference panel that are...</td>\n",
       "      <td>Integer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Count</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Auxiliary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>int</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genecov_qc_format_passed</th>\n",
       "      <td>14</td>\n",
       "      <td>human_genetics</td>\n",
       "      <td>Passed FASTQ format validity</td>\n",
       "      <td>Indicates proper formatting of the input FASTQ...</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Auxiliary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genecov_qc_r1_eq_r2_passed</th>\n",
       "      <td>14</td>\n",
       "      <td>human_genetics</td>\n",
       "      <td>Passed paired-end count</td>\n",
       "      <td>Indicates number of bases in R1 file equal to ...</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Auxiliary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genecov_qc_r1_r2_ids_match_passed</th>\n",
       "      <td>14</td>\n",
       "      <td>human_genetics</td>\n",
       "      <td>Passed paired-end match</td>\n",
       "      <td>Indicates R1 read identifiers match R2 read id...</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Auxiliary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genecov_qc_bases_passed</th>\n",
       "      <td>14</td>\n",
       "      <td>human_genetics</td>\n",
       "      <td>Passed Bases sequenced</td>\n",
       "      <td>Indicates number of total bases sequenced grea...</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Auxiliary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genecov_qc_bases_dedup_passed</th>\n",
       "      <td>14</td>\n",
       "      <td>human_genetics</td>\n",
       "      <td>Passed Deduplicated bases</td>\n",
       "      <td>Indicates number of deduplicated bases greater...</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Auxiliary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genecov_qc_bases_dedup_mapped_passed</th>\n",
       "      <td>14</td>\n",
       "      <td>human_genetics</td>\n",
       "      <td>Passed Deduplicated bases aligned</td>\n",
       "      <td>Indicates number of deduplicated bases aligned...</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Auxiliary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genecov_qc_effective_coverage_passed</th>\n",
       "      <td>14</td>\n",
       "      <td>human_genetics</td>\n",
       "      <td>Passed Effective coverage</td>\n",
       "      <td>Indicates effective coverage greater than 0. f...</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Auxiliary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genecov_qc_fraction_contamination_passed</th>\n",
       "      <td>14</td>\n",
       "      <td>human_genetics</td>\n",
       "      <td>Passed DNA contamination</td>\n",
       "      <td>Indicates contamination by DNA less than 0.06....</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Auxiliary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genecov_qc_snps_passed</th>\n",
       "      <td>14</td>\n",
       "      <td>human_genetics</td>\n",
       "      <td>Passed SNPs covered</td>\n",
       "      <td>Indicates number of variants covered by at lea...</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Auxiliary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>object</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>heterozygosity_proportion</th>\n",
       "      <td>14</td>\n",
       "      <td>human_genetics</td>\n",
       "      <td>Heterozygosity proportion</td>\n",
       "      <td>Estimated heterozygosity proportion across a s...</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Precent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>float</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pass_het_check</th>\n",
       "      <td>14</td>\n",
       "      <td>human_genetics</td>\n",
       "      <td>Passed Heterozygosity proportion</td>\n",
       "      <td>Indicates heterozygosity proportion is less th...</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Supporting</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>bool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_qc_gp_pass_rate</th>\n",
       "      <td>14</td>\n",
       "      <td>human_genetics</td>\n",
       "      <td>Genotype Probabilities pass rate</td>\n",
       "      <td>Proportion of genotypes with maximum GP greate...</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Precent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Supporting</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>float</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_qc_gp_pass_check</th>\n",
       "      <td>14</td>\n",
       "      <td>human_genetics</td>\n",
       "      <td>Passed Genotype Probabilities</td>\n",
       "      <td>Indicates proportion of GP greater than 0.9 is...</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Supporting</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>bool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genetic_sex</th>\n",
       "      <td>14</td>\n",
       "      <td>human_genetics</td>\n",
       "      <td>DNA sex</td>\n",
       "      <td>Sex as determined by the percentage of read-al...</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>int</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_x_proportion</th>\n",
       "      <td>14</td>\n",
       "      <td>human_genetics</td>\n",
       "      <td>Proportion of alignment to chromosome X</td>\n",
       "      <td>percentage of reads aligned to chromosome X</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Precent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>float</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_y_proportion</th>\n",
       "      <td>14</td>\n",
       "      <td>human_genetics</td>\n",
       "      <td>Proportion of alignment to chromosome Y</td>\n",
       "      <td>percentage of reads aligned to chromosome Y</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Precent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>float</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_qc_gender_match_check</th>\n",
       "      <td>14</td>\n",
       "      <td>human_genetics</td>\n",
       "      <td>Passed gender check</td>\n",
       "      <td>Indicates inferred_gender and submitted_gender...</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Supporting</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>bool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>variants_qc_parquet</th>\n",
       "      <td>14</td>\n",
       "      <td>variants_qc</td>\n",
       "      <td>Variants QC report</td>\n",
       "      <td>QC report for each SNP</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>parquet</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Supporting</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>relatives</th>\n",
       "      <td>14</td>\n",
       "      <td>relationship</td>\n",
       "      <td>Relatives based on IBD estimation</td>\n",
       "      <td>List of second and first degree relatives acco...</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>relationship_king_tsv</th>\n",
       "      <td>14</td>\n",
       "      <td>relationship</td>\n",
       "      <td>KING kingship file</td>\n",
       "      <td>KING kinship coefficient matrix for pairs of p...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tsv</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>relationship_ibs_txt</th>\n",
       "      <td>14</td>\n",
       "      <td>relationship</td>\n",
       "      <td>PLINK IBS file</td>\n",
       "      <td>PLINK IBS coefficient matrix for paris of part...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>txt</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_1_raw_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BED file for the Raw genotype data for chr1</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_1_raw_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK FAM file accompanying Raw .bed for chr1</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_1_raw_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BIM file for the Raw genotype data for chr1</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_2_raw_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BED file for the Raw genotype data for chr2</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_2_raw_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK FAM file accompanying Raw .bed for chr2</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_2_raw_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BIM file for the Raw genotype data for chr2</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_3_raw_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BED file for the Raw genotype data for chr3</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_3_raw_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK FAM file accompanying Raw .bed for chr3</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_3_raw_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BIM file for the Raw genotype data for chr3</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_4_raw_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BED file for the Raw genotype data for chr4</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_4_raw_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK FAM file accompanying Raw .bed for chr4</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_4_raw_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BIM file for the Raw genotype data for chr4</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_5_raw_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BED file for the Raw genotype data for chr5</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_5_raw_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK FAM file accompanying Raw .bed for chr5</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_5_raw_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BIM file for the Raw genotype data for chr5</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_6_raw_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BED file for the Raw genotype data for chr6</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_6_raw_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK FAM file accompanying Raw .bed for chr6</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_6_raw_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BIM file for the Raw genotype data for chr6</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_7_raw_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BED file for the Raw genotype data for chr7</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_7_raw_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK FAM file accompanying Raw .bed for chr7</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_7_raw_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BIM file for the Raw genotype data for chr7</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_8_raw_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BED file for the Raw genotype data for chr8</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_8_raw_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK FAM file accompanying Raw .bed for chr8</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_8_raw_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BIM file for the Raw genotype data for chr8</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_9_raw_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BED file for the Raw genotype data for chr9</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_9_raw_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK FAM file accompanying Raw .bed for chr9</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_9_raw_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BIM file for the Raw genotype data for chr9</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_10_raw_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BED file for the Raw genotype data for c...</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_10_raw_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK FAM file accompanying Raw .bed for chr10</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_10_raw_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BIM file for the Raw genotype data for c...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_11_raw_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BED file for the Raw genotype data for c...</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_11_raw_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK FAM file accompanying Raw .bed for chr11</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_11_raw_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BIM file for the Raw genotype data for c...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_12_raw_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BED file for the Raw genotype data for c...</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_12_raw_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK FAM file accompanying Raw .bed for chr12</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_12_raw_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BIM file for the Raw genotype data for c...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_13_raw_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BED file for the Raw genotype data for c...</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_13_raw_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK FAM file accompanying Raw .bed for chr13</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_13_raw_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BIM file for the Raw genotype data for c...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_14_raw_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BED file for the Raw genotype data for c...</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_14_raw_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK FAM file accompanying Raw .bed for chr14</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_14_raw_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BIM file for the Raw genotype data for c...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_15_raw_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BED file for the Raw genotype data for c...</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_15_raw_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK FAM file accompanying Raw .bed for chr15</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_15_raw_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BIM file for the Raw genotype data for c...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_16_raw_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BED file for the Raw genotype data for c...</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_16_raw_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK FAM file accompanying Raw .bed for chr16</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_16_raw_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BIM file for the Raw genotype data for c...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_17_raw_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BED file for the Raw genotype data for c...</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_17_raw_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK FAM file accompanying Raw .bed for chr17</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_17_raw_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BIM file for the Raw genotype data for c...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_18_raw_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BED file for the Raw genotype data for c...</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_18_raw_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK FAM file accompanying Raw .bed for chr18</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_18_raw_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BIM file for the Raw genotype data for c...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_19_raw_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BED file for the Raw genotype data for c...</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_19_raw_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK FAM file accompanying Raw .bed for chr19</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_19_raw_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BIM file for the Raw genotype data for c...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_20_raw_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BED file for the Raw genotype data for c...</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_20_raw_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK FAM file accompanying Raw .bed for chr20</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_20_raw_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BIM file for the Raw genotype data for c...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_21_raw_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BED file for the Raw genotype data for c...</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_21_raw_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK FAM file accompanying Raw .bed for chr21</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_21_raw_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BIM file for the Raw genotype data for c...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_22_raw_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BED file for the Raw genotype data for c...</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_22_raw_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK FAM file accompanying Raw .bed for chr22</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_22_raw_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>PLINK BIM file for the Raw genotype data for c...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_1_postqc_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BED file for post QC genotype data for chr1</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_1_postqc_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK FAM file accompanying QC .bed for chr1</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_1_postqc_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BIM file for the post QC genotype data f...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_2_postqc_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BED file for post QC genotype data for chr2</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_2_postqc_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK FAM file accompanying QC .bed for chr2</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_2_postqc_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BIM file for the post QC genotype data f...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_3_postqc_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BED file for post QC genotype data for chr3</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_3_postqc_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK FAM file accompanying QC .bed for chr3</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_3_postqc_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BIM file for the post QC genotype data f...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_4_postqc_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BED file for post QC genotype data for chr4</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_4_postqc_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK FAM file accompanying QC .bed for chr4</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_4_postqc_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BIM file for the post QC genotype data f...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_5_postqc_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BED file for post QC genotype data for chr5</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_5_postqc_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK FAM file accompanying QC .bed for chr5</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_5_postqc_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BIM file for the post QC genotype data f...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_6_postqc_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BED file for post QC genotype data for chr6</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_6_postqc_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK FAM file accompanying QC .bed for chr6</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_6_postqc_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BIM file for the post QC genotype data f...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_7_postqc_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BED file for post QC genotype data for chr7</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_7_postqc_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK FAM file accompanying QC .bed for chr7</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_7_postqc_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BIM file for the post QC genotype data f...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_8_postqc_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BED file for post QC genotype data for chr8</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_8_postqc_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK FAM file accompanying QC .bed for chr8</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_8_postqc_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BIM file for the post QC genotype data f...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_9_postqc_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BED file for post QC genotype data for chr9</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_9_postqc_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK FAM file accompanying QC .bed for chr9</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_9_postqc_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BIM file for the post QC genotype data f...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_10_postqc_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BED file for post QC genotype data for c...</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_10_postqc_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK FAM file accompanying QC .bed for chr10</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_10_postqc_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BIM file for the post QC genotype data f...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_11_postqc_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BED file for post QC genotype data for c...</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_11_postqc_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK FAM file accompanying QC .bed for chr11</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_11_postqc_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BIM file for the post QC genotype data f...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_12_postqc_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BED file for post QC genotype data for c...</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_12_postqc_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK FAM file accompanying QC .bed for chr12</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_12_postqc_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BIM file for the post QC genotype data f...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_13_postqc_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BED file for post QC genotype data for c...</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_13_postqc_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK FAM file accompanying QC .bed for chr13</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_13_postqc_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BIM file for the post QC genotype data f...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_14_postqc_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BED file for post QC genotype data for c...</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_14_postqc_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK FAM file accompanying QC .bed for chr14</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_14_postqc_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BIM file for the post QC genotype data f...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_15_postqc_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BED file for post QC genotype data for c...</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_15_postqc_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK FAM file accompanying QC .bed for chr15</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_15_postqc_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BIM file for the post QC genotype data f...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_16_postqc_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BED file for post QC genotype data for c...</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_16_postqc_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK FAM file accompanying QC .bed for chr16</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_16_postqc_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BIM file for the post QC genotype data f...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_17_postqc_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BED file for post QC genotype data for c...</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_17_postqc_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK FAM file accompanying QC .bed for chr17</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_17_postqc_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BIM file for the post QC genotype data f...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_18_postqc_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BED file for post QC genotype data for c...</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_18_postqc_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK FAM file accompanying QC .bed for chr18</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_18_postqc_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BIM file for the post QC genotype data f...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_19_postqc_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BED file for post QC genotype data for c...</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_19_postqc_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK FAM file accompanying QC .bed for chr19</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_19_postqc_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BIM file for the post QC genotype data f...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_20_postqc_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BED file for post QC genotype data for c...</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_20_postqc_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK FAM file accompanying QC .bed for chr20</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_20_postqc_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BIM file for the post QC genotype data f...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_21_postqc_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BED file for post QC genotype data for c...</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_21_postqc_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK FAM file accompanying QC .bed for chr21</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_21_postqc_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BIM file for the post QC genotype data f...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_22_postqc_bed</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BED file for post QC genotype data for c...</td>\n",
       "      <td>Raw genotype calls (per chromosome) in binary ...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bed</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_22_postqc_fam</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK FAM file accompanying QC .bed for chr22</td>\n",
       "      <td>PLINK sample information file accompanying a ....</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fam</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr_22_postqc_bim</th>\n",
       "      <td>14</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>PLINK BIM file for the post QC genotype data f...</td>\n",
       "      <td>PLINK variant information file (per chromosome...</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bim</td>\n",
       "      <td>human_genetics/human_genetics.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc1</th>\n",
       "      <td>14</td>\n",
       "      <td>pca</td>\n",
       "      <td>Principal component 1</td>\n",
       "      <td>Score for projected principal component 1</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/pca/pca.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>float</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc2</th>\n",
       "      <td>14</td>\n",
       "      <td>pca</td>\n",
       "      <td>Principal component 2</td>\n",
       "      <td>Score for projected principal component 2</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/pca/pca.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>float</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc3</th>\n",
       "      <td>14</td>\n",
       "      <td>pca</td>\n",
       "      <td>Principal component 3</td>\n",
       "      <td>Score for projected principal component 3</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/pca/pca.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>float</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc4</th>\n",
       "      <td>14</td>\n",
       "      <td>pca</td>\n",
       "      <td>Principal component 4</td>\n",
       "      <td>Score for projected principal component 4</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/pca/pca.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>float</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc5</th>\n",
       "      <td>14</td>\n",
       "      <td>pca</td>\n",
       "      <td>Principal component 5</td>\n",
       "      <td>Score for projected principal component 5</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/pca/pca.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>float</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc6</th>\n",
       "      <td>14</td>\n",
       "      <td>pca</td>\n",
       "      <td>Principal component 6</td>\n",
       "      <td>Score for projected principal component 6</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/pca/pca.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>float</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc7</th>\n",
       "      <td>14</td>\n",
       "      <td>pca</td>\n",
       "      <td>Principal component 7</td>\n",
       "      <td>Score for projected principal component 7</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/pca/pca.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>float</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc8</th>\n",
       "      <td>14</td>\n",
       "      <td>pca</td>\n",
       "      <td>Principal component 8</td>\n",
       "      <td>Score for projected principal component 8</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/pca/pca.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>float</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc9</th>\n",
       "      <td>14</td>\n",
       "      <td>pca</td>\n",
       "      <td>Principal component 9</td>\n",
       "      <td>Score for projected principal component 9</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/pca/pca.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>float</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc10</th>\n",
       "      <td>14</td>\n",
       "      <td>pca</td>\n",
       "      <td>Principal component 10</td>\n",
       "      <td>Score for projected principal component 10</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/pca/pca.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>float</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc11</th>\n",
       "      <td>14</td>\n",
       "      <td>pca</td>\n",
       "      <td>Principal component 11</td>\n",
       "      <td>Score for projected principal component 11</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/pca/pca.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>float</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc12</th>\n",
       "      <td>14</td>\n",
       "      <td>pca</td>\n",
       "      <td>Principal component 12</td>\n",
       "      <td>Score for projected principal component 12</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/pca/pca.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>float</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc13</th>\n",
       "      <td>14</td>\n",
       "      <td>pca</td>\n",
       "      <td>Principal component 13</td>\n",
       "      <td>Score for projected principal component 13</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/pca/pca.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>float</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc14</th>\n",
       "      <td>14</td>\n",
       "      <td>pca</td>\n",
       "      <td>Principal component 14</td>\n",
       "      <td>Score for projected principal component 14</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/pca/pca.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>float</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc15</th>\n",
       "      <td>14</td>\n",
       "      <td>pca</td>\n",
       "      <td>Principal component 15</td>\n",
       "      <td>Score for projected principal component 15</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/pca/pca.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>float</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc16</th>\n",
       "      <td>14</td>\n",
       "      <td>pca</td>\n",
       "      <td>Principal component 16</td>\n",
       "      <td>Score for projected principal component 16</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/pca/pca.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>float</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc17</th>\n",
       "      <td>14</td>\n",
       "      <td>pca</td>\n",
       "      <td>Principal component 17</td>\n",
       "      <td>Score for projected principal component 17</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/pca/pca.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>float</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc18</th>\n",
       "      <td>14</td>\n",
       "      <td>pca</td>\n",
       "      <td>Principal component 18</td>\n",
       "      <td>Score for projected principal component 18</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/pca/pca.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>float</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc19</th>\n",
       "      <td>14</td>\n",
       "      <td>pca</td>\n",
       "      <td>Principal component 19</td>\n",
       "      <td>Score for projected principal component 19</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/pca/pca.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>float</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pc20</th>\n",
       "      <td>14</td>\n",
       "      <td>pca</td>\n",
       "      <td>Principal component 20</td>\n",
       "      <td>Score for projected principal component 20</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/pca/pca.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>float</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>used_in_pca_calculation</th>\n",
       "      <td>14</td>\n",
       "      <td>pca</td>\n",
       "      <td>Included in principal component analysis</td>\n",
       "      <td>Indicates samples was in the input for princip...</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human_genetics/pca/pca.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>bool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pca_loadings_tsv</th>\n",
       "      <td>14</td>\n",
       "      <td>pca</td>\n",
       "      <td>Principal component loading file</td>\n",
       "      <td>Principal component loading file</td>\n",
       "      <td>Bulk file (group)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tsv</td>\n",
       "      <td>human_genetics/pca/pca.parquet</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Accruing</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Primary</td>\n",
       "      <td>Both sexes</td>\n",
       "      <td>Single</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Genetics</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          folder_id       feature_set  \\\n",
       "tabular_field_name                                                      \n",
       "collection_date                                  14    human_genetics   \n",
       "version                                          14    human_genetics   \n",
       "gencove_fastq_r1                                 14         fastq_bam   \n",
       "gencove_fastq_r2                                 14         fastq_bam   \n",
       "gencove_bam                                      14         fastq_bam   \n",
       "gencove_bai                                      14         fastq_bam   \n",
       "gencove_vcf                                      14         fastq_bam   \n",
       "genecov_qc_bases                                 14    human_genetics   \n",
       "genecov_qc_bases_dedup                           14    human_genetics   \n",
       "gencove_qc_bases_dedup_mapped                    14    human_genetics   \n",
       "genecov_qc_effective_coverage                    14    human_genetics   \n",
       "genecov_qc_fraction_contamination                14    human_genetics   \n",
       "genecov_qc_snps                                  14    human_genetics   \n",
       "genecov_qc_format_passed                         14    human_genetics   \n",
       "genecov_qc_r1_eq_r2_passed                       14    human_genetics   \n",
       "genecov_qc_r1_r2_ids_match_passed                14    human_genetics   \n",
       "genecov_qc_bases_passed                          14    human_genetics   \n",
       "genecov_qc_bases_dedup_passed                    14    human_genetics   \n",
       "genecov_qc_bases_dedup_mapped_passed             14    human_genetics   \n",
       "genecov_qc_effective_coverage_passed             14    human_genetics   \n",
       "genecov_qc_fraction_contamination_passed         14    human_genetics   \n",
       "genecov_qc_snps_passed                           14    human_genetics   \n",
       "heterozygosity_proportion                        14    human_genetics   \n",
       "pass_het_check                                   14    human_genetics   \n",
       "sample_qc_gp_pass_rate                           14    human_genetics   \n",
       "sample_qc_gp_pass_check                          14    human_genetics   \n",
       "genetic_sex                                      14    human_genetics   \n",
       "chr_x_proportion                                 14    human_genetics   \n",
       "chr_y_proportion                                 14    human_genetics   \n",
       "sample_qc_gender_match_check                     14    human_genetics   \n",
       "variants_qc_parquet                              14       variants_qc   \n",
       "relatives                                        14      relationship   \n",
       "relationship_king_tsv                            14      relationship   \n",
       "relationship_ibs_txt                             14      relationship   \n",
       "chr_1_raw_bed                                    14     genotypes_raw   \n",
       "chr_1_raw_fam                                    14     genotypes_raw   \n",
       "chr_1_raw_bim                                    14     genotypes_raw   \n",
       "chr_2_raw_bed                                    14     genotypes_raw   \n",
       "chr_2_raw_fam                                    14     genotypes_raw   \n",
       "chr_2_raw_bim                                    14     genotypes_raw   \n",
       "chr_3_raw_bed                                    14     genotypes_raw   \n",
       "chr_3_raw_fam                                    14     genotypes_raw   \n",
       "chr_3_raw_bim                                    14     genotypes_raw   \n",
       "chr_4_raw_bed                                    14     genotypes_raw   \n",
       "chr_4_raw_fam                                    14     genotypes_raw   \n",
       "chr_4_raw_bim                                    14     genotypes_raw   \n",
       "chr_5_raw_bed                                    14     genotypes_raw   \n",
       "chr_5_raw_fam                                    14     genotypes_raw   \n",
       "chr_5_raw_bim                                    14     genotypes_raw   \n",
       "chr_6_raw_bed                                    14     genotypes_raw   \n",
       "chr_6_raw_fam                                    14     genotypes_raw   \n",
       "chr_6_raw_bim                                    14     genotypes_raw   \n",
       "chr_7_raw_bed                                    14     genotypes_raw   \n",
       "chr_7_raw_fam                                    14     genotypes_raw   \n",
       "chr_7_raw_bim                                    14     genotypes_raw   \n",
       "chr_8_raw_bed                                    14     genotypes_raw   \n",
       "chr_8_raw_fam                                    14     genotypes_raw   \n",
       "chr_8_raw_bim                                    14     genotypes_raw   \n",
       "chr_9_raw_bed                                    14     genotypes_raw   \n",
       "chr_9_raw_fam                                    14     genotypes_raw   \n",
       "chr_9_raw_bim                                    14     genotypes_raw   \n",
       "chr_10_raw_bed                                   14     genotypes_raw   \n",
       "chr_10_raw_fam                                   14     genotypes_raw   \n",
       "chr_10_raw_bim                                   14     genotypes_raw   \n",
       "chr_11_raw_bed                                   14     genotypes_raw   \n",
       "chr_11_raw_fam                                   14     genotypes_raw   \n",
       "chr_11_raw_bim                                   14     genotypes_raw   \n",
       "chr_12_raw_bed                                   14     genotypes_raw   \n",
       "chr_12_raw_fam                                   14     genotypes_raw   \n",
       "chr_12_raw_bim                                   14     genotypes_raw   \n",
       "chr_13_raw_bed                                   14     genotypes_raw   \n",
       "chr_13_raw_fam                                   14     genotypes_raw   \n",
       "chr_13_raw_bim                                   14     genotypes_raw   \n",
       "chr_14_raw_bed                                   14     genotypes_raw   \n",
       "chr_14_raw_fam                                   14     genotypes_raw   \n",
       "chr_14_raw_bim                                   14     genotypes_raw   \n",
       "chr_15_raw_bed                                   14     genotypes_raw   \n",
       "chr_15_raw_fam                                   14     genotypes_raw   \n",
       "chr_15_raw_bim                                   14     genotypes_raw   \n",
       "chr_16_raw_bed                                   14     genotypes_raw   \n",
       "chr_16_raw_fam                                   14     genotypes_raw   \n",
       "chr_16_raw_bim                                   14     genotypes_raw   \n",
       "chr_17_raw_bed                                   14     genotypes_raw   \n",
       "chr_17_raw_fam                                   14     genotypes_raw   \n",
       "chr_17_raw_bim                                   14     genotypes_raw   \n",
       "chr_18_raw_bed                                   14     genotypes_raw   \n",
       "chr_18_raw_fam                                   14     genotypes_raw   \n",
       "chr_18_raw_bim                                   14     genotypes_raw   \n",
       "chr_19_raw_bed                                   14     genotypes_raw   \n",
       "chr_19_raw_fam                                   14     genotypes_raw   \n",
       "chr_19_raw_bim                                   14     genotypes_raw   \n",
       "chr_20_raw_bed                                   14     genotypes_raw   \n",
       "chr_20_raw_fam                                   14     genotypes_raw   \n",
       "chr_20_raw_bim                                   14     genotypes_raw   \n",
       "chr_21_raw_bed                                   14     genotypes_raw   \n",
       "chr_21_raw_fam                                   14     genotypes_raw   \n",
       "chr_21_raw_bim                                   14     genotypes_raw   \n",
       "chr_22_raw_bed                                   14     genotypes_raw   \n",
       "chr_22_raw_fam                                   14     genotypes_raw   \n",
       "chr_22_raw_bim                                   14     genotypes_raw   \n",
       "chr_1_postqc_bed                                 14  genotypes_postqc   \n",
       "chr_1_postqc_fam                                 14  genotypes_postqc   \n",
       "chr_1_postqc_bim                                 14  genotypes_postqc   \n",
       "chr_2_postqc_bed                                 14  genotypes_postqc   \n",
       "chr_2_postqc_fam                                 14  genotypes_postqc   \n",
       "chr_2_postqc_bim                                 14  genotypes_postqc   \n",
       "chr_3_postqc_bed                                 14  genotypes_postqc   \n",
       "chr_3_postqc_fam                                 14  genotypes_postqc   \n",
       "chr_3_postqc_bim                                 14  genotypes_postqc   \n",
       "chr_4_postqc_bed                                 14  genotypes_postqc   \n",
       "chr_4_postqc_fam                                 14  genotypes_postqc   \n",
       "chr_4_postqc_bim                                 14  genotypes_postqc   \n",
       "chr_5_postqc_bed                                 14  genotypes_postqc   \n",
       "chr_5_postqc_fam                                 14  genotypes_postqc   \n",
       "chr_5_postqc_bim                                 14  genotypes_postqc   \n",
       "chr_6_postqc_bed                                 14  genotypes_postqc   \n",
       "chr_6_postqc_fam                                 14  genotypes_postqc   \n",
       "chr_6_postqc_bim                                 14  genotypes_postqc   \n",
       "chr_7_postqc_bed                                 14  genotypes_postqc   \n",
       "chr_7_postqc_fam                                 14  genotypes_postqc   \n",
       "chr_7_postqc_bim                                 14  genotypes_postqc   \n",
       "chr_8_postqc_bed                                 14  genotypes_postqc   \n",
       "chr_8_postqc_fam                                 14  genotypes_postqc   \n",
       "chr_8_postqc_bim                                 14  genotypes_postqc   \n",
       "chr_9_postqc_bed                                 14  genotypes_postqc   \n",
       "chr_9_postqc_fam                                 14  genotypes_postqc   \n",
       "chr_9_postqc_bim                                 14  genotypes_postqc   \n",
       "chr_10_postqc_bed                                14  genotypes_postqc   \n",
       "chr_10_postqc_fam                                14  genotypes_postqc   \n",
       "chr_10_postqc_bim                                14  genotypes_postqc   \n",
       "chr_11_postqc_bed                                14  genotypes_postqc   \n",
       "chr_11_postqc_fam                                14  genotypes_postqc   \n",
       "chr_11_postqc_bim                                14  genotypes_postqc   \n",
       "chr_12_postqc_bed                                14  genotypes_postqc   \n",
       "chr_12_postqc_fam                                14  genotypes_postqc   \n",
       "chr_12_postqc_bim                                14  genotypes_postqc   \n",
       "chr_13_postqc_bed                                14  genotypes_postqc   \n",
       "chr_13_postqc_fam                                14  genotypes_postqc   \n",
       "chr_13_postqc_bim                                14  genotypes_postqc   \n",
       "chr_14_postqc_bed                                14  genotypes_postqc   \n",
       "chr_14_postqc_fam                                14  genotypes_postqc   \n",
       "chr_14_postqc_bim                                14  genotypes_postqc   \n",
       "chr_15_postqc_bed                                14  genotypes_postqc   \n",
       "chr_15_postqc_fam                                14  genotypes_postqc   \n",
       "chr_15_postqc_bim                                14  genotypes_postqc   \n",
       "chr_16_postqc_bed                                14  genotypes_postqc   \n",
       "chr_16_postqc_fam                                14  genotypes_postqc   \n",
       "chr_16_postqc_bim                                14  genotypes_postqc   \n",
       "chr_17_postqc_bed                                14  genotypes_postqc   \n",
       "chr_17_postqc_fam                                14  genotypes_postqc   \n",
       "chr_17_postqc_bim                                14  genotypes_postqc   \n",
       "chr_18_postqc_bed                                14  genotypes_postqc   \n",
       "chr_18_postqc_fam                                14  genotypes_postqc   \n",
       "chr_18_postqc_bim                                14  genotypes_postqc   \n",
       "chr_19_postqc_bed                                14  genotypes_postqc   \n",
       "chr_19_postqc_fam                                14  genotypes_postqc   \n",
       "chr_19_postqc_bim                                14  genotypes_postqc   \n",
       "chr_20_postqc_bed                                14  genotypes_postqc   \n",
       "chr_20_postqc_fam                                14  genotypes_postqc   \n",
       "chr_20_postqc_bim                                14  genotypes_postqc   \n",
       "chr_21_postqc_bed                                14  genotypes_postqc   \n",
       "chr_21_postqc_fam                                14  genotypes_postqc   \n",
       "chr_21_postqc_bim                                14  genotypes_postqc   \n",
       "chr_22_postqc_bed                                14  genotypes_postqc   \n",
       "chr_22_postqc_fam                                14  genotypes_postqc   \n",
       "chr_22_postqc_bim                                14  genotypes_postqc   \n",
       "pc1                                              14               pca   \n",
       "pc2                                              14               pca   \n",
       "pc3                                              14               pca   \n",
       "pc4                                              14               pca   \n",
       "pc5                                              14               pca   \n",
       "pc6                                              14               pca   \n",
       "pc7                                              14               pca   \n",
       "pc8                                              14               pca   \n",
       "pc9                                              14               pca   \n",
       "pc10                                             14               pca   \n",
       "pc11                                             14               pca   \n",
       "pc12                                             14               pca   \n",
       "pc13                                             14               pca   \n",
       "pc14                                             14               pca   \n",
       "pc15                                             14               pca   \n",
       "pc16                                             14               pca   \n",
       "pc17                                             14               pca   \n",
       "pc18                                             14               pca   \n",
       "pc19                                             14               pca   \n",
       "pc20                                             14               pca   \n",
       "used_in_pca_calculation                          14               pca   \n",
       "pca_loadings_tsv                                 14               pca   \n",
       "\n",
       "                                                                               field_string  \\\n",
       "tabular_field_name                                                                            \n",
       "collection_date                                                             Collection date   \n",
       "version                                                                     Gencove version   \n",
       "gencove_fastq_r1                                                                   R1_FASTQ   \n",
       "gencove_fastq_r2                                                                   R2_FASTQ   \n",
       "gencove_bam                                                                             BAM   \n",
       "gencove_bai                                                                     BAM indices   \n",
       "gencove_vcf                                            Imputation Variant Call Format (VCF)   \n",
       "genecov_qc_bases                                                            Bases sequenced   \n",
       "genecov_qc_bases_dedup                                                   Deduplicated bases   \n",
       "gencove_qc_bases_dedup_mapped                                    Deduplicated bases aligned   \n",
       "genecov_qc_effective_coverage                                            Effective coverage   \n",
       "genecov_qc_fraction_contamination                                         DNA contamination   \n",
       "genecov_qc_snps                                                                SNPs covered   \n",
       "genecov_qc_format_passed                                       Passed FASTQ format validity   \n",
       "genecov_qc_r1_eq_r2_passed                                          Passed paired-end count   \n",
       "genecov_qc_r1_r2_ids_match_passed                                   Passed paired-end match   \n",
       "genecov_qc_bases_passed                                              Passed Bases sequenced   \n",
       "genecov_qc_bases_dedup_passed                                     Passed Deduplicated bases   \n",
       "genecov_qc_bases_dedup_mapped_passed                      Passed Deduplicated bases aligned   \n",
       "genecov_qc_effective_coverage_passed                              Passed Effective coverage   \n",
       "genecov_qc_fraction_contamination_passed                           Passed DNA contamination   \n",
       "genecov_qc_snps_passed                                                  Passed SNPs covered   \n",
       "heterozygosity_proportion                                         Heterozygosity proportion   \n",
       "pass_het_check                                             Passed Heterozygosity proportion   \n",
       "sample_qc_gp_pass_rate                                     Genotype Probabilities pass rate   \n",
       "sample_qc_gp_pass_check                                       Passed Genotype Probabilities   \n",
       "genetic_sex                                                                         DNA sex   \n",
       "chr_x_proportion                                    Proportion of alignment to chromosome X   \n",
       "chr_y_proportion                                    Proportion of alignment to chromosome Y   \n",
       "sample_qc_gender_match_check                                            Passed gender check   \n",
       "variants_qc_parquet                                                      Variants QC report   \n",
       "relatives                                                 Relatives based on IBD estimation   \n",
       "relationship_king_tsv                                                    KING kingship file   \n",
       "relationship_ibs_txt                                                         PLINK IBS file   \n",
       "chr_1_raw_bed                             PLINK BED file for the Raw genotype data for chr1   \n",
       "chr_1_raw_fam                                 PLINK FAM file accompanying Raw .bed for chr1   \n",
       "chr_1_raw_bim                             PLINK BIM file for the Raw genotype data for chr1   \n",
       "chr_2_raw_bed                             PLINK BED file for the Raw genotype data for chr2   \n",
       "chr_2_raw_fam                                 PLINK FAM file accompanying Raw .bed for chr2   \n",
       "chr_2_raw_bim                             PLINK BIM file for the Raw genotype data for chr2   \n",
       "chr_3_raw_bed                             PLINK BED file for the Raw genotype data for chr3   \n",
       "chr_3_raw_fam                                 PLINK FAM file accompanying Raw .bed for chr3   \n",
       "chr_3_raw_bim                             PLINK BIM file for the Raw genotype data for chr3   \n",
       "chr_4_raw_bed                             PLINK BED file for the Raw genotype data for chr4   \n",
       "chr_4_raw_fam                                 PLINK FAM file accompanying Raw .bed for chr4   \n",
       "chr_4_raw_bim                             PLINK BIM file for the Raw genotype data for chr4   \n",
       "chr_5_raw_bed                             PLINK BED file for the Raw genotype data for chr5   \n",
       "chr_5_raw_fam                                 PLINK FAM file accompanying Raw .bed for chr5   \n",
       "chr_5_raw_bim                             PLINK BIM file for the Raw genotype data for chr5   \n",
       "chr_6_raw_bed                             PLINK BED file for the Raw genotype data for chr6   \n",
       "chr_6_raw_fam                                 PLINK FAM file accompanying Raw .bed for chr6   \n",
       "chr_6_raw_bim                             PLINK BIM file for the Raw genotype data for chr6   \n",
       "chr_7_raw_bed                             PLINK BED file for the Raw genotype data for chr7   \n",
       "chr_7_raw_fam                                 PLINK FAM file accompanying Raw .bed for chr7   \n",
       "chr_7_raw_bim                             PLINK BIM file for the Raw genotype data for chr7   \n",
       "chr_8_raw_bed                             PLINK BED file for the Raw genotype data for chr8   \n",
       "chr_8_raw_fam                                 PLINK FAM file accompanying Raw .bed for chr8   \n",
       "chr_8_raw_bim                             PLINK BIM file for the Raw genotype data for chr8   \n",
       "chr_9_raw_bed                             PLINK BED file for the Raw genotype data for chr9   \n",
       "chr_9_raw_fam                                 PLINK FAM file accompanying Raw .bed for chr9   \n",
       "chr_9_raw_bim                             PLINK BIM file for the Raw genotype data for chr9   \n",
       "chr_10_raw_bed                            PLINK BED file for the Raw genotype data for c...   \n",
       "chr_10_raw_fam                               PLINK FAM file accompanying Raw .bed for chr10   \n",
       "chr_10_raw_bim                            PLINK BIM file for the Raw genotype data for c...   \n",
       "chr_11_raw_bed                            PLINK BED file for the Raw genotype data for c...   \n",
       "chr_11_raw_fam                               PLINK FAM file accompanying Raw .bed for chr11   \n",
       "chr_11_raw_bim                            PLINK BIM file for the Raw genotype data for c...   \n",
       "chr_12_raw_bed                            PLINK BED file for the Raw genotype data for c...   \n",
       "chr_12_raw_fam                               PLINK FAM file accompanying Raw .bed for chr12   \n",
       "chr_12_raw_bim                            PLINK BIM file for the Raw genotype data for c...   \n",
       "chr_13_raw_bed                            PLINK BED file for the Raw genotype data for c...   \n",
       "chr_13_raw_fam                               PLINK FAM file accompanying Raw .bed for chr13   \n",
       "chr_13_raw_bim                            PLINK BIM file for the Raw genotype data for c...   \n",
       "chr_14_raw_bed                            PLINK BED file for the Raw genotype data for c...   \n",
       "chr_14_raw_fam                               PLINK FAM file accompanying Raw .bed for chr14   \n",
       "chr_14_raw_bim                            PLINK BIM file for the Raw genotype data for c...   \n",
       "chr_15_raw_bed                            PLINK BED file for the Raw genotype data for c...   \n",
       "chr_15_raw_fam                               PLINK FAM file accompanying Raw .bed for chr15   \n",
       "chr_15_raw_bim                            PLINK BIM file for the Raw genotype data for c...   \n",
       "chr_16_raw_bed                            PLINK BED file for the Raw genotype data for c...   \n",
       "chr_16_raw_fam                               PLINK FAM file accompanying Raw .bed for chr16   \n",
       "chr_16_raw_bim                            PLINK BIM file for the Raw genotype data for c...   \n",
       "chr_17_raw_bed                            PLINK BED file for the Raw genotype data for c...   \n",
       "chr_17_raw_fam                               PLINK FAM file accompanying Raw .bed for chr17   \n",
       "chr_17_raw_bim                            PLINK BIM file for the Raw genotype data for c...   \n",
       "chr_18_raw_bed                            PLINK BED file for the Raw genotype data for c...   \n",
       "chr_18_raw_fam                               PLINK FAM file accompanying Raw .bed for chr18   \n",
       "chr_18_raw_bim                            PLINK BIM file for the Raw genotype data for c...   \n",
       "chr_19_raw_bed                            PLINK BED file for the Raw genotype data for c...   \n",
       "chr_19_raw_fam                               PLINK FAM file accompanying Raw .bed for chr19   \n",
       "chr_19_raw_bim                            PLINK BIM file for the Raw genotype data for c...   \n",
       "chr_20_raw_bed                            PLINK BED file for the Raw genotype data for c...   \n",
       "chr_20_raw_fam                               PLINK FAM file accompanying Raw .bed for chr20   \n",
       "chr_20_raw_bim                            PLINK BIM file for the Raw genotype data for c...   \n",
       "chr_21_raw_bed                            PLINK BED file for the Raw genotype data for c...   \n",
       "chr_21_raw_fam                               PLINK FAM file accompanying Raw .bed for chr21   \n",
       "chr_21_raw_bim                            PLINK BIM file for the Raw genotype data for c...   \n",
       "chr_22_raw_bed                            PLINK BED file for the Raw genotype data for c...   \n",
       "chr_22_raw_fam                               PLINK FAM file accompanying Raw .bed for chr22   \n",
       "chr_22_raw_bim                            PLINK BIM file for the Raw genotype data for c...   \n",
       "chr_1_postqc_bed                          PLINK BED file for post QC genotype data for chr1   \n",
       "chr_1_postqc_fam                               PLINK FAM file accompanying QC .bed for chr1   \n",
       "chr_1_postqc_bim                          PLINK BIM file for the post QC genotype data f...   \n",
       "chr_2_postqc_bed                          PLINK BED file for post QC genotype data for chr2   \n",
       "chr_2_postqc_fam                               PLINK FAM file accompanying QC .bed for chr2   \n",
       "chr_2_postqc_bim                          PLINK BIM file for the post QC genotype data f...   \n",
       "chr_3_postqc_bed                          PLINK BED file for post QC genotype data for chr3   \n",
       "chr_3_postqc_fam                               PLINK FAM file accompanying QC .bed for chr3   \n",
       "chr_3_postqc_bim                          PLINK BIM file for the post QC genotype data f...   \n",
       "chr_4_postqc_bed                          PLINK BED file for post QC genotype data for chr4   \n",
       "chr_4_postqc_fam                               PLINK FAM file accompanying QC .bed for chr4   \n",
       "chr_4_postqc_bim                          PLINK BIM file for the post QC genotype data f...   \n",
       "chr_5_postqc_bed                          PLINK BED file for post QC genotype data for chr5   \n",
       "chr_5_postqc_fam                               PLINK FAM file accompanying QC .bed for chr5   \n",
       "chr_5_postqc_bim                          PLINK BIM file for the post QC genotype data f...   \n",
       "chr_6_postqc_bed                          PLINK BED file for post QC genotype data for chr6   \n",
       "chr_6_postqc_fam                               PLINK FAM file accompanying QC .bed for chr6   \n",
       "chr_6_postqc_bim                          PLINK BIM file for the post QC genotype data f...   \n",
       "chr_7_postqc_bed                          PLINK BED file for post QC genotype data for chr7   \n",
       "chr_7_postqc_fam                               PLINK FAM file accompanying QC .bed for chr7   \n",
       "chr_7_postqc_bim                          PLINK BIM file for the post QC genotype data f...   \n",
       "chr_8_postqc_bed                          PLINK BED file for post QC genotype data for chr8   \n",
       "chr_8_postqc_fam                               PLINK FAM file accompanying QC .bed for chr8   \n",
       "chr_8_postqc_bim                          PLINK BIM file for the post QC genotype data f...   \n",
       "chr_9_postqc_bed                          PLINK BED file for post QC genotype data for chr9   \n",
       "chr_9_postqc_fam                               PLINK FAM file accompanying QC .bed for chr9   \n",
       "chr_9_postqc_bim                          PLINK BIM file for the post QC genotype data f...   \n",
       "chr_10_postqc_bed                         PLINK BED file for post QC genotype data for c...   \n",
       "chr_10_postqc_fam                             PLINK FAM file accompanying QC .bed for chr10   \n",
       "chr_10_postqc_bim                         PLINK BIM file for the post QC genotype data f...   \n",
       "chr_11_postqc_bed                         PLINK BED file for post QC genotype data for c...   \n",
       "chr_11_postqc_fam                             PLINK FAM file accompanying QC .bed for chr11   \n",
       "chr_11_postqc_bim                         PLINK BIM file for the post QC genotype data f...   \n",
       "chr_12_postqc_bed                         PLINK BED file for post QC genotype data for c...   \n",
       "chr_12_postqc_fam                             PLINK FAM file accompanying QC .bed for chr12   \n",
       "chr_12_postqc_bim                         PLINK BIM file for the post QC genotype data f...   \n",
       "chr_13_postqc_bed                         PLINK BED file for post QC genotype data for c...   \n",
       "chr_13_postqc_fam                             PLINK FAM file accompanying QC .bed for chr13   \n",
       "chr_13_postqc_bim                         PLINK BIM file for the post QC genotype data f...   \n",
       "chr_14_postqc_bed                         PLINK BED file for post QC genotype data for c...   \n",
       "chr_14_postqc_fam                             PLINK FAM file accompanying QC .bed for chr14   \n",
       "chr_14_postqc_bim                         PLINK BIM file for the post QC genotype data f...   \n",
       "chr_15_postqc_bed                         PLINK BED file for post QC genotype data for c...   \n",
       "chr_15_postqc_fam                             PLINK FAM file accompanying QC .bed for chr15   \n",
       "chr_15_postqc_bim                         PLINK BIM file for the post QC genotype data f...   \n",
       "chr_16_postqc_bed                         PLINK BED file for post QC genotype data for c...   \n",
       "chr_16_postqc_fam                             PLINK FAM file accompanying QC .bed for chr16   \n",
       "chr_16_postqc_bim                         PLINK BIM file for the post QC genotype data f...   \n",
       "chr_17_postqc_bed                         PLINK BED file for post QC genotype data for c...   \n",
       "chr_17_postqc_fam                             PLINK FAM file accompanying QC .bed for chr17   \n",
       "chr_17_postqc_bim                         PLINK BIM file for the post QC genotype data f...   \n",
       "chr_18_postqc_bed                         PLINK BED file for post QC genotype data for c...   \n",
       "chr_18_postqc_fam                             PLINK FAM file accompanying QC .bed for chr18   \n",
       "chr_18_postqc_bim                         PLINK BIM file for the post QC genotype data f...   \n",
       "chr_19_postqc_bed                         PLINK BED file for post QC genotype data for c...   \n",
       "chr_19_postqc_fam                             PLINK FAM file accompanying QC .bed for chr19   \n",
       "chr_19_postqc_bim                         PLINK BIM file for the post QC genotype data f...   \n",
       "chr_20_postqc_bed                         PLINK BED file for post QC genotype data for c...   \n",
       "chr_20_postqc_fam                             PLINK FAM file accompanying QC .bed for chr20   \n",
       "chr_20_postqc_bim                         PLINK BIM file for the post QC genotype data f...   \n",
       "chr_21_postqc_bed                         PLINK BED file for post QC genotype data for c...   \n",
       "chr_21_postqc_fam                             PLINK FAM file accompanying QC .bed for chr21   \n",
       "chr_21_postqc_bim                         PLINK BIM file for the post QC genotype data f...   \n",
       "chr_22_postqc_bed                         PLINK BED file for post QC genotype data for c...   \n",
       "chr_22_postqc_fam                             PLINK FAM file accompanying QC .bed for chr22   \n",
       "chr_22_postqc_bim                         PLINK BIM file for the post QC genotype data f...   \n",
       "pc1                                                                   Principal component 1   \n",
       "pc2                                                                   Principal component 2   \n",
       "pc3                                                                   Principal component 3   \n",
       "pc4                                                                   Principal component 4   \n",
       "pc5                                                                   Principal component 5   \n",
       "pc6                                                                   Principal component 6   \n",
       "pc7                                                                   Principal component 7   \n",
       "pc8                                                                   Principal component 8   \n",
       "pc9                                                                   Principal component 9   \n",
       "pc10                                                                 Principal component 10   \n",
       "pc11                                                                 Principal component 11   \n",
       "pc12                                                                 Principal component 12   \n",
       "pc13                                                                 Principal component 13   \n",
       "pc14                                                                 Principal component 14   \n",
       "pc15                                                                 Principal component 15   \n",
       "pc16                                                                 Principal component 16   \n",
       "pc17                                                                 Principal component 17   \n",
       "pc18                                                                 Principal component 18   \n",
       "pc19                                                                 Principal component 19   \n",
       "pc20                                                                 Principal component 20   \n",
       "used_in_pca_calculation                            Included in principal component analysis   \n",
       "pca_loadings_tsv                                           Principal component loading file   \n",
       "\n",
       "                                                                         description_string  \\\n",
       "tabular_field_name                                                                            \n",
       "collection_date                           The date of downloading Gencove results from t...   \n",
       "version                                                          Gencove API version 1 or 2   \n",
       "gencove_fastq_r1                          Per sample FASTQ file, a text file that contai...   \n",
       "gencove_fastq_r2                          Per sample FASTQ file, a text file that contai...   \n",
       "gencove_bam                               Per sample Binary Alignment Map (BAM) file for...   \n",
       "gencove_bai                                                 Per sample indices for BAM file   \n",
       "gencove_vcf                               Per sample VCF, a text file storing imputed ge...   \n",
       "genecov_qc_bases                          number of total bases sequenced. from genecov ...   \n",
       "genecov_qc_bases_dedup                    number of deduplicated bases. from genecov qc ...   \n",
       "gencove_qc_bases_dedup_mapped             number of deduplicated bases that have aligned...   \n",
       "genecov_qc_effective_coverage                      effective coverage. from genecov qc file   \n",
       "genecov_qc_fraction_contamination         contamination by DNA from another sample of th...   \n",
       "genecov_qc_snps                           number of variants in reference panel that are...   \n",
       "genecov_qc_format_passed                  Indicates proper formatting of the input FASTQ...   \n",
       "genecov_qc_r1_eq_r2_passed                Indicates number of bases in R1 file equal to ...   \n",
       "genecov_qc_r1_r2_ids_match_passed         Indicates R1 read identifiers match R2 read id...   \n",
       "genecov_qc_bases_passed                   Indicates number of total bases sequenced grea...   \n",
       "genecov_qc_bases_dedup_passed             Indicates number of deduplicated bases greater...   \n",
       "genecov_qc_bases_dedup_mapped_passed      Indicates number of deduplicated bases aligned...   \n",
       "genecov_qc_effective_coverage_passed      Indicates effective coverage greater than 0. f...   \n",
       "genecov_qc_fraction_contamination_passed  Indicates contamination by DNA less than 0.06....   \n",
       "genecov_qc_snps_passed                    Indicates number of variants covered by at lea...   \n",
       "heterozygosity_proportion                 Estimated heterozygosity proportion across a s...   \n",
       "pass_het_check                            Indicates heterozygosity proportion is less th...   \n",
       "sample_qc_gp_pass_rate                    Proportion of genotypes with maximum GP greate...   \n",
       "sample_qc_gp_pass_check                   Indicates proportion of GP greater than 0.9 is...   \n",
       "genetic_sex                               Sex as determined by the percentage of read-al...   \n",
       "chr_x_proportion                                percentage of reads aligned to chromosome X   \n",
       "chr_y_proportion                                percentage of reads aligned to chromosome Y   \n",
       "sample_qc_gender_match_check              Indicates inferred_gender and submitted_gender...   \n",
       "variants_qc_parquet                                                  QC report for each SNP   \n",
       "relatives                                 List of second and first degree relatives acco...   \n",
       "relationship_king_tsv                     KING kinship coefficient matrix for pairs of p...   \n",
       "relationship_ibs_txt                      PLINK IBS coefficient matrix for paris of part...   \n",
       "chr_1_raw_bed                             Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_1_raw_fam                             PLINK sample information file accompanying a ....   \n",
       "chr_1_raw_bim                             PLINK variant information file (per chromosome...   \n",
       "chr_2_raw_bed                             Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_2_raw_fam                             PLINK sample information file accompanying a ....   \n",
       "chr_2_raw_bim                             PLINK variant information file (per chromosome...   \n",
       "chr_3_raw_bed                             Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_3_raw_fam                             PLINK sample information file accompanying a ....   \n",
       "chr_3_raw_bim                             PLINK variant information file (per chromosome...   \n",
       "chr_4_raw_bed                             Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_4_raw_fam                             PLINK sample information file accompanying a ....   \n",
       "chr_4_raw_bim                             PLINK variant information file (per chromosome...   \n",
       "chr_5_raw_bed                             Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_5_raw_fam                             PLINK sample information file accompanying a ....   \n",
       "chr_5_raw_bim                             PLINK variant information file (per chromosome...   \n",
       "chr_6_raw_bed                             Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_6_raw_fam                             PLINK sample information file accompanying a ....   \n",
       "chr_6_raw_bim                             PLINK variant information file (per chromosome...   \n",
       "chr_7_raw_bed                             Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_7_raw_fam                             PLINK sample information file accompanying a ....   \n",
       "chr_7_raw_bim                             PLINK variant information file (per chromosome...   \n",
       "chr_8_raw_bed                             Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_8_raw_fam                             PLINK sample information file accompanying a ....   \n",
       "chr_8_raw_bim                             PLINK variant information file (per chromosome...   \n",
       "chr_9_raw_bed                             Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_9_raw_fam                             PLINK sample information file accompanying a ....   \n",
       "chr_9_raw_bim                             PLINK variant information file (per chromosome...   \n",
       "chr_10_raw_bed                            Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_10_raw_fam                            PLINK sample information file accompanying a ....   \n",
       "chr_10_raw_bim                            PLINK variant information file (per chromosome...   \n",
       "chr_11_raw_bed                            Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_11_raw_fam                            PLINK sample information file accompanying a ....   \n",
       "chr_11_raw_bim                            PLINK variant information file (per chromosome...   \n",
       "chr_12_raw_bed                            Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_12_raw_fam                            PLINK sample information file accompanying a ....   \n",
       "chr_12_raw_bim                            PLINK variant information file (per chromosome...   \n",
       "chr_13_raw_bed                            Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_13_raw_fam                            PLINK sample information file accompanying a ....   \n",
       "chr_13_raw_bim                            PLINK variant information file (per chromosome...   \n",
       "chr_14_raw_bed                            Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_14_raw_fam                            PLINK sample information file accompanying a ....   \n",
       "chr_14_raw_bim                            PLINK variant information file (per chromosome...   \n",
       "chr_15_raw_bed                            Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_15_raw_fam                            PLINK sample information file accompanying a ....   \n",
       "chr_15_raw_bim                            PLINK variant information file (per chromosome...   \n",
       "chr_16_raw_bed                            Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_16_raw_fam                            PLINK sample information file accompanying a ....   \n",
       "chr_16_raw_bim                            PLINK variant information file (per chromosome...   \n",
       "chr_17_raw_bed                            Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_17_raw_fam                            PLINK sample information file accompanying a ....   \n",
       "chr_17_raw_bim                            PLINK variant information file (per chromosome...   \n",
       "chr_18_raw_bed                            Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_18_raw_fam                            PLINK sample information file accompanying a ....   \n",
       "chr_18_raw_bim                            PLINK variant information file (per chromosome...   \n",
       "chr_19_raw_bed                            Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_19_raw_fam                            PLINK sample information file accompanying a ....   \n",
       "chr_19_raw_bim                            PLINK variant information file (per chromosome...   \n",
       "chr_20_raw_bed                            Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_20_raw_fam                            PLINK sample information file accompanying a ....   \n",
       "chr_20_raw_bim                            PLINK variant information file (per chromosome...   \n",
       "chr_21_raw_bed                            Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_21_raw_fam                            PLINK sample information file accompanying a ....   \n",
       "chr_21_raw_bim                            PLINK variant information file (per chromosome...   \n",
       "chr_22_raw_bed                            Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_22_raw_fam                            PLINK sample information file accompanying a ....   \n",
       "chr_22_raw_bim                            PLINK variant information file (per chromosome...   \n",
       "chr_1_postqc_bed                          Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_1_postqc_fam                          PLINK sample information file accompanying a ....   \n",
       "chr_1_postqc_bim                          PLINK variant information file (per chromosome...   \n",
       "chr_2_postqc_bed                          Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_2_postqc_fam                          PLINK sample information file accompanying a ....   \n",
       "chr_2_postqc_bim                          PLINK variant information file (per chromosome...   \n",
       "chr_3_postqc_bed                          Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_3_postqc_fam                          PLINK sample information file accompanying a ....   \n",
       "chr_3_postqc_bim                          PLINK variant information file (per chromosome...   \n",
       "chr_4_postqc_bed                          Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_4_postqc_fam                          PLINK sample information file accompanying a ....   \n",
       "chr_4_postqc_bim                          PLINK variant information file (per chromosome...   \n",
       "chr_5_postqc_bed                          Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_5_postqc_fam                          PLINK sample information file accompanying a ....   \n",
       "chr_5_postqc_bim                          PLINK variant information file (per chromosome...   \n",
       "chr_6_postqc_bed                          Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_6_postqc_fam                          PLINK sample information file accompanying a ....   \n",
       "chr_6_postqc_bim                          PLINK variant information file (per chromosome...   \n",
       "chr_7_postqc_bed                          Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_7_postqc_fam                          PLINK sample information file accompanying a ....   \n",
       "chr_7_postqc_bim                          PLINK variant information file (per chromosome...   \n",
       "chr_8_postqc_bed                          Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_8_postqc_fam                          PLINK sample information file accompanying a ....   \n",
       "chr_8_postqc_bim                          PLINK variant information file (per chromosome...   \n",
       "chr_9_postqc_bed                          Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_9_postqc_fam                          PLINK sample information file accompanying a ....   \n",
       "chr_9_postqc_bim                          PLINK variant information file (per chromosome...   \n",
       "chr_10_postqc_bed                         Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_10_postqc_fam                         PLINK sample information file accompanying a ....   \n",
       "chr_10_postqc_bim                         PLINK variant information file (per chromosome...   \n",
       "chr_11_postqc_bed                         Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_11_postqc_fam                         PLINK sample information file accompanying a ....   \n",
       "chr_11_postqc_bim                         PLINK variant information file (per chromosome...   \n",
       "chr_12_postqc_bed                         Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_12_postqc_fam                         PLINK sample information file accompanying a ....   \n",
       "chr_12_postqc_bim                         PLINK variant information file (per chromosome...   \n",
       "chr_13_postqc_bed                         Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_13_postqc_fam                         PLINK sample information file accompanying a ....   \n",
       "chr_13_postqc_bim                         PLINK variant information file (per chromosome...   \n",
       "chr_14_postqc_bed                         Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_14_postqc_fam                         PLINK sample information file accompanying a ....   \n",
       "chr_14_postqc_bim                         PLINK variant information file (per chromosome...   \n",
       "chr_15_postqc_bed                         Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_15_postqc_fam                         PLINK sample information file accompanying a ....   \n",
       "chr_15_postqc_bim                         PLINK variant information file (per chromosome...   \n",
       "chr_16_postqc_bed                         Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_16_postqc_fam                         PLINK sample information file accompanying a ....   \n",
       "chr_16_postqc_bim                         PLINK variant information file (per chromosome...   \n",
       "chr_17_postqc_bed                         Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_17_postqc_fam                         PLINK sample information file accompanying a ....   \n",
       "chr_17_postqc_bim                         PLINK variant information file (per chromosome...   \n",
       "chr_18_postqc_bed                         Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_18_postqc_fam                         PLINK sample information file accompanying a ....   \n",
       "chr_18_postqc_bim                         PLINK variant information file (per chromosome...   \n",
       "chr_19_postqc_bed                         Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_19_postqc_fam                         PLINK sample information file accompanying a ....   \n",
       "chr_19_postqc_bim                         PLINK variant information file (per chromosome...   \n",
       "chr_20_postqc_bed                         Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_20_postqc_fam                         PLINK sample information file accompanying a ....   \n",
       "chr_20_postqc_bim                         PLINK variant information file (per chromosome...   \n",
       "chr_21_postqc_bed                         Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_21_postqc_fam                         PLINK sample information file accompanying a ....   \n",
       "chr_21_postqc_bim                         PLINK variant information file (per chromosome...   \n",
       "chr_22_postqc_bed                         Raw genotype calls (per chromosome) in binary ...   \n",
       "chr_22_postqc_fam                         PLINK sample information file accompanying a ....   \n",
       "chr_22_postqc_bim                         PLINK variant information file (per chromosome...   \n",
       "pc1                                               Score for projected principal component 1   \n",
       "pc2                                               Score for projected principal component 2   \n",
       "pc3                                               Score for projected principal component 3   \n",
       "pc4                                               Score for projected principal component 4   \n",
       "pc5                                               Score for projected principal component 5   \n",
       "pc6                                               Score for projected principal component 6   \n",
       "pc7                                               Score for projected principal component 7   \n",
       "pc8                                               Score for projected principal component 8   \n",
       "pc9                                               Score for projected principal component 9   \n",
       "pc10                                             Score for projected principal component 10   \n",
       "pc11                                             Score for projected principal component 11   \n",
       "pc12                                             Score for projected principal component 12   \n",
       "pc13                                             Score for projected principal component 13   \n",
       "pc14                                             Score for projected principal component 14   \n",
       "pc15                                             Score for projected principal component 15   \n",
       "pc16                                             Score for projected principal component 16   \n",
       "pc17                                             Score for projected principal component 17   \n",
       "pc18                                             Score for projected principal component 18   \n",
       "pc19                                             Score for projected principal component 19   \n",
       "pc20                                             Score for projected principal component 20   \n",
       "used_in_pca_calculation                   Indicates samples was in the input for princip...   \n",
       "pca_loadings_tsv                                           Principal component loading file   \n",
       "\n",
       "                                                       field_type  \\\n",
       "tabular_field_name                                                  \n",
       "collection_date                                              Date   \n",
       "version                                      Categorical (single)   \n",
       "gencove_fastq_r1                          Fastq file (individual)   \n",
       "gencove_fastq_r2                          Fastq file (individual)   \n",
       "gencove_bam                                Bulk file (individual)   \n",
       "gencove_bai                                Bulk file (individual)   \n",
       "gencove_vcf                                Bulk file (individual)   \n",
       "genecov_qc_bases                                          Integer   \n",
       "genecov_qc_bases_dedup                                    Integer   \n",
       "gencove_qc_bases_dedup_mapped                             Integer   \n",
       "genecov_qc_effective_coverage                          Continuous   \n",
       "genecov_qc_fraction_contamination                      Continuous   \n",
       "genecov_qc_snps                                           Integer   \n",
       "genecov_qc_format_passed                     Categorical (single)   \n",
       "genecov_qc_r1_eq_r2_passed                   Categorical (single)   \n",
       "genecov_qc_r1_r2_ids_match_passed            Categorical (single)   \n",
       "genecov_qc_bases_passed                      Categorical (single)   \n",
       "genecov_qc_bases_dedup_passed                Categorical (single)   \n",
       "genecov_qc_bases_dedup_mapped_passed         Categorical (single)   \n",
       "genecov_qc_effective_coverage_passed         Categorical (single)   \n",
       "genecov_qc_fraction_contamination_passed     Categorical (single)   \n",
       "genecov_qc_snps_passed                       Categorical (single)   \n",
       "heterozygosity_proportion                              Continuous   \n",
       "pass_het_check                               Categorical (single)   \n",
       "sample_qc_gp_pass_rate                                 Continuous   \n",
       "sample_qc_gp_pass_check                      Categorical (single)   \n",
       "genetic_sex                                  Categorical (single)   \n",
       "chr_x_proportion                                       Continuous   \n",
       "chr_y_proportion                                       Continuous   \n",
       "sample_qc_gender_match_check                 Categorical (single)   \n",
       "variants_qc_parquet                             Bulk file (group)   \n",
       "relatives                                                    Text   \n",
       "relationship_king_tsv                           Bulk file (group)   \n",
       "relationship_ibs_txt                            Bulk file (group)   \n",
       "chr_1_raw_bed                                   Bulk file (group)   \n",
       "chr_1_raw_fam                                   Bulk file (group)   \n",
       "chr_1_raw_bim                                   Bulk file (group)   \n",
       "chr_2_raw_bed                                   Bulk file (group)   \n",
       "chr_2_raw_fam                                   Bulk file (group)   \n",
       "chr_2_raw_bim                                   Bulk file (group)   \n",
       "chr_3_raw_bed                                   Bulk file (group)   \n",
       "chr_3_raw_fam                                   Bulk file (group)   \n",
       "chr_3_raw_bim                                   Bulk file (group)   \n",
       "chr_4_raw_bed                                   Bulk file (group)   \n",
       "chr_4_raw_fam                                   Bulk file (group)   \n",
       "chr_4_raw_bim                                   Bulk file (group)   \n",
       "chr_5_raw_bed                                   Bulk file (group)   \n",
       "chr_5_raw_fam                                   Bulk file (group)   \n",
       "chr_5_raw_bim                                   Bulk file (group)   \n",
       "chr_6_raw_bed                                   Bulk file (group)   \n",
       "chr_6_raw_fam                                   Bulk file (group)   \n",
       "chr_6_raw_bim                                   Bulk file (group)   \n",
       "chr_7_raw_bed                                   Bulk file (group)   \n",
       "chr_7_raw_fam                                   Bulk file (group)   \n",
       "chr_7_raw_bim                                   Bulk file (group)   \n",
       "chr_8_raw_bed                                   Bulk file (group)   \n",
       "chr_8_raw_fam                                   Bulk file (group)   \n",
       "chr_8_raw_bim                                   Bulk file (group)   \n",
       "chr_9_raw_bed                                   Bulk file (group)   \n",
       "chr_9_raw_fam                                   Bulk file (group)   \n",
       "chr_9_raw_bim                                   Bulk file (group)   \n",
       "chr_10_raw_bed                                  Bulk file (group)   \n",
       "chr_10_raw_fam                                  Bulk file (group)   \n",
       "chr_10_raw_bim                                  Bulk file (group)   \n",
       "chr_11_raw_bed                                  Bulk file (group)   \n",
       "chr_11_raw_fam                                  Bulk file (group)   \n",
       "chr_11_raw_bim                                  Bulk file (group)   \n",
       "chr_12_raw_bed                                  Bulk file (group)   \n",
       "chr_12_raw_fam                                  Bulk file (group)   \n",
       "chr_12_raw_bim                                  Bulk file (group)   \n",
       "chr_13_raw_bed                                  Bulk file (group)   \n",
       "chr_13_raw_fam                                  Bulk file (group)   \n",
       "chr_13_raw_bim                                  Bulk file (group)   \n",
       "chr_14_raw_bed                                  Bulk file (group)   \n",
       "chr_14_raw_fam                                  Bulk file (group)   \n",
       "chr_14_raw_bim                                  Bulk file (group)   \n",
       "chr_15_raw_bed                                  Bulk file (group)   \n",
       "chr_15_raw_fam                                  Bulk file (group)   \n",
       "chr_15_raw_bim                                  Bulk file (group)   \n",
       "chr_16_raw_bed                                  Bulk file (group)   \n",
       "chr_16_raw_fam                                  Bulk file (group)   \n",
       "chr_16_raw_bim                                  Bulk file (group)   \n",
       "chr_17_raw_bed                                  Bulk file (group)   \n",
       "chr_17_raw_fam                                  Bulk file (group)   \n",
       "chr_17_raw_bim                                  Bulk file (group)   \n",
       "chr_18_raw_bed                                  Bulk file (group)   \n",
       "chr_18_raw_fam                                  Bulk file (group)   \n",
       "chr_18_raw_bim                                  Bulk file (group)   \n",
       "chr_19_raw_bed                                  Bulk file (group)   \n",
       "chr_19_raw_fam                                  Bulk file (group)   \n",
       "chr_19_raw_bim                                  Bulk file (group)   \n",
       "chr_20_raw_bed                                  Bulk file (group)   \n",
       "chr_20_raw_fam                                  Bulk file (group)   \n",
       "chr_20_raw_bim                                  Bulk file (group)   \n",
       "chr_21_raw_bed                                  Bulk file (group)   \n",
       "chr_21_raw_fam                                  Bulk file (group)   \n",
       "chr_21_raw_bim                                  Bulk file (group)   \n",
       "chr_22_raw_bed                                  Bulk file (group)   \n",
       "chr_22_raw_fam                                  Bulk file (group)   \n",
       "chr_22_raw_bim                                  Bulk file (group)   \n",
       "chr_1_postqc_bed                                Bulk file (group)   \n",
       "chr_1_postqc_fam                                Bulk file (group)   \n",
       "chr_1_postqc_bim                                Bulk file (group)   \n",
       "chr_2_postqc_bed                                Bulk file (group)   \n",
       "chr_2_postqc_fam                                Bulk file (group)   \n",
       "chr_2_postqc_bim                                Bulk file (group)   \n",
       "chr_3_postqc_bed                                Bulk file (group)   \n",
       "chr_3_postqc_fam                                Bulk file (group)   \n",
       "chr_3_postqc_bim                                Bulk file (group)   \n",
       "chr_4_postqc_bed                                Bulk file (group)   \n",
       "chr_4_postqc_fam                                Bulk file (group)   \n",
       "chr_4_postqc_bim                                Bulk file (group)   \n",
       "chr_5_postqc_bed                                Bulk file (group)   \n",
       "chr_5_postqc_fam                                Bulk file (group)   \n",
       "chr_5_postqc_bim                                Bulk file (group)   \n",
       "chr_6_postqc_bed                                Bulk file (group)   \n",
       "chr_6_postqc_fam                                Bulk file (group)   \n",
       "chr_6_postqc_bim                                Bulk file (group)   \n",
       "chr_7_postqc_bed                                Bulk file (group)   \n",
       "chr_7_postqc_fam                                Bulk file (group)   \n",
       "chr_7_postqc_bim                                Bulk file (group)   \n",
       "chr_8_postqc_bed                                Bulk file (group)   \n",
       "chr_8_postqc_fam                                Bulk file (group)   \n",
       "chr_8_postqc_bim                                Bulk file (group)   \n",
       "chr_9_postqc_bed                                Bulk file (group)   \n",
       "chr_9_postqc_fam                                Bulk file (group)   \n",
       "chr_9_postqc_bim                                Bulk file (group)   \n",
       "chr_10_postqc_bed                               Bulk file (group)   \n",
       "chr_10_postqc_fam                               Bulk file (group)   \n",
       "chr_10_postqc_bim                               Bulk file (group)   \n",
       "chr_11_postqc_bed                               Bulk file (group)   \n",
       "chr_11_postqc_fam                               Bulk file (group)   \n",
       "chr_11_postqc_bim                               Bulk file (group)   \n",
       "chr_12_postqc_bed                               Bulk file (group)   \n",
       "chr_12_postqc_fam                               Bulk file (group)   \n",
       "chr_12_postqc_bim                               Bulk file (group)   \n",
       "chr_13_postqc_bed                               Bulk file (group)   \n",
       "chr_13_postqc_fam                               Bulk file (group)   \n",
       "chr_13_postqc_bim                               Bulk file (group)   \n",
       "chr_14_postqc_bed                               Bulk file (group)   \n",
       "chr_14_postqc_fam                               Bulk file (group)   \n",
       "chr_14_postqc_bim                               Bulk file (group)   \n",
       "chr_15_postqc_bed                               Bulk file (group)   \n",
       "chr_15_postqc_fam                               Bulk file (group)   \n",
       "chr_15_postqc_bim                               Bulk file (group)   \n",
       "chr_16_postqc_bed                               Bulk file (group)   \n",
       "chr_16_postqc_fam                               Bulk file (group)   \n",
       "chr_16_postqc_bim                               Bulk file (group)   \n",
       "chr_17_postqc_bed                               Bulk file (group)   \n",
       "chr_17_postqc_fam                               Bulk file (group)   \n",
       "chr_17_postqc_bim                               Bulk file (group)   \n",
       "chr_18_postqc_bed                               Bulk file (group)   \n",
       "chr_18_postqc_fam                               Bulk file (group)   \n",
       "chr_18_postqc_bim                               Bulk file (group)   \n",
       "chr_19_postqc_bed                               Bulk file (group)   \n",
       "chr_19_postqc_fam                               Bulk file (group)   \n",
       "chr_19_postqc_bim                               Bulk file (group)   \n",
       "chr_20_postqc_bed                               Bulk file (group)   \n",
       "chr_20_postqc_fam                               Bulk file (group)   \n",
       "chr_20_postqc_bim                               Bulk file (group)   \n",
       "chr_21_postqc_bed                               Bulk file (group)   \n",
       "chr_21_postqc_fam                               Bulk file (group)   \n",
       "chr_21_postqc_bim                               Bulk file (group)   \n",
       "chr_22_postqc_bed                               Bulk file (group)   \n",
       "chr_22_postqc_fam                               Bulk file (group)   \n",
       "chr_22_postqc_bim                               Bulk file (group)   \n",
       "pc1                                                    Continuous   \n",
       "pc2                                                    Continuous   \n",
       "pc3                                                    Continuous   \n",
       "pc4                                                    Continuous   \n",
       "pc5                                                    Continuous   \n",
       "pc6                                                    Continuous   \n",
       "pc7                                                    Continuous   \n",
       "pc8                                                    Continuous   \n",
       "pc9                                                    Continuous   \n",
       "pc10                                                   Continuous   \n",
       "pc11                                                   Continuous   \n",
       "pc12                                                   Continuous   \n",
       "pc13                                                   Continuous   \n",
       "pc14                                                   Continuous   \n",
       "pc15                                                   Continuous   \n",
       "pc16                                                   Continuous   \n",
       "pc17                                                   Continuous   \n",
       "pc18                                                   Continuous   \n",
       "pc19                                                   Continuous   \n",
       "pc20                                                   Continuous   \n",
       "used_in_pca_calculation                      Categorical (single)   \n",
       "pca_loadings_tsv                                Bulk file (group)   \n",
       "\n",
       "                                          bulk_dictionary bulk_file_extension  \\\n",
       "tabular_field_name                                                              \n",
       "collection_date                                       NaN                 NaN   \n",
       "version                                               NaN                 NaN   \n",
       "gencove_fastq_r1                                      NaN         R1.fastq.gz   \n",
       "gencove_fastq_r2                                      NaN         R2.fastq.gz   \n",
       "gencove_bam                                           NaN                 bam   \n",
       "gencove_bai                                           NaN             bam.bai   \n",
       "gencove_vcf                                           NaN              vcf.gz   \n",
       "genecov_qc_bases                                      NaN                 NaN   \n",
       "genecov_qc_bases_dedup                                NaN                 NaN   \n",
       "gencove_qc_bases_dedup_mapped                         NaN                 NaN   \n",
       "genecov_qc_effective_coverage                         NaN                 NaN   \n",
       "genecov_qc_fraction_contamination                     NaN                 NaN   \n",
       "genecov_qc_snps                                       NaN                 NaN   \n",
       "genecov_qc_format_passed                              NaN                 NaN   \n",
       "genecov_qc_r1_eq_r2_passed                            NaN                 NaN   \n",
       "genecov_qc_r1_r2_ids_match_passed                     NaN                 NaN   \n",
       "genecov_qc_bases_passed                               NaN                 NaN   \n",
       "genecov_qc_bases_dedup_passed                         NaN                 NaN   \n",
       "genecov_qc_bases_dedup_mapped_passed                  NaN                 NaN   \n",
       "genecov_qc_effective_coverage_passed                  NaN                 NaN   \n",
       "genecov_qc_fraction_contamination_passed              NaN                 NaN   \n",
       "genecov_qc_snps_passed                                NaN                 NaN   \n",
       "heterozygosity_proportion                             NaN                 NaN   \n",
       "pass_het_check                                        NaN                 NaN   \n",
       "sample_qc_gp_pass_rate                                NaN                 NaN   \n",
       "sample_qc_gp_pass_check                               NaN                 NaN   \n",
       "genetic_sex                                           NaN                 NaN   \n",
       "chr_x_proportion                                      NaN                 NaN   \n",
       "chr_y_proportion                                      NaN                 NaN   \n",
       "sample_qc_gender_match_check                          NaN                 NaN   \n",
       "variants_qc_parquet                                   NaN             parquet   \n",
       "relatives                                             NaN                 NaN   \n",
       "relationship_king_tsv                                 NaN                 tsv   \n",
       "relationship_ibs_txt                                  NaN                 txt   \n",
       "chr_1_raw_bed                                         NaN                 bed   \n",
       "chr_1_raw_fam                                         NaN                 fam   \n",
       "chr_1_raw_bim                                         NaN                 bim   \n",
       "chr_2_raw_bed                                         NaN                 bed   \n",
       "chr_2_raw_fam                                         NaN                 fam   \n",
       "chr_2_raw_bim                                         NaN                 bim   \n",
       "chr_3_raw_bed                                         NaN                 bed   \n",
       "chr_3_raw_fam                                         NaN                 fam   \n",
       "chr_3_raw_bim                                         NaN                 bim   \n",
       "chr_4_raw_bed                                         NaN                 bed   \n",
       "chr_4_raw_fam                                         NaN                 fam   \n",
       "chr_4_raw_bim                                         NaN                 bim   \n",
       "chr_5_raw_bed                                         NaN                 bed   \n",
       "chr_5_raw_fam                                         NaN                 fam   \n",
       "chr_5_raw_bim                                         NaN                 bim   \n",
       "chr_6_raw_bed                                         NaN                 bed   \n",
       "chr_6_raw_fam                                         NaN                 fam   \n",
       "chr_6_raw_bim                                         NaN                 bim   \n",
       "chr_7_raw_bed                                         NaN                 bed   \n",
       "chr_7_raw_fam                                         NaN                 fam   \n",
       "chr_7_raw_bim                                         NaN                 bim   \n",
       "chr_8_raw_bed                                         NaN                 bed   \n",
       "chr_8_raw_fam                                         NaN                 fam   \n",
       "chr_8_raw_bim                                         NaN                 bim   \n",
       "chr_9_raw_bed                                         NaN                 bed   \n",
       "chr_9_raw_fam                                         NaN                 fam   \n",
       "chr_9_raw_bim                                         NaN                 bim   \n",
       "chr_10_raw_bed                                        NaN                 bed   \n",
       "chr_10_raw_fam                                        NaN                 fam   \n",
       "chr_10_raw_bim                                        NaN                 bim   \n",
       "chr_11_raw_bed                                        NaN                 bed   \n",
       "chr_11_raw_fam                                        NaN                 fam   \n",
       "chr_11_raw_bim                                        NaN                 bim   \n",
       "chr_12_raw_bed                                        NaN                 bed   \n",
       "chr_12_raw_fam                                        NaN                 fam   \n",
       "chr_12_raw_bim                                        NaN                 bim   \n",
       "chr_13_raw_bed                                        NaN                 bed   \n",
       "chr_13_raw_fam                                        NaN                 fam   \n",
       "chr_13_raw_bim                                        NaN                 bim   \n",
       "chr_14_raw_bed                                        NaN                 bed   \n",
       "chr_14_raw_fam                                        NaN                 fam   \n",
       "chr_14_raw_bim                                        NaN                 bim   \n",
       "chr_15_raw_bed                                        NaN                 bed   \n",
       "chr_15_raw_fam                                        NaN                 fam   \n",
       "chr_15_raw_bim                                        NaN                 bim   \n",
       "chr_16_raw_bed                                        NaN                 bed   \n",
       "chr_16_raw_fam                                        NaN                 fam   \n",
       "chr_16_raw_bim                                        NaN                 bim   \n",
       "chr_17_raw_bed                                        NaN                 bed   \n",
       "chr_17_raw_fam                                        NaN                 fam   \n",
       "chr_17_raw_bim                                        NaN                 bim   \n",
       "chr_18_raw_bed                                        NaN                 bed   \n",
       "chr_18_raw_fam                                        NaN                 fam   \n",
       "chr_18_raw_bim                                        NaN                 bim   \n",
       "chr_19_raw_bed                                        NaN                 bed   \n",
       "chr_19_raw_fam                                        NaN                 fam   \n",
       "chr_19_raw_bim                                        NaN                 bim   \n",
       "chr_20_raw_bed                                        NaN                 bed   \n",
       "chr_20_raw_fam                                        NaN                 fam   \n",
       "chr_20_raw_bim                                        NaN                 bim   \n",
       "chr_21_raw_bed                                        NaN                 bed   \n",
       "chr_21_raw_fam                                        NaN                 fam   \n",
       "chr_21_raw_bim                                        NaN                 bim   \n",
       "chr_22_raw_bed                                        NaN                 bed   \n",
       "chr_22_raw_fam                                        NaN                 fam   \n",
       "chr_22_raw_bim                                        NaN                 bim   \n",
       "chr_1_postqc_bed                                      NaN                 bed   \n",
       "chr_1_postqc_fam                                      NaN                 fam   \n",
       "chr_1_postqc_bim                                      NaN                 bim   \n",
       "chr_2_postqc_bed                                      NaN                 bed   \n",
       "chr_2_postqc_fam                                      NaN                 fam   \n",
       "chr_2_postqc_bim                                      NaN                 bim   \n",
       "chr_3_postqc_bed                                      NaN                 bed   \n",
       "chr_3_postqc_fam                                      NaN                 fam   \n",
       "chr_3_postqc_bim                                      NaN                 bim   \n",
       "chr_4_postqc_bed                                      NaN                 bed   \n",
       "chr_4_postqc_fam                                      NaN                 fam   \n",
       "chr_4_postqc_bim                                      NaN                 bim   \n",
       "chr_5_postqc_bed                                      NaN                 bed   \n",
       "chr_5_postqc_fam                                      NaN                 fam   \n",
       "chr_5_postqc_bim                                      NaN                 bim   \n",
       "chr_6_postqc_bed                                      NaN                 bed   \n",
       "chr_6_postqc_fam                                      NaN                 fam   \n",
       "chr_6_postqc_bim                                      NaN                 bim   \n",
       "chr_7_postqc_bed                                      NaN                 bed   \n",
       "chr_7_postqc_fam                                      NaN                 fam   \n",
       "chr_7_postqc_bim                                      NaN                 bim   \n",
       "chr_8_postqc_bed                                      NaN                 bed   \n",
       "chr_8_postqc_fam                                      NaN                 fam   \n",
       "chr_8_postqc_bim                                      NaN                 bim   \n",
       "chr_9_postqc_bed                                      NaN                 bed   \n",
       "chr_9_postqc_fam                                      NaN                 fam   \n",
       "chr_9_postqc_bim                                      NaN                 bim   \n",
       "chr_10_postqc_bed                                     NaN                 bed   \n",
       "chr_10_postqc_fam                                     NaN                 fam   \n",
       "chr_10_postqc_bim                                     NaN                 bim   \n",
       "chr_11_postqc_bed                                     NaN                 bed   \n",
       "chr_11_postqc_fam                                     NaN                 fam   \n",
       "chr_11_postqc_bim                                     NaN                 bim   \n",
       "chr_12_postqc_bed                                     NaN                 bed   \n",
       "chr_12_postqc_fam                                     NaN                 fam   \n",
       "chr_12_postqc_bim                                     NaN                 bim   \n",
       "chr_13_postqc_bed                                     NaN                 bed   \n",
       "chr_13_postqc_fam                                     NaN                 fam   \n",
       "chr_13_postqc_bim                                     NaN                 bim   \n",
       "chr_14_postqc_bed                                     NaN                 bed   \n",
       "chr_14_postqc_fam                                     NaN                 fam   \n",
       "chr_14_postqc_bim                                     NaN                 bim   \n",
       "chr_15_postqc_bed                                     NaN                 bed   \n",
       "chr_15_postqc_fam                                     NaN                 fam   \n",
       "chr_15_postqc_bim                                     NaN                 bim   \n",
       "chr_16_postqc_bed                                     NaN                 bed   \n",
       "chr_16_postqc_fam                                     NaN                 fam   \n",
       "chr_16_postqc_bim                                     NaN                 bim   \n",
       "chr_17_postqc_bed                                     NaN                 bed   \n",
       "chr_17_postqc_fam                                     NaN                 fam   \n",
       "chr_17_postqc_bim                                     NaN                 bim   \n",
       "chr_18_postqc_bed                                     NaN                 bed   \n",
       "chr_18_postqc_fam                                     NaN                 fam   \n",
       "chr_18_postqc_bim                                     NaN                 bim   \n",
       "chr_19_postqc_bed                                     NaN                 bed   \n",
       "chr_19_postqc_fam                                     NaN                 fam   \n",
       "chr_19_postqc_bim                                     NaN                 bim   \n",
       "chr_20_postqc_bed                                     NaN                 bed   \n",
       "chr_20_postqc_fam                                     NaN                 fam   \n",
       "chr_20_postqc_bim                                     NaN                 bim   \n",
       "chr_21_postqc_bed                                     NaN                 bed   \n",
       "chr_21_postqc_fam                                     NaN                 fam   \n",
       "chr_21_postqc_bim                                     NaN                 bim   \n",
       "chr_22_postqc_bed                                     NaN                 bed   \n",
       "chr_22_postqc_fam                                     NaN                 fam   \n",
       "chr_22_postqc_bim                                     NaN                 bim   \n",
       "pc1                                                   NaN                 NaN   \n",
       "pc2                                                   NaN                 NaN   \n",
       "pc3                                                   NaN                 NaN   \n",
       "pc4                                                   NaN                 NaN   \n",
       "pc5                                                   NaN                 NaN   \n",
       "pc6                                                   NaN                 NaN   \n",
       "pc7                                                   NaN                 NaN   \n",
       "pc8                                                   NaN                 NaN   \n",
       "pc9                                                   NaN                 NaN   \n",
       "pc10                                                  NaN                 NaN   \n",
       "pc11                                                  NaN                 NaN   \n",
       "pc12                                                  NaN                 NaN   \n",
       "pc13                                                  NaN                 NaN   \n",
       "pc14                                                  NaN                 NaN   \n",
       "pc15                                                  NaN                 NaN   \n",
       "pc16                                                  NaN                 NaN   \n",
       "pc17                                                  NaN                 NaN   \n",
       "pc18                                                  NaN                 NaN   \n",
       "pc19                                                  NaN                 NaN   \n",
       "pc20                                                  NaN                 NaN   \n",
       "used_in_pca_calculation                               NaN                 NaN   \n",
       "pca_loadings_tsv                                      NaN                 tsv   \n",
       "\n",
       "                                                              relative_location  \\\n",
       "tabular_field_name                                                                \n",
       "collection_date                           human_genetics/human_genetics.parquet   \n",
       "version                                   human_genetics/human_genetics.parquet   \n",
       "gencove_fastq_r1                          human_genetics/human_genetics.parquet   \n",
       "gencove_fastq_r2                          human_genetics/human_genetics.parquet   \n",
       "gencove_bam                               human_genetics/human_genetics.parquet   \n",
       "gencove_bai                               human_genetics/human_genetics.parquet   \n",
       "gencove_vcf                               human_genetics/human_genetics.parquet   \n",
       "genecov_qc_bases                          human_genetics/human_genetics.parquet   \n",
       "genecov_qc_bases_dedup                    human_genetics/human_genetics.parquet   \n",
       "gencove_qc_bases_dedup_mapped             human_genetics/human_genetics.parquet   \n",
       "genecov_qc_effective_coverage             human_genetics/human_genetics.parquet   \n",
       "genecov_qc_fraction_contamination         human_genetics/human_genetics.parquet   \n",
       "genecov_qc_snps                           human_genetics/human_genetics.parquet   \n",
       "genecov_qc_format_passed                  human_genetics/human_genetics.parquet   \n",
       "genecov_qc_r1_eq_r2_passed                human_genetics/human_genetics.parquet   \n",
       "genecov_qc_r1_r2_ids_match_passed         human_genetics/human_genetics.parquet   \n",
       "genecov_qc_bases_passed                   human_genetics/human_genetics.parquet   \n",
       "genecov_qc_bases_dedup_passed             human_genetics/human_genetics.parquet   \n",
       "genecov_qc_bases_dedup_mapped_passed      human_genetics/human_genetics.parquet   \n",
       "genecov_qc_effective_coverage_passed      human_genetics/human_genetics.parquet   \n",
       "genecov_qc_fraction_contamination_passed  human_genetics/human_genetics.parquet   \n",
       "genecov_qc_snps_passed                    human_genetics/human_genetics.parquet   \n",
       "heterozygosity_proportion                 human_genetics/human_genetics.parquet   \n",
       "pass_het_check                            human_genetics/human_genetics.parquet   \n",
       "sample_qc_gp_pass_rate                    human_genetics/human_genetics.parquet   \n",
       "sample_qc_gp_pass_check                   human_genetics/human_genetics.parquet   \n",
       "genetic_sex                               human_genetics/human_genetics.parquet   \n",
       "chr_x_proportion                          human_genetics/human_genetics.parquet   \n",
       "chr_y_proportion                          human_genetics/human_genetics.parquet   \n",
       "sample_qc_gender_match_check              human_genetics/human_genetics.parquet   \n",
       "variants_qc_parquet                       human_genetics/human_genetics.parquet   \n",
       "relatives                                 human_genetics/human_genetics.parquet   \n",
       "relationship_king_tsv                     human_genetics/human_genetics.parquet   \n",
       "relationship_ibs_txt                      human_genetics/human_genetics.parquet   \n",
       "chr_1_raw_bed                             human_genetics/human_genetics.parquet   \n",
       "chr_1_raw_fam                             human_genetics/human_genetics.parquet   \n",
       "chr_1_raw_bim                             human_genetics/human_genetics.parquet   \n",
       "chr_2_raw_bed                             human_genetics/human_genetics.parquet   \n",
       "chr_2_raw_fam                             human_genetics/human_genetics.parquet   \n",
       "chr_2_raw_bim                             human_genetics/human_genetics.parquet   \n",
       "chr_3_raw_bed                             human_genetics/human_genetics.parquet   \n",
       "chr_3_raw_fam                             human_genetics/human_genetics.parquet   \n",
       "chr_3_raw_bim                             human_genetics/human_genetics.parquet   \n",
       "chr_4_raw_bed                             human_genetics/human_genetics.parquet   \n",
       "chr_4_raw_fam                             human_genetics/human_genetics.parquet   \n",
       "chr_4_raw_bim                             human_genetics/human_genetics.parquet   \n",
       "chr_5_raw_bed                             human_genetics/human_genetics.parquet   \n",
       "chr_5_raw_fam                             human_genetics/human_genetics.parquet   \n",
       "chr_5_raw_bim                             human_genetics/human_genetics.parquet   \n",
       "chr_6_raw_bed                             human_genetics/human_genetics.parquet   \n",
       "chr_6_raw_fam                             human_genetics/human_genetics.parquet   \n",
       "chr_6_raw_bim                             human_genetics/human_genetics.parquet   \n",
       "chr_7_raw_bed                             human_genetics/human_genetics.parquet   \n",
       "chr_7_raw_fam                             human_genetics/human_genetics.parquet   \n",
       "chr_7_raw_bim                             human_genetics/human_genetics.parquet   \n",
       "chr_8_raw_bed                             human_genetics/human_genetics.parquet   \n",
       "chr_8_raw_fam                             human_genetics/human_genetics.parquet   \n",
       "chr_8_raw_bim                             human_genetics/human_genetics.parquet   \n",
       "chr_9_raw_bed                             human_genetics/human_genetics.parquet   \n",
       "chr_9_raw_fam                             human_genetics/human_genetics.parquet   \n",
       "chr_9_raw_bim                             human_genetics/human_genetics.parquet   \n",
       "chr_10_raw_bed                            human_genetics/human_genetics.parquet   \n",
       "chr_10_raw_fam                            human_genetics/human_genetics.parquet   \n",
       "chr_10_raw_bim                            human_genetics/human_genetics.parquet   \n",
       "chr_11_raw_bed                            human_genetics/human_genetics.parquet   \n",
       "chr_11_raw_fam                            human_genetics/human_genetics.parquet   \n",
       "chr_11_raw_bim                            human_genetics/human_genetics.parquet   \n",
       "chr_12_raw_bed                            human_genetics/human_genetics.parquet   \n",
       "chr_12_raw_fam                            human_genetics/human_genetics.parquet   \n",
       "chr_12_raw_bim                            human_genetics/human_genetics.parquet   \n",
       "chr_13_raw_bed                            human_genetics/human_genetics.parquet   \n",
       "chr_13_raw_fam                            human_genetics/human_genetics.parquet   \n",
       "chr_13_raw_bim                            human_genetics/human_genetics.parquet   \n",
       "chr_14_raw_bed                            human_genetics/human_genetics.parquet   \n",
       "chr_14_raw_fam                            human_genetics/human_genetics.parquet   \n",
       "chr_14_raw_bim                            human_genetics/human_genetics.parquet   \n",
       "chr_15_raw_bed                            human_genetics/human_genetics.parquet   \n",
       "chr_15_raw_fam                            human_genetics/human_genetics.parquet   \n",
       "chr_15_raw_bim                            human_genetics/human_genetics.parquet   \n",
       "chr_16_raw_bed                            human_genetics/human_genetics.parquet   \n",
       "chr_16_raw_fam                            human_genetics/human_genetics.parquet   \n",
       "chr_16_raw_bim                            human_genetics/human_genetics.parquet   \n",
       "chr_17_raw_bed                            human_genetics/human_genetics.parquet   \n",
       "chr_17_raw_fam                            human_genetics/human_genetics.parquet   \n",
       "chr_17_raw_bim                            human_genetics/human_genetics.parquet   \n",
       "chr_18_raw_bed                            human_genetics/human_genetics.parquet   \n",
       "chr_18_raw_fam                            human_genetics/human_genetics.parquet   \n",
       "chr_18_raw_bim                            human_genetics/human_genetics.parquet   \n",
       "chr_19_raw_bed                            human_genetics/human_genetics.parquet   \n",
       "chr_19_raw_fam                            human_genetics/human_genetics.parquet   \n",
       "chr_19_raw_bim                            human_genetics/human_genetics.parquet   \n",
       "chr_20_raw_bed                            human_genetics/human_genetics.parquet   \n",
       "chr_20_raw_fam                            human_genetics/human_genetics.parquet   \n",
       "chr_20_raw_bim                            human_genetics/human_genetics.parquet   \n",
       "chr_21_raw_bed                            human_genetics/human_genetics.parquet   \n",
       "chr_21_raw_fam                            human_genetics/human_genetics.parquet   \n",
       "chr_21_raw_bim                            human_genetics/human_genetics.parquet   \n",
       "chr_22_raw_bed                            human_genetics/human_genetics.parquet   \n",
       "chr_22_raw_fam                            human_genetics/human_genetics.parquet   \n",
       "chr_22_raw_bim                            human_genetics/human_genetics.parquet   \n",
       "chr_1_postqc_bed                          human_genetics/human_genetics.parquet   \n",
       "chr_1_postqc_fam                          human_genetics/human_genetics.parquet   \n",
       "chr_1_postqc_bim                          human_genetics/human_genetics.parquet   \n",
       "chr_2_postqc_bed                          human_genetics/human_genetics.parquet   \n",
       "chr_2_postqc_fam                          human_genetics/human_genetics.parquet   \n",
       "chr_2_postqc_bim                          human_genetics/human_genetics.parquet   \n",
       "chr_3_postqc_bed                          human_genetics/human_genetics.parquet   \n",
       "chr_3_postqc_fam                          human_genetics/human_genetics.parquet   \n",
       "chr_3_postqc_bim                          human_genetics/human_genetics.parquet   \n",
       "chr_4_postqc_bed                          human_genetics/human_genetics.parquet   \n",
       "chr_4_postqc_fam                          human_genetics/human_genetics.parquet   \n",
       "chr_4_postqc_bim                          human_genetics/human_genetics.parquet   \n",
       "chr_5_postqc_bed                          human_genetics/human_genetics.parquet   \n",
       "chr_5_postqc_fam                          human_genetics/human_genetics.parquet   \n",
       "chr_5_postqc_bim                          human_genetics/human_genetics.parquet   \n",
       "chr_6_postqc_bed                          human_genetics/human_genetics.parquet   \n",
       "chr_6_postqc_fam                          human_genetics/human_genetics.parquet   \n",
       "chr_6_postqc_bim                          human_genetics/human_genetics.parquet   \n",
       "chr_7_postqc_bed                          human_genetics/human_genetics.parquet   \n",
       "chr_7_postqc_fam                          human_genetics/human_genetics.parquet   \n",
       "chr_7_postqc_bim                          human_genetics/human_genetics.parquet   \n",
       "chr_8_postqc_bed                          human_genetics/human_genetics.parquet   \n",
       "chr_8_postqc_fam                          human_genetics/human_genetics.parquet   \n",
       "chr_8_postqc_bim                          human_genetics/human_genetics.parquet   \n",
       "chr_9_postqc_bed                          human_genetics/human_genetics.parquet   \n",
       "chr_9_postqc_fam                          human_genetics/human_genetics.parquet   \n",
       "chr_9_postqc_bim                          human_genetics/human_genetics.parquet   \n",
       "chr_10_postqc_bed                         human_genetics/human_genetics.parquet   \n",
       "chr_10_postqc_fam                         human_genetics/human_genetics.parquet   \n",
       "chr_10_postqc_bim                         human_genetics/human_genetics.parquet   \n",
       "chr_11_postqc_bed                         human_genetics/human_genetics.parquet   \n",
       "chr_11_postqc_fam                         human_genetics/human_genetics.parquet   \n",
       "chr_11_postqc_bim                         human_genetics/human_genetics.parquet   \n",
       "chr_12_postqc_bed                         human_genetics/human_genetics.parquet   \n",
       "chr_12_postqc_fam                         human_genetics/human_genetics.parquet   \n",
       "chr_12_postqc_bim                         human_genetics/human_genetics.parquet   \n",
       "chr_13_postqc_bed                         human_genetics/human_genetics.parquet   \n",
       "chr_13_postqc_fam                         human_genetics/human_genetics.parquet   \n",
       "chr_13_postqc_bim                         human_genetics/human_genetics.parquet   \n",
       "chr_14_postqc_bed                         human_genetics/human_genetics.parquet   \n",
       "chr_14_postqc_fam                         human_genetics/human_genetics.parquet   \n",
       "chr_14_postqc_bim                         human_genetics/human_genetics.parquet   \n",
       "chr_15_postqc_bed                         human_genetics/human_genetics.parquet   \n",
       "chr_15_postqc_fam                         human_genetics/human_genetics.parquet   \n",
       "chr_15_postqc_bim                         human_genetics/human_genetics.parquet   \n",
       "chr_16_postqc_bed                         human_genetics/human_genetics.parquet   \n",
       "chr_16_postqc_fam                         human_genetics/human_genetics.parquet   \n",
       "chr_16_postqc_bim                         human_genetics/human_genetics.parquet   \n",
       "chr_17_postqc_bed                         human_genetics/human_genetics.parquet   \n",
       "chr_17_postqc_fam                         human_genetics/human_genetics.parquet   \n",
       "chr_17_postqc_bim                         human_genetics/human_genetics.parquet   \n",
       "chr_18_postqc_bed                         human_genetics/human_genetics.parquet   \n",
       "chr_18_postqc_fam                         human_genetics/human_genetics.parquet   \n",
       "chr_18_postqc_bim                         human_genetics/human_genetics.parquet   \n",
       "chr_19_postqc_bed                         human_genetics/human_genetics.parquet   \n",
       "chr_19_postqc_fam                         human_genetics/human_genetics.parquet   \n",
       "chr_19_postqc_bim                         human_genetics/human_genetics.parquet   \n",
       "chr_20_postqc_bed                         human_genetics/human_genetics.parquet   \n",
       "chr_20_postqc_fam                         human_genetics/human_genetics.parquet   \n",
       "chr_20_postqc_bim                         human_genetics/human_genetics.parquet   \n",
       "chr_21_postqc_bed                         human_genetics/human_genetics.parquet   \n",
       "chr_21_postqc_fam                         human_genetics/human_genetics.parquet   \n",
       "chr_21_postqc_bim                         human_genetics/human_genetics.parquet   \n",
       "chr_22_postqc_bed                         human_genetics/human_genetics.parquet   \n",
       "chr_22_postqc_fam                         human_genetics/human_genetics.parquet   \n",
       "chr_22_postqc_bim                         human_genetics/human_genetics.parquet   \n",
       "pc1                                              human_genetics/pca/pca.parquet   \n",
       "pc2                                              human_genetics/pca/pca.parquet   \n",
       "pc3                                              human_genetics/pca/pca.parquet   \n",
       "pc4                                              human_genetics/pca/pca.parquet   \n",
       "pc5                                              human_genetics/pca/pca.parquet   \n",
       "pc6                                              human_genetics/pca/pca.parquet   \n",
       "pc7                                              human_genetics/pca/pca.parquet   \n",
       "pc8                                              human_genetics/pca/pca.parquet   \n",
       "pc9                                              human_genetics/pca/pca.parquet   \n",
       "pc10                                             human_genetics/pca/pca.parquet   \n",
       "pc11                                             human_genetics/pca/pca.parquet   \n",
       "pc12                                             human_genetics/pca/pca.parquet   \n",
       "pc13                                             human_genetics/pca/pca.parquet   \n",
       "pc14                                             human_genetics/pca/pca.parquet   \n",
       "pc15                                             human_genetics/pca/pca.parquet   \n",
       "pc16                                             human_genetics/pca/pca.parquet   \n",
       "pc17                                             human_genetics/pca/pca.parquet   \n",
       "pc18                                             human_genetics/pca/pca.parquet   \n",
       "pc19                                             human_genetics/pca/pca.parquet   \n",
       "pc20                                             human_genetics/pca/pca.parquet   \n",
       "used_in_pca_calculation                          human_genetics/pca/pca.parquet   \n",
       "pca_loadings_tsv                                 human_genetics/pca/pca.parquet   \n",
       "\n",
       "                                          data_coding stability    units  \\\n",
       "tabular_field_name                                                         \n",
       "collection_date                                   NaN  Accruing     Time   \n",
       "version                                           NaN  Accruing      NaN   \n",
       "gencove_fastq_r1                                  NaN  Accruing     Text   \n",
       "gencove_fastq_r2                                  NaN  Accruing     Text   \n",
       "gencove_bam                                       NaN  Accruing     Text   \n",
       "gencove_bai                                       NaN  Accruing     Text   \n",
       "gencove_vcf                                       NaN  Accruing     Text   \n",
       "genecov_qc_bases                                  NaN  Accruing    Count   \n",
       "genecov_qc_bases_dedup                            NaN  Accruing    Count   \n",
       "gencove_qc_bases_dedup_mapped                     NaN  Accruing    Count   \n",
       "genecov_qc_effective_coverage                     NaN  Accruing  Precent   \n",
       "genecov_qc_fraction_contamination                 NaN  Accruing  Precent   \n",
       "genecov_qc_snps                                   NaN  Accruing    Count   \n",
       "genecov_qc_format_passed                          NaN  Accruing  Boolean   \n",
       "genecov_qc_r1_eq_r2_passed                        NaN  Accruing  Boolean   \n",
       "genecov_qc_r1_r2_ids_match_passed                 NaN  Accruing  Boolean   \n",
       "genecov_qc_bases_passed                           NaN  Accruing  Boolean   \n",
       "genecov_qc_bases_dedup_passed                     NaN  Accruing  Boolean   \n",
       "genecov_qc_bases_dedup_mapped_passed              NaN  Accruing  Boolean   \n",
       "genecov_qc_effective_coverage_passed              NaN  Accruing  Boolean   \n",
       "genecov_qc_fraction_contamination_passed          NaN  Accruing  Boolean   \n",
       "genecov_qc_snps_passed                            NaN  Accruing  Boolean   \n",
       "heterozygosity_proportion                         NaN  Accruing  Precent   \n",
       "pass_het_check                                    NaN  Accruing  Boolean   \n",
       "sample_qc_gp_pass_rate                            NaN  Accruing  Precent   \n",
       "sample_qc_gp_pass_check                           NaN  Accruing  Boolean   \n",
       "genetic_sex                                       NaN  Accruing  Boolean   \n",
       "chr_x_proportion                                  NaN  Accruing  Precent   \n",
       "chr_y_proportion                                  NaN  Accruing  Precent   \n",
       "sample_qc_gender_match_check                      NaN  Accruing  Boolean   \n",
       "variants_qc_parquet                               NaN  Accruing     Text   \n",
       "relatives                                         NaN  Accruing     Text   \n",
       "relationship_king_tsv                             NaN  Accruing     Text   \n",
       "relationship_ibs_txt                              NaN  Accruing     Text   \n",
       "chr_1_raw_bed                                     NaN  Accruing     Text   \n",
       "chr_1_raw_fam                                     NaN  Accruing     Text   \n",
       "chr_1_raw_bim                                     NaN  Accruing     Text   \n",
       "chr_2_raw_bed                                     NaN  Accruing     Text   \n",
       "chr_2_raw_fam                                     NaN  Accruing     Text   \n",
       "chr_2_raw_bim                                     NaN  Accruing     Text   \n",
       "chr_3_raw_bed                                     NaN  Accruing     Text   \n",
       "chr_3_raw_fam                                     NaN  Accruing     Text   \n",
       "chr_3_raw_bim                                     NaN  Accruing     Text   \n",
       "chr_4_raw_bed                                     NaN  Accruing     Text   \n",
       "chr_4_raw_fam                                     NaN  Accruing     Text   \n",
       "chr_4_raw_bim                                     NaN  Accruing     Text   \n",
       "chr_5_raw_bed                                     NaN  Accruing     Text   \n",
       "chr_5_raw_fam                                     NaN  Accruing     Text   \n",
       "chr_5_raw_bim                                     NaN  Accruing     Text   \n",
       "chr_6_raw_bed                                     NaN  Accruing     Text   \n",
       "chr_6_raw_fam                                     NaN  Accruing     Text   \n",
       "chr_6_raw_bim                                     NaN  Accruing     Text   \n",
       "chr_7_raw_bed                                     NaN  Accruing     Text   \n",
       "chr_7_raw_fam                                     NaN  Accruing     Text   \n",
       "chr_7_raw_bim                                     NaN  Accruing     Text   \n",
       "chr_8_raw_bed                                     NaN  Accruing     Text   \n",
       "chr_8_raw_fam                                     NaN  Accruing     Text   \n",
       "chr_8_raw_bim                                     NaN  Accruing     Text   \n",
       "chr_9_raw_bed                                     NaN  Accruing     Text   \n",
       "chr_9_raw_fam                                     NaN  Accruing     Text   \n",
       "chr_9_raw_bim                                     NaN  Accruing     Text   \n",
       "chr_10_raw_bed                                    NaN  Accruing     Text   \n",
       "chr_10_raw_fam                                    NaN  Accruing     Text   \n",
       "chr_10_raw_bim                                    NaN  Accruing     Text   \n",
       "chr_11_raw_bed                                    NaN  Accruing     Text   \n",
       "chr_11_raw_fam                                    NaN  Accruing     Text   \n",
       "chr_11_raw_bim                                    NaN  Accruing     Text   \n",
       "chr_12_raw_bed                                    NaN  Accruing     Text   \n",
       "chr_12_raw_fam                                    NaN  Accruing     Text   \n",
       "chr_12_raw_bim                                    NaN  Accruing     Text   \n",
       "chr_13_raw_bed                                    NaN  Accruing     Text   \n",
       "chr_13_raw_fam                                    NaN  Accruing     Text   \n",
       "chr_13_raw_bim                                    NaN  Accruing     Text   \n",
       "chr_14_raw_bed                                    NaN  Accruing     Text   \n",
       "chr_14_raw_fam                                    NaN  Accruing     Text   \n",
       "chr_14_raw_bim                                    NaN  Accruing     Text   \n",
       "chr_15_raw_bed                                    NaN  Accruing     Text   \n",
       "chr_15_raw_fam                                    NaN  Accruing     Text   \n",
       "chr_15_raw_bim                                    NaN  Accruing     Text   \n",
       "chr_16_raw_bed                                    NaN  Accruing     Text   \n",
       "chr_16_raw_fam                                    NaN  Accruing     Text   \n",
       "chr_16_raw_bim                                    NaN  Accruing     Text   \n",
       "chr_17_raw_bed                                    NaN  Accruing     Text   \n",
       "chr_17_raw_fam                                    NaN  Accruing     Text   \n",
       "chr_17_raw_bim                                    NaN  Accruing     Text   \n",
       "chr_18_raw_bed                                    NaN  Accruing     Text   \n",
       "chr_18_raw_fam                                    NaN  Accruing     Text   \n",
       "chr_18_raw_bim                                    NaN  Accruing     Text   \n",
       "chr_19_raw_bed                                    NaN  Accruing     Text   \n",
       "chr_19_raw_fam                                    NaN  Accruing     Text   \n",
       "chr_19_raw_bim                                    NaN  Accruing     Text   \n",
       "chr_20_raw_bed                                    NaN  Accruing     Text   \n",
       "chr_20_raw_fam                                    NaN  Accruing     Text   \n",
       "chr_20_raw_bim                                    NaN  Accruing     Text   \n",
       "chr_21_raw_bed                                    NaN  Accruing     Text   \n",
       "chr_21_raw_fam                                    NaN  Accruing     Text   \n",
       "chr_21_raw_bim                                    NaN  Accruing     Text   \n",
       "chr_22_raw_bed                                    NaN  Accruing     Text   \n",
       "chr_22_raw_fam                                    NaN  Accruing     Text   \n",
       "chr_22_raw_bim                                    NaN  Accruing     Text   \n",
       "chr_1_postqc_bed                                  NaN  Accruing     Text   \n",
       "chr_1_postqc_fam                                  NaN  Accruing     Text   \n",
       "chr_1_postqc_bim                                  NaN  Accruing     Text   \n",
       "chr_2_postqc_bed                                  NaN  Accruing     Text   \n",
       "chr_2_postqc_fam                                  NaN  Accruing     Text   \n",
       "chr_2_postqc_bim                                  NaN  Accruing     Text   \n",
       "chr_3_postqc_bed                                  NaN  Accruing     Text   \n",
       "chr_3_postqc_fam                                  NaN  Accruing     Text   \n",
       "chr_3_postqc_bim                                  NaN  Accruing     Text   \n",
       "chr_4_postqc_bed                                  NaN  Accruing     Text   \n",
       "chr_4_postqc_fam                                  NaN  Accruing     Text   \n",
       "chr_4_postqc_bim                                  NaN  Accruing     Text   \n",
       "chr_5_postqc_bed                                  NaN  Accruing     Text   \n",
       "chr_5_postqc_fam                                  NaN  Accruing     Text   \n",
       "chr_5_postqc_bim                                  NaN  Accruing     Text   \n",
       "chr_6_postqc_bed                                  NaN  Accruing     Text   \n",
       "chr_6_postqc_fam                                  NaN  Accruing     Text   \n",
       "chr_6_postqc_bim                                  NaN  Accruing     Text   \n",
       "chr_7_postqc_bed                                  NaN  Accruing     Text   \n",
       "chr_7_postqc_fam                                  NaN  Accruing     Text   \n",
       "chr_7_postqc_bim                                  NaN  Accruing     Text   \n",
       "chr_8_postqc_bed                                  NaN  Accruing     Text   \n",
       "chr_8_postqc_fam                                  NaN  Accruing     Text   \n",
       "chr_8_postqc_bim                                  NaN  Accruing     Text   \n",
       "chr_9_postqc_bed                                  NaN  Accruing     Text   \n",
       "chr_9_postqc_fam                                  NaN  Accruing     Text   \n",
       "chr_9_postqc_bim                                  NaN  Accruing     Text   \n",
       "chr_10_postqc_bed                                 NaN  Accruing     Text   \n",
       "chr_10_postqc_fam                                 NaN  Accruing     Text   \n",
       "chr_10_postqc_bim                                 NaN  Accruing     Text   \n",
       "chr_11_postqc_bed                                 NaN  Accruing     Text   \n",
       "chr_11_postqc_fam                                 NaN  Accruing     Text   \n",
       "chr_11_postqc_bim                                 NaN  Accruing     Text   \n",
       "chr_12_postqc_bed                                 NaN  Accruing     Text   \n",
       "chr_12_postqc_fam                                 NaN  Accruing     Text   \n",
       "chr_12_postqc_bim                                 NaN  Accruing     Text   \n",
       "chr_13_postqc_bed                                 NaN  Accruing     Text   \n",
       "chr_13_postqc_fam                                 NaN  Accruing     Text   \n",
       "chr_13_postqc_bim                                 NaN  Accruing     Text   \n",
       "chr_14_postqc_bed                                 NaN  Accruing     Text   \n",
       "chr_14_postqc_fam                                 NaN  Accruing     Text   \n",
       "chr_14_postqc_bim                                 NaN  Accruing     Text   \n",
       "chr_15_postqc_bed                                 NaN  Accruing     Text   \n",
       "chr_15_postqc_fam                                 NaN  Accruing     Text   \n",
       "chr_15_postqc_bim                                 NaN  Accruing     Text   \n",
       "chr_16_postqc_bed                                 NaN  Accruing     Text   \n",
       "chr_16_postqc_fam                                 NaN  Accruing     Text   \n",
       "chr_16_postqc_bim                                 NaN  Accruing     Text   \n",
       "chr_17_postqc_bed                                 NaN  Accruing     Text   \n",
       "chr_17_postqc_fam                                 NaN  Accruing     Text   \n",
       "chr_17_postqc_bim                                 NaN  Accruing     Text   \n",
       "chr_18_postqc_bed                                 NaN  Accruing     Text   \n",
       "chr_18_postqc_fam                                 NaN  Accruing     Text   \n",
       "chr_18_postqc_bim                                 NaN  Accruing     Text   \n",
       "chr_19_postqc_bed                                 NaN  Accruing     Text   \n",
       "chr_19_postqc_fam                                 NaN  Accruing     Text   \n",
       "chr_19_postqc_bim                                 NaN  Accruing     Text   \n",
       "chr_20_postqc_bed                                 NaN  Accruing     Text   \n",
       "chr_20_postqc_fam                                 NaN  Accruing     Text   \n",
       "chr_20_postqc_bim                                 NaN  Accruing     Text   \n",
       "chr_21_postqc_bed                                 NaN  Accruing     Text   \n",
       "chr_21_postqc_fam                                 NaN  Accruing     Text   \n",
       "chr_21_postqc_bim                                 NaN  Accruing     Text   \n",
       "chr_22_postqc_bed                                 NaN  Accruing     Text   \n",
       "chr_22_postqc_fam                                 NaN  Accruing     Text   \n",
       "chr_22_postqc_bim                                 NaN  Accruing     Text   \n",
       "pc1                                               NaN  Accruing      NaN   \n",
       "pc2                                               NaN  Accruing      NaN   \n",
       "pc3                                               NaN  Accruing      NaN   \n",
       "pc4                                               NaN  Accruing      NaN   \n",
       "pc5                                               NaN  Accruing      NaN   \n",
       "pc6                                               NaN  Accruing      NaN   \n",
       "pc7                                               NaN  Accruing      NaN   \n",
       "pc8                                               NaN  Accruing      NaN   \n",
       "pc9                                               NaN  Accruing      NaN   \n",
       "pc10                                              NaN  Accruing      NaN   \n",
       "pc11                                              NaN  Accruing      NaN   \n",
       "pc12                                              NaN  Accruing      NaN   \n",
       "pc13                                              NaN  Accruing      NaN   \n",
       "pc14                                              NaN  Accruing      NaN   \n",
       "pc15                                              NaN  Accruing      NaN   \n",
       "pc16                                              NaN  Accruing      NaN   \n",
       "pc17                                              NaN  Accruing      NaN   \n",
       "pc18                                              NaN  Accruing      NaN   \n",
       "pc19                                              NaN  Accruing      NaN   \n",
       "pc20                                              NaN  Accruing      NaN   \n",
       "used_in_pca_calculation                           NaN  Accruing  Boolean   \n",
       "pca_loadings_tsv                                  NaN  Accruing     Text   \n",
       "\n",
       "                                          sampling_rate      strata  \\\n",
       "tabular_field_name                                                    \n",
       "collection_date                                     NaN   Auxiliary   \n",
       "version                                             NaN   Auxiliary   \n",
       "gencove_fastq_r1                                    NaN     Primary   \n",
       "gencove_fastq_r2                                    NaN     Primary   \n",
       "gencove_bam                                         NaN     Primary   \n",
       "gencove_bai                                         NaN     Primary   \n",
       "gencove_vcf                                         NaN     Primary   \n",
       "genecov_qc_bases                                    NaN   Auxiliary   \n",
       "genecov_qc_bases_dedup                              NaN   Auxiliary   \n",
       "gencove_qc_bases_dedup_mapped                       NaN   Auxiliary   \n",
       "genecov_qc_effective_coverage                       NaN   Auxiliary   \n",
       "genecov_qc_fraction_contamination                   NaN   Auxiliary   \n",
       "genecov_qc_snps                                     NaN   Auxiliary   \n",
       "genecov_qc_format_passed                            NaN   Auxiliary   \n",
       "genecov_qc_r1_eq_r2_passed                          NaN   Auxiliary   \n",
       "genecov_qc_r1_r2_ids_match_passed                   NaN   Auxiliary   \n",
       "genecov_qc_bases_passed                             NaN   Auxiliary   \n",
       "genecov_qc_bases_dedup_passed                       NaN   Auxiliary   \n",
       "genecov_qc_bases_dedup_mapped_passed                NaN   Auxiliary   \n",
       "genecov_qc_effective_coverage_passed                NaN   Auxiliary   \n",
       "genecov_qc_fraction_contamination_passed            NaN   Auxiliary   \n",
       "genecov_qc_snps_passed                              NaN   Auxiliary   \n",
       "heterozygosity_proportion                           NaN     Primary   \n",
       "pass_het_check                                      NaN  Supporting   \n",
       "sample_qc_gp_pass_rate                              NaN  Supporting   \n",
       "sample_qc_gp_pass_check                             NaN  Supporting   \n",
       "genetic_sex                                         NaN     Primary   \n",
       "chr_x_proportion                                    NaN     Primary   \n",
       "chr_y_proportion                                    NaN     Primary   \n",
       "sample_qc_gender_match_check                        NaN  Supporting   \n",
       "variants_qc_parquet                                 NaN  Supporting   \n",
       "relatives                                           NaN     Primary   \n",
       "relationship_king_tsv                               NaN     Primary   \n",
       "relationship_ibs_txt                                NaN     Primary   \n",
       "chr_1_raw_bed                                       NaN     Primary   \n",
       "chr_1_raw_fam                                       NaN     Primary   \n",
       "chr_1_raw_bim                                       NaN     Primary   \n",
       "chr_2_raw_bed                                       NaN     Primary   \n",
       "chr_2_raw_fam                                       NaN     Primary   \n",
       "chr_2_raw_bim                                       NaN     Primary   \n",
       "chr_3_raw_bed                                       NaN     Primary   \n",
       "chr_3_raw_fam                                       NaN     Primary   \n",
       "chr_3_raw_bim                                       NaN     Primary   \n",
       "chr_4_raw_bed                                       NaN     Primary   \n",
       "chr_4_raw_fam                                       NaN     Primary   \n",
       "chr_4_raw_bim                                       NaN     Primary   \n",
       "chr_5_raw_bed                                       NaN     Primary   \n",
       "chr_5_raw_fam                                       NaN     Primary   \n",
       "chr_5_raw_bim                                       NaN     Primary   \n",
       "chr_6_raw_bed                                       NaN     Primary   \n",
       "chr_6_raw_fam                                       NaN     Primary   \n",
       "chr_6_raw_bim                                       NaN     Primary   \n",
       "chr_7_raw_bed                                       NaN     Primary   \n",
       "chr_7_raw_fam                                       NaN     Primary   \n",
       "chr_7_raw_bim                                       NaN     Primary   \n",
       "chr_8_raw_bed                                       NaN     Primary   \n",
       "chr_8_raw_fam                                       NaN     Primary   \n",
       "chr_8_raw_bim                                       NaN     Primary   \n",
       "chr_9_raw_bed                                       NaN     Primary   \n",
       "chr_9_raw_fam                                       NaN     Primary   \n",
       "chr_9_raw_bim                                       NaN     Primary   \n",
       "chr_10_raw_bed                                      NaN     Primary   \n",
       "chr_10_raw_fam                                      NaN     Primary   \n",
       "chr_10_raw_bim                                      NaN     Primary   \n",
       "chr_11_raw_bed                                      NaN     Primary   \n",
       "chr_11_raw_fam                                      NaN     Primary   \n",
       "chr_11_raw_bim                                      NaN     Primary   \n",
       "chr_12_raw_bed                                      NaN     Primary   \n",
       "chr_12_raw_fam                                      NaN     Primary   \n",
       "chr_12_raw_bim                                      NaN     Primary   \n",
       "chr_13_raw_bed                                      NaN     Primary   \n",
       "chr_13_raw_fam                                      NaN     Primary   \n",
       "chr_13_raw_bim                                      NaN     Primary   \n",
       "chr_14_raw_bed                                      NaN     Primary   \n",
       "chr_14_raw_fam                                      NaN     Primary   \n",
       "chr_14_raw_bim                                      NaN     Primary   \n",
       "chr_15_raw_bed                                      NaN     Primary   \n",
       "chr_15_raw_fam                                      NaN     Primary   \n",
       "chr_15_raw_bim                                      NaN     Primary   \n",
       "chr_16_raw_bed                                      NaN     Primary   \n",
       "chr_16_raw_fam                                      NaN     Primary   \n",
       "chr_16_raw_bim                                      NaN     Primary   \n",
       "chr_17_raw_bed                                      NaN     Primary   \n",
       "chr_17_raw_fam                                      NaN     Primary   \n",
       "chr_17_raw_bim                                      NaN     Primary   \n",
       "chr_18_raw_bed                                      NaN     Primary   \n",
       "chr_18_raw_fam                                      NaN     Primary   \n",
       "chr_18_raw_bim                                      NaN     Primary   \n",
       "chr_19_raw_bed                                      NaN     Primary   \n",
       "chr_19_raw_fam                                      NaN     Primary   \n",
       "chr_19_raw_bim                                      NaN     Primary   \n",
       "chr_20_raw_bed                                      NaN     Primary   \n",
       "chr_20_raw_fam                                      NaN     Primary   \n",
       "chr_20_raw_bim                                      NaN     Primary   \n",
       "chr_21_raw_bed                                      NaN     Primary   \n",
       "chr_21_raw_fam                                      NaN     Primary   \n",
       "chr_21_raw_bim                                      NaN     Primary   \n",
       "chr_22_raw_bed                                      NaN     Primary   \n",
       "chr_22_raw_fam                                      NaN     Primary   \n",
       "chr_22_raw_bim                                      NaN     Primary   \n",
       "chr_1_postqc_bed                                    NaN     Primary   \n",
       "chr_1_postqc_fam                                    NaN     Primary   \n",
       "chr_1_postqc_bim                                    NaN     Primary   \n",
       "chr_2_postqc_bed                                    NaN     Primary   \n",
       "chr_2_postqc_fam                                    NaN     Primary   \n",
       "chr_2_postqc_bim                                    NaN     Primary   \n",
       "chr_3_postqc_bed                                    NaN     Primary   \n",
       "chr_3_postqc_fam                                    NaN     Primary   \n",
       "chr_3_postqc_bim                                    NaN     Primary   \n",
       "chr_4_postqc_bed                                    NaN     Primary   \n",
       "chr_4_postqc_fam                                    NaN     Primary   \n",
       "chr_4_postqc_bim                                    NaN     Primary   \n",
       "chr_5_postqc_bed                                    NaN     Primary   \n",
       "chr_5_postqc_fam                                    NaN     Primary   \n",
       "chr_5_postqc_bim                                    NaN     Primary   \n",
       "chr_6_postqc_bed                                    NaN     Primary   \n",
       "chr_6_postqc_fam                                    NaN     Primary   \n",
       "chr_6_postqc_bim                                    NaN     Primary   \n",
       "chr_7_postqc_bed                                    NaN     Primary   \n",
       "chr_7_postqc_fam                                    NaN     Primary   \n",
       "chr_7_postqc_bim                                    NaN     Primary   \n",
       "chr_8_postqc_bed                                    NaN     Primary   \n",
       "chr_8_postqc_fam                                    NaN     Primary   \n",
       "chr_8_postqc_bim                                    NaN     Primary   \n",
       "chr_9_postqc_bed                                    NaN     Primary   \n",
       "chr_9_postqc_fam                                    NaN     Primary   \n",
       "chr_9_postqc_bim                                    NaN     Primary   \n",
       "chr_10_postqc_bed                                   NaN     Primary   \n",
       "chr_10_postqc_fam                                   NaN     Primary   \n",
       "chr_10_postqc_bim                                   NaN     Primary   \n",
       "chr_11_postqc_bed                                   NaN     Primary   \n",
       "chr_11_postqc_fam                                   NaN     Primary   \n",
       "chr_11_postqc_bim                                   NaN     Primary   \n",
       "chr_12_postqc_bed                                   NaN     Primary   \n",
       "chr_12_postqc_fam                                   NaN     Primary   \n",
       "chr_12_postqc_bim                                   NaN     Primary   \n",
       "chr_13_postqc_bed                                   NaN     Primary   \n",
       "chr_13_postqc_fam                                   NaN     Primary   \n",
       "chr_13_postqc_bim                                   NaN     Primary   \n",
       "chr_14_postqc_bed                                   NaN     Primary   \n",
       "chr_14_postqc_fam                                   NaN     Primary   \n",
       "chr_14_postqc_bim                                   NaN     Primary   \n",
       "chr_15_postqc_bed                                   NaN     Primary   \n",
       "chr_15_postqc_fam                                   NaN     Primary   \n",
       "chr_15_postqc_bim                                   NaN     Primary   \n",
       "chr_16_postqc_bed                                   NaN     Primary   \n",
       "chr_16_postqc_fam                                   NaN     Primary   \n",
       "chr_16_postqc_bim                                   NaN     Primary   \n",
       "chr_17_postqc_bed                                   NaN     Primary   \n",
       "chr_17_postqc_fam                                   NaN     Primary   \n",
       "chr_17_postqc_bim                                   NaN     Primary   \n",
       "chr_18_postqc_bed                                   NaN     Primary   \n",
       "chr_18_postqc_fam                                   NaN     Primary   \n",
       "chr_18_postqc_bim                                   NaN     Primary   \n",
       "chr_19_postqc_bed                                   NaN     Primary   \n",
       "chr_19_postqc_fam                                   NaN     Primary   \n",
       "chr_19_postqc_bim                                   NaN     Primary   \n",
       "chr_20_postqc_bed                                   NaN     Primary   \n",
       "chr_20_postqc_fam                                   NaN     Primary   \n",
       "chr_20_postqc_bim                                   NaN     Primary   \n",
       "chr_21_postqc_bed                                   NaN     Primary   \n",
       "chr_21_postqc_fam                                   NaN     Primary   \n",
       "chr_21_postqc_bim                                   NaN     Primary   \n",
       "chr_22_postqc_bed                                   NaN     Primary   \n",
       "chr_22_postqc_fam                                   NaN     Primary   \n",
       "chr_22_postqc_bim                                   NaN     Primary   \n",
       "pc1                                                 NaN     Primary   \n",
       "pc2                                                 NaN     Primary   \n",
       "pc3                                                 NaN     Primary   \n",
       "pc4                                                 NaN     Primary   \n",
       "pc5                                                 NaN     Primary   \n",
       "pc6                                                 NaN     Primary   \n",
       "pc7                                                 NaN     Primary   \n",
       "pc8                                                 NaN     Primary   \n",
       "pc9                                                 NaN     Primary   \n",
       "pc10                                                NaN     Primary   \n",
       "pc11                                                NaN     Primary   \n",
       "pc12                                                NaN     Primary   \n",
       "pc13                                                NaN     Primary   \n",
       "pc14                                                NaN     Primary   \n",
       "pc15                                                NaN     Primary   \n",
       "pc16                                                NaN     Primary   \n",
       "pc17                                                NaN     Primary   \n",
       "pc18                                                NaN     Primary   \n",
       "pc19                                                NaN     Primary   \n",
       "pc20                                                NaN     Primary   \n",
       "used_in_pca_calculation                             NaN     Primary   \n",
       "pca_loadings_tsv                                    NaN     Primary   \n",
       "\n",
       "                                               sexed   array  completed  \\\n",
       "tabular_field_name                                                        \n",
       "collection_date                           Both sexes  Single        NaN   \n",
       "version                                   Both sexes  Single        NaN   \n",
       "gencove_fastq_r1                          Both sexes  Single        NaN   \n",
       "gencove_fastq_r2                          Both sexes  Single        NaN   \n",
       "gencove_bam                               Both sexes  Single        NaN   \n",
       "gencove_bai                               Both sexes  Single        NaN   \n",
       "gencove_vcf                               Both sexes  Single        NaN   \n",
       "genecov_qc_bases                          Both sexes  Single        NaN   \n",
       "genecov_qc_bases_dedup                    Both sexes  Single        NaN   \n",
       "gencove_qc_bases_dedup_mapped             Both sexes  Single        NaN   \n",
       "genecov_qc_effective_coverage             Both sexes  Single        NaN   \n",
       "genecov_qc_fraction_contamination         Both sexes  Single        NaN   \n",
       "genecov_qc_snps                           Both sexes  Single        NaN   \n",
       "genecov_qc_format_passed                  Both sexes  Single        NaN   \n",
       "genecov_qc_r1_eq_r2_passed                Both sexes  Single        NaN   \n",
       "genecov_qc_r1_r2_ids_match_passed         Both sexes  Single        NaN   \n",
       "genecov_qc_bases_passed                   Both sexes  Single        NaN   \n",
       "genecov_qc_bases_dedup_passed             Both sexes  Single        NaN   \n",
       "genecov_qc_bases_dedup_mapped_passed      Both sexes  Single        NaN   \n",
       "genecov_qc_effective_coverage_passed      Both sexes  Single        NaN   \n",
       "genecov_qc_fraction_contamination_passed  Both sexes  Single        NaN   \n",
       "genecov_qc_snps_passed                    Both sexes  Single        NaN   \n",
       "heterozygosity_proportion                 Both sexes  Single        NaN   \n",
       "pass_het_check                            Both sexes  Single        NaN   \n",
       "sample_qc_gp_pass_rate                    Both sexes  Single        NaN   \n",
       "sample_qc_gp_pass_check                   Both sexes  Single        NaN   \n",
       "genetic_sex                               Both sexes  Single        NaN   \n",
       "chr_x_proportion                          Both sexes  Single        NaN   \n",
       "chr_y_proportion                          Both sexes  Single        NaN   \n",
       "sample_qc_gender_match_check              Both sexes  Single        NaN   \n",
       "variants_qc_parquet                       Both sexes  Single        NaN   \n",
       "relatives                                 Both sexes  Single        NaN   \n",
       "relationship_king_tsv                     Both sexes  Single        NaN   \n",
       "relationship_ibs_txt                      Both sexes  Single        NaN   \n",
       "chr_1_raw_bed                             Both sexes  Single        NaN   \n",
       "chr_1_raw_fam                             Both sexes  Single        NaN   \n",
       "chr_1_raw_bim                             Both sexes  Single        NaN   \n",
       "chr_2_raw_bed                             Both sexes  Single        NaN   \n",
       "chr_2_raw_fam                             Both sexes  Single        NaN   \n",
       "chr_2_raw_bim                             Both sexes  Single        NaN   \n",
       "chr_3_raw_bed                             Both sexes  Single        NaN   \n",
       "chr_3_raw_fam                             Both sexes  Single        NaN   \n",
       "chr_3_raw_bim                             Both sexes  Single        NaN   \n",
       "chr_4_raw_bed                             Both sexes  Single        NaN   \n",
       "chr_4_raw_fam                             Both sexes  Single        NaN   \n",
       "chr_4_raw_bim                             Both sexes  Single        NaN   \n",
       "chr_5_raw_bed                             Both sexes  Single        NaN   \n",
       "chr_5_raw_fam                             Both sexes  Single        NaN   \n",
       "chr_5_raw_bim                             Both sexes  Single        NaN   \n",
       "chr_6_raw_bed                             Both sexes  Single        NaN   \n",
       "chr_6_raw_fam                             Both sexes  Single        NaN   \n",
       "chr_6_raw_bim                             Both sexes  Single        NaN   \n",
       "chr_7_raw_bed                             Both sexes  Single        NaN   \n",
       "chr_7_raw_fam                             Both sexes  Single        NaN   \n",
       "chr_7_raw_bim                             Both sexes  Single        NaN   \n",
       "chr_8_raw_bed                             Both sexes  Single        NaN   \n",
       "chr_8_raw_fam                             Both sexes  Single        NaN   \n",
       "chr_8_raw_bim                             Both sexes  Single        NaN   \n",
       "chr_9_raw_bed                             Both sexes  Single        NaN   \n",
       "chr_9_raw_fam                             Both sexes  Single        NaN   \n",
       "chr_9_raw_bim                             Both sexes  Single        NaN   \n",
       "chr_10_raw_bed                            Both sexes  Single        NaN   \n",
       "chr_10_raw_fam                            Both sexes  Single        NaN   \n",
       "chr_10_raw_bim                            Both sexes  Single        NaN   \n",
       "chr_11_raw_bed                            Both sexes  Single        NaN   \n",
       "chr_11_raw_fam                            Both sexes  Single        NaN   \n",
       "chr_11_raw_bim                            Both sexes  Single        NaN   \n",
       "chr_12_raw_bed                            Both sexes  Single        NaN   \n",
       "chr_12_raw_fam                            Both sexes  Single        NaN   \n",
       "chr_12_raw_bim                            Both sexes  Single        NaN   \n",
       "chr_13_raw_bed                            Both sexes  Single        NaN   \n",
       "chr_13_raw_fam                            Both sexes  Single        NaN   \n",
       "chr_13_raw_bim                            Both sexes  Single        NaN   \n",
       "chr_14_raw_bed                            Both sexes  Single        NaN   \n",
       "chr_14_raw_fam                            Both sexes  Single        NaN   \n",
       "chr_14_raw_bim                            Both sexes  Single        NaN   \n",
       "chr_15_raw_bed                            Both sexes  Single        NaN   \n",
       "chr_15_raw_fam                            Both sexes  Single        NaN   \n",
       "chr_15_raw_bim                            Both sexes  Single        NaN   \n",
       "chr_16_raw_bed                            Both sexes  Single        NaN   \n",
       "chr_16_raw_fam                            Both sexes  Single        NaN   \n",
       "chr_16_raw_bim                            Both sexes  Single        NaN   \n",
       "chr_17_raw_bed                            Both sexes  Single        NaN   \n",
       "chr_17_raw_fam                            Both sexes  Single        NaN   \n",
       "chr_17_raw_bim                            Both sexes  Single        NaN   \n",
       "chr_18_raw_bed                            Both sexes  Single        NaN   \n",
       "chr_18_raw_fam                            Both sexes  Single        NaN   \n",
       "chr_18_raw_bim                            Both sexes  Single        NaN   \n",
       "chr_19_raw_bed                            Both sexes  Single        NaN   \n",
       "chr_19_raw_fam                            Both sexes  Single        NaN   \n",
       "chr_19_raw_bim                            Both sexes  Single        NaN   \n",
       "chr_20_raw_bed                            Both sexes  Single        NaN   \n",
       "chr_20_raw_fam                            Both sexes  Single        NaN   \n",
       "chr_20_raw_bim                            Both sexes  Single        NaN   \n",
       "chr_21_raw_bed                            Both sexes  Single        NaN   \n",
       "chr_21_raw_fam                            Both sexes  Single        NaN   \n",
       "chr_21_raw_bim                            Both sexes  Single        NaN   \n",
       "chr_22_raw_bed                            Both sexes  Single        NaN   \n",
       "chr_22_raw_fam                            Both sexes  Single        NaN   \n",
       "chr_22_raw_bim                            Both sexes  Single        NaN   \n",
       "chr_1_postqc_bed                          Both sexes  Single        NaN   \n",
       "chr_1_postqc_fam                          Both sexes  Single        NaN   \n",
       "chr_1_postqc_bim                          Both sexes  Single        NaN   \n",
       "chr_2_postqc_bed                          Both sexes  Single        NaN   \n",
       "chr_2_postqc_fam                          Both sexes  Single        NaN   \n",
       "chr_2_postqc_bim                          Both sexes  Single        NaN   \n",
       "chr_3_postqc_bed                          Both sexes  Single        NaN   \n",
       "chr_3_postqc_fam                          Both sexes  Single        NaN   \n",
       "chr_3_postqc_bim                          Both sexes  Single        NaN   \n",
       "chr_4_postqc_bed                          Both sexes  Single        NaN   \n",
       "chr_4_postqc_fam                          Both sexes  Single        NaN   \n",
       "chr_4_postqc_bim                          Both sexes  Single        NaN   \n",
       "chr_5_postqc_bed                          Both sexes  Single        NaN   \n",
       "chr_5_postqc_fam                          Both sexes  Single        NaN   \n",
       "chr_5_postqc_bim                          Both sexes  Single        NaN   \n",
       "chr_6_postqc_bed                          Both sexes  Single        NaN   \n",
       "chr_6_postqc_fam                          Both sexes  Single        NaN   \n",
       "chr_6_postqc_bim                          Both sexes  Single        NaN   \n",
       "chr_7_postqc_bed                          Both sexes  Single        NaN   \n",
       "chr_7_postqc_fam                          Both sexes  Single        NaN   \n",
       "chr_7_postqc_bim                          Both sexes  Single        NaN   \n",
       "chr_8_postqc_bed                          Both sexes  Single        NaN   \n",
       "chr_8_postqc_fam                          Both sexes  Single        NaN   \n",
       "chr_8_postqc_bim                          Both sexes  Single        NaN   \n",
       "chr_9_postqc_bed                          Both sexes  Single        NaN   \n",
       "chr_9_postqc_fam                          Both sexes  Single        NaN   \n",
       "chr_9_postqc_bim                          Both sexes  Single        NaN   \n",
       "chr_10_postqc_bed                         Both sexes  Single        NaN   \n",
       "chr_10_postqc_fam                         Both sexes  Single        NaN   \n",
       "chr_10_postqc_bim                         Both sexes  Single        NaN   \n",
       "chr_11_postqc_bed                         Both sexes  Single        NaN   \n",
       "chr_11_postqc_fam                         Both sexes  Single        NaN   \n",
       "chr_11_postqc_bim                         Both sexes  Single        NaN   \n",
       "chr_12_postqc_bed                         Both sexes  Single        NaN   \n",
       "chr_12_postqc_fam                         Both sexes  Single        NaN   \n",
       "chr_12_postqc_bim                         Both sexes  Single        NaN   \n",
       "chr_13_postqc_bed                         Both sexes  Single        NaN   \n",
       "chr_13_postqc_fam                         Both sexes  Single        NaN   \n",
       "chr_13_postqc_bim                         Both sexes  Single        NaN   \n",
       "chr_14_postqc_bed                         Both sexes  Single        NaN   \n",
       "chr_14_postqc_fam                         Both sexes  Single        NaN   \n",
       "chr_14_postqc_bim                         Both sexes  Single        NaN   \n",
       "chr_15_postqc_bed                         Both sexes  Single        NaN   \n",
       "chr_15_postqc_fam                         Both sexes  Single        NaN   \n",
       "chr_15_postqc_bim                         Both sexes  Single        NaN   \n",
       "chr_16_postqc_bed                         Both sexes  Single        NaN   \n",
       "chr_16_postqc_fam                         Both sexes  Single        NaN   \n",
       "chr_16_postqc_bim                         Both sexes  Single        NaN   \n",
       "chr_17_postqc_bed                         Both sexes  Single        NaN   \n",
       "chr_17_postqc_fam                         Both sexes  Single        NaN   \n",
       "chr_17_postqc_bim                         Both sexes  Single        NaN   \n",
       "chr_18_postqc_bed                         Both sexes  Single        NaN   \n",
       "chr_18_postqc_fam                         Both sexes  Single        NaN   \n",
       "chr_18_postqc_bim                         Both sexes  Single        NaN   \n",
       "chr_19_postqc_bed                         Both sexes  Single        NaN   \n",
       "chr_19_postqc_fam                         Both sexes  Single        NaN   \n",
       "chr_19_postqc_bim                         Both sexes  Single        NaN   \n",
       "chr_20_postqc_bed                         Both sexes  Single        NaN   \n",
       "chr_20_postqc_fam                         Both sexes  Single        NaN   \n",
       "chr_20_postqc_bim                         Both sexes  Single        NaN   \n",
       "chr_21_postqc_bed                         Both sexes  Single        NaN   \n",
       "chr_21_postqc_fam                         Both sexes  Single        NaN   \n",
       "chr_21_postqc_bim                         Both sexes  Single        NaN   \n",
       "chr_22_postqc_bed                         Both sexes  Single        NaN   \n",
       "chr_22_postqc_fam                         Both sexes  Single        NaN   \n",
       "chr_22_postqc_bim                         Both sexes  Single        NaN   \n",
       "pc1                                       Both sexes  Single        NaN   \n",
       "pc2                                       Both sexes  Single        NaN   \n",
       "pc3                                       Both sexes  Single        NaN   \n",
       "pc4                                       Both sexes  Single        NaN   \n",
       "pc5                                       Both sexes  Single        NaN   \n",
       "pc6                                       Both sexes  Single        NaN   \n",
       "pc7                                       Both sexes  Single        NaN   \n",
       "pc8                                       Both sexes  Single        NaN   \n",
       "pc9                                       Both sexes  Single        NaN   \n",
       "pc10                                      Both sexes  Single        NaN   \n",
       "pc11                                      Both sexes  Single        NaN   \n",
       "pc12                                      Both sexes  Single        NaN   \n",
       "pc13                                      Both sexes  Single        NaN   \n",
       "pc14                                      Both sexes  Single        NaN   \n",
       "pc15                                      Both sexes  Single        NaN   \n",
       "pc16                                      Both sexes  Single        NaN   \n",
       "pc17                                      Both sexes  Single        NaN   \n",
       "pc18                                      Both sexes  Single        NaN   \n",
       "pc19                                      Both sexes  Single        NaN   \n",
       "pc20                                      Both sexes  Single        NaN   \n",
       "used_in_pca_calculation                   Both sexes  Single        NaN   \n",
       "pca_loadings_tsv                          Both sexes  Single        NaN   \n",
       "\n",
       "                                          debut  transformation  \\\n",
       "tabular_field_name                                                \n",
       "collection_date                             NaN             NaN   \n",
       "version                                     NaN             NaN   \n",
       "gencove_fastq_r1                            NaN             NaN   \n",
       "gencove_fastq_r2                            NaN             NaN   \n",
       "gencove_bam                                 NaN             NaN   \n",
       "gencove_bai                                 NaN             NaN   \n",
       "gencove_vcf                                 NaN             NaN   \n",
       "genecov_qc_bases                            NaN             NaN   \n",
       "genecov_qc_bases_dedup                      NaN             NaN   \n",
       "gencove_qc_bases_dedup_mapped               NaN             NaN   \n",
       "genecov_qc_effective_coverage               NaN             NaN   \n",
       "genecov_qc_fraction_contamination           NaN             NaN   \n",
       "genecov_qc_snps                             NaN             NaN   \n",
       "genecov_qc_format_passed                    NaN             NaN   \n",
       "genecov_qc_r1_eq_r2_passed                  NaN             NaN   \n",
       "genecov_qc_r1_r2_ids_match_passed           NaN             NaN   \n",
       "genecov_qc_bases_passed                     NaN             NaN   \n",
       "genecov_qc_bases_dedup_passed               NaN             NaN   \n",
       "genecov_qc_bases_dedup_mapped_passed        NaN             NaN   \n",
       "genecov_qc_effective_coverage_passed        NaN             NaN   \n",
       "genecov_qc_fraction_contamination_passed    NaN             NaN   \n",
       "genecov_qc_snps_passed                      NaN             NaN   \n",
       "heterozygosity_proportion                   NaN             NaN   \n",
       "pass_het_check                              NaN             NaN   \n",
       "sample_qc_gp_pass_rate                      NaN             NaN   \n",
       "sample_qc_gp_pass_check                     NaN             NaN   \n",
       "genetic_sex                                 NaN             NaN   \n",
       "chr_x_proportion                            NaN             NaN   \n",
       "chr_y_proportion                            NaN             NaN   \n",
       "sample_qc_gender_match_check                NaN             NaN   \n",
       "variants_qc_parquet                         NaN             NaN   \n",
       "relatives                                   NaN             NaN   \n",
       "relationship_king_tsv                       NaN             NaN   \n",
       "relationship_ibs_txt                        NaN             NaN   \n",
       "chr_1_raw_bed                               NaN             NaN   \n",
       "chr_1_raw_fam                               NaN             NaN   \n",
       "chr_1_raw_bim                               NaN             NaN   \n",
       "chr_2_raw_bed                               NaN             NaN   \n",
       "chr_2_raw_fam                               NaN             NaN   \n",
       "chr_2_raw_bim                               NaN             NaN   \n",
       "chr_3_raw_bed                               NaN             NaN   \n",
       "chr_3_raw_fam                               NaN             NaN   \n",
       "chr_3_raw_bim                               NaN             NaN   \n",
       "chr_4_raw_bed                               NaN             NaN   \n",
       "chr_4_raw_fam                               NaN             NaN   \n",
       "chr_4_raw_bim                               NaN             NaN   \n",
       "chr_5_raw_bed                               NaN             NaN   \n",
       "chr_5_raw_fam                               NaN             NaN   \n",
       "chr_5_raw_bim                               NaN             NaN   \n",
       "chr_6_raw_bed                               NaN             NaN   \n",
       "chr_6_raw_fam                               NaN             NaN   \n",
       "chr_6_raw_bim                               NaN             NaN   \n",
       "chr_7_raw_bed                               NaN             NaN   \n",
       "chr_7_raw_fam                               NaN             NaN   \n",
       "chr_7_raw_bim                               NaN             NaN   \n",
       "chr_8_raw_bed                               NaN             NaN   \n",
       "chr_8_raw_fam                               NaN             NaN   \n",
       "chr_8_raw_bim                               NaN             NaN   \n",
       "chr_9_raw_bed                               NaN             NaN   \n",
       "chr_9_raw_fam                               NaN             NaN   \n",
       "chr_9_raw_bim                               NaN             NaN   \n",
       "chr_10_raw_bed                              NaN             NaN   \n",
       "chr_10_raw_fam                              NaN             NaN   \n",
       "chr_10_raw_bim                              NaN             NaN   \n",
       "chr_11_raw_bed                              NaN             NaN   \n",
       "chr_11_raw_fam                              NaN             NaN   \n",
       "chr_11_raw_bim                              NaN             NaN   \n",
       "chr_12_raw_bed                              NaN             NaN   \n",
       "chr_12_raw_fam                              NaN             NaN   \n",
       "chr_12_raw_bim                              NaN             NaN   \n",
       "chr_13_raw_bed                              NaN             NaN   \n",
       "chr_13_raw_fam                              NaN             NaN   \n",
       "chr_13_raw_bim                              NaN             NaN   \n",
       "chr_14_raw_bed                              NaN             NaN   \n",
       "chr_14_raw_fam                              NaN             NaN   \n",
       "chr_14_raw_bim                              NaN             NaN   \n",
       "chr_15_raw_bed                              NaN             NaN   \n",
       "chr_15_raw_fam                              NaN             NaN   \n",
       "chr_15_raw_bim                              NaN             NaN   \n",
       "chr_16_raw_bed                              NaN             NaN   \n",
       "chr_16_raw_fam                              NaN             NaN   \n",
       "chr_16_raw_bim                              NaN             NaN   \n",
       "chr_17_raw_bed                              NaN             NaN   \n",
       "chr_17_raw_fam                              NaN             NaN   \n",
       "chr_17_raw_bim                              NaN             NaN   \n",
       "chr_18_raw_bed                              NaN             NaN   \n",
       "chr_18_raw_fam                              NaN             NaN   \n",
       "chr_18_raw_bim                              NaN             NaN   \n",
       "chr_19_raw_bed                              NaN             NaN   \n",
       "chr_19_raw_fam                              NaN             NaN   \n",
       "chr_19_raw_bim                              NaN             NaN   \n",
       "chr_20_raw_bed                              NaN             NaN   \n",
       "chr_20_raw_fam                              NaN             NaN   \n",
       "chr_20_raw_bim                              NaN             NaN   \n",
       "chr_21_raw_bed                              NaN             NaN   \n",
       "chr_21_raw_fam                              NaN             NaN   \n",
       "chr_21_raw_bim                              NaN             NaN   \n",
       "chr_22_raw_bed                              NaN             NaN   \n",
       "chr_22_raw_fam                              NaN             NaN   \n",
       "chr_22_raw_bim                              NaN             NaN   \n",
       "chr_1_postqc_bed                            NaN             NaN   \n",
       "chr_1_postqc_fam                            NaN             NaN   \n",
       "chr_1_postqc_bim                            NaN             NaN   \n",
       "chr_2_postqc_bed                            NaN             NaN   \n",
       "chr_2_postqc_fam                            NaN             NaN   \n",
       "chr_2_postqc_bim                            NaN             NaN   \n",
       "chr_3_postqc_bed                            NaN             NaN   \n",
       "chr_3_postqc_fam                            NaN             NaN   \n",
       "chr_3_postqc_bim                            NaN             NaN   \n",
       "chr_4_postqc_bed                            NaN             NaN   \n",
       "chr_4_postqc_fam                            NaN             NaN   \n",
       "chr_4_postqc_bim                            NaN             NaN   \n",
       "chr_5_postqc_bed                            NaN             NaN   \n",
       "chr_5_postqc_fam                            NaN             NaN   \n",
       "chr_5_postqc_bim                            NaN             NaN   \n",
       "chr_6_postqc_bed                            NaN             NaN   \n",
       "chr_6_postqc_fam                            NaN             NaN   \n",
       "chr_6_postqc_bim                            NaN             NaN   \n",
       "chr_7_postqc_bed                            NaN             NaN   \n",
       "chr_7_postqc_fam                            NaN             NaN   \n",
       "chr_7_postqc_bim                            NaN             NaN   \n",
       "chr_8_postqc_bed                            NaN             NaN   \n",
       "chr_8_postqc_fam                            NaN             NaN   \n",
       "chr_8_postqc_bim                            NaN             NaN   \n",
       "chr_9_postqc_bed                            NaN             NaN   \n",
       "chr_9_postqc_fam                            NaN             NaN   \n",
       "chr_9_postqc_bim                            NaN             NaN   \n",
       "chr_10_postqc_bed                           NaN             NaN   \n",
       "chr_10_postqc_fam                           NaN             NaN   \n",
       "chr_10_postqc_bim                           NaN             NaN   \n",
       "chr_11_postqc_bed                           NaN             NaN   \n",
       "chr_11_postqc_fam                           NaN             NaN   \n",
       "chr_11_postqc_bim                           NaN             NaN   \n",
       "chr_12_postqc_bed                           NaN             NaN   \n",
       "chr_12_postqc_fam                           NaN             NaN   \n",
       "chr_12_postqc_bim                           NaN             NaN   \n",
       "chr_13_postqc_bed                           NaN             NaN   \n",
       "chr_13_postqc_fam                           NaN             NaN   \n",
       "chr_13_postqc_bim                           NaN             NaN   \n",
       "chr_14_postqc_bed                           NaN             NaN   \n",
       "chr_14_postqc_fam                           NaN             NaN   \n",
       "chr_14_postqc_bim                           NaN             NaN   \n",
       "chr_15_postqc_bed                           NaN             NaN   \n",
       "chr_15_postqc_fam                           NaN             NaN   \n",
       "chr_15_postqc_bim                           NaN             NaN   \n",
       "chr_16_postqc_bed                           NaN             NaN   \n",
       "chr_16_postqc_fam                           NaN             NaN   \n",
       "chr_16_postqc_bim                           NaN             NaN   \n",
       "chr_17_postqc_bed                           NaN             NaN   \n",
       "chr_17_postqc_fam                           NaN             NaN   \n",
       "chr_17_postqc_bim                           NaN             NaN   \n",
       "chr_18_postqc_bed                           NaN             NaN   \n",
       "chr_18_postqc_fam                           NaN             NaN   \n",
       "chr_18_postqc_bim                           NaN             NaN   \n",
       "chr_19_postqc_bed                           NaN             NaN   \n",
       "chr_19_postqc_fam                           NaN             NaN   \n",
       "chr_19_postqc_bim                           NaN             NaN   \n",
       "chr_20_postqc_bed                           NaN             NaN   \n",
       "chr_20_postqc_fam                           NaN             NaN   \n",
       "chr_20_postqc_bim                           NaN             NaN   \n",
       "chr_21_postqc_bed                           NaN             NaN   \n",
       "chr_21_postqc_fam                           NaN             NaN   \n",
       "chr_21_postqc_bim                           NaN             NaN   \n",
       "chr_22_postqc_bed                           NaN             NaN   \n",
       "chr_22_postqc_fam                           NaN             NaN   \n",
       "chr_22_postqc_bim                           NaN             NaN   \n",
       "pc1                                         NaN             NaN   \n",
       "pc2                                         NaN             NaN   \n",
       "pc3                                         NaN             NaN   \n",
       "pc4                                         NaN             NaN   \n",
       "pc5                                         NaN             NaN   \n",
       "pc6                                         NaN             NaN   \n",
       "pc7                                         NaN             NaN   \n",
       "pc8                                         NaN             NaN   \n",
       "pc9                                         NaN             NaN   \n",
       "pc10                                        NaN             NaN   \n",
       "pc11                                        NaN             NaN   \n",
       "pc12                                        NaN             NaN   \n",
       "pc13                                        NaN             NaN   \n",
       "pc14                                        NaN             NaN   \n",
       "pc15                                        NaN             NaN   \n",
       "pc16                                        NaN             NaN   \n",
       "pc17                                        NaN             NaN   \n",
       "pc18                                        NaN             NaN   \n",
       "pc19                                        NaN             NaN   \n",
       "pc20                                        NaN             NaN   \n",
       "used_in_pca_calculation                     NaN             NaN   \n",
       "pca_loadings_tsv                            NaN             NaN   \n",
       "\n",
       "                                            list_of_tags    pandas_dtype  \n",
       "tabular_field_name                                                        \n",
       "collection_date                           Human Genetics  datetime64[ns]  \n",
       "version                                   Human Genetics        category  \n",
       "gencove_fastq_r1                          Human Genetics          string  \n",
       "gencove_fastq_r2                          Human Genetics          string  \n",
       "gencove_bam                               Human Genetics          string  \n",
       "gencove_bai                               Human Genetics          string  \n",
       "gencove_vcf                               Human Genetics          string  \n",
       "genecov_qc_bases                          Human Genetics             int  \n",
       "genecov_qc_bases_dedup                    Human Genetics             int  \n",
       "gencove_qc_bases_dedup_mapped             Human Genetics             int  \n",
       "genecov_qc_effective_coverage             Human Genetics           float  \n",
       "genecov_qc_fraction_contamination         Human Genetics           float  \n",
       "genecov_qc_snps                           Human Genetics             int  \n",
       "genecov_qc_format_passed                  Human Genetics          object  \n",
       "genecov_qc_r1_eq_r2_passed                Human Genetics          object  \n",
       "genecov_qc_r1_r2_ids_match_passed         Human Genetics          object  \n",
       "genecov_qc_bases_passed                   Human Genetics          object  \n",
       "genecov_qc_bases_dedup_passed             Human Genetics          object  \n",
       "genecov_qc_bases_dedup_mapped_passed      Human Genetics          object  \n",
       "genecov_qc_effective_coverage_passed      Human Genetics          object  \n",
       "genecov_qc_fraction_contamination_passed  Human Genetics          object  \n",
       "genecov_qc_snps_passed                    Human Genetics          object  \n",
       "heterozygosity_proportion                 Human Genetics           float  \n",
       "pass_het_check                            Human Genetics            bool  \n",
       "sample_qc_gp_pass_rate                    Human Genetics           float  \n",
       "sample_qc_gp_pass_check                   Human Genetics            bool  \n",
       "genetic_sex                               Human Genetics             int  \n",
       "chr_x_proportion                          Human Genetics           float  \n",
       "chr_y_proportion                          Human Genetics           float  \n",
       "sample_qc_gender_match_check              Human Genetics            bool  \n",
       "variants_qc_parquet                       Human Genetics        category  \n",
       "relatives                                 Human Genetics          string  \n",
       "relationship_king_tsv                     Human Genetics        category  \n",
       "relationship_ibs_txt                      Human Genetics        category  \n",
       "chr_1_raw_bed                             Human Genetics        category  \n",
       "chr_1_raw_fam                             Human Genetics        category  \n",
       "chr_1_raw_bim                             Human Genetics        category  \n",
       "chr_2_raw_bed                             Human Genetics        category  \n",
       "chr_2_raw_fam                             Human Genetics        category  \n",
       "chr_2_raw_bim                             Human Genetics        category  \n",
       "chr_3_raw_bed                             Human Genetics        category  \n",
       "chr_3_raw_fam                             Human Genetics        category  \n",
       "chr_3_raw_bim                             Human Genetics        category  \n",
       "chr_4_raw_bed                             Human Genetics        category  \n",
       "chr_4_raw_fam                             Human Genetics        category  \n",
       "chr_4_raw_bim                             Human Genetics        category  \n",
       "chr_5_raw_bed                             Human Genetics        category  \n",
       "chr_5_raw_fam                             Human Genetics        category  \n",
       "chr_5_raw_bim                             Human Genetics        category  \n",
       "chr_6_raw_bed                             Human Genetics        category  \n",
       "chr_6_raw_fam                             Human Genetics        category  \n",
       "chr_6_raw_bim                             Human Genetics        category  \n",
       "chr_7_raw_bed                             Human Genetics        category  \n",
       "chr_7_raw_fam                             Human Genetics        category  \n",
       "chr_7_raw_bim                             Human Genetics        category  \n",
       "chr_8_raw_bed                             Human Genetics        category  \n",
       "chr_8_raw_fam                             Human Genetics        category  \n",
       "chr_8_raw_bim                             Human Genetics        category  \n",
       "chr_9_raw_bed                             Human Genetics        category  \n",
       "chr_9_raw_fam                             Human Genetics        category  \n",
       "chr_9_raw_bim                             Human Genetics        category  \n",
       "chr_10_raw_bed                            Human Genetics        category  \n",
       "chr_10_raw_fam                            Human Genetics        category  \n",
       "chr_10_raw_bim                            Human Genetics        category  \n",
       "chr_11_raw_bed                            Human Genetics        category  \n",
       "chr_11_raw_fam                            Human Genetics        category  \n",
       "chr_11_raw_bim                            Human Genetics        category  \n",
       "chr_12_raw_bed                            Human Genetics        category  \n",
       "chr_12_raw_fam                            Human Genetics        category  \n",
       "chr_12_raw_bim                            Human Genetics        category  \n",
       "chr_13_raw_bed                            Human Genetics        category  \n",
       "chr_13_raw_fam                            Human Genetics        category  \n",
       "chr_13_raw_bim                            Human Genetics        category  \n",
       "chr_14_raw_bed                            Human Genetics        category  \n",
       "chr_14_raw_fam                            Human Genetics        category  \n",
       "chr_14_raw_bim                            Human Genetics        category  \n",
       "chr_15_raw_bed                            Human Genetics        category  \n",
       "chr_15_raw_fam                            Human Genetics        category  \n",
       "chr_15_raw_bim                            Human Genetics        category  \n",
       "chr_16_raw_bed                            Human Genetics        category  \n",
       "chr_16_raw_fam                            Human Genetics        category  \n",
       "chr_16_raw_bim                            Human Genetics        category  \n",
       "chr_17_raw_bed                            Human Genetics        category  \n",
       "chr_17_raw_fam                            Human Genetics        category  \n",
       "chr_17_raw_bim                            Human Genetics        category  \n",
       "chr_18_raw_bed                            Human Genetics        category  \n",
       "chr_18_raw_fam                            Human Genetics        category  \n",
       "chr_18_raw_bim                            Human Genetics        category  \n",
       "chr_19_raw_bed                            Human Genetics        category  \n",
       "chr_19_raw_fam                            Human Genetics        category  \n",
       "chr_19_raw_bim                            Human Genetics        category  \n",
       "chr_20_raw_bed                            Human Genetics        category  \n",
       "chr_20_raw_fam                            Human Genetics        category  \n",
       "chr_20_raw_bim                            Human Genetics        category  \n",
       "chr_21_raw_bed                            Human Genetics        category  \n",
       "chr_21_raw_fam                            Human Genetics        category  \n",
       "chr_21_raw_bim                            Human Genetics        category  \n",
       "chr_22_raw_bed                            Human Genetics        category  \n",
       "chr_22_raw_fam                            Human Genetics        category  \n",
       "chr_22_raw_bim                            Human Genetics        category  \n",
       "chr_1_postqc_bed                          Human Genetics        category  \n",
       "chr_1_postqc_fam                          Human Genetics        category  \n",
       "chr_1_postqc_bim                          Human Genetics        category  \n",
       "chr_2_postqc_bed                          Human Genetics        category  \n",
       "chr_2_postqc_fam                          Human Genetics        category  \n",
       "chr_2_postqc_bim                          Human Genetics        category  \n",
       "chr_3_postqc_bed                          Human Genetics        category  \n",
       "chr_3_postqc_fam                          Human Genetics        category  \n",
       "chr_3_postqc_bim                          Human Genetics        category  \n",
       "chr_4_postqc_bed                          Human Genetics        category  \n",
       "chr_4_postqc_fam                          Human Genetics        category  \n",
       "chr_4_postqc_bim                          Human Genetics        category  \n",
       "chr_5_postqc_bed                          Human Genetics        category  \n",
       "chr_5_postqc_fam                          Human Genetics        category  \n",
       "chr_5_postqc_bim                          Human Genetics        category  \n",
       "chr_6_postqc_bed                          Human Genetics        category  \n",
       "chr_6_postqc_fam                          Human Genetics        category  \n",
       "chr_6_postqc_bim                          Human Genetics        category  \n",
       "chr_7_postqc_bed                          Human Genetics        category  \n",
       "chr_7_postqc_fam                          Human Genetics        category  \n",
       "chr_7_postqc_bim                          Human Genetics        category  \n",
       "chr_8_postqc_bed                          Human Genetics        category  \n",
       "chr_8_postqc_fam                          Human Genetics        category  \n",
       "chr_8_postqc_bim                          Human Genetics        category  \n",
       "chr_9_postqc_bed                          Human Genetics        category  \n",
       "chr_9_postqc_fam                          Human Genetics        category  \n",
       "chr_9_postqc_bim                          Human Genetics        category  \n",
       "chr_10_postqc_bed                         Human Genetics        category  \n",
       "chr_10_postqc_fam                         Human Genetics        category  \n",
       "chr_10_postqc_bim                         Human Genetics        category  \n",
       "chr_11_postqc_bed                         Human Genetics        category  \n",
       "chr_11_postqc_fam                         Human Genetics        category  \n",
       "chr_11_postqc_bim                         Human Genetics        category  \n",
       "chr_12_postqc_bed                         Human Genetics        category  \n",
       "chr_12_postqc_fam                         Human Genetics        category  \n",
       "chr_12_postqc_bim                         Human Genetics        category  \n",
       "chr_13_postqc_bed                         Human Genetics        category  \n",
       "chr_13_postqc_fam                         Human Genetics        category  \n",
       "chr_13_postqc_bim                         Human Genetics        category  \n",
       "chr_14_postqc_bed                         Human Genetics        category  \n",
       "chr_14_postqc_fam                         Human Genetics        category  \n",
       "chr_14_postqc_bim                         Human Genetics        category  \n",
       "chr_15_postqc_bed                         Human Genetics        category  \n",
       "chr_15_postqc_fam                         Human Genetics        category  \n",
       "chr_15_postqc_bim                         Human Genetics        category  \n",
       "chr_16_postqc_bed                         Human Genetics        category  \n",
       "chr_16_postqc_fam                         Human Genetics        category  \n",
       "chr_16_postqc_bim                         Human Genetics        category  \n",
       "chr_17_postqc_bed                         Human Genetics        category  \n",
       "chr_17_postqc_fam                         Human Genetics        category  \n",
       "chr_17_postqc_bim                         Human Genetics        category  \n",
       "chr_18_postqc_bed                         Human Genetics        category  \n",
       "chr_18_postqc_fam                         Human Genetics        category  \n",
       "chr_18_postqc_bim                         Human Genetics        category  \n",
       "chr_19_postqc_bed                         Human Genetics        category  \n",
       "chr_19_postqc_fam                         Human Genetics        category  \n",
       "chr_19_postqc_bim                         Human Genetics        category  \n",
       "chr_20_postqc_bed                         Human Genetics        category  \n",
       "chr_20_postqc_fam                         Human Genetics        category  \n",
       "chr_20_postqc_bim                         Human Genetics        category  \n",
       "chr_21_postqc_bed                         Human Genetics        category  \n",
       "chr_21_postqc_fam                         Human Genetics        category  \n",
       "chr_21_postqc_bim                         Human Genetics        category  \n",
       "chr_22_postqc_bed                         Human Genetics        category  \n",
       "chr_22_postqc_fam                         Human Genetics        category  \n",
       "chr_22_postqc_bim                         Human Genetics        category  \n",
       "pc1                                       Human Genetics           float  \n",
       "pc2                                       Human Genetics           float  \n",
       "pc3                                       Human Genetics           float  \n",
       "pc4                                       Human Genetics           float  \n",
       "pc5                                       Human Genetics           float  \n",
       "pc6                                       Human Genetics           float  \n",
       "pc7                                       Human Genetics           float  \n",
       "pc8                                       Human Genetics           float  \n",
       "pc9                                       Human Genetics           float  \n",
       "pc10                                      Human Genetics           float  \n",
       "pc11                                      Human Genetics           float  \n",
       "pc12                                      Human Genetics           float  \n",
       "pc13                                      Human Genetics           float  \n",
       "pc14                                      Human Genetics           float  \n",
       "pc15                                      Human Genetics           float  \n",
       "pc16                                      Human Genetics           float  \n",
       "pc17                                      Human Genetics           float  \n",
       "pc18                                      Human Genetics           float  \n",
       "pc19                                      Human Genetics           float  \n",
       "pc20                                      Human Genetics           float  \n",
       "used_in_pca_calculation                   Human Genetics            bool  \n",
       "pca_loadings_tsv                          Human Genetics        category  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pl.dict"
   ]
  }
 ],
 "metadata": {
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
