{
 "cells": [
  {
   "cell_type": "raw",
   "id": "dabbc44e",
   "metadata": {},
   "source": [
    "---\n",
    "output-file: 13-gut_microbiome.html\n",
    "title: 13-gut_microbiome\n",
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "044cd96d",
   "metadata": {},
   "source": [
    "# Gut microbiome dataset  \n",
    "\n",
    "Metagenomics is the study of genetic material from environmental samples, including microbial communities. It involves sequencing the DNA of all microorganisms in the sample, rather than isolating individual organisms. Metagenomics enables the identification and functional analysis of microorganisms in diverse environments, including soil, water, and the human body.\n",
    "\n",
    "This dataset maps out the human gut microbiota per participant via shotgun metagenomic sequencing given stool samples. It is then compared to known references of gut flora to measure prevalence of specific microbes.\n",
    "\n",
    "Gut microbiome metagenomics can be used to identify potential biomarkers of disease, develop personalized treatment strategies, and better understand the complex relationship between the gut microbiota and human physiology. It has already been providing insights into the role of gut microbes in various diseases such as obesity, diabetes and IBD.\n",
    "\n",
    "To measure the genetic makeup of the human gut flora given stool samples via metagenomics, the following steps done:\n",
    "\n",
    "1. Collection of stool sample: For each visit, a stool sample is collected from the individual and stored appropriately to preserve the microbial community.\n",
    "2. DNA extraction: DNA is extracted from the stool sample using specialized techniques to isolate the microbial DNA from other materials present in the sample.\n",
    "3. DNA fragmentation and sequencing: The extracted DNA is then fragmented into small pieces and sequenced using high-throughput sequencing technologies.\n",
    "4. Quality control: The resulting raw sequencing data is then pre-processed, removing low-quality reads and artifacts of the sequencing methodology.\n",
    "5. Taxonomic classification: The processed sequencing data is then compared to databases of known microbial sequences to identify and classify the microbial species presence and their respective abundances in the sample.\n",
    "\n",
    "### Data availability:\n",
    "The information is stored in multiple parquet files:\n",
    "\n",
    "- `metadata`: Sequencing and QC statistics.\n",
    "- `urs`: Segal Lab relative abundance.\n",
    "- `metaphlan_*`: 8 tables with MetaPhlAn 4 relative abundances, separated by taxonomic levels.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "739a66f4",
   "metadata": {},
   "outputs": [],
   "source": [
    "#| echo: false\n",
    "import pandas as pd\n",
    "pd.set_option(\"display.max_rows\", 500)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cb150bb3",
   "metadata": {},
   "outputs": [],
   "source": [
    "from pheno_utils import PhenoLoader"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "51cf9e4e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "DataLoader for gut_microbiome with\n",
       "45 fields\n",
       "2 tables: ['gut_microbiome', 'age_sex']"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dl = PhenoLoader('gut_microbiome')\n",
    "dl"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "1550c326",
   "metadata": {},
   "source": [
    "# Data dictionary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "a111e8c4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>field_string</th>\n",
       "      <th>description_string</th>\n",
       "      <th>parent_dataframe</th>\n",
       "      <th>relative_location</th>\n",
       "      <th>value_type</th>\n",
       "      <th>units</th>\n",
       "      <th>sampling_rate</th>\n",
       "      <th>item_type</th>\n",
       "      <th>array</th>\n",
       "      <th>cohorts</th>\n",
       "      <th>data_type</th>\n",
       "      <th>debut</th>\n",
       "      <th>pandas_dtype</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tabular_field_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>collection_timestamp</th>\n",
       "      <td>Sampled timestamp</td>\n",
       "      <td>Time sample was given</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Date</td>\n",
       "      <td>Time</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>datetime64[ns, Asia/Jerusalem]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>collection_date</th>\n",
       "      <td>Sampled date</td>\n",
       "      <td>Date sample was given</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Date</td>\n",
       "      <td>Time</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>datetime64[ns]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>timezone</th>\n",
       "      <td>Timezone</td>\n",
       "      <td>Timezone</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_name</th>\n",
       "      <td>Sample name</td>\n",
       "      <td>Sample Name</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>urs_abundance_parquet</th>\n",
       "      <td>URS abundances</td>\n",
       "      <td>Aggregated abundance (mapped to the URS refere...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Binary string</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Molecular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaphlan_abundance_kingdom_parquet</th>\n",
       "      <td>MetaPhlAn 4 abundance (kingdom)</td>\n",
       "      <td>Aggregated abundance (MetaPhlAn 4) at the king...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Binary string</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Molecular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaphlan_abundance_phylum_parquet</th>\n",
       "      <td>MetaPhlAn 4 abundance (phylum)</td>\n",
       "      <td>Aggregated abundance (MetaPhlAn 4) at the phyl...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Binary string</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Molecular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaphlan_abundance_class_parquet</th>\n",
       "      <td>MetaPhlAn 4 abundance (class)</td>\n",
       "      <td>Aggregated abundance (MetaPhlAn 4) at the clas...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Binary string</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Molecular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaphlan_abundance_order_parquet</th>\n",
       "      <td>MetaPhlAn 4 abundance (order)</td>\n",
       "      <td>Aggregated abundance (MetaPhlAn 4) at the orde...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Binary string</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Molecular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaphlan_abundance_family_parquet</th>\n",
       "      <td>MetaPhlAn 4 abundance (famiily)</td>\n",
       "      <td>Aggregated abundance (MetaPhlAn 4) at the fami...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Binary string</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Molecular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaphlan_abundance_genus_parquet</th>\n",
       "      <td>MetaPhlAn 4 abundance (genus)</td>\n",
       "      <td>Aggregated abundance (MetaPhlAn 4) at the genu...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Binary string</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Molecular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaphlan_abundance_species_parquet</th>\n",
       "      <td>MetaPhlAn 4 abundance (species)</td>\n",
       "      <td>Aggregated abundance (MetaPhlAn 4) at the spec...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Binary string</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Molecular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaphlan_abundance_strain_parquet</th>\n",
       "      <td>MetaPhlAn 4 abundance (strain)</td>\n",
       "      <td>Aggregated abundance (MetaPhlAn 4) at the stra...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Binary string</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Molecular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>raw_fastq</th>\n",
       "      <td>Raw FASTQ path</td>\n",
       "      <td>The original FASTQ file</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Binary string</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Molecular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>trimmed_fastq</th>\n",
       "      <td>Post QC FASTQ path</td>\n",
       "      <td>FASTQ file post Trimmomatic (removing adaptors...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Binary string</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Molecular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>non_human_fastq</th>\n",
       "      <td>Non-human FASTQ path</td>\n",
       "      <td>FASTQ file for non-human reads (Post Trimmomatic)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Binary string</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Molecular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fastqc_html</th>\n",
       "      <td>FASTQC HTML path</td>\n",
       "      <td>FASTQC Output for Post QC FASTQ - HTML</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Binary string</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fastqc_zip</th>\n",
       "      <td>FASTQC zip path</td>\n",
       "      <td>FASTQC Output for Post QC FASTQ - Zip</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Binary string</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>human_variants_bed</th>\n",
       "      <td>Human variants plink (BED)</td>\n",
       "      <td>Plink file containing all the human alleles ex...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Binary string</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Molecular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>human_variants_bim</th>\n",
       "      <td>Human variants plink (BIM)</td>\n",
       "      <td>Plink file containing all the human variants e...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Binary string</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Molecular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>human_variants_fam</th>\n",
       "      <td>Human variants plink (FAM)</td>\n",
       "      <td>Plink file containing only the specific sample</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Binary string</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Molecular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>urs_mapped_bam</th>\n",
       "      <td>URS mapped bam path</td>\n",
       "      <td>Reads aligned to the URS reference</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Binary string</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Molecular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>urs_mapped_single_bacteria_per_read_bam</th>\n",
       "      <td>URS mapped bam path (best match only)</td>\n",
       "      <td>Reads aligned to the URS reference, but only t...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Binary string</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Molecular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>urs_mapped_metrics_txt</th>\n",
       "      <td>URS Bowtie2 metric file path</td>\n",
       "      <td>Bowtie2 metric output for the URS alignment</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Binary string</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>urs_mapped_results_csv</th>\n",
       "      <td>URS abundances path</td>\n",
       "      <td>URS relative abundances</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Binary string</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>metaphlan4_results_txt</th>\n",
       "      <td>MetaPhlAn 4 abundance path</td>\n",
       "      <td>MetaPhlAn4 relative abundances</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Binary string</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>duplicate_reads_count</th>\n",
       "      <td>Duplicate reads count</td>\n",
       "      <td>#Reads that are exact duplciates</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Integer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>int</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>excessive_duplicate_reads</th>\n",
       "      <td>Excessive duplicate reads</td>\n",
       "      <td>Whether the sample contain an abnormal amount ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>raw_read_count</th>\n",
       "      <td>Raw reads count</td>\n",
       "      <td>#Reads in the original FASTQ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Integer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>int</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>trimmed_read_count</th>\n",
       "      <td>Post QC reads count</td>\n",
       "      <td>#Reads in the Post QC FASTQ</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Integer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>int</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>human_variant_count</th>\n",
       "      <td>Detectable human variants count</td>\n",
       "      <td>#Variants in the Plink files</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Integer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>int</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>human_read_count</th>\n",
       "      <td>Human reads count</td>\n",
       "      <td>#Reads successfuly aligned to humans</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Integer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>int</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>urs_mapped_read_count</th>\n",
       "      <td>URS mapped reads count</td>\n",
       "      <td>#Reads mapped to the URS reference panel (not ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Integer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>int</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>wgs_compare_error_total</th>\n",
       "      <td>WGS vs stool compare - #Errors</td>\n",
       "      <td>#Errors when comparing homozygous SNPs found i...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Integer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>int</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>wgs_compare_errors_percent</th>\n",
       "      <td>WGS vs stool compare - %Errors</td>\n",
       "      <td>%Errors when comparing homozygous SNPs found i...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Continuous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>float</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>wgs_compare_count_homozygous</th>\n",
       "      <td>WGS vs stool compare - #Homozygous SNPs</td>\n",
       "      <td>#homozygous SNPs found in stool vs WGS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Integer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>int</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>wgs_compare_count_variants</th>\n",
       "      <td>WGS vs stool compare - #SNPs</td>\n",
       "      <td>#SNPs found in stool vs WGS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Integer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>int</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>wgs_compare_pass</th>\n",
       "      <td>WGS vs stool compare - Passed Validation</td>\n",
       "      <td>Passed Stool Vs WGS validation</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>raw_read_length</th>\n",
       "      <td>Legnth of raw reads</td>\n",
       "      <td>Length of reads (from the Raw FASTQ file)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Integer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>int</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>single_ended</th>\n",
       "      <td>Is single ended</td>\n",
       "      <td>Single ended (SE) or pair ended (PE)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>is_genotek</th>\n",
       "      <td>Vial type</td>\n",
       "      <td>Type of vial used for stool collecting</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Boolean</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>run_name</th>\n",
       "      <td>Batch name</td>\n",
       "      <td>The run-name of the sample's batch(s)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>wgs_dna_code</th>\n",
       "      <td>WGS DNA code</td>\n",
       "      <td>WGS DNA code</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/gut_microbiome/gut_microbiome.parquet</td>\n",
       "      <td>Integer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Metadata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>int</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                     field_string  \\\n",
       "tabular_field_name                                                                  \n",
       "collection_timestamp                                            Sampled timestamp   \n",
       "collection_date                                                      Sampled date   \n",
       "timezone                                                                 Timezone   \n",
       "sample_name                                                           Sample name   \n",
       "urs_abundance_parquet                                              URS abundances   \n",
       "metaphlan_abundance_kingdom_parquet               MetaPhlAn 4 abundance (kingdom)   \n",
       "metaphlan_abundance_phylum_parquet                 MetaPhlAn 4 abundance (phylum)   \n",
       "metaphlan_abundance_class_parquet                   MetaPhlAn 4 abundance (class)   \n",
       "metaphlan_abundance_order_parquet                   MetaPhlAn 4 abundance (order)   \n",
       "metaphlan_abundance_family_parquet                MetaPhlAn 4 abundance (famiily)   \n",
       "metaphlan_abundance_genus_parquet                   MetaPhlAn 4 abundance (genus)   \n",
       "metaphlan_abundance_species_parquet               MetaPhlAn 4 abundance (species)   \n",
       "metaphlan_abundance_strain_parquet                 MetaPhlAn 4 abundance (strain)   \n",
       "raw_fastq                                                          Raw FASTQ path   \n",
       "trimmed_fastq                                                  Post QC FASTQ path   \n",
       "non_human_fastq                                              Non-human FASTQ path   \n",
       "fastqc_html                                                      FASTQC HTML path   \n",
       "fastqc_zip                                                        FASTQC zip path   \n",
       "human_variants_bed                                     Human variants plink (BED)   \n",
       "human_variants_bim                                     Human variants plink (BIM)   \n",
       "human_variants_fam                                     Human variants plink (FAM)   \n",
       "urs_mapped_bam                                                URS mapped bam path   \n",
       "urs_mapped_single_bacteria_per_read_bam     URS mapped bam path (best match only)   \n",
       "urs_mapped_metrics_txt                               URS Bowtie2 metric file path   \n",
       "urs_mapped_results_csv                                        URS abundances path   \n",
       "metaphlan4_results_txt                                 MetaPhlAn 4 abundance path   \n",
       "duplicate_reads_count                                       Duplicate reads count   \n",
       "excessive_duplicate_reads                               Excessive duplicate reads   \n",
       "raw_read_count                                                    Raw reads count   \n",
       "trimmed_read_count                                            Post QC reads count   \n",
       "human_variant_count                             Detectable human variants count     \n",
       "human_read_count                                                Human reads count   \n",
       "urs_mapped_read_count                                      URS mapped reads count   \n",
       "wgs_compare_error_total                           WGS vs stool compare - #Errors    \n",
       "wgs_compare_errors_percent                        WGS vs stool compare - %Errors    \n",
       "wgs_compare_count_homozygous             WGS vs stool compare - #Homozygous SNPs    \n",
       "wgs_compare_count_variants                           WGS vs stool compare - #SNPs   \n",
       "wgs_compare_pass                         WGS vs stool compare - Passed Validation   \n",
       "raw_read_length                                               Legnth of raw reads   \n",
       "single_ended                                                      Is single ended   \n",
       "is_genotek                                                              Vial type   \n",
       "run_name                                                               Batch name   \n",
       "wgs_dna_code                                                         WGS DNA code   \n",
       "\n",
       "                                                                        description_string  \\\n",
       "tabular_field_name                                                                           \n",
       "collection_timestamp                                                 Time sample was given   \n",
       "collection_date                                                      Date sample was given   \n",
       "timezone                                                                          Timezone   \n",
       "sample_name                                                                    Sample Name   \n",
       "urs_abundance_parquet                    Aggregated abundance (mapped to the URS refere...   \n",
       "metaphlan_abundance_kingdom_parquet      Aggregated abundance (MetaPhlAn 4) at the king...   \n",
       "metaphlan_abundance_phylum_parquet       Aggregated abundance (MetaPhlAn 4) at the phyl...   \n",
       "metaphlan_abundance_class_parquet        Aggregated abundance (MetaPhlAn 4) at the clas...   \n",
       "metaphlan_abundance_order_parquet        Aggregated abundance (MetaPhlAn 4) at the orde...   \n",
       "metaphlan_abundance_family_parquet       Aggregated abundance (MetaPhlAn 4) at the fami...   \n",
       "metaphlan_abundance_genus_parquet        Aggregated abundance (MetaPhlAn 4) at the genu...   \n",
       "metaphlan_abundance_species_parquet      Aggregated abundance (MetaPhlAn 4) at the spec...   \n",
       "metaphlan_abundance_strain_parquet       Aggregated abundance (MetaPhlAn 4) at the stra...   \n",
       "raw_fastq                                                          The original FASTQ file   \n",
       "trimmed_fastq                            FASTQ file post Trimmomatic (removing adaptors...   \n",
       "non_human_fastq                          FASTQ file for non-human reads (Post Trimmomatic)   \n",
       "fastqc_html                                         FASTQC Output for Post QC FASTQ - HTML   \n",
       "fastqc_zip                                           FASTQC Output for Post QC FASTQ - Zip   \n",
       "human_variants_bed                       Plink file containing all the human alleles ex...   \n",
       "human_variants_bim                       Plink file containing all the human variants e...   \n",
       "human_variants_fam                          Plink file containing only the specific sample   \n",
       "urs_mapped_bam                                          Reads aligned to the URS reference   \n",
       "urs_mapped_single_bacteria_per_read_bam  Reads aligned to the URS reference, but only t...   \n",
       "urs_mapped_metrics_txt                         Bowtie2 metric output for the URS alignment   \n",
       "urs_mapped_results_csv                                             URS relative abundances   \n",
       "metaphlan4_results_txt                                      MetaPhlAn4 relative abundances   \n",
       "duplicate_reads_count                                     #Reads that are exact duplciates   \n",
       "excessive_duplicate_reads                Whether the sample contain an abnormal amount ...   \n",
       "raw_read_count                                                #Reads in the original FASTQ   \n",
       "trimmed_read_count                                             #Reads in the Post QC FASTQ   \n",
       "human_variant_count                                           #Variants in the Plink files   \n",
       "human_read_count                                      #Reads successfuly aligned to humans   \n",
       "urs_mapped_read_count                    #Reads mapped to the URS reference panel (not ...   \n",
       "wgs_compare_error_total                  #Errors when comparing homozygous SNPs found i...   \n",
       "wgs_compare_errors_percent               %Errors when comparing homozygous SNPs found i...   \n",
       "wgs_compare_count_homozygous                        #homozygous SNPs found in stool vs WGS   \n",
       "wgs_compare_count_variants                                     #SNPs found in stool vs WGS   \n",
       "wgs_compare_pass                                            Passed Stool Vs WGS validation   \n",
       "raw_read_length                                  Length of reads (from the Raw FASTQ file)   \n",
       "single_ended                                          Single ended (SE) or pair ended (PE)   \n",
       "is_genotek                                          Type of vial used for stool collecting   \n",
       "run_name                                             The run-name of the sample's batch(s)   \n",
       "wgs_dna_code                                                                  WGS DNA code   \n",
       "\n",
       "                                         parent_dataframe  \\\n",
       "tabular_field_name                                          \n",
       "collection_timestamp                                  NaN   \n",
       "collection_date                                       NaN   \n",
       "timezone                                              NaN   \n",
       "sample_name                                           NaN   \n",
       "urs_abundance_parquet                                 NaN   \n",
       "metaphlan_abundance_kingdom_parquet                   NaN   \n",
       "metaphlan_abundance_phylum_parquet                    NaN   \n",
       "metaphlan_abundance_class_parquet                     NaN   \n",
       "metaphlan_abundance_order_parquet                     NaN   \n",
       "metaphlan_abundance_family_parquet                    NaN   \n",
       "metaphlan_abundance_genus_parquet                     NaN   \n",
       "metaphlan_abundance_species_parquet                   NaN   \n",
       "metaphlan_abundance_strain_parquet                    NaN   \n",
       "raw_fastq                                             NaN   \n",
       "trimmed_fastq                                         NaN   \n",
       "non_human_fastq                                       NaN   \n",
       "fastqc_html                                           NaN   \n",
       "fastqc_zip                                            NaN   \n",
       "human_variants_bed                                    NaN   \n",
       "human_variants_bim                                    NaN   \n",
       "human_variants_fam                                    NaN   \n",
       "urs_mapped_bam                                        NaN   \n",
       "urs_mapped_single_bacteria_per_read_bam               NaN   \n",
       "urs_mapped_metrics_txt                                NaN   \n",
       "urs_mapped_results_csv                                NaN   \n",
       "metaphlan4_results_txt                                NaN   \n",
       "duplicate_reads_count                                 NaN   \n",
       "excessive_duplicate_reads                             NaN   \n",
       "raw_read_count                                        NaN   \n",
       "trimmed_read_count                                    NaN   \n",
       "human_variant_count                                   NaN   \n",
       "human_read_count                                      NaN   \n",
       "urs_mapped_read_count                                 NaN   \n",
       "wgs_compare_error_total                               NaN   \n",
       "wgs_compare_errors_percent                            NaN   \n",
       "wgs_compare_count_homozygous                          NaN   \n",
       "wgs_compare_count_variants                            NaN   \n",
       "wgs_compare_pass                                      NaN   \n",
       "raw_read_length                                       NaN   \n",
       "single_ended                                          NaN   \n",
       "is_genotek                                            NaN   \n",
       "run_name                                              NaN   \n",
       "wgs_dna_code                                          NaN   \n",
       "\n",
       "                                                              relative_location  \\\n",
       "tabular_field_name                                                                \n",
       "collection_timestamp                     /gut_microbiome/gut_microbiome.parquet   \n",
       "collection_date                          /gut_microbiome/gut_microbiome.parquet   \n",
       "timezone                                 /gut_microbiome/gut_microbiome.parquet   \n",
       "sample_name                              /gut_microbiome/gut_microbiome.parquet   \n",
       "urs_abundance_parquet                    /gut_microbiome/gut_microbiome.parquet   \n",
       "metaphlan_abundance_kingdom_parquet      /gut_microbiome/gut_microbiome.parquet   \n",
       "metaphlan_abundance_phylum_parquet       /gut_microbiome/gut_microbiome.parquet   \n",
       "metaphlan_abundance_class_parquet        /gut_microbiome/gut_microbiome.parquet   \n",
       "metaphlan_abundance_order_parquet        /gut_microbiome/gut_microbiome.parquet   \n",
       "metaphlan_abundance_family_parquet       /gut_microbiome/gut_microbiome.parquet   \n",
       "metaphlan_abundance_genus_parquet        /gut_microbiome/gut_microbiome.parquet   \n",
       "metaphlan_abundance_species_parquet      /gut_microbiome/gut_microbiome.parquet   \n",
       "metaphlan_abundance_strain_parquet       /gut_microbiome/gut_microbiome.parquet   \n",
       "raw_fastq                                /gut_microbiome/gut_microbiome.parquet   \n",
       "trimmed_fastq                            /gut_microbiome/gut_microbiome.parquet   \n",
       "non_human_fastq                          /gut_microbiome/gut_microbiome.parquet   \n",
       "fastqc_html                              /gut_microbiome/gut_microbiome.parquet   \n",
       "fastqc_zip                               /gut_microbiome/gut_microbiome.parquet   \n",
       "human_variants_bed                       /gut_microbiome/gut_microbiome.parquet   \n",
       "human_variants_bim                       /gut_microbiome/gut_microbiome.parquet   \n",
       "human_variants_fam                       /gut_microbiome/gut_microbiome.parquet   \n",
       "urs_mapped_bam                           /gut_microbiome/gut_microbiome.parquet   \n",
       "urs_mapped_single_bacteria_per_read_bam  /gut_microbiome/gut_microbiome.parquet   \n",
       "urs_mapped_metrics_txt                   /gut_microbiome/gut_microbiome.parquet   \n",
       "urs_mapped_results_csv                   /gut_microbiome/gut_microbiome.parquet   \n",
       "metaphlan4_results_txt                   /gut_microbiome/gut_microbiome.parquet   \n",
       "duplicate_reads_count                    /gut_microbiome/gut_microbiome.parquet   \n",
       "excessive_duplicate_reads                /gut_microbiome/gut_microbiome.parquet   \n",
       "raw_read_count                           /gut_microbiome/gut_microbiome.parquet   \n",
       "trimmed_read_count                       /gut_microbiome/gut_microbiome.parquet   \n",
       "human_variant_count                      /gut_microbiome/gut_microbiome.parquet   \n",
       "human_read_count                         /gut_microbiome/gut_microbiome.parquet   \n",
       "urs_mapped_read_count                    /gut_microbiome/gut_microbiome.parquet   \n",
       "wgs_compare_error_total                  /gut_microbiome/gut_microbiome.parquet   \n",
       "wgs_compare_errors_percent               /gut_microbiome/gut_microbiome.parquet   \n",
       "wgs_compare_count_homozygous             /gut_microbiome/gut_microbiome.parquet   \n",
       "wgs_compare_count_variants               /gut_microbiome/gut_microbiome.parquet   \n",
       "wgs_compare_pass                         /gut_microbiome/gut_microbiome.parquet   \n",
       "raw_read_length                          /gut_microbiome/gut_microbiome.parquet   \n",
       "single_ended                             /gut_microbiome/gut_microbiome.parquet   \n",
       "is_genotek                               /gut_microbiome/gut_microbiome.parquet   \n",
       "run_name                                 /gut_microbiome/gut_microbiome.parquet   \n",
       "wgs_dna_code                             /gut_microbiome/gut_microbiome.parquet   \n",
       "\n",
       "                                                   value_type units  \\\n",
       "tabular_field_name                                                    \n",
       "collection_timestamp                                     Date  Time   \n",
       "collection_date                                          Date  Time   \n",
       "timezone                                                 Text   NaN   \n",
       "sample_name                                              Text   NaN   \n",
       "urs_abundance_parquet                           Binary string   NaN   \n",
       "metaphlan_abundance_kingdom_parquet             Binary string   NaN   \n",
       "metaphlan_abundance_phylum_parquet              Binary string   NaN   \n",
       "metaphlan_abundance_class_parquet               Binary string   NaN   \n",
       "metaphlan_abundance_order_parquet               Binary string   NaN   \n",
       "metaphlan_abundance_family_parquet              Binary string   NaN   \n",
       "metaphlan_abundance_genus_parquet               Binary string   NaN   \n",
       "metaphlan_abundance_species_parquet             Binary string   NaN   \n",
       "metaphlan_abundance_strain_parquet              Binary string   NaN   \n",
       "raw_fastq                                       Binary string   NaN   \n",
       "trimmed_fastq                                   Binary string   NaN   \n",
       "non_human_fastq                                 Binary string   NaN   \n",
       "fastqc_html                                     Binary string   NaN   \n",
       "fastqc_zip                                      Binary string   NaN   \n",
       "human_variants_bed                              Binary string   NaN   \n",
       "human_variants_bim                              Binary string   NaN   \n",
       "human_variants_fam                              Binary string   NaN   \n",
       "urs_mapped_bam                                  Binary string   NaN   \n",
       "urs_mapped_single_bacteria_per_read_bam         Binary string   NaN   \n",
       "urs_mapped_metrics_txt                          Binary string   NaN   \n",
       "urs_mapped_results_csv                          Binary string   NaN   \n",
       "metaphlan4_results_txt                          Binary string   NaN   \n",
       "duplicate_reads_count                                 Integer   NaN   \n",
       "excessive_duplicate_reads                             Boolean   NaN   \n",
       "raw_read_count                                        Integer   NaN   \n",
       "trimmed_read_count                                    Integer   NaN   \n",
       "human_variant_count                                   Integer   NaN   \n",
       "human_read_count                                      Integer   NaN   \n",
       "urs_mapped_read_count                                 Integer   NaN   \n",
       "wgs_compare_error_total                               Integer   NaN   \n",
       "wgs_compare_errors_percent                         Continuous   NaN   \n",
       "wgs_compare_count_homozygous                          Integer   NaN   \n",
       "wgs_compare_count_variants                            Integer   NaN   \n",
       "wgs_compare_pass                         Categorical (single)   NaN   \n",
       "raw_read_length                                       Integer   NaN   \n",
       "single_ended                                          Boolean   NaN   \n",
       "is_genotek                                            Boolean   NaN   \n",
       "run_name                                                 Text   NaN   \n",
       "wgs_dna_code                                          Integer   NaN   \n",
       "\n",
       "                                         sampling_rate item_type   array  \\\n",
       "tabular_field_name                                                         \n",
       "collection_timestamp                               NaN      Data  Single   \n",
       "collection_date                                    NaN      Data  Single   \n",
       "timezone                                           NaN      Data  Single   \n",
       "sample_name                                        NaN      Data  Single   \n",
       "urs_abundance_parquet                              NaN      Bulk  Single   \n",
       "metaphlan_abundance_kingdom_parquet                NaN      Bulk  Single   \n",
       "metaphlan_abundance_phylum_parquet                 NaN      Bulk  Single   \n",
       "metaphlan_abundance_class_parquet                  NaN      Bulk  Single   \n",
       "metaphlan_abundance_order_parquet                  NaN      Bulk  Single   \n",
       "metaphlan_abundance_family_parquet                 NaN      Bulk  Single   \n",
       "metaphlan_abundance_genus_parquet                  NaN      Bulk  Single   \n",
       "metaphlan_abundance_species_parquet                NaN      Bulk  Single   \n",
       "metaphlan_abundance_strain_parquet                 NaN      Bulk  Single   \n",
       "raw_fastq                                          NaN      Bulk  Single   \n",
       "trimmed_fastq                                      NaN      Bulk  Single   \n",
       "non_human_fastq                                    NaN      Bulk  Single   \n",
       "fastqc_html                                        NaN      Bulk  Single   \n",
       "fastqc_zip                                         NaN      Bulk  Single   \n",
       "human_variants_bed                                 NaN      Bulk  Single   \n",
       "human_variants_bim                                 NaN      Bulk  Single   \n",
       "human_variants_fam                                 NaN      Bulk  Single   \n",
       "urs_mapped_bam                                     NaN      Bulk  Single   \n",
       "urs_mapped_single_bacteria_per_read_bam            NaN      Bulk  Single   \n",
       "urs_mapped_metrics_txt                             NaN      Bulk  Single   \n",
       "urs_mapped_results_csv                             NaN      Bulk  Single   \n",
       "metaphlan4_results_txt                             NaN      Bulk  Single   \n",
       "duplicate_reads_count                              NaN      Data  Single   \n",
       "excessive_duplicate_reads                          NaN      Data  Single   \n",
       "raw_read_count                                     NaN      Data  Single   \n",
       "trimmed_read_count                                 NaN      Data  Single   \n",
       "human_variant_count                                NaN      Data  Single   \n",
       "human_read_count                                   NaN      Data  Single   \n",
       "urs_mapped_read_count                              NaN      Data  Single   \n",
       "wgs_compare_error_total                            NaN      Data  Single   \n",
       "wgs_compare_errors_percent                         NaN      Data  Single   \n",
       "wgs_compare_count_homozygous                       NaN      Data  Single   \n",
       "wgs_compare_count_variants                         NaN      Data  Single   \n",
       "wgs_compare_pass                                   NaN      Data  Single   \n",
       "raw_read_length                                    NaN      Data  Single   \n",
       "single_ended                                       NaN      Data  Single   \n",
       "is_genotek                                         NaN      Data  Single   \n",
       "run_name                                           NaN      Data  Single   \n",
       "wgs_dna_code                                       NaN      Data  Single   \n",
       "\n",
       "                                        cohorts  data_type  debut  \\\n",
       "tabular_field_name                                                  \n",
       "collection_timestamp                        10k   Metadata    NaN   \n",
       "collection_date                             10k   Metadata    NaN   \n",
       "timezone                                    10k   Metadata    NaN   \n",
       "sample_name                                 10k   Metadata    NaN   \n",
       "urs_abundance_parquet                       10k  Molecular    NaN   \n",
       "metaphlan_abundance_kingdom_parquet         10k  Molecular    NaN   \n",
       "metaphlan_abundance_phylum_parquet          10k  Molecular    NaN   \n",
       "metaphlan_abundance_class_parquet           10k  Molecular    NaN   \n",
       "metaphlan_abundance_order_parquet           10k  Molecular    NaN   \n",
       "metaphlan_abundance_family_parquet          10k  Molecular    NaN   \n",
       "metaphlan_abundance_genus_parquet           10k  Molecular    NaN   \n",
       "metaphlan_abundance_species_parquet         10k  Molecular    NaN   \n",
       "metaphlan_abundance_strain_parquet          10k  Molecular    NaN   \n",
       "raw_fastq                                   10k  Molecular    NaN   \n",
       "trimmed_fastq                               10k  Molecular    NaN   \n",
       "non_human_fastq                             10k  Molecular    NaN   \n",
       "fastqc_html                                 10k   Metadata    NaN   \n",
       "fastqc_zip                                  10k   Metadata    NaN   \n",
       "human_variants_bed                          10k  Molecular    NaN   \n",
       "human_variants_bim                          10k  Molecular    NaN   \n",
       "human_variants_fam                          10k  Molecular    NaN   \n",
       "urs_mapped_bam                              10k  Molecular    NaN   \n",
       "urs_mapped_single_bacteria_per_read_bam     10k  Molecular    NaN   \n",
       "urs_mapped_metrics_txt                      10k   Metadata    NaN   \n",
       "urs_mapped_results_csv                      10k    Tabular    NaN   \n",
       "metaphlan4_results_txt                      10k    Tabular    NaN   \n",
       "duplicate_reads_count                       10k   Metadata    NaN   \n",
       "excessive_duplicate_reads                   10k   Metadata    NaN   \n",
       "raw_read_count                              10k   Metadata    NaN   \n",
       "trimmed_read_count                          10k   Metadata    NaN   \n",
       "human_variant_count                         10k   Metadata    NaN   \n",
       "human_read_count                            10k   Metadata    NaN   \n",
       "urs_mapped_read_count                       10k   Metadata    NaN   \n",
       "wgs_compare_error_total                     10k   Metadata    NaN   \n",
       "wgs_compare_errors_percent                  10k   Metadata    NaN   \n",
       "wgs_compare_count_homozygous                10k   Metadata    NaN   \n",
       "wgs_compare_count_variants                  10k   Metadata    NaN   \n",
       "wgs_compare_pass                            10k   Metadata    NaN   \n",
       "raw_read_length                             10k   Metadata    NaN   \n",
       "single_ended                                10k   Metadata    NaN   \n",
       "is_genotek                                  10k   Metadata    NaN   \n",
       "run_name                                    10k   Metadata    NaN   \n",
       "wgs_dna_code                                10k   Metadata    NaN   \n",
       "\n",
       "                                                           pandas_dtype  \n",
       "tabular_field_name                                                       \n",
       "collection_timestamp                     datetime64[ns, Asia/Jerusalem]  \n",
       "collection_date                                          datetime64[ns]  \n",
       "timezone                                                         string  \n",
       "sample_name                                                      string  \n",
       "urs_abundance_parquet                                          category  \n",
       "metaphlan_abundance_kingdom_parquet                            category  \n",
       "metaphlan_abundance_phylum_parquet                             category  \n",
       "metaphlan_abundance_class_parquet                              category  \n",
       "metaphlan_abundance_order_parquet                              category  \n",
       "metaphlan_abundance_family_parquet                             category  \n",
       "metaphlan_abundance_genus_parquet                              category  \n",
       "metaphlan_abundance_species_parquet                            category  \n",
       "metaphlan_abundance_strain_parquet                             category  \n",
       "raw_fastq                                                        string  \n",
       "trimmed_fastq                                                    string  \n",
       "non_human_fastq                                                  string  \n",
       "fastqc_html                                                      string  \n",
       "fastqc_zip                                                       string  \n",
       "human_variants_bed                                               string  \n",
       "human_variants_bim                                               string  \n",
       "human_variants_fam                                               string  \n",
       "urs_mapped_bam                                                   string  \n",
       "urs_mapped_single_bacteria_per_read_bam                          string  \n",
       "urs_mapped_metrics_txt                                           string  \n",
       "urs_mapped_results_csv                                           string  \n",
       "metaphlan4_results_txt                                           string  \n",
       "duplicate_reads_count                                               int  \n",
       "excessive_duplicate_reads                                          bool  \n",
       "raw_read_count                                                      int  \n",
       "trimmed_read_count                                                  int  \n",
       "human_variant_count                                                 int  \n",
       "human_read_count                                                    int  \n",
       "urs_mapped_read_count                                               int  \n",
       "wgs_compare_error_total                                             int  \n",
       "wgs_compare_errors_percent                                        float  \n",
       "wgs_compare_count_homozygous                                        int  \n",
       "wgs_compare_count_variants                                          int  \n",
       "wgs_compare_pass                                               category  \n",
       "raw_read_length                                                     int  \n",
       "single_ended                                                       bool  \n",
       "is_genotek                                                         bool  \n",
       "run_name                                                         string  \n",
       "wgs_dna_code                                                        int  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dl.dict"
   ]
  }
 ],
 "metadata": {},
 "nbformat": 4,
 "nbformat_minor": 5
}
