{
 "cells": [
  {
   "cell_type": "raw",
   "id": "7e09670c",
   "metadata": {},
   "source": [
    "---\n",
    "output-file: 14-human_genetics.html\n",
    "title: 14-human_genetics\n",
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4bf7574c",
   "metadata": {},
   "source": [
    "# Human genetics (Gencove) \n",
    "\n",
    "The Human Phenotype Project collects genomic variation data on all its participants. The genomic data together with the Human Phenotype Project  deep-phenotypes allows to investigate the progression of disease, and to explore personalized treatments. We genotype millions of positions by low-pass sequencing combined with imputation using gencove platform technologies. Genotype imputation is a process of statistically inferring unobserved genotypes using known haplotypes in a population. The performance of Gencove genotype imputation is very high ( accuracy > 98% ) (Wasik et al. 2021). \n",
    "\n",
    "### Data availability:\n",
    "The information is stored in a number of statistics parquet files:\n",
    "- `main.parquet`: sample metadata, including QC statistics, paths to PLINK variant files (raw and post-QC), and principal components (PCs).\n",
    "- `variant_qc.parquet`: variant QC statistics.\n",
    "- `relatives/plink_ibs.parquet`: IBS calculated by PLINK for pairs of participants.\n",
    "- `relatives/king_kinship.parquet`: King kinship coefficients for pairs of participants.\n",
    "\n",
    "And a PLINK text file `pca/eigenvec.var` containing the principal component loadings."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ab2710ac",
   "metadata": {},
   "outputs": [],
   "source": [
    "#| echo: false\n",
    "import pandas as pd\n",
    "pd.set_option(\"display.max_rows\", 500)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a92edbcb",
   "metadata": {},
   "outputs": [],
   "source": [
    "from pheno_utils import PhenoLoader"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "51cf9e4e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "DataLoader for human_genetics with\n",
       "190 fields\n",
       "3 tables: ['human_genetics', 'pca', 'age_sex']"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# We load the data but:\n",
    "# - Skip the king matrix (large and slow to load)\n",
    "# - Skip the eigenvectors of the PCA (PLINK output, not a parquet file)\n",
    "\n",
    "dl = PhenoLoader('human_genetics')\n",
    "dl"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "8ba8c3c4",
   "metadata": {},
   "source": [
    "# Data dictionary\n",
    "For some datasets, we can also take a look at the data dictionary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "06436c04",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>field_string</th>\n",
       "      <th>description_string</th>\n",
       "      <th>parent_dataframe</th>\n",
       "      <th>relative_location</th>\n",
       "      <th>value_type</th>\n",
       "      <th>units</th>\n",
       "      <th>sampling_rate</th>\n",
       "      <th>item_type</th>\n",
       "      <th>array</th>\n",
       "      <th>cohorts</th>\n",
       "      <th>data_type</th>\n",
       "      <th>debut</th>\n",
       "      <th>pandas_dtype</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tabular_field_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>collection_date</th>\n",
       "      <td>Collection date</td>\n",
       "      <td>The date of downloading Gencove results from t...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/human_genetics/human_genetics.parquet</td>\n",
       "      <td>Time</td>\n",
       "      <td>Time</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>datetime64[ns]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>version</th>\n",
       "      <td>Gencove version</td>\n",
       "      <td>Gencove API version 1 or 2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/human_genetics/human_genetics.parquet</td>\n",
       "      <td>Categorical (single)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Data</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Tabular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>category</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gencove_fastq_r1</th>\n",
       "      <td>R1_FASTQ</td>\n",
       "      <td>Per sample FASTQ file, a text file that contai...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/human_genetics/human_genetics.parquet</td>\n",
       "      <td>String</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Text file</td>\n",
       "      <td>NaN</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gencove_fastq_r2</th>\n",
       "      <td>R2_FASTQ</td>\n",
       "      <td>Per sample FASTQ file, a text file that contai...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/human_genetics/human_genetics.parquet</td>\n",
       "      <td>String</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Text file</td>\n",
       "      <td>NaN</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gencove_bam</th>\n",
       "      <td>BAM</td>\n",
       "      <td>Per sample Binary Alignment Map (BAM) file for...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/human_genetics/human_genetics.parquet</td>\n",
       "      <td>String</td>\n",
       "      <td>Text</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bulk</td>\n",
       "      <td>Single</td>\n",
       "      <td>10k</td>\n",
       "      <td>Compressed binary file</td>\n",
       "      <td>NaN</td>\n",
       "      <td>string</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       field_string  \\\n",
       "tabular_field_name                    \n",
       "collection_date     Collection date   \n",
       "version             Gencove version   \n",
       "gencove_fastq_r1           R1_FASTQ   \n",
       "gencove_fastq_r2           R2_FASTQ   \n",
       "gencove_bam                     BAM   \n",
       "\n",
       "                                                   description_string  \\\n",
       "tabular_field_name                                                      \n",
       "collection_date     The date of downloading Gencove results from t...   \n",
       "version                                    Gencove API version 1 or 2   \n",
       "gencove_fastq_r1    Per sample FASTQ file, a text file that contai...   \n",
       "gencove_fastq_r2    Per sample FASTQ file, a text file that contai...   \n",
       "gencove_bam         Per sample Binary Alignment Map (BAM) file for...   \n",
       "\n",
       "                    parent_dataframe                       relative_location  \\\n",
       "tabular_field_name                                                             \n",
       "collection_date                  NaN  /human_genetics/human_genetics.parquet   \n",
       "version                          NaN  /human_genetics/human_genetics.parquet   \n",
       "gencove_fastq_r1                 NaN  /human_genetics/human_genetics.parquet   \n",
       "gencove_fastq_r2                 NaN  /human_genetics/human_genetics.parquet   \n",
       "gencove_bam                      NaN  /human_genetics/human_genetics.parquet   \n",
       "\n",
       "                              value_type units  sampling_rate item_type  \\\n",
       "tabular_field_name                                                        \n",
       "collection_date                     Time  Time            NaN      Data   \n",
       "version             Categorical (single)   NaN            NaN      Data   \n",
       "gencove_fastq_r1                  String  Text            NaN      Bulk   \n",
       "gencove_fastq_r2                  String  Text            NaN      Bulk   \n",
       "gencove_bam                       String  Text            NaN      Bulk   \n",
       "\n",
       "                     array cohorts               data_type  debut  \\\n",
       "tabular_field_name                                                  \n",
       "collection_date     Single     10k                 Tabular    NaN   \n",
       "version             Single     10k                 Tabular    NaN   \n",
       "gencove_fastq_r1    Single     10k               Text file    NaN   \n",
       "gencove_fastq_r2    Single     10k               Text file    NaN   \n",
       "gencove_bam         Single     10k  Compressed binary file    NaN   \n",
       "\n",
       "                      pandas_dtype  \n",
       "tabular_field_name                  \n",
       "collection_date     datetime64[ns]  \n",
       "version                   category  \n",
       "gencove_fastq_r1            string  \n",
       "gencove_fastq_r2            string  \n",
       "gencove_bam                 string  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dl.dict.head()"
   ]
  }
 ],
 "metadata": {},
 "nbformat": 4,
 "nbformat_minor": 5
}
