{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import dask.dataframe as dd\n",
    "from collections import defaultdict\n",
    "import traceback\n",
    "import os\n",
    "import json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "BEFORE_2023 = True\n",
    "SHOW_FEATURE_SETS = False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# path to datasets\n",
    "ver = \"v12\"\n",
    "# count_bucket_name = \"pheno-master-data-collection-v1.2-stage-eu-west-1\"\n",
    "# count_bucket_name = \"pheno-master-data-collection-v1-prod-eu-west-1\"\n",
    "# count_bucket_name = \"pheno-academic-collection-v1-prod-eu-west-1\"\n",
    "# count_bucket_name = \"pheno-academic-collection-v1-prod-us-east-1\"\n",
    "# count_bucket_name = \"pheno-master-data-collection-v1.2-prod-eu-west-1\"\n",
    "count_bucket_name = \"pheno-academic-collection-v1.2-prod-eu-west-1\"\n",
    "master_path = f\"s3://{count_bucket_name}\"\n",
    "output_path = f's3://ds-users/mariag/data_counts/{count_bucket_name}_{ver}_before_2023_{BEFORE_2023}_show_feature_{SHOW_FEATURE_SETS}.csv'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset_id</th>\n",
       "      <th>dataset</th>\n",
       "      <th>feature_set</th>\n",
       "      <th>Data owner</th>\n",
       "      <th>status</th>\n",
       "      <th>pheno-big-data-prod-us-east-1_relative_path</th>\n",
       "      <th>contains_sensitive</th>\n",
       "      <th>notes</th>\n",
       "      <th>count_parquet_relative_path</th>\n",
       "      <th>count_field_name</th>\n",
       "      <th>data points</th>\n",
       "      <th>participants</th>\n",
       "      <th>first_date</th>\n",
       "      <th>last_date</th>\n",
       "      <th>dict_check</th>\n",
       "      <th>demo</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>population</td>\n",
       "      <td>population</td>\n",
       "      <td>Maria</td>\n",
       "      <td>DS Owner Finished for v1.2</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>/population/population.parquet</td>\n",
       "      <td>sex</td>\n",
       "      <td>11179</td>\n",
       "      <td>11179</td>\n",
       "      <td>2018-12-05</td>\n",
       "      <td>2023-03-19</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>events</td>\n",
       "      <td>events</td>\n",
       "      <td>Maria</td>\n",
       "      <td>DS Owner Finished for v1.2</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>/events/events.parquet</td>\n",
       "      <td>research_stage_date</td>\n",
       "      <td>23300</td>\n",
       "      <td>11179</td>\n",
       "      <td>2018-12-05</td>\n",
       "      <td>2023-03-19</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>anthropometrics</td>\n",
       "      <td>anthropometrics</td>\n",
       "      <td>Adam</td>\n",
       "      <td>DS Owner Finished for v1.2</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>/anthropometrics/anthropometrics.parquet</td>\n",
       "      <td>bmi</td>\n",
       "      <td>13568</td>\n",
       "      <td>10781</td>\n",
       "      <td>2018-11-22</td>\n",
       "      <td>2023-02-07</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>fundus</td>\n",
       "      <td>fundus</td>\n",
       "      <td>Hagai</td>\n",
       "      <td>DS Owner Finished for v1.2</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>/fundus/fundus.parquet</td>\n",
       "      <td>fundus_image_left, fundus_image_right</td>\n",
       "      <td>7079</td>\n",
       "      <td>7049</td>\n",
       "      <td>2021-02-17</td>\n",
       "      <td>2022-12-04</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3</td>\n",
       "      <td>fundus</td>\n",
       "      <td>microvasculature</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>DS Owner Finished for v1.2</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>/fundus/microvasculature.parquet</td>\n",
       "      <td>vessel_density_left, vessel_density_right</td>\n",
       "      <td>7079</td>\n",
       "      <td>7049</td>\n",
       "      <td>2021-02-17</td>\n",
       "      <td>2022-12-04</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  dataset_id          dataset       feature_set Data owner  \\\n",
       "0          0       population        population      Maria   \n",
       "1          1           events            events      Maria   \n",
       "2          2  anthropometrics   anthropometrics       Adam   \n",
       "3          3           fundus            fundus      Hagai   \n",
       "4          3           fundus  microvasculature       <NA>   \n",
       "\n",
       "                       status pheno-big-data-prod-us-east-1_relative_path  \\\n",
       "0  DS Owner Finished for v1.2                                        <NA>   \n",
       "1  DS Owner Finished for v1.2                                        <NA>   \n",
       "2  DS Owner Finished for v1.2                                        <NA>   \n",
       "3  DS Owner Finished for v1.2                                        <NA>   \n",
       "4  DS Owner Finished for v1.2                                        <NA>   \n",
       "\n",
       "  contains_sensitive notes               count_parquet_relative_path  \\\n",
       "0               <NA>  <NA>            /population/population.parquet   \n",
       "1               <NA>  <NA>                    /events/events.parquet   \n",
       "2               <NA>  <NA>  /anthropometrics/anthropometrics.parquet   \n",
       "3               <NA>  <NA>                    /fundus/fundus.parquet   \n",
       "4               <NA>  <NA>          /fundus/microvasculature.parquet   \n",
       "\n",
       "                            count_field_name data points participants  \\\n",
       "0                                        sex       11179        11179   \n",
       "1                        research_stage_date       23300        11179   \n",
       "2                                        bmi       13568        10781   \n",
       "3      fundus_image_left, fundus_image_right        7079         7049   \n",
       "4  vessel_density_left, vessel_density_right        7079         7049   \n",
       "\n",
       "   first_date   last_date dict_check  demo  \n",
       "0  2018-12-05  2023-03-19       <NA>  <NA>  \n",
       "1  2018-12-05  2023-03-19       <NA>  <NA>  \n",
       "2  2018-11-22  2023-02-07       <NA>  <NA>  \n",
       "3  2021-02-17  2022-12-04       <NA>  <NA>  \n",
       "4  2021-02-17  2022-12-04       <NA>  <NA>  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# link to the google sheet\n",
    "sheet_id = '1l8gf56ifdBwAptohozsdVidx9dUS-yNx2qxamYsooTw'\n",
    "sheet_name = 'Sheet1'\n",
    "url = f\"https://docs.google.com/spreadsheets/d/{sheet_id}/gviz/tq?tqx=out:csv&sheet={sheet_name}\"\n",
    "df_feature_set = pd.read_csv(url, dtype=str).fillna(pd.NA)\n",
    "\n",
    "    \n",
    "df_feature_set[['dataset_id', 'dataset']] = df_feature_set[['dataset_id', 'dataset']].ffill()\n",
    "df_feature_set_ready = df_feature_set[pd.notna(df_feature_set.status)].query('status != \"Not relevant v1.2\"')\n",
    "df_feature_set_ready.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(115, 16)\n",
      "(109, 16)\n",
      "(109, 16)\n"
     ]
    }
   ],
   "source": [
    "# check for duplicates and remove features without parquet path or count field\n",
    "print(df_feature_set_ready.shape)\n",
    "df = df_feature_set_ready.query('@pd.notna(count_parquet_relative_path) and @pd.notna(count_field_name)') # in @none\n",
    "print(df.shape)\n",
    "df_no_dup = df.drop_duplicates(['dataset', 'feature_set', 'count_parquet_relative_path', 'count_field_name'])\n",
    "print(df_no_dup.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "sleep s3://pheno-academic-collection-v1.2-prod-eu-west-1/sleep/raw/sleep.parquet\n",
      "Traceback (most recent call last):\n",
      "  File \"/tmp/ipykernel_3603/1176001438.py\", line 9, in <module>\n",
      "    df_parquet = pd.read_parquet(full_path).reset_index()\n",
      "                 ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 670, in read_parquet\n",
      "    return impl.read(\n",
      "           ^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 272, in read\n",
      "    pa_table = self.api.parquet.read_table(\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2956, in read_table\n",
      "    dataset = _ParquetDatasetV2(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2507, in __init__\n",
      "    self._dataset = ds.dataset(path_or_paths, filesystem=filesystem,\n",
      "                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 782, in dataset\n",
      "    return _filesystem_dataset(source, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 465, in _filesystem_dataset\n",
      "    fs, paths_or_selector = _ensure_single_source(source, filesystem)\n",
      "                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 441, in _ensure_single_source\n",
      "    raise FileNotFoundError(path)\n",
      "FileNotFoundError: pheno-academic-collection-v1.2-prod-eu-west-1/sleep/raw/sleep.parquet\n",
      "\n",
      "sleep s3://pheno-academic-collection-v1.2-prod-eu-west-1/sleep/raw/hrv.parquet\n",
      "Traceback (most recent call last):\n",
      "  File \"/tmp/ipykernel_3603/1176001438.py\", line 9, in <module>\n",
      "    df_parquet = pd.read_parquet(full_path).reset_index()\n",
      "                 ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 670, in read_parquet\n",
      "    return impl.read(\n",
      "           ^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 272, in read\n",
      "    pa_table = self.api.parquet.read_table(\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2956, in read_table\n",
      "    dataset = _ParquetDatasetV2(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2507, in __init__\n",
      "    self._dataset = ds.dataset(path_or_paths, filesystem=filesystem,\n",
      "                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 782, in dataset\n",
      "    return _filesystem_dataset(source, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 465, in _filesystem_dataset\n",
      "    fs, paths_or_selector = _ensure_single_source(source, filesystem)\n",
      "                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 441, in _ensure_single_source\n",
      "    raise FileNotFoundError(path)\n",
      "FileNotFoundError: pheno-academic-collection-v1.2-prod-eu-west-1/sleep/raw/hrv.parquet\n",
      "\n",
      "curated_phenotypes s3://pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/menopause.parquet\n",
      "Traceback (most recent call last):\n",
      "  File \"/tmp/ipykernel_3603/1176001438.py\", line 9, in <module>\n",
      "    df_parquet = pd.read_parquet(full_path).reset_index()\n",
      "                 ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 670, in read_parquet\n",
      "    return impl.read(\n",
      "           ^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 272, in read\n",
      "    pa_table = self.api.parquet.read_table(\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2956, in read_table\n",
      "    dataset = _ParquetDatasetV2(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2507, in __init__\n",
      "    self._dataset = ds.dataset(path_or_paths, filesystem=filesystem,\n",
      "                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 782, in dataset\n",
      "    return _filesystem_dataset(source, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 465, in _filesystem_dataset\n",
      "    fs, paths_or_selector = _ensure_single_source(source, filesystem)\n",
      "                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 441, in _ensure_single_source\n",
      "    raise FileNotFoundError(path)\n",
      "FileNotFoundError: pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/menopause.parquet\n",
      "\n",
      "curated_phenotypes s3://pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/osteoporosis.parquet\n",
      "Traceback (most recent call last):\n",
      "  File \"/tmp/ipykernel_3603/1176001438.py\", line 9, in <module>\n",
      "    df_parquet = pd.read_parquet(full_path).reset_index()\n",
      "                 ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 670, in read_parquet\n",
      "    return impl.read(\n",
      "           ^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 272, in read\n",
      "    pa_table = self.api.parquet.read_table(\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2956, in read_table\n",
      "    dataset = _ParquetDatasetV2(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2507, in __init__\n",
      "    self._dataset = ds.dataset(path_or_paths, filesystem=filesystem,\n",
      "                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 782, in dataset\n",
      "    return _filesystem_dataset(source, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 465, in _filesystem_dataset\n",
      "    fs, paths_or_selector = _ensure_single_source(source, filesystem)\n",
      "                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 441, in _ensure_single_source\n",
      "    raise FileNotFoundError(path)\n",
      "FileNotFoundError: pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/osteoporosis.parquet\n",
      "\n",
      "curated_phenotypes s3://pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/migraine.parquet\n",
      "Traceback (most recent call last):\n",
      "  File \"/tmp/ipykernel_3603/1176001438.py\", line 9, in <module>\n",
      "    df_parquet = pd.read_parquet(full_path).reset_index()\n",
      "                 ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 670, in read_parquet\n",
      "    return impl.read(\n",
      "           ^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 272, in read\n",
      "    pa_table = self.api.parquet.read_table(\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2956, in read_table\n",
      "    dataset = _ParquetDatasetV2(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2507, in __init__\n",
      "    self._dataset = ds.dataset(path_or_paths, filesystem=filesystem,\n",
      "                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 782, in dataset\n",
      "    return _filesystem_dataset(source, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 465, in _filesystem_dataset\n",
      "    fs, paths_or_selector = _ensure_single_source(source, filesystem)\n",
      "                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 441, in _ensure_single_source\n",
      "    raise FileNotFoundError(path)\n",
      "FileNotFoundError: pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/migraine.parquet\n",
      "\n",
      "curated_phenotypes s3://pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/depression.parquet\n",
      "Traceback (most recent call last):\n",
      "  File \"/tmp/ipykernel_3603/1176001438.py\", line 9, in <module>\n",
      "    df_parquet = pd.read_parquet(full_path).reset_index()\n",
      "                 ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 670, in read_parquet\n",
      "    return impl.read(\n",
      "           ^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 272, in read\n",
      "    pa_table = self.api.parquet.read_table(\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2956, in read_table\n",
      "    dataset = _ParquetDatasetV2(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2507, in __init__\n",
      "    self._dataset = ds.dataset(path_or_paths, filesystem=filesystem,\n",
      "                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 782, in dataset\n",
      "    return _filesystem_dataset(source, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 465, in _filesystem_dataset\n",
      "    fs, paths_or_selector = _ensure_single_source(source, filesystem)\n",
      "                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 441, in _ensure_single_source\n",
      "    raise FileNotFoundError(path)\n",
      "FileNotFoundError: pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/depression.parquet\n",
      "\n",
      "curated_phenotypes s3://pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/adhd.parquet\n",
      "Traceback (most recent call last):\n",
      "  File \"/tmp/ipykernel_3603/1176001438.py\", line 9, in <module>\n",
      "    df_parquet = pd.read_parquet(full_path).reset_index()\n",
      "                 ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 670, in read_parquet\n",
      "    return impl.read(\n",
      "           ^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 272, in read\n",
      "    pa_table = self.api.parquet.read_table(\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2956, in read_table\n",
      "    dataset = _ParquetDatasetV2(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2507, in __init__\n",
      "    self._dataset = ds.dataset(path_or_paths, filesystem=filesystem,\n",
      "                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 782, in dataset\n",
      "    return _filesystem_dataset(source, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 465, in _filesystem_dataset\n",
      "    fs, paths_or_selector = _ensure_single_source(source, filesystem)\n",
      "                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 441, in _ensure_single_source\n",
      "    raise FileNotFoundError(path)\n",
      "FileNotFoundError: pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/adhd.parquet\n",
      "\n",
      "curated_phenotypes s3://pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/anxiety.parquet\n",
      "Traceback (most recent call last):\n",
      "  File \"/tmp/ipykernel_3603/1176001438.py\", line 9, in <module>\n",
      "    df_parquet = pd.read_parquet(full_path).reset_index()\n",
      "                 ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 670, in read_parquet\n",
      "    return impl.read(\n",
      "           ^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 272, in read\n",
      "    pa_table = self.api.parquet.read_table(\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2956, in read_table\n",
      "    dataset = _ParquetDatasetV2(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2507, in __init__\n",
      "    self._dataset = ds.dataset(path_or_paths, filesystem=filesystem,\n",
      "                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 782, in dataset\n",
      "    return _filesystem_dataset(source, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 465, in _filesystem_dataset\n",
      "    fs, paths_or_selector = _ensure_single_source(source, filesystem)\n",
      "                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 441, in _ensure_single_source\n",
      "    raise FileNotFoundError(path)\n",
      "FileNotFoundError: pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/anxiety.parquet\n",
      "\n",
      "curated_phenotypes s3://pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/abdominal_adiposity.parquet\n",
      "Traceback (most recent call last):\n",
      "  File \"/tmp/ipykernel_3603/1176001438.py\", line 9, in <module>\n",
      "    df_parquet = pd.read_parquet(full_path).reset_index()\n",
      "                 ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 670, in read_parquet\n",
      "    return impl.read(\n",
      "           ^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 272, in read\n",
      "    pa_table = self.api.parquet.read_table(\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2956, in read_table\n",
      "    dataset = _ParquetDatasetV2(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2507, in __init__\n",
      "    self._dataset = ds.dataset(path_or_paths, filesystem=filesystem,\n",
      "                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 782, in dataset\n",
      "    return _filesystem_dataset(source, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 465, in _filesystem_dataset\n",
      "    fs, paths_or_selector = _ensure_single_source(source, filesystem)\n",
      "                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 441, in _ensure_single_source\n",
      "    raise FileNotFoundError(path)\n",
      "FileNotFoundError: pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/abdominal_adiposity.parquet\n",
      "\n",
      "curated_phenotypes s3://pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/bmi.parquet\n",
      "Traceback (most recent call last):\n",
      "  File \"/tmp/ipykernel_3603/1176001438.py\", line 9, in <module>\n",
      "    df_parquet = pd.read_parquet(full_path).reset_index()\n",
      "                 ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 670, in read_parquet\n",
      "    return impl.read(\n",
      "           ^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 272, in read\n",
      "    pa_table = self.api.parquet.read_table(\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2956, in read_table\n",
      "    dataset = _ParquetDatasetV2(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2507, in __init__\n",
      "    self._dataset = ds.dataset(path_or_paths, filesystem=filesystem,\n",
      "                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 782, in dataset\n",
      "    return _filesystem_dataset(source, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 465, in _filesystem_dataset\n",
      "    fs, paths_or_selector = _ensure_single_source(source, filesystem)\n",
      "                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 441, in _ensure_single_source\n",
      "    raise FileNotFoundError(path)\n",
      "FileNotFoundError: pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/bmi.parquet\n",
      "\n",
      "curated_phenotypes s3://pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/osa.parquet\n",
      "Traceback (most recent call last):\n",
      "  File \"/tmp/ipykernel_3603/1176001438.py\", line 9, in <module>\n",
      "    df_parquet = pd.read_parquet(full_path).reset_index()\n",
      "                 ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 670, in read_parquet\n",
      "    return impl.read(\n",
      "           ^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 272, in read\n",
      "    pa_table = self.api.parquet.read_table(\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2956, in read_table\n",
      "    dataset = _ParquetDatasetV2(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2507, in __init__\n",
      "    self._dataset = ds.dataset(path_or_paths, filesystem=filesystem,\n",
      "                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 782, in dataset\n",
      "    return _filesystem_dataset(source, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 465, in _filesystem_dataset\n",
      "    fs, paths_or_selector = _ensure_single_source(source, filesystem)\n",
      "                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 441, in _ensure_single_source\n",
      "    raise FileNotFoundError(path)\n",
      "FileNotFoundError: pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/osa.parquet\n",
      "\n",
      "curated_phenotypes s3://pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/ckd.parquet\n",
      "Traceback (most recent call last):\n",
      "  File \"/tmp/ipykernel_3603/1176001438.py\", line 9, in <module>\n",
      "    df_parquet = pd.read_parquet(full_path).reset_index()\n",
      "                 ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 670, in read_parquet\n",
      "    return impl.read(\n",
      "           ^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 272, in read\n",
      "    pa_table = self.api.parquet.read_table(\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2956, in read_table\n",
      "    dataset = _ParquetDatasetV2(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2507, in __init__\n",
      "    self._dataset = ds.dataset(path_or_paths, filesystem=filesystem,\n",
      "                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 782, in dataset\n",
      "    return _filesystem_dataset(source, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 465, in _filesystem_dataset\n",
      "    fs, paths_or_selector = _ensure_single_source(source, filesystem)\n",
      "                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 441, in _ensure_single_source\n",
      "    raise FileNotFoundError(path)\n",
      "FileNotFoundError: pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/ckd.parquet\n",
      "\n",
      "curated_phenotypes s3://pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/mafld.parquet\n",
      "Traceback (most recent call last):\n",
      "  File \"/tmp/ipykernel_3603/1176001438.py\", line 9, in <module>\n",
      "    df_parquet = pd.read_parquet(full_path).reset_index()\n",
      "                 ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 670, in read_parquet\n",
      "    return impl.read(\n",
      "           ^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 272, in read\n",
      "    pa_table = self.api.parquet.read_table(\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2956, in read_table\n",
      "    dataset = _ParquetDatasetV2(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2507, in __init__\n",
      "    self._dataset = ds.dataset(path_or_paths, filesystem=filesystem,\n",
      "                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 782, in dataset\n",
      "    return _filesystem_dataset(source, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 465, in _filesystem_dataset\n",
      "    fs, paths_or_selector = _ensure_single_source(source, filesystem)\n",
      "                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 441, in _ensure_single_source\n",
      "    raise FileNotFoundError(path)\n",
      "FileNotFoundError: pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/mafld.parquet\n",
      "\n",
      "curated_phenotypes s3://pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/diabetes.parquet\n",
      "Traceback (most recent call last):\n",
      "  File \"/tmp/ipykernel_3603/1176001438.py\", line 9, in <module>\n",
      "    df_parquet = pd.read_parquet(full_path).reset_index()\n",
      "                 ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 670, in read_parquet\n",
      "    return impl.read(\n",
      "           ^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 272, in read\n",
      "    pa_table = self.api.parquet.read_table(\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2956, in read_table\n",
      "    dataset = _ParquetDatasetV2(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2507, in __init__\n",
      "    self._dataset = ds.dataset(path_or_paths, filesystem=filesystem,\n",
      "                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 782, in dataset\n",
      "    return _filesystem_dataset(source, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 465, in _filesystem_dataset\n",
      "    fs, paths_or_selector = _ensure_single_source(source, filesystem)\n",
      "                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 441, in _ensure_single_source\n",
      "    raise FileNotFoundError(path)\n",
      "FileNotFoundError: pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/diabetes.parquet\n",
      "\n",
      "curated_phenotypes s3://pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/ischemic_heart_disease.parquet\n",
      "Traceback (most recent call last):\n",
      "  File \"/tmp/ipykernel_3603/1176001438.py\", line 9, in <module>\n",
      "    df_parquet = pd.read_parquet(full_path).reset_index()\n",
      "                 ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 670, in read_parquet\n",
      "    return impl.read(\n",
      "           ^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 272, in read\n",
      "    pa_table = self.api.parquet.read_table(\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2956, in read_table\n",
      "    dataset = _ParquetDatasetV2(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2507, in __init__\n",
      "    self._dataset = ds.dataset(path_or_paths, filesystem=filesystem,\n",
      "                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 782, in dataset\n",
      "    return _filesystem_dataset(source, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 465, in _filesystem_dataset\n",
      "    fs, paths_or_selector = _ensure_single_source(source, filesystem)\n",
      "                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 441, in _ensure_single_source\n",
      "    raise FileNotFoundError(path)\n",
      "FileNotFoundError: pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/ischemic_heart_disease.parquet\n",
      "\n",
      "curated_phenotypes s3://pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/sleep_quality.parquet\n",
      "Traceback (most recent call last):\n",
      "  File \"/tmp/ipykernel_3603/1176001438.py\", line 9, in <module>\n",
      "    df_parquet = pd.read_parquet(full_path).reset_index()\n",
      "                 ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 670, in read_parquet\n",
      "    return impl.read(\n",
      "           ^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 272, in read\n",
      "    pa_table = self.api.parquet.read_table(\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2956, in read_table\n",
      "    dataset = _ParquetDatasetV2(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2507, in __init__\n",
      "    self._dataset = ds.dataset(path_or_paths, filesystem=filesystem,\n",
      "                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 782, in dataset\n",
      "    return _filesystem_dataset(source, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 465, in _filesystem_dataset\n",
      "    fs, paths_or_selector = _ensure_single_source(source, filesystem)\n",
      "                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 441, in _ensure_single_source\n",
      "    raise FileNotFoundError(path)\n",
      "FileNotFoundError: pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/sleep_quality.parquet\n",
      "\n",
      "curated_phenotypes s3://pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/hypertension.parquet\n",
      "Traceback (most recent call last):\n",
      "  File \"/tmp/ipykernel_3603/1176001438.py\", line 9, in <module>\n",
      "    df_parquet = pd.read_parquet(full_path).reset_index()\n",
      "                 ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 670, in read_parquet\n",
      "    return impl.read(\n",
      "           ^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 272, in read\n",
      "    pa_table = self.api.parquet.read_table(\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2956, in read_table\n",
      "    dataset = _ParquetDatasetV2(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2507, in __init__\n",
      "    self._dataset = ds.dataset(path_or_paths, filesystem=filesystem,\n",
      "                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 782, in dataset\n",
      "    return _filesystem_dataset(source, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 465, in _filesystem_dataset\n",
      "    fs, paths_or_selector = _ensure_single_source(source, filesystem)\n",
      "                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 441, in _ensure_single_source\n",
      "    raise FileNotFoundError(path)\n",
      "FileNotFoundError: pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/hypertension.parquet\n",
      "\n",
      "curated_phenotypes s3://pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/hyperlipidemia.parquet\n",
      "Traceback (most recent call last):\n",
      "  File \"/tmp/ipykernel_3603/1176001438.py\", line 9, in <module>\n",
      "    df_parquet = pd.read_parquet(full_path).reset_index()\n",
      "                 ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 670, in read_parquet\n",
      "    return impl.read(\n",
      "           ^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 272, in read\n",
      "    pa_table = self.api.parquet.read_table(\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2956, in read_table\n",
      "    dataset = _ParquetDatasetV2(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2507, in __init__\n",
      "    self._dataset = ds.dataset(path_or_paths, filesystem=filesystem,\n",
      "                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 782, in dataset\n",
      "    return _filesystem_dataset(source, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 465, in _filesystem_dataset\n",
      "    fs, paths_or_selector = _ensure_single_source(source, filesystem)\n",
      "                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 441, in _ensure_single_source\n",
      "    raise FileNotFoundError(path)\n",
      "FileNotFoundError: pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/hyperlipidemia.parquet\n",
      "\n",
      "curated_phenotypes s3://pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/nmsc.parquet\n",
      "Traceback (most recent call last):\n",
      "  File \"/tmp/ipykernel_3603/1176001438.py\", line 9, in <module>\n",
      "    df_parquet = pd.read_parquet(full_path).reset_index()\n",
      "                 ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 670, in read_parquet\n",
      "    return impl.read(\n",
      "           ^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 272, in read\n",
      "    pa_table = self.api.parquet.read_table(\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2956, in read_table\n",
      "    dataset = _ParquetDatasetV2(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2507, in __init__\n",
      "    self._dataset = ds.dataset(path_or_paths, filesystem=filesystem,\n",
      "                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 782, in dataset\n",
      "    return _filesystem_dataset(source, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 465, in _filesystem_dataset\n",
      "    fs, paths_or_selector = _ensure_single_source(source, filesystem)\n",
      "                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 441, in _ensure_single_source\n",
      "    raise FileNotFoundError(path)\n",
      "FileNotFoundError: pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/nmsc.parquet\n",
      "\n",
      "curated_phenotypes s3://pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/endometriosis.parquet\n",
      "Traceback (most recent call last):\n",
      "  File \"/tmp/ipykernel_3603/1176001438.py\", line 9, in <module>\n",
      "    df_parquet = pd.read_parquet(full_path).reset_index()\n",
      "                 ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 670, in read_parquet\n",
      "    return impl.read(\n",
      "           ^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pandas/io/parquet.py\", line 272, in read\n",
      "    pa_table = self.api.parquet.read_table(\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2956, in read_table\n",
      "    dataset = _ParquetDatasetV2(\n",
      "              ^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/parquet/core.py\", line 2507, in __init__\n",
      "    self._dataset = ds.dataset(path_or_paths, filesystem=filesystem,\n",
      "                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 782, in dataset\n",
      "    return _filesystem_dataset(source, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 465, in _filesystem_dataset\n",
      "    fs, paths_or_selector = _ensure_single_source(source, filesystem)\n",
      "                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"/home/ec2-user/miniconda3/lib/python3.11/site-packages/pyarrow/dataset.py\", line 441, in _ensure_single_source\n",
      "    raise FileNotFoundError(path)\n",
      "FileNotFoundError: pheno-academic-collection-v1.2-prod-eu-west-1/curated_phenotypes/endometriosis.parquet\n",
      "\n"
     ]
    }
   ],
   "source": [
    "count_dict = defaultdict(dict)\n",
    "for index, row in df_no_dup.iterrows():\n",
    "    try:\n",
    "        dataset = row['dataset']\n",
    "        if dataset == 'curated_phenotypes':\n",
    "            continue\n",
    "        feature_set = row['feature_set']\n",
    "        \n",
    "        #read parquet file\n",
    "        full_path = f\"{master_path}{row['count_parquet_relative_path']}\"\n",
    "        df_parquet = pd.read_parquet(full_path).reset_index()\n",
    "        \n",
    "        if dataset=='medications':\n",
    "            row['count_field_name'] = 'medication'\n",
    "        res_dict =  {\n",
    "            \"df_parquet_shape\": df_parquet.shape[0],\n",
    "            \"count_field_name\": row['count_field_name'],\n",
    "            \"count_parquet_relative_path\": row['count_parquet_relative_path'],\n",
    "            \"particioant_id_nunique\": df_parquet['participant_id'].nunique()\n",
    "            }\n",
    "        \n",
    "        max_col = ''\n",
    "        max_count = 0\n",
    "        \n",
    "        for col in row['count_field_name'].split(','):\n",
    "            # remove white space\n",
    "            col = col.strip()\n",
    "            if col not in df_parquet.columns:\n",
    "                print (f\"ERROR: {col} not in {full_path} : {df_parquet.columns}\")\n",
    "                \n",
    "            no_na_df = df_parquet.dropna(subset=[col]).copy()\n",
    "            \n",
    "            if dataset != 'events': \n",
    "                date_col = 'collection_date'\n",
    "            else: \n",
    "                date_col = 'research_stage_date'\n",
    "                \n",
    "            if date_col not in no_na_df.columns:\n",
    "                date_col = 'collection_timestamp'\n",
    "            if date_col not in no_na_df.columns:\n",
    "                first_date = None\n",
    "                last_date = None\n",
    "            else:   \n",
    "                if BEFORE_2023:\n",
    "                    # filter date_col before 2023-01-01\n",
    "                    no_na_df[date_col] = pd.to_datetime(no_na_df[date_col])\n",
    "                    no_na_df = no_na_df[no_na_df[date_col] < '2023-01-01']\n",
    "                first_date = no_na_df[date_col].min() # first date\n",
    "                last_date = no_na_df[date_col].max() # last date\n",
    "\n",
    "                    \n",
    "            nunique_participants_not_na = no_na_df.participant_id.nunique() # number of unique participants\n",
    "            count_col_notna = no_na_df.shape[0] # number of not na entries\n",
    "            \n",
    "            if no_na_df.shape[0] > max_count:\n",
    "                max_count = no_na_df.shape[0]\n",
    "                max_col = col\n",
    "                \n",
    "                \n",
    "            res_dict[col] = {\n",
    "                \"count_col_notna\": count_col_notna,  \n",
    "                \"nunique_participants_not_na\" : nunique_participants_not_na,\n",
    "                \"first_date\": first_date,\n",
    "                \"last_date\": last_date,\n",
    "            }\n",
    "        del df_parquet\n",
    "        res_dict['max_col'] = max_col\n",
    "        count_dict[f\"{int(row['dataset_id'])}-{dataset}\"][feature_set] = res_dict\n",
    "        del no_na_df\n",
    "        \n",
    "        \n",
    "    except Exception as e: \n",
    "        print(row['dataset'], full_path)\n",
    "        print(traceback.format_exc())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "full_data = list()\n",
    "for k, v in count_dict.items():\n",
    "    for kk, vv in v.items():\n",
    "        data = list()\n",
    "        data.append(k)\n",
    "        data.append(kk)\n",
    "        col = vv['max_col']\n",
    "        data.append(vv[col][\"count_col_notna\"])\n",
    "        data.append(vv[col][\"nunique_participants_not_na\"])\n",
    "        data.append(vv[col][\"first_date\"])\n",
    "        data.append(vv[col][\"last_date\"])\n",
    "        full_data.append(data)\n",
    "        \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[['0-population', 'population', 12014, 12014, None, None],\n",
       " ['1-events',\n",
       "  'events',\n",
       "  20723,\n",
       "  10434,\n",
       "  Timestamp('2018-12-05 00:00:00'),\n",
       "  Timestamp('2022-12-29 00:00:00')],\n",
       " ['2-anthropometrics',\n",
       "  'anthropometrics',\n",
       "  12925,\n",
       "  10387,\n",
       "  Timestamp('2018-11-22 00:00:00'),\n",
       "  Timestamp('2022-12-29 00:00:00')],\n",
       " ['3-fundus',\n",
       "  'fundus',\n",
       "  7068,\n",
       "  7038,\n",
       "  Timestamp('2021-02-17 00:00:00'),\n",
       "  Timestamp('2022-12-04 00:00:00')],\n",
       " ['3-fundus', 'microvasculature', 7068, 7038, None, None],\n",
       " ['3-fundus',\n",
       "  'images',\n",
       "  7068,\n",
       "  7038,\n",
       "  Timestamp('2021-02-17 00:00:00'),\n",
       "  Timestamp('2022-12-04 00:00:00')],\n",
       " ['3-fundus', 'segmented_images', 7068, 7038, None, None],\n",
       " ['4-liver_ultrasound',\n",
       "  'liver_ultrasound',\n",
       "  26684,\n",
       "  8229,\n",
       "  Timestamp('2020-02-20 00:00:00'),\n",
       "  Timestamp('2022-12-29 00:00:00')],\n",
       " ['5-diet_logging',\n",
       "  'diet_logging',\n",
       "  149912,\n",
       "  9531,\n",
       "  Timestamp('2019-09-01 00:00:00'),\n",
       "  Timestamp('2022-12-31 00:00:00')],\n",
       " ['5-diet_logging',\n",
       "  'diet_logging_events',\n",
       "  2640312,\n",
       "  10197,\n",
       "  Timestamp('2019-09-01 00:00:00'),\n",
       "  Timestamp('2023-03-21 00:00:00')],\n",
       " ['5-diet_logging',\n",
       "  'raw_diet_logging_events',\n",
       "  6483638,\n",
       "  10696,\n",
       "  Timestamp('2019-09-01 00:00:00'),\n",
       "  Timestamp('2023-03-21 00:00:00')],\n",
       " ['6-hand_grip',\n",
       "  'hand_grip',\n",
       "  12894,\n",
       "  10371,\n",
       "  Timestamp('2018-12-27 00:00:00'),\n",
       "  Timestamp('2022-12-29 00:00:00')],\n",
       " ['7-blood_pressure',\n",
       "  'blood_pressure',\n",
       "  12946,\n",
       "  10398,\n",
       "  Timestamp('2018-12-27 00:00:00'),\n",
       "  Timestamp('2022-12-29 00:00:00')],\n",
       " ['8-serum_lipidomics',\n",
       "  'serum_lipidomics',\n",
       "  6321,\n",
       "  6144,\n",
       "  Timestamp('2019-02-19 00:00:00'),\n",
       "  Timestamp('2021-08-08 00:00:00')],\n",
       " ['9-sleep',\n",
       "  'sleep',\n",
       "  21038,\n",
       "  7009,\n",
       "  Timestamp('2020-01-15 00:00:00'),\n",
       "  Timestamp('2022-12-27 00:00:00')],\n",
       " ['9-sleep', 'hrv', 20030, 6926, None, None],\n",
       " ['9-sleep',\n",
       "  'channels_time_series',\n",
       "  21072,\n",
       "  6973,\n",
       "  Timestamp('2020-01-15 00:00:00'),\n",
       "  Timestamp('2022-12-27 00:00:00')],\n",
       " ['9-sleep',\n",
       "  'channels_edf_file',\n",
       "  21069,\n",
       "  6973,\n",
       "  Timestamp('2020-01-15 00:00:00'),\n",
       "  Timestamp('2022-12-27 00:00:00')],\n",
       " ['9-sleep',\n",
       "  'events_time_series',\n",
       "  21372,\n",
       "  7062,\n",
       "  Timestamp('2020-01-15 00:00:00'),\n",
       "  Timestamp('2022-12-27 00:00:00')],\n",
       " ['10-vascular_health',\n",
       "  'vascular_health',\n",
       "  11387,\n",
       "  9729,\n",
       "  Timestamp('2018-12-27 00:00:00'),\n",
       "  Timestamp('2022-12-29 00:00:00')],\n",
       " ['11-body_composition',\n",
       "  'body_composition',\n",
       "  8990,\n",
       "  8411,\n",
       "  Timestamp('2020-06-09 00:00:00'),\n",
       "  Timestamp('2022-12-29 00:00:00')],\n",
       " ['12-carotid_ultrasound',\n",
       "  'carotid_ultrasound',\n",
       "  8518,\n",
       "  7988,\n",
       "  Timestamp('2020-07-23 00:00:00'),\n",
       "  Timestamp('2022-12-29 00:00:00')],\n",
       " ['13-gut_microbiome',\n",
       "  'gut_microbiome',\n",
       "  10694,\n",
       "  8838,\n",
       "  Timestamp('2019-02-14 00:00:00'),\n",
       "  Timestamp('2022-12-28 00:00:00')],\n",
       " ['13-gut_microbiome',\n",
       "  'raw_fastq',\n",
       "  10694,\n",
       "  8838,\n",
       "  Timestamp('2019-02-14 00:00:00'),\n",
       "  Timestamp('2022-12-28 00:00:00')],\n",
       " ['13-gut_microbiome',\n",
       "  'trimmed_fastq',\n",
       "  10694,\n",
       "  8838,\n",
       "  Timestamp('2019-02-14 00:00:00'),\n",
       "  Timestamp('2022-12-28 00:00:00')],\n",
       " ['13-gut_microbiome',\n",
       "  'nonhuman_fastq',\n",
       "  10694,\n",
       "  8838,\n",
       "  Timestamp('2019-02-14 00:00:00'),\n",
       "  Timestamp('2022-12-28 00:00:00')],\n",
       " ['13-gut_microbiome',\n",
       "  'fastqc',\n",
       "  10694,\n",
       "  8838,\n",
       "  Timestamp('2019-02-14 00:00:00'),\n",
       "  Timestamp('2022-12-28 00:00:00')],\n",
       " ['13-gut_microbiome',\n",
       "  'human_plink',\n",
       "  10694,\n",
       "  8838,\n",
       "  Timestamp('2019-02-14 00:00:00'),\n",
       "  Timestamp('2022-12-28 00:00:00')],\n",
       " ['13-gut_microbiome',\n",
       "  'urs_abundances_single',\n",
       "  10694,\n",
       "  8838,\n",
       "  Timestamp('2019-02-14 00:00:00'),\n",
       "  Timestamp('2022-12-28 00:00:00')],\n",
       " ['13-gut_microbiome',\n",
       "  'urs_abundances_aggregated',\n",
       "  10694,\n",
       "  8838,\n",
       "  Timestamp('2019-02-14 00:00:00'),\n",
       "  Timestamp('2022-12-28 00:00:00')],\n",
       " ['13-gut_microbiome',\n",
       "  'metaphlan_abundances_single',\n",
       "  10694,\n",
       "  8838,\n",
       "  Timestamp('2019-02-14 00:00:00'),\n",
       "  Timestamp('2022-12-28 00:00:00')],\n",
       " ['13-gut_microbiome',\n",
       "  'metaphlan_abundances_aggregated',\n",
       "  10694,\n",
       "  8838,\n",
       "  Timestamp('2019-02-14 00:00:00'),\n",
       "  Timestamp('2022-12-28 00:00:00')],\n",
       " ['14-human_genetics',\n",
       "  'human_genetics',\n",
       "  8821,\n",
       "  8821,\n",
       "  Timestamp('2019-03-11 00:00:00'),\n",
       "  Timestamp('2022-12-21 00:00:00')],\n",
       " ['14-human_genetics', 'pca', 8820, 8820, None, None],\n",
       " ['14-human_genetics',\n",
       "  'variants_qc',\n",
       "  8821,\n",
       "  8821,\n",
       "  Timestamp('2019-03-11 00:00:00'),\n",
       "  Timestamp('2022-12-21 00:00:00')],\n",
       " ['14-human_genetics',\n",
       "  'fastq_bam',\n",
       "  8821,\n",
       "  8821,\n",
       "  Timestamp('2019-03-11 00:00:00'),\n",
       "  Timestamp('2022-12-21 00:00:00')],\n",
       " ['14-human_genetics',\n",
       "  'genotypes_raw',\n",
       "  8821,\n",
       "  8821,\n",
       "  Timestamp('2019-03-11 00:00:00'),\n",
       "  Timestamp('2022-12-21 00:00:00')],\n",
       " ['14-human_genetics',\n",
       "  'genotypes_postqc',\n",
       "  8820,\n",
       "  8820,\n",
       "  Timestamp('2019-03-11 00:00:00'),\n",
       "  Timestamp('2022-12-21 00:00:00')],\n",
       " ['14-human_genetics',\n",
       "  'relationship',\n",
       "  8821,\n",
       "  8821,\n",
       "  Timestamp('2019-03-11 00:00:00'),\n",
       "  Timestamp('2022-12-21 00:00:00')],\n",
       " ['15-ecg',\n",
       "  'ecg',\n",
       "  9702,\n",
       "  8085,\n",
       "  Timestamp('2019-08-13 00:00:00'),\n",
       "  Timestamp('2022-12-29 00:00:00')],\n",
       " ['15-ecg',\n",
       "  'raw_timeseries_wide',\n",
       "  9441,\n",
       "  7888,\n",
       "  Timestamp('2019-08-13 00:00:00'),\n",
       "  Timestamp('2022-12-29 00:00:00')],\n",
       " ['15-ecg',\n",
       "  'processed_timeseries_wide',\n",
       "  9172,\n",
       "  7711,\n",
       "  Timestamp('2019-08-13 00:00:00'),\n",
       "  Timestamp('2022-12-29 00:00:00')],\n",
       " ['15-ecg',\n",
       "  'raw_timeseries_long',\n",
       "  9441,\n",
       "  7888,\n",
       "  Timestamp('2019-08-13 00:00:00'),\n",
       "  Timestamp('2022-12-29 00:00:00')],\n",
       " ['15-ecg',\n",
       "  'processed_timeseries_long',\n",
       "  9172,\n",
       "  7711,\n",
       "  Timestamp('2019-08-13 00:00:00'),\n",
       "  Timestamp('2022-12-29 00:00:00')],\n",
       " ['16-blood_tests',\n",
       "  'blood_tests',\n",
       "  29074,\n",
       "  10252,\n",
       "  Timestamp('2010-01-06 00:00:00'),\n",
       "  Timestamp('2022-12-29 00:00:00')],\n",
       " ['17-cgm',\n",
       "  'cgm',\n",
       "  9497,\n",
       "  9431,\n",
       "  Timestamp('2019-01-07 00:14:00+0200', tz='Asia/Jerusalem'),\n",
       "  Timestamp('2022-12-30 00:14:00+0200', tz='Asia/Jerusalem')],\n",
       " ['17-cgm',\n",
       "  'timeseries',\n",
       "  9497,\n",
       "  9431,\n",
       "  Timestamp('2019-01-07 00:14:00+0200', tz='Asia/Jerusalem'),\n",
       "  Timestamp('2022-12-30 00:14:00+0200', tz='Asia/Jerusalem')],\n",
       " ['17-cgm', 'iglu', 9497, 9431, None, None],\n",
       " ['17-cgm',\n",
       "  'iglu_daily',\n",
       "  104845,\n",
       "  9431,\n",
       "  Timestamp('2019-01-07 00:00:00'),\n",
       "  Timestamp('2022-12-31 00:00:00')],\n",
       " ['22-bone_density',\n",
       "  'bone_density',\n",
       "  9032,\n",
       "  8459,\n",
       "  Timestamp('2020-06-09 00:00:00'),\n",
       "  Timestamp('2022-12-29 00:00:00')],\n",
       " ['18-medications',\n",
       "  'medications',\n",
       "  43417,\n",
       "  10787,\n",
       "  Timestamp('2018-10-18 00:00:00'),\n",
       "  Timestamp('2022-12-31 00:00:00')],\n",
       " ['21-medical_conditions',\n",
       "  'medical_conditions',\n",
       "  22413,\n",
       "  10792,\n",
       "  Timestamp('2018-11-21 00:00:00'),\n",
       "  Timestamp('2022-12-31 00:00:00')],\n",
       " ['49-medical_symptoms',\n",
       "  'initial_medical',\n",
       "  7416,\n",
       "  7383,\n",
       "  Timestamp('2020-02-26 00:00:00'),\n",
       "  Timestamp('2022-12-31 00:00:00')],\n",
       " ['49-medical_symptoms',\n",
       "  'follow_up_ukbb',\n",
       "  5710,\n",
       "  4471,\n",
       "  Timestamp('2020-02-17 00:00:00'),\n",
       "  Timestamp('2022-12-31 00:00:00')],\n",
       " ['49-medical_symptoms',\n",
       "  'digestive_health',\n",
       "  9172,\n",
       "  8719,\n",
       "  Timestamp('2019-01-09 00:00:00'),\n",
       "  Timestamp('2022-12-31 00:00:00')],\n",
       " ['50-medical_procedures',\n",
       "  'initial_medical',\n",
       "  7416,\n",
       "  7383,\n",
       "  Timestamp('2020-02-26 00:00:00'),\n",
       "  Timestamp('2022-12-31 00:00:00')],\n",
       " ['50-medical_procedures',\n",
       "  'follow_up_medical',\n",
       "  7572,\n",
       "  5869,\n",
       "  Timestamp('2021-04-22 00:00:00'),\n",
       "  Timestamp('2022-12-29 00:00:00')],\n",
       " ['43-covid',\n",
       "  'covid',\n",
       "  14988,\n",
       "  9120,\n",
       "  Timestamp('2021-04-22 00:00:00'),\n",
       "  Timestamp('2022-12-29 00:00:00')],\n",
       " ['44-sex_specific_factors',\n",
       "  'menstrual_cycle_survey',\n",
       "  1405,\n",
       "  1405,\n",
       "  Timestamp('2022-06-12 00:00:00'),\n",
       "  Timestamp('2022-12-29 00:00:00')],\n",
       " ['44-sex_specific_factors',\n",
       "  'initial_medical',\n",
       "  7416,\n",
       "  7383,\n",
       "  Timestamp('2020-02-26 00:00:00'),\n",
       "  Timestamp('2022-12-31 00:00:00')],\n",
       " ['44-sex_specific_factors',\n",
       "  'ukbb',\n",
       "  9151,\n",
       "  8759,\n",
       "  Timestamp('2019-01-09 00:00:00'),\n",
       "  Timestamp('2022-12-31 00:00:00')],\n",
       " ['44-sex_specific_factors',\n",
       "  'follow_up_ukbb',\n",
       "  5710,\n",
       "  4471,\n",
       "  Timestamp('2020-02-17 00:00:00'),\n",
       "  Timestamp('2022-12-31 00:00:00')],\n",
       " ['44-sex_specific_factors',\n",
       "  'follow_up_medical',\n",
       "  7572,\n",
       "  5869,\n",
       "  Timestamp('2021-04-22 00:00:00'),\n",
       "  Timestamp('2022-12-29 00:00:00')],\n",
       " ['52-family_history',\n",
       "  'initial_medical',\n",
       "  7416,\n",
       "  7383,\n",
       "  Timestamp('2020-02-26 00:00:00'),\n",
       "  Timestamp('2022-12-31 00:00:00')],\n",
       " ['52-family_history',\n",
       "  'ukbb',\n",
       "  14861,\n",
       "  9262,\n",
       "  Timestamp('2019-01-09 00:00:00'),\n",
       "  Timestamp('2022-12-31 00:00:00')],\n",
       " ['51-children',\n",
       "  'children',\n",
       "  7416,\n",
       "  7383,\n",
       "  Timestamp('2020-02-26 00:00:00'),\n",
       "  Timestamp('2022-12-31 00:00:00')],\n",
       " ['53-sociodemographics',\n",
       "  'ukbb',\n",
       "  14861,\n",
       "  9262,\n",
       "  Timestamp('2019-01-09 00:00:00'),\n",
       "  Timestamp('2022-12-31 00:00:00')],\n",
       " ['53-sociodemographics',\n",
       "  'initial_medical',\n",
       "  7416,\n",
       "  7383,\n",
       "  Timestamp('2020-02-26 00:00:00'),\n",
       "  Timestamp('2022-12-31 00:00:00')],\n",
       " ['55-lifestyle_and_environment',\n",
       "  'lifestyle_and_environment',\n",
       "  14861,\n",
       "  9262,\n",
       "  Timestamp('2019-01-09 00:00:00'),\n",
       "  Timestamp('2022-12-31 00:00:00')],\n",
       " ['57-psychological_and_social_health',\n",
       "  'psychological_and_social_health',\n",
       "  14861,\n",
       "  9262,\n",
       "  Timestamp('2019-01-09 00:00:00'),\n",
       "  Timestamp('2022-12-31 00:00:00')],\n",
       " ['58-health_and_medical_history',\n",
       "  'ukbb',\n",
       "  14861,\n",
       "  9262,\n",
       "  Timestamp('2019-01-09 00:00:00'),\n",
       "  Timestamp('2022-12-31 00:00:00')],\n",
       " ['58-health_and_medical_history',\n",
       "  'initial_medical',\n",
       "  10197,\n",
       "  10197,\n",
       "  Timestamp('2018-11-21 00:00:00'),\n",
       "  Timestamp('2022-12-31 00:00:00')],\n",
       " ['45-participant_relation',\n",
       "  'participant relation',\n",
       "  3739,\n",
       "  3106,\n",
       "  Timestamp('2019-02-05 00:00:00'),\n",
       "  Timestamp('2022-12-29 00:00:00')],\n",
       " ['24-rna_seq',\n",
       "  'rna_seq',\n",
       "  3958,\n",
       "  3863,\n",
       "  Timestamp('2021-02-28 00:00:00'),\n",
       "  Timestamp('2022-06-21 00:00:00')],\n",
       " ['24-rna_seq',\n",
       "  'raw_fastq',\n",
       "  3958,\n",
       "  3863,\n",
       "  Timestamp('2021-02-28 00:00:00'),\n",
       "  Timestamp('2022-06-21 00:00:00')],\n",
       " ['24-rna_seq',\n",
       "  'trimmed_fastq',\n",
       "  3958,\n",
       "  3863,\n",
       "  Timestamp('2021-02-28 00:00:00'),\n",
       "  Timestamp('2022-06-21 00:00:00')],\n",
       " ['24-rna_seq',\n",
       "  'aligned_bam',\n",
       "  3958,\n",
       "  3863,\n",
       "  Timestamp('2021-02-28 00:00:00'),\n",
       "  Timestamp('2022-06-21 00:00:00')],\n",
       " ['24-rna_seq',\n",
       "  'deduplicated_bam',\n",
       "  3958,\n",
       "  3863,\n",
       "  Timestamp('2021-02-28 00:00:00'),\n",
       "  Timestamp('2022-06-21 00:00:00')],\n",
       " ['24-rna_seq',\n",
       "  'multiqc',\n",
       "  3958,\n",
       "  3863,\n",
       "  Timestamp('2021-02-28 00:00:00'),\n",
       "  Timestamp('2022-06-21 00:00:00')],\n",
       " ['29-nightingale_metabolomics',\n",
       "  'nightingale_metabolomics',\n",
       "  1644,\n",
       "  1552,\n",
       "  Timestamp('2019-05-07 00:00:00'),\n",
       "  Timestamp('2022-08-02 00:00:00')],\n",
       " ['29-nightingale_metabolomics',\n",
       "  'bimarker_qc_tags',\n",
       "  1644,\n",
       "  1552,\n",
       "  Timestamp('2019-05-07 00:00:00'),\n",
       "  Timestamp('2022-08-02 00:00:00')],\n",
       " ['70-oral_microbiome',\n",
       "  'oral_microbiome',\n",
       "  8816,\n",
       "  8816,\n",
       "  Timestamp('2019-03-11 00:00:00'),\n",
       "  Timestamp('2022-12-21 00:00:00')],\n",
       " ['70-oral_microbiome',\n",
       "  'raw_fastq',\n",
       "  8816,\n",
       "  8816,\n",
       "  Timestamp('2019-03-11 00:00:00'),\n",
       "  Timestamp('2022-12-21 00:00:00')],\n",
       " ['70-oral_microbiome',\n",
       "  'trimmed_fastq',\n",
       "  8816,\n",
       "  8816,\n",
       "  Timestamp('2019-03-11 00:00:00'),\n",
       "  Timestamp('2022-12-21 00:00:00')],\n",
       " ['70-oral_microbiome',\n",
       "  'fastqc',\n",
       "  8816,\n",
       "  8816,\n",
       "  Timestamp('2019-03-11 00:00:00'),\n",
       "  Timestamp('2022-12-21 00:00:00')],\n",
       " ['70-oral_microbiome',\n",
       "  'metaphlan',\n",
       "  8816,\n",
       "  8816,\n",
       "  Timestamp('2019-03-11 00:00:00'),\n",
       "  Timestamp('2022-12-21 00:00:00')],\n",
       " ['70-oral_microbiome',\n",
       "  'metaphlan_aggregated',\n",
       "  8816,\n",
       "  8816,\n",
       "  Timestamp('2019-03-11 00:00:00'),\n",
       "  Timestamp('2022-12-21 00:00:00')],\n",
       " ['70-oral_microbiome',\n",
       "  'humann',\n",
       "  8816,\n",
       "  8816,\n",
       "  Timestamp('2019-03-11 00:00:00'),\n",
       "  Timestamp('2022-12-21 00:00:00')],\n",
       " ['70-oral_microbiome',\n",
       "  'humann_aggregated',\n",
       "  8816,\n",
       "  8816,\n",
       "  Timestamp('2019-03-11 00:00:00'),\n",
       "  Timestamp('2022-12-21 00:00:00')]]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset</th>\n",
       "      <th>feature_set</th>\n",
       "      <th>count_col_notna</th>\n",
       "      <th>nunique_participants_not_na</th>\n",
       "      <th>first_date</th>\n",
       "      <th>last_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0-population</td>\n",
       "      <td>population</td>\n",
       "      <td>12014</td>\n",
       "      <td>12014</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1-events</td>\n",
       "      <td>events</td>\n",
       "      <td>20723</td>\n",
       "      <td>10434</td>\n",
       "      <td>2018-12-05 00:00:00</td>\n",
       "      <td>2022-12-29 00:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2-anthropometrics</td>\n",
       "      <td>anthropometrics</td>\n",
       "      <td>12925</td>\n",
       "      <td>10387</td>\n",
       "      <td>2018-11-22 00:00:00</td>\n",
       "      <td>2022-12-29 00:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3-fundus</td>\n",
       "      <td>fundus</td>\n",
       "      <td>7068</td>\n",
       "      <td>7038</td>\n",
       "      <td>2021-02-17 00:00:00</td>\n",
       "      <td>2022-12-04 00:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3-fundus</td>\n",
       "      <td>microvasculature</td>\n",
       "      <td>7068</td>\n",
       "      <td>7038</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             dataset       feature_set  count_col_notna  \\\n",
       "0       0-population        population            12014   \n",
       "1           1-events            events            20723   \n",
       "2  2-anthropometrics   anthropometrics            12925   \n",
       "3           3-fundus            fundus             7068   \n",
       "4           3-fundus  microvasculature             7068   \n",
       "\n",
       "   nunique_participants_not_na           first_date            last_date  \n",
       "0                        12014                 None                 None  \n",
       "1                        10434  2018-12-05 00:00:00  2022-12-29 00:00:00  \n",
       "2                        10387  2018-11-22 00:00:00  2022-12-29 00:00:00  \n",
       "3                         7038  2021-02-17 00:00:00  2022-12-04 00:00:00  \n",
       "4                         7038                 None                 None  "
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_res = pd.DataFrame(full_data)\n",
    "df_res.columns = ['dataset', 'feature_set', 'count_col_notna', 'nunique_participants_not_na', 'first_date', 'last_date']\n",
    "df_res.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df_res.to_csv('pheno_stage_data_collection_v1_2_full_count.csv', index=False)\n",
    "# df_res.to_csv('pheno_stage_data_collection_v1_2_before_2023_count.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset</th>\n",
       "      <th>feature_set</th>\n",
       "      <th>count_col_notna</th>\n",
       "      <th>nunique_participants_not_na</th>\n",
       "      <th>first_date</th>\n",
       "      <th>last_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>17-cgm</td>\n",
       "      <td>cgm</td>\n",
       "      <td>9497</td>\n",
       "      <td>9431</td>\n",
       "      <td>2019-01-07 00:14:00+02:00</td>\n",
       "      <td>2022-12-30 00:14:00+02:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>17-cgm</td>\n",
       "      <td>timeseries</td>\n",
       "      <td>9497</td>\n",
       "      <td>9431</td>\n",
       "      <td>2019-01-07 00:14:00+02:00</td>\n",
       "      <td>2022-12-30 00:14:00+02:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>17-cgm</td>\n",
       "      <td>iglu</td>\n",
       "      <td>9497</td>\n",
       "      <td>9431</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>17-cgm</td>\n",
       "      <td>iglu_daily</td>\n",
       "      <td>104845</td>\n",
       "      <td>9431</td>\n",
       "      <td>2019-01-07 00:00:00</td>\n",
       "      <td>2022-12-31 00:00:00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   dataset feature_set  count_col_notna  nunique_participants_not_na  \\\n",
       "45  17-cgm         cgm             9497                         9431   \n",
       "46  17-cgm  timeseries             9497                         9431   \n",
       "47  17-cgm        iglu             9497                         9431   \n",
       "48  17-cgm  iglu_daily           104845                         9431   \n",
       "\n",
       "                   first_date                  last_date  \n",
       "45  2019-01-07 00:14:00+02:00  2022-12-30 00:14:00+02:00  \n",
       "46  2019-01-07 00:14:00+02:00  2022-12-30 00:14:00+02:00  \n",
       "47                       None                       None  \n",
       "48        2019-01-07 00:00:00        2022-12-31 00:00:00  "
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_res.query('dataset == \"17-cgm\"')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "if BEFORE_2023:\n",
    "    df_res.loc[df_res['dataset']== '0-population', 'count_col_notna'] =  df_res.loc[df_res['dataset']== '1-events', 'nunique_participants_not_na'].values\n",
    "    df_res.loc[df_res['dataset']== '0-population', 'nunique_participants_not_na'] =  df_res.loc[df_res['dataset']== '1-events', 'nunique_participants_not_na'].values\n",
    "    k = 'population'\n",
    "    v = 'events'\n",
    "    df_res.loc[df_res['feature_set']== k, 'first_date'] =  df_res.loc[df_res['feature_set']== v, 'first_date'].values\n",
    "    df_res.loc[df_res['feature_set']== k, 'last_date'] = df_res.loc[df_res['feature_set']== v, 'last_date'].values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset</th>\n",
       "      <th>feature_set</th>\n",
       "      <th>count_col_notna</th>\n",
       "      <th>nunique_participants_not_na</th>\n",
       "      <th>first_date</th>\n",
       "      <th>last_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>70-oral_microbiome</td>\n",
       "      <td>humann</td>\n",
       "      <td>8816</td>\n",
       "      <td>8816</td>\n",
       "      <td>2019-03-11 00:00:00</td>\n",
       "      <td>2022-12-21 00:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>70-oral_microbiome</td>\n",
       "      <td>humann_aggregated</td>\n",
       "      <td>8816</td>\n",
       "      <td>8816</td>\n",
       "      <td>2019-03-11 00:00:00</td>\n",
       "      <td>2022-12-21 00:00:00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               dataset        feature_set  count_col_notna  \\\n",
       "87  70-oral_microbiome             humann             8816   \n",
       "88  70-oral_microbiome  humann_aggregated             8816   \n",
       "\n",
       "    nunique_participants_not_na           first_date            last_date  \n",
       "87                         8816  2019-03-11 00:00:00  2022-12-21 00:00:00  \n",
       "88                         8816  2019-03-11 00:00:00  2022-12-21 00:00:00  "
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "df_res.loc[df_res['dataset']== '100-curated_phenotypes', 'first_date'] =  df_res.loc[df_res['dataset']== '1-events', 'first_date'].values[0]\n",
    "df_res.loc[df_res['dataset']== '100-curated_phenotypes', 'last_date'] = df_res.loc[df_res['dataset']== '1-events','last_date'].values[0]\n",
    "\n",
    "if BEFORE_2023:\n",
    "    df_res.loc[df_res['dataset'] == '100-curated_phenotypes', 'count_col_notna'] =  df_res.loc[df_res['dataset'] == '100-curated_phenotypes', 'count_col_notna'].clip(upper=df_res.loc[df_res['dataset'] == '1-events', 'count_col_notna'].iloc[0])\n",
    "    df_res.loc[df_res['dataset'] == '100-curated_phenotypes', 'nunique_participants_not_na'] =  df_res.loc[df_res['dataset'] == '100-curated_phenotypes', 'nunique_participants_not_na'].clip(upper=df_res.loc[df_res['dataset'] == '1-events', 'nunique_participants_not_na'].iloc[0])\n",
    "    \n",
    "\n",
    "df_res.tail(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "if BEFORE_2023 and not SHOW_FEATURE_SETS:\n",
    "    df_res = df_res[pd.notna(df_res['last_date'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset</th>\n",
       "      <th>feature_set</th>\n",
       "      <th>count_col_notna</th>\n",
       "      <th>nunique_participants_not_na</th>\n",
       "      <th>first_date</th>\n",
       "      <th>last_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0-population</td>\n",
       "      <td>population</td>\n",
       "      <td>12014</td>\n",
       "      <td>12014</td>\n",
       "      <td>2018-12-05 00:00:00</td>\n",
       "      <td>2022-12-29 00:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1-events</td>\n",
       "      <td>events</td>\n",
       "      <td>20723</td>\n",
       "      <td>10434</td>\n",
       "      <td>2018-12-05 00:00:00</td>\n",
       "      <td>2022-12-29 00:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2-anthropometrics</td>\n",
       "      <td>anthropometrics</td>\n",
       "      <td>12925</td>\n",
       "      <td>10387</td>\n",
       "      <td>2018-11-22 00:00:00</td>\n",
       "      <td>2022-12-29 00:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3-fundus</td>\n",
       "      <td>fundus</td>\n",
       "      <td>7068</td>\n",
       "      <td>7038</td>\n",
       "      <td>2021-02-17 00:00:00</td>\n",
       "      <td>2022-12-04 00:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3-fundus</td>\n",
       "      <td>microvasculature</td>\n",
       "      <td>7068</td>\n",
       "      <td>7038</td>\n",
       "      <td>2021-02-17 00:00:00</td>\n",
       "      <td>2022-12-04 00:00:00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             dataset       feature_set  count_col_notna  \\\n",
       "0       0-population        population            12014   \n",
       "1           1-events            events            20723   \n",
       "2  2-anthropometrics   anthropometrics            12925   \n",
       "3           3-fundus            fundus             7068   \n",
       "4           3-fundus  microvasculature             7068   \n",
       "\n",
       "   nunique_participants_not_na           first_date            last_date  \n",
       "0                        12014  2018-12-05 00:00:00  2022-12-29 00:00:00  \n",
       "1                        10434  2018-12-05 00:00:00  2022-12-29 00:00:00  \n",
       "2                        10387  2018-11-22 00:00:00  2022-12-29 00:00:00  \n",
       "3                         7038  2021-02-17 00:00:00  2022-12-04 00:00:00  \n",
       "4                         7038  2021-02-17 00:00:00  2022-12-04 00:00:00  "
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "dict_dataset = {\n",
    "    'population': 'events',\n",
    "    'microvasculature': 'fundus',\n",
    "    'segmented_images': 'fundus',\n",
    "    'hrv': 'sleep',\n",
    "    'hrv_raw': 'sleep',\n",
    "    'pca': 'human_genetics',\n",
    "    'iglu': 'iglu_daily', \n",
    "    \n",
    "}\n",
    "for k, v in dict_dataset.items():\n",
    "    df_res.loc[df_res['feature_set']== k, 'first_date'] =  df_res.loc[df_res['feature_set']== v, 'first_date'].values\n",
    "    df_res.loc[df_res['feature_set']== k, 'last_date'] = df_res.loc[df_res['feature_set']== v, 'last_date'].values\n",
    "\n",
    "df_res.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset</th>\n",
       "      <th>feature_set</th>\n",
       "      <th>count_col_notna</th>\n",
       "      <th>nunique_participants_not_na</th>\n",
       "      <th>first_date</th>\n",
       "      <th>last_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>70-oral_microbiome</td>\n",
       "      <td>humann</td>\n",
       "      <td>8816</td>\n",
       "      <td>8816</td>\n",
       "      <td>2019-03-11 00:00:00</td>\n",
       "      <td>2022-12-21 00:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>70-oral_microbiome</td>\n",
       "      <td>humann_aggregated</td>\n",
       "      <td>8816</td>\n",
       "      <td>8816</td>\n",
       "      <td>2019-03-11 00:00:00</td>\n",
       "      <td>2022-12-21 00:00:00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               dataset        feature_set  count_col_notna  \\\n",
       "87  70-oral_microbiome             humann             8816   \n",
       "88  70-oral_microbiome  humann_aggregated             8816   \n",
       "\n",
       "    nunique_participants_not_na           first_date            last_date  \n",
       "87                         8816  2019-03-11 00:00:00  2022-12-21 00:00:00  \n",
       "88                         8816  2019-03-11 00:00:00  2022-12-21 00:00:00  "
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_res.tail(2)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset</th>\n",
       "      <th>feature_set</th>\n",
       "      <th>count_col_notna</th>\n",
       "      <th>nunique_participants_not_na</th>\n",
       "      <th>first_date</th>\n",
       "      <th>last_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>17-cgm</td>\n",
       "      <td>cgm</td>\n",
       "      <td>9497</td>\n",
       "      <td>9431</td>\n",
       "      <td>2019-01-07 00:14:00+02:00</td>\n",
       "      <td>2022-12-30 00:14:00+02:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>17-cgm</td>\n",
       "      <td>timeseries</td>\n",
       "      <td>9497</td>\n",
       "      <td>9431</td>\n",
       "      <td>2019-01-07 00:14:00+02:00</td>\n",
       "      <td>2022-12-30 00:14:00+02:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>17-cgm</td>\n",
       "      <td>iglu</td>\n",
       "      <td>9497</td>\n",
       "      <td>9431</td>\n",
       "      <td>2019-01-07 00:00:00</td>\n",
       "      <td>2022-12-31 00:00:00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>17-cgm</td>\n",
       "      <td>iglu_daily</td>\n",
       "      <td>104845</td>\n",
       "      <td>9431</td>\n",
       "      <td>2019-01-07 00:00:00</td>\n",
       "      <td>2022-12-31 00:00:00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   dataset feature_set  count_col_notna  nunique_participants_not_na  \\\n",
       "45  17-cgm         cgm             9497                         9431   \n",
       "46  17-cgm  timeseries             9497                         9431   \n",
       "47  17-cgm        iglu             9497                         9431   \n",
       "48  17-cgm  iglu_daily           104845                         9431   \n",
       "\n",
       "                   first_date                  last_date  \n",
       "45  2019-01-07 00:14:00+02:00  2022-12-30 00:14:00+02:00  \n",
       "46  2019-01-07 00:14:00+02:00  2022-12-30 00:14:00+02:00  \n",
       "47        2019-01-07 00:00:00        2022-12-31 00:00:00  \n",
       "48        2019-01-07 00:00:00        2022-12-31 00:00:00  "
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_res.query('dataset == \"17-cgm\"')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "import datetime\n",
    "\n",
    "# Create a function to convert the date string to a datetime object\n",
    "def convert_date_string(date_string):\n",
    "    if date_string is None or date_string == 'None' or date_string == 'NaT' or date_string == 'nan':\n",
    "        return None\n",
    "    else:\n",
    "        date_string_without_timezone = date_string[:19]\n",
    "        datetime_object = datetime.datetime.strptime(date_string_without_timezone, '%Y-%m-%d %H:%M:%S')\n",
    "        date = datetime_object.date()\n",
    "        return date\n",
    "    \n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_convert = df_res.copy()\n",
    "\n",
    "\n",
    "df_convert[\"first_date\"] = df_convert[\"first_date\"].apply(str)\n",
    "df_convert[\"last_date\"] = df_convert[\"last_date\"].apply(str)\n",
    "\n",
    "# Convert all the dates in the DataFrame to datetime objects\n",
    "df_convert[\"first_date\"] = df_convert[\"first_date\"].apply(convert_date_string)\n",
    "df_convert[\"last_date\"] = df_convert[\"last_date\"].apply(convert_date_string)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset</th>\n",
       "      <th>feature_set</th>\n",
       "      <th>count_col_notna</th>\n",
       "      <th>nunique_participants_not_na</th>\n",
       "      <th>first_date</th>\n",
       "      <th>last_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>17-cgm</td>\n",
       "      <td>cgm</td>\n",
       "      <td>9497</td>\n",
       "      <td>9431</td>\n",
       "      <td>2019-01-07</td>\n",
       "      <td>2022-12-30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>17-cgm</td>\n",
       "      <td>timeseries</td>\n",
       "      <td>9497</td>\n",
       "      <td>9431</td>\n",
       "      <td>2019-01-07</td>\n",
       "      <td>2022-12-30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>17-cgm</td>\n",
       "      <td>iglu</td>\n",
       "      <td>9497</td>\n",
       "      <td>9431</td>\n",
       "      <td>2019-01-07</td>\n",
       "      <td>2022-12-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>17-cgm</td>\n",
       "      <td>iglu_daily</td>\n",
       "      <td>104845</td>\n",
       "      <td>9431</td>\n",
       "      <td>2019-01-07</td>\n",
       "      <td>2022-12-31</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   dataset feature_set  count_col_notna  nunique_participants_not_na  \\\n",
       "45  17-cgm         cgm             9497                         9431   \n",
       "46  17-cgm  timeseries             9497                         9431   \n",
       "47  17-cgm        iglu             9497                         9431   \n",
       "48  17-cgm  iglu_daily           104845                         9431   \n",
       "\n",
       "    first_date   last_date  \n",
       "45  2019-01-07  2022-12-30  \n",
       "46  2019-01-07  2022-12-30  \n",
       "47  2019-01-07  2022-12-31  \n",
       "48  2019-01-07  2022-12-31  "
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_convert.query('dataset == \"17-cgm\"')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(89, 6)"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_convert.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset</th>\n",
       "      <th>feature_set</th>\n",
       "      <th>count_col_notna</th>\n",
       "      <th>nunique_participants_not_na</th>\n",
       "      <th>first_date</th>\n",
       "      <th>last_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0-population</td>\n",
       "      <td>population</td>\n",
       "      <td>12014</td>\n",
       "      <td>12014</td>\n",
       "      <td>2018-12-05</td>\n",
       "      <td>2022-12-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1-events</td>\n",
       "      <td>events</td>\n",
       "      <td>20723</td>\n",
       "      <td>10434</td>\n",
       "      <td>2018-12-05</td>\n",
       "      <td>2022-12-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2-anthropometrics</td>\n",
       "      <td>anthropometrics</td>\n",
       "      <td>12925</td>\n",
       "      <td>10387</td>\n",
       "      <td>2018-11-22</td>\n",
       "      <td>2022-12-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3-fundus</td>\n",
       "      <td>fundus</td>\n",
       "      <td>7068</td>\n",
       "      <td>7038</td>\n",
       "      <td>2021-02-17</td>\n",
       "      <td>2022-12-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3-fundus</td>\n",
       "      <td>microvasculature</td>\n",
       "      <td>7068</td>\n",
       "      <td>7038</td>\n",
       "      <td>2021-02-17</td>\n",
       "      <td>2022-12-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>3-fundus</td>\n",
       "      <td>images</td>\n",
       "      <td>7068</td>\n",
       "      <td>7038</td>\n",
       "      <td>2021-02-17</td>\n",
       "      <td>2022-12-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>3-fundus</td>\n",
       "      <td>segmented_images</td>\n",
       "      <td>7068</td>\n",
       "      <td>7038</td>\n",
       "      <td>2021-02-17</td>\n",
       "      <td>2022-12-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>4-liver_ultrasound</td>\n",
       "      <td>liver_ultrasound</td>\n",
       "      <td>26684</td>\n",
       "      <td>8229</td>\n",
       "      <td>2020-02-20</td>\n",
       "      <td>2022-12-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>5-diet_logging</td>\n",
       "      <td>diet_logging</td>\n",
       "      <td>149912</td>\n",
       "      <td>9531</td>\n",
       "      <td>2019-09-01</td>\n",
       "      <td>2022-12-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>5-diet_logging</td>\n",
       "      <td>diet_logging_events</td>\n",
       "      <td>2640312</td>\n",
       "      <td>10197</td>\n",
       "      <td>2019-09-01</td>\n",
       "      <td>2023-03-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>5-diet_logging</td>\n",
       "      <td>raw_diet_logging_events</td>\n",
       "      <td>6483638</td>\n",
       "      <td>10696</td>\n",
       "      <td>2019-09-01</td>\n",
       "      <td>2023-03-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>6-hand_grip</td>\n",
       "      <td>hand_grip</td>\n",
       "      <td>12894</td>\n",
       "      <td>10371</td>\n",
       "      <td>2018-12-27</td>\n",
       "      <td>2022-12-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>7-blood_pressure</td>\n",
       "      <td>blood_pressure</td>\n",
       "      <td>12946</td>\n",
       "      <td>10398</td>\n",
       "      <td>2018-12-27</td>\n",
       "      <td>2022-12-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>8-serum_lipidomics</td>\n",
       "      <td>serum_lipidomics</td>\n",
       "      <td>6321</td>\n",
       "      <td>6144</td>\n",
       "      <td>2019-02-19</td>\n",
       "      <td>2021-08-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>9-sleep</td>\n",
       "      <td>sleep</td>\n",
       "      <td>21038</td>\n",
       "      <td>7009</td>\n",
       "      <td>2020-01-15</td>\n",
       "      <td>2022-12-27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>9-sleep</td>\n",
       "      <td>hrv</td>\n",
       "      <td>20030</td>\n",
       "      <td>6926</td>\n",
       "      <td>2020-01-15</td>\n",
       "      <td>2022-12-27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>9-sleep</td>\n",
       "      <td>channels_time_series</td>\n",
       "      <td>21072</td>\n",
       "      <td>6973</td>\n",
       "      <td>2020-01-15</td>\n",
       "      <td>2022-12-27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>9-sleep</td>\n",
       "      <td>channels_edf_file</td>\n",
       "      <td>21069</td>\n",
       "      <td>6973</td>\n",
       "      <td>2020-01-15</td>\n",
       "      <td>2022-12-27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>9-sleep</td>\n",
       "      <td>events_time_series</td>\n",
       "      <td>21372</td>\n",
       "      <td>7062</td>\n",
       "      <td>2020-01-15</td>\n",
       "      <td>2022-12-27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>10-vascular_health</td>\n",
       "      <td>vascular_health</td>\n",
       "      <td>11387</td>\n",
       "      <td>9729</td>\n",
       "      <td>2018-12-27</td>\n",
       "      <td>2022-12-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>11-body_composition</td>\n",
       "      <td>body_composition</td>\n",
       "      <td>8990</td>\n",
       "      <td>8411</td>\n",
       "      <td>2020-06-09</td>\n",
       "      <td>2022-12-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>12-carotid_ultrasound</td>\n",
       "      <td>carotid_ultrasound</td>\n",
       "      <td>8518</td>\n",
       "      <td>7988</td>\n",
       "      <td>2020-07-23</td>\n",
       "      <td>2022-12-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>13-gut_microbiome</td>\n",
       "      <td>gut_microbiome</td>\n",
       "      <td>10694</td>\n",
       "      <td>8838</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>2022-12-28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>13-gut_microbiome</td>\n",
       "      <td>raw_fastq</td>\n",
       "      <td>10694</td>\n",
       "      <td>8838</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>2022-12-28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>13-gut_microbiome</td>\n",
       "      <td>trimmed_fastq</td>\n",
       "      <td>10694</td>\n",
       "      <td>8838</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>2022-12-28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>13-gut_microbiome</td>\n",
       "      <td>nonhuman_fastq</td>\n",
       "      <td>10694</td>\n",
       "      <td>8838</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>2022-12-28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>13-gut_microbiome</td>\n",
       "      <td>fastqc</td>\n",
       "      <td>10694</td>\n",
       "      <td>8838</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>2022-12-28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>13-gut_microbiome</td>\n",
       "      <td>human_plink</td>\n",
       "      <td>10694</td>\n",
       "      <td>8838</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>2022-12-28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>13-gut_microbiome</td>\n",
       "      <td>urs_abundances_single</td>\n",
       "      <td>10694</td>\n",
       "      <td>8838</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>2022-12-28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>13-gut_microbiome</td>\n",
       "      <td>urs_abundances_aggregated</td>\n",
       "      <td>10694</td>\n",
       "      <td>8838</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>2022-12-28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>13-gut_microbiome</td>\n",
       "      <td>metaphlan_abundances_single</td>\n",
       "      <td>10694</td>\n",
       "      <td>8838</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>2022-12-28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>13-gut_microbiome</td>\n",
       "      <td>metaphlan_abundances_aggregated</td>\n",
       "      <td>10694</td>\n",
       "      <td>8838</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>2022-12-28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>14-human_genetics</td>\n",
       "      <td>human_genetics</td>\n",
       "      <td>8821</td>\n",
       "      <td>8821</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>14-human_genetics</td>\n",
       "      <td>pca</td>\n",
       "      <td>8820</td>\n",
       "      <td>8820</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>14-human_genetics</td>\n",
       "      <td>variants_qc</td>\n",
       "      <td>8821</td>\n",
       "      <td>8821</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>14-human_genetics</td>\n",
       "      <td>fastq_bam</td>\n",
       "      <td>8821</td>\n",
       "      <td>8821</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>14-human_genetics</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>8821</td>\n",
       "      <td>8821</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>14-human_genetics</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>8820</td>\n",
       "      <td>8820</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>14-human_genetics</td>\n",
       "      <td>relationship</td>\n",
       "      <td>8821</td>\n",
       "      <td>8821</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>15-ecg</td>\n",
       "      <td>ecg</td>\n",
       "      <td>9702</td>\n",
       "      <td>8085</td>\n",
       "      <td>2019-08-13</td>\n",
       "      <td>2022-12-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>15-ecg</td>\n",
       "      <td>raw_timeseries_wide</td>\n",
       "      <td>9441</td>\n",
       "      <td>7888</td>\n",
       "      <td>2019-08-13</td>\n",
       "      <td>2022-12-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>15-ecg</td>\n",
       "      <td>processed_timeseries_wide</td>\n",
       "      <td>9172</td>\n",
       "      <td>7711</td>\n",
       "      <td>2019-08-13</td>\n",
       "      <td>2022-12-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>15-ecg</td>\n",
       "      <td>raw_timeseries_long</td>\n",
       "      <td>9441</td>\n",
       "      <td>7888</td>\n",
       "      <td>2019-08-13</td>\n",
       "      <td>2022-12-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>15-ecg</td>\n",
       "      <td>processed_timeseries_long</td>\n",
       "      <td>9172</td>\n",
       "      <td>7711</td>\n",
       "      <td>2019-08-13</td>\n",
       "      <td>2022-12-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>16-blood_tests</td>\n",
       "      <td>blood_tests</td>\n",
       "      <td>29074</td>\n",
       "      <td>10252</td>\n",
       "      <td>2010-01-06</td>\n",
       "      <td>2022-12-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>17-cgm</td>\n",
       "      <td>cgm</td>\n",
       "      <td>9497</td>\n",
       "      <td>9431</td>\n",
       "      <td>2019-01-07</td>\n",
       "      <td>2022-12-30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>17-cgm</td>\n",
       "      <td>timeseries</td>\n",
       "      <td>9497</td>\n",
       "      <td>9431</td>\n",
       "      <td>2019-01-07</td>\n",
       "      <td>2022-12-30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>17-cgm</td>\n",
       "      <td>iglu</td>\n",
       "      <td>9497</td>\n",
       "      <td>9431</td>\n",
       "      <td>2019-01-07</td>\n",
       "      <td>2022-12-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>17-cgm</td>\n",
       "      <td>iglu_daily</td>\n",
       "      <td>104845</td>\n",
       "      <td>9431</td>\n",
       "      <td>2019-01-07</td>\n",
       "      <td>2022-12-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>22-bone_density</td>\n",
       "      <td>bone_density</td>\n",
       "      <td>9032</td>\n",
       "      <td>8459</td>\n",
       "      <td>2020-06-09</td>\n",
       "      <td>2022-12-29</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  dataset                      feature_set  count_col_notna  \\\n",
       "0            0-population                       population            12014   \n",
       "1                1-events                           events            20723   \n",
       "2       2-anthropometrics                  anthropometrics            12925   \n",
       "3                3-fundus                           fundus             7068   \n",
       "4                3-fundus                 microvasculature             7068   \n",
       "5                3-fundus                           images             7068   \n",
       "6                3-fundus                 segmented_images             7068   \n",
       "7      4-liver_ultrasound                 liver_ultrasound            26684   \n",
       "8          5-diet_logging                     diet_logging           149912   \n",
       "9          5-diet_logging              diet_logging_events          2640312   \n",
       "10         5-diet_logging          raw_diet_logging_events          6483638   \n",
       "11            6-hand_grip                        hand_grip            12894   \n",
       "12       7-blood_pressure                   blood_pressure            12946   \n",
       "13     8-serum_lipidomics                 serum_lipidomics             6321   \n",
       "14                9-sleep                            sleep            21038   \n",
       "15                9-sleep                              hrv            20030   \n",
       "16                9-sleep             channels_time_series            21072   \n",
       "17                9-sleep                channels_edf_file            21069   \n",
       "18                9-sleep               events_time_series            21372   \n",
       "19     10-vascular_health                  vascular_health            11387   \n",
       "20    11-body_composition                 body_composition             8990   \n",
       "21  12-carotid_ultrasound               carotid_ultrasound             8518   \n",
       "22      13-gut_microbiome                   gut_microbiome            10694   \n",
       "23      13-gut_microbiome                        raw_fastq            10694   \n",
       "24      13-gut_microbiome                    trimmed_fastq            10694   \n",
       "25      13-gut_microbiome                   nonhuman_fastq            10694   \n",
       "26      13-gut_microbiome                           fastqc            10694   \n",
       "27      13-gut_microbiome                      human_plink            10694   \n",
       "28      13-gut_microbiome            urs_abundances_single            10694   \n",
       "29      13-gut_microbiome        urs_abundances_aggregated            10694   \n",
       "30      13-gut_microbiome      metaphlan_abundances_single            10694   \n",
       "31      13-gut_microbiome  metaphlan_abundances_aggregated            10694   \n",
       "32      14-human_genetics                   human_genetics             8821   \n",
       "33      14-human_genetics                              pca             8820   \n",
       "34      14-human_genetics                      variants_qc             8821   \n",
       "35      14-human_genetics                        fastq_bam             8821   \n",
       "36      14-human_genetics                    genotypes_raw             8821   \n",
       "37      14-human_genetics                 genotypes_postqc             8820   \n",
       "38      14-human_genetics                     relationship             8821   \n",
       "39                 15-ecg                              ecg             9702   \n",
       "40                 15-ecg              raw_timeseries_wide             9441   \n",
       "41                 15-ecg        processed_timeseries_wide             9172   \n",
       "42                 15-ecg              raw_timeseries_long             9441   \n",
       "43                 15-ecg        processed_timeseries_long             9172   \n",
       "44         16-blood_tests                      blood_tests            29074   \n",
       "45                 17-cgm                              cgm             9497   \n",
       "46                 17-cgm                       timeseries             9497   \n",
       "47                 17-cgm                             iglu             9497   \n",
       "48                 17-cgm                       iglu_daily           104845   \n",
       "49        22-bone_density                     bone_density             9032   \n",
       "\n",
       "    nunique_participants_not_na  first_date   last_date  \n",
       "0                         12014  2018-12-05  2022-12-29  \n",
       "1                         10434  2018-12-05  2022-12-29  \n",
       "2                         10387  2018-11-22  2022-12-29  \n",
       "3                          7038  2021-02-17  2022-12-04  \n",
       "4                          7038  2021-02-17  2022-12-04  \n",
       "5                          7038  2021-02-17  2022-12-04  \n",
       "6                          7038  2021-02-17  2022-12-04  \n",
       "7                          8229  2020-02-20  2022-12-29  \n",
       "8                          9531  2019-09-01  2022-12-31  \n",
       "9                         10197  2019-09-01  2023-03-21  \n",
       "10                        10696  2019-09-01  2023-03-21  \n",
       "11                        10371  2018-12-27  2022-12-29  \n",
       "12                        10398  2018-12-27  2022-12-29  \n",
       "13                         6144  2019-02-19  2021-08-08  \n",
       "14                         7009  2020-01-15  2022-12-27  \n",
       "15                         6926  2020-01-15  2022-12-27  \n",
       "16                         6973  2020-01-15  2022-12-27  \n",
       "17                         6973  2020-01-15  2022-12-27  \n",
       "18                         7062  2020-01-15  2022-12-27  \n",
       "19                         9729  2018-12-27  2022-12-29  \n",
       "20                         8411  2020-06-09  2022-12-29  \n",
       "21                         7988  2020-07-23  2022-12-29  \n",
       "22                         8838  2019-02-14  2022-12-28  \n",
       "23                         8838  2019-02-14  2022-12-28  \n",
       "24                         8838  2019-02-14  2022-12-28  \n",
       "25                         8838  2019-02-14  2022-12-28  \n",
       "26                         8838  2019-02-14  2022-12-28  \n",
       "27                         8838  2019-02-14  2022-12-28  \n",
       "28                         8838  2019-02-14  2022-12-28  \n",
       "29                         8838  2019-02-14  2022-12-28  \n",
       "30                         8838  2019-02-14  2022-12-28  \n",
       "31                         8838  2019-02-14  2022-12-28  \n",
       "32                         8821  2019-03-11  2022-12-21  \n",
       "33                         8820  2019-03-11  2022-12-21  \n",
       "34                         8821  2019-03-11  2022-12-21  \n",
       "35                         8821  2019-03-11  2022-12-21  \n",
       "36                         8821  2019-03-11  2022-12-21  \n",
       "37                         8820  2019-03-11  2022-12-21  \n",
       "38                         8821  2019-03-11  2022-12-21  \n",
       "39                         8085  2019-08-13  2022-12-29  \n",
       "40                         7888  2019-08-13  2022-12-29  \n",
       "41                         7711  2019-08-13  2022-12-29  \n",
       "42                         7888  2019-08-13  2022-12-29  \n",
       "43                         7711  2019-08-13  2022-12-29  \n",
       "44                        10252  2010-01-06  2022-12-29  \n",
       "45                         9431  2019-01-07  2022-12-30  \n",
       "46                         9431  2019-01-07  2022-12-30  \n",
       "47                         9431  2019-01-07  2022-12-31  \n",
       "48                         9431  2019-01-07  2022-12-31  \n",
       "49                         8459  2020-06-09  2022-12-29  "
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_convert.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset</th>\n",
       "      <th>feature_set</th>\n",
       "      <th>count_col_notna</th>\n",
       "      <th>nunique_participants_not_na</th>\n",
       "      <th>first_date</th>\n",
       "      <th>last_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>15-ecg</td>\n",
       "      <td>ecg</td>\n",
       "      <td>9702</td>\n",
       "      <td>8085</td>\n",
       "      <td>2019-08-13</td>\n",
       "      <td>2022-12-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>15-ecg</td>\n",
       "      <td>raw_timeseries_wide</td>\n",
       "      <td>9441</td>\n",
       "      <td>7888</td>\n",
       "      <td>2019-08-13</td>\n",
       "      <td>2022-12-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>15-ecg</td>\n",
       "      <td>processed_timeseries_wide</td>\n",
       "      <td>9172</td>\n",
       "      <td>7711</td>\n",
       "      <td>2019-08-13</td>\n",
       "      <td>2022-12-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>15-ecg</td>\n",
       "      <td>raw_timeseries_long</td>\n",
       "      <td>9441</td>\n",
       "      <td>7888</td>\n",
       "      <td>2019-08-13</td>\n",
       "      <td>2022-12-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>15-ecg</td>\n",
       "      <td>processed_timeseries_long</td>\n",
       "      <td>9172</td>\n",
       "      <td>7711</td>\n",
       "      <td>2019-08-13</td>\n",
       "      <td>2022-12-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>16-blood_tests</td>\n",
       "      <td>blood_tests</td>\n",
       "      <td>29074</td>\n",
       "      <td>10252</td>\n",
       "      <td>2010-01-06</td>\n",
       "      <td>2022-12-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>17-cgm</td>\n",
       "      <td>cgm</td>\n",
       "      <td>9497</td>\n",
       "      <td>9431</td>\n",
       "      <td>2019-01-07</td>\n",
       "      <td>2022-12-30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>17-cgm</td>\n",
       "      <td>timeseries</td>\n",
       "      <td>9497</td>\n",
       "      <td>9431</td>\n",
       "      <td>2019-01-07</td>\n",
       "      <td>2022-12-30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>17-cgm</td>\n",
       "      <td>iglu</td>\n",
       "      <td>9497</td>\n",
       "      <td>9431</td>\n",
       "      <td>2019-01-07</td>\n",
       "      <td>2022-12-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>17-cgm</td>\n",
       "      <td>iglu_daily</td>\n",
       "      <td>104845</td>\n",
       "      <td>9431</td>\n",
       "      <td>2019-01-07</td>\n",
       "      <td>2022-12-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>22-bone_density</td>\n",
       "      <td>bone_density</td>\n",
       "      <td>9032</td>\n",
       "      <td>8459</td>\n",
       "      <td>2020-06-09</td>\n",
       "      <td>2022-12-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>18-medications</td>\n",
       "      <td>medications</td>\n",
       "      <td>43417</td>\n",
       "      <td>10787</td>\n",
       "      <td>2018-10-18</td>\n",
       "      <td>2022-12-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>21-medical_conditions</td>\n",
       "      <td>medical_conditions</td>\n",
       "      <td>22413</td>\n",
       "      <td>10792</td>\n",
       "      <td>2018-11-21</td>\n",
       "      <td>2022-12-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>49-medical_symptoms</td>\n",
       "      <td>initial_medical</td>\n",
       "      <td>7416</td>\n",
       "      <td>7383</td>\n",
       "      <td>2020-02-26</td>\n",
       "      <td>2022-12-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>49-medical_symptoms</td>\n",
       "      <td>follow_up_ukbb</td>\n",
       "      <td>5710</td>\n",
       "      <td>4471</td>\n",
       "      <td>2020-02-17</td>\n",
       "      <td>2022-12-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>49-medical_symptoms</td>\n",
       "      <td>digestive_health</td>\n",
       "      <td>9172</td>\n",
       "      <td>8719</td>\n",
       "      <td>2019-01-09</td>\n",
       "      <td>2022-12-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>50-medical_procedures</td>\n",
       "      <td>initial_medical</td>\n",
       "      <td>7416</td>\n",
       "      <td>7383</td>\n",
       "      <td>2020-02-26</td>\n",
       "      <td>2022-12-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>50-medical_procedures</td>\n",
       "      <td>follow_up_medical</td>\n",
       "      <td>7572</td>\n",
       "      <td>5869</td>\n",
       "      <td>2021-04-22</td>\n",
       "      <td>2022-12-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>43-covid</td>\n",
       "      <td>covid</td>\n",
       "      <td>14988</td>\n",
       "      <td>9120</td>\n",
       "      <td>2021-04-22</td>\n",
       "      <td>2022-12-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>44-sex_specific_factors</td>\n",
       "      <td>menstrual_cycle_survey</td>\n",
       "      <td>1405</td>\n",
       "      <td>1405</td>\n",
       "      <td>2022-06-12</td>\n",
       "      <td>2022-12-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>44-sex_specific_factors</td>\n",
       "      <td>initial_medical</td>\n",
       "      <td>7416</td>\n",
       "      <td>7383</td>\n",
       "      <td>2020-02-26</td>\n",
       "      <td>2022-12-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>44-sex_specific_factors</td>\n",
       "      <td>ukbb</td>\n",
       "      <td>9151</td>\n",
       "      <td>8759</td>\n",
       "      <td>2019-01-09</td>\n",
       "      <td>2022-12-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>44-sex_specific_factors</td>\n",
       "      <td>follow_up_ukbb</td>\n",
       "      <td>5710</td>\n",
       "      <td>4471</td>\n",
       "      <td>2020-02-17</td>\n",
       "      <td>2022-12-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>44-sex_specific_factors</td>\n",
       "      <td>follow_up_medical</td>\n",
       "      <td>7572</td>\n",
       "      <td>5869</td>\n",
       "      <td>2021-04-22</td>\n",
       "      <td>2022-12-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>52-family_history</td>\n",
       "      <td>initial_medical</td>\n",
       "      <td>7416</td>\n",
       "      <td>7383</td>\n",
       "      <td>2020-02-26</td>\n",
       "      <td>2022-12-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>52-family_history</td>\n",
       "      <td>ukbb</td>\n",
       "      <td>14861</td>\n",
       "      <td>9262</td>\n",
       "      <td>2019-01-09</td>\n",
       "      <td>2022-12-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>51-children</td>\n",
       "      <td>children</td>\n",
       "      <td>7416</td>\n",
       "      <td>7383</td>\n",
       "      <td>2020-02-26</td>\n",
       "      <td>2022-12-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>53-sociodemographics</td>\n",
       "      <td>ukbb</td>\n",
       "      <td>14861</td>\n",
       "      <td>9262</td>\n",
       "      <td>2019-01-09</td>\n",
       "      <td>2022-12-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>53-sociodemographics</td>\n",
       "      <td>initial_medical</td>\n",
       "      <td>7416</td>\n",
       "      <td>7383</td>\n",
       "      <td>2020-02-26</td>\n",
       "      <td>2022-12-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>55-lifestyle_and_environment</td>\n",
       "      <td>lifestyle_and_environment</td>\n",
       "      <td>14861</td>\n",
       "      <td>9262</td>\n",
       "      <td>2019-01-09</td>\n",
       "      <td>2022-12-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>57-psychological_and_social_health</td>\n",
       "      <td>psychological_and_social_health</td>\n",
       "      <td>14861</td>\n",
       "      <td>9262</td>\n",
       "      <td>2019-01-09</td>\n",
       "      <td>2022-12-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>58-health_and_medical_history</td>\n",
       "      <td>ukbb</td>\n",
       "      <td>14861</td>\n",
       "      <td>9262</td>\n",
       "      <td>2019-01-09</td>\n",
       "      <td>2022-12-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>58-health_and_medical_history</td>\n",
       "      <td>initial_medical</td>\n",
       "      <td>10197</td>\n",
       "      <td>10197</td>\n",
       "      <td>2018-11-21</td>\n",
       "      <td>2022-12-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>45-participant_relation</td>\n",
       "      <td>participant relation</td>\n",
       "      <td>3739</td>\n",
       "      <td>3106</td>\n",
       "      <td>2019-02-05</td>\n",
       "      <td>2022-12-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>24-rna_seq</td>\n",
       "      <td>rna_seq</td>\n",
       "      <td>3958</td>\n",
       "      <td>3863</td>\n",
       "      <td>2021-02-28</td>\n",
       "      <td>2022-06-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>24-rna_seq</td>\n",
       "      <td>raw_fastq</td>\n",
       "      <td>3958</td>\n",
       "      <td>3863</td>\n",
       "      <td>2021-02-28</td>\n",
       "      <td>2022-06-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>24-rna_seq</td>\n",
       "      <td>trimmed_fastq</td>\n",
       "      <td>3958</td>\n",
       "      <td>3863</td>\n",
       "      <td>2021-02-28</td>\n",
       "      <td>2022-06-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>24-rna_seq</td>\n",
       "      <td>aligned_bam</td>\n",
       "      <td>3958</td>\n",
       "      <td>3863</td>\n",
       "      <td>2021-02-28</td>\n",
       "      <td>2022-06-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>24-rna_seq</td>\n",
       "      <td>deduplicated_bam</td>\n",
       "      <td>3958</td>\n",
       "      <td>3863</td>\n",
       "      <td>2021-02-28</td>\n",
       "      <td>2022-06-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>24-rna_seq</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>3958</td>\n",
       "      <td>3863</td>\n",
       "      <td>2021-02-28</td>\n",
       "      <td>2022-06-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>29-nightingale_metabolomics</td>\n",
       "      <td>nightingale_metabolomics</td>\n",
       "      <td>1644</td>\n",
       "      <td>1552</td>\n",
       "      <td>2019-05-07</td>\n",
       "      <td>2022-08-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>29-nightingale_metabolomics</td>\n",
       "      <td>bimarker_qc_tags</td>\n",
       "      <td>1644</td>\n",
       "      <td>1552</td>\n",
       "      <td>2019-05-07</td>\n",
       "      <td>2022-08-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>70-oral_microbiome</td>\n",
       "      <td>oral_microbiome</td>\n",
       "      <td>8816</td>\n",
       "      <td>8816</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>70-oral_microbiome</td>\n",
       "      <td>raw_fastq</td>\n",
       "      <td>8816</td>\n",
       "      <td>8816</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>70-oral_microbiome</td>\n",
       "      <td>trimmed_fastq</td>\n",
       "      <td>8816</td>\n",
       "      <td>8816</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>70-oral_microbiome</td>\n",
       "      <td>fastqc</td>\n",
       "      <td>8816</td>\n",
       "      <td>8816</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>70-oral_microbiome</td>\n",
       "      <td>metaphlan</td>\n",
       "      <td>8816</td>\n",
       "      <td>8816</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>70-oral_microbiome</td>\n",
       "      <td>metaphlan_aggregated</td>\n",
       "      <td>8816</td>\n",
       "      <td>8816</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>70-oral_microbiome</td>\n",
       "      <td>humann</td>\n",
       "      <td>8816</td>\n",
       "      <td>8816</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>70-oral_microbiome</td>\n",
       "      <td>humann_aggregated</td>\n",
       "      <td>8816</td>\n",
       "      <td>8816</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               dataset                      feature_set  \\\n",
       "39                              15-ecg                              ecg   \n",
       "40                              15-ecg              raw_timeseries_wide   \n",
       "41                              15-ecg        processed_timeseries_wide   \n",
       "42                              15-ecg              raw_timeseries_long   \n",
       "43                              15-ecg        processed_timeseries_long   \n",
       "44                      16-blood_tests                      blood_tests   \n",
       "45                              17-cgm                              cgm   \n",
       "46                              17-cgm                       timeseries   \n",
       "47                              17-cgm                             iglu   \n",
       "48                              17-cgm                       iglu_daily   \n",
       "49                     22-bone_density                     bone_density   \n",
       "50                      18-medications                      medications   \n",
       "51               21-medical_conditions               medical_conditions   \n",
       "52                 49-medical_symptoms                  initial_medical   \n",
       "53                 49-medical_symptoms                   follow_up_ukbb   \n",
       "54                 49-medical_symptoms                 digestive_health   \n",
       "55               50-medical_procedures                  initial_medical   \n",
       "56               50-medical_procedures                follow_up_medical   \n",
       "57                            43-covid                            covid   \n",
       "58             44-sex_specific_factors           menstrual_cycle_survey   \n",
       "59             44-sex_specific_factors                  initial_medical   \n",
       "60             44-sex_specific_factors                             ukbb   \n",
       "61             44-sex_specific_factors                   follow_up_ukbb   \n",
       "62             44-sex_specific_factors                follow_up_medical   \n",
       "63                   52-family_history                  initial_medical   \n",
       "64                   52-family_history                             ukbb   \n",
       "65                         51-children                         children   \n",
       "66                53-sociodemographics                             ukbb   \n",
       "67                53-sociodemographics                  initial_medical   \n",
       "68        55-lifestyle_and_environment        lifestyle_and_environment   \n",
       "69  57-psychological_and_social_health  psychological_and_social_health   \n",
       "70       58-health_and_medical_history                             ukbb   \n",
       "71       58-health_and_medical_history                  initial_medical   \n",
       "72             45-participant_relation             participant relation   \n",
       "73                          24-rna_seq                          rna_seq   \n",
       "74                          24-rna_seq                        raw_fastq   \n",
       "75                          24-rna_seq                    trimmed_fastq   \n",
       "76                          24-rna_seq                      aligned_bam   \n",
       "77                          24-rna_seq                 deduplicated_bam   \n",
       "78                          24-rna_seq                          multiqc   \n",
       "79         29-nightingale_metabolomics         nightingale_metabolomics   \n",
       "80         29-nightingale_metabolomics                 bimarker_qc_tags   \n",
       "81                  70-oral_microbiome                  oral_microbiome   \n",
       "82                  70-oral_microbiome                        raw_fastq   \n",
       "83                  70-oral_microbiome                    trimmed_fastq   \n",
       "84                  70-oral_microbiome                           fastqc   \n",
       "85                  70-oral_microbiome                        metaphlan   \n",
       "86                  70-oral_microbiome             metaphlan_aggregated   \n",
       "87                  70-oral_microbiome                           humann   \n",
       "88                  70-oral_microbiome                humann_aggregated   \n",
       "\n",
       "    count_col_notna  nunique_participants_not_na  first_date   last_date  \n",
       "39             9702                         8085  2019-08-13  2022-12-29  \n",
       "40             9441                         7888  2019-08-13  2022-12-29  \n",
       "41             9172                         7711  2019-08-13  2022-12-29  \n",
       "42             9441                         7888  2019-08-13  2022-12-29  \n",
       "43             9172                         7711  2019-08-13  2022-12-29  \n",
       "44            29074                        10252  2010-01-06  2022-12-29  \n",
       "45             9497                         9431  2019-01-07  2022-12-30  \n",
       "46             9497                         9431  2019-01-07  2022-12-30  \n",
       "47             9497                         9431  2019-01-07  2022-12-31  \n",
       "48           104845                         9431  2019-01-07  2022-12-31  \n",
       "49             9032                         8459  2020-06-09  2022-12-29  \n",
       "50            43417                        10787  2018-10-18  2022-12-31  \n",
       "51            22413                        10792  2018-11-21  2022-12-31  \n",
       "52             7416                         7383  2020-02-26  2022-12-31  \n",
       "53             5710                         4471  2020-02-17  2022-12-31  \n",
       "54             9172                         8719  2019-01-09  2022-12-31  \n",
       "55             7416                         7383  2020-02-26  2022-12-31  \n",
       "56             7572                         5869  2021-04-22  2022-12-29  \n",
       "57            14988                         9120  2021-04-22  2022-12-29  \n",
       "58             1405                         1405  2022-06-12  2022-12-29  \n",
       "59             7416                         7383  2020-02-26  2022-12-31  \n",
       "60             9151                         8759  2019-01-09  2022-12-31  \n",
       "61             5710                         4471  2020-02-17  2022-12-31  \n",
       "62             7572                         5869  2021-04-22  2022-12-29  \n",
       "63             7416                         7383  2020-02-26  2022-12-31  \n",
       "64            14861                         9262  2019-01-09  2022-12-31  \n",
       "65             7416                         7383  2020-02-26  2022-12-31  \n",
       "66            14861                         9262  2019-01-09  2022-12-31  \n",
       "67             7416                         7383  2020-02-26  2022-12-31  \n",
       "68            14861                         9262  2019-01-09  2022-12-31  \n",
       "69            14861                         9262  2019-01-09  2022-12-31  \n",
       "70            14861                         9262  2019-01-09  2022-12-31  \n",
       "71            10197                        10197  2018-11-21  2022-12-31  \n",
       "72             3739                         3106  2019-02-05  2022-12-29  \n",
       "73             3958                         3863  2021-02-28  2022-06-21  \n",
       "74             3958                         3863  2021-02-28  2022-06-21  \n",
       "75             3958                         3863  2021-02-28  2022-06-21  \n",
       "76             3958                         3863  2021-02-28  2022-06-21  \n",
       "77             3958                         3863  2021-02-28  2022-06-21  \n",
       "78             3958                         3863  2021-02-28  2022-06-21  \n",
       "79             1644                         1552  2019-05-07  2022-08-02  \n",
       "80             1644                         1552  2019-05-07  2022-08-02  \n",
       "81             8816                         8816  2019-03-11  2022-12-21  \n",
       "82             8816                         8816  2019-03-11  2022-12-21  \n",
       "83             8816                         8816  2019-03-11  2022-12-21  \n",
       "84             8816                         8816  2019-03-11  2022-12-21  \n",
       "85             8816                         8816  2019-03-11  2022-12-21  \n",
       "86             8816                         8816  2019-03-11  2022-12-21  \n",
       "87             8816                         8816  2019-03-11  2022-12-21  \n",
       "88             8816                         8816  2019-03-11  2022-12-21  "
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_convert.tail(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Function to split 'dataset' column\n",
    "def split_dataset(value):\n",
    "    try:\n",
    "        id_, dataset = value.split('-', 1)\n",
    "    except ValueError:\n",
    "        id_ = None\n",
    "        dataset = value\n",
    "    return pd.Series([id_, dataset])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df_convert.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3603/3306964056.py:2: FutureWarning: Returning a DataFrame from Series.apply when the supplied function returns a Series is deprecated and will be removed in a future version.\n",
      "  df[['id', 'dataset']] = df['dataset'].apply(split_dataset)\n"
     ]
    }
   ],
   "source": [
    "\n",
    "# Apply the function\n",
    "df[['id', 'dataset']] = df['dataset'].apply(split_dataset)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset</th>\n",
       "      <th>feature_set</th>\n",
       "      <th>count_col_notna</th>\n",
       "      <th>nunique_participants_not_na</th>\n",
       "      <th>first_date</th>\n",
       "      <th>last_date</th>\n",
       "      <th>id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>population</td>\n",
       "      <td>population</td>\n",
       "      <td>12014</td>\n",
       "      <td>12014</td>\n",
       "      <td>2018-12-05</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>events</td>\n",
       "      <td>events</td>\n",
       "      <td>20723</td>\n",
       "      <td>10434</td>\n",
       "      <td>2018-12-05</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>anthropometrics</td>\n",
       "      <td>anthropometrics</td>\n",
       "      <td>12925</td>\n",
       "      <td>10387</td>\n",
       "      <td>2018-11-22</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>fundus</td>\n",
       "      <td>fundus</td>\n",
       "      <td>7068</td>\n",
       "      <td>7038</td>\n",
       "      <td>2021-02-17</td>\n",
       "      <td>2022-12-04</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>fundus</td>\n",
       "      <td>microvasculature</td>\n",
       "      <td>7068</td>\n",
       "      <td>7038</td>\n",
       "      <td>2021-02-17</td>\n",
       "      <td>2022-12-04</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           dataset       feature_set  count_col_notna  \\\n",
       "0       population        population            12014   \n",
       "1           events            events            20723   \n",
       "2  anthropometrics   anthropometrics            12925   \n",
       "3           fundus            fundus             7068   \n",
       "4           fundus  microvasculature             7068   \n",
       "\n",
       "   nunique_participants_not_na  first_date   last_date id  \n",
       "0                        12014  2018-12-05  2022-12-29  0  \n",
       "1                        10434  2018-12-05  2022-12-29  1  \n",
       "2                        10387  2018-11-22  2022-12-29  2  \n",
       "3                         7038  2021-02-17  2022-12-04  3  \n",
       "4                         7038  2021-02-17  2022-12-04  3  "
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset</th>\n",
       "      <th>feature_set</th>\n",
       "      <th>count_col_notna</th>\n",
       "      <th>nunique_participants_not_na</th>\n",
       "      <th>first_date</th>\n",
       "      <th>last_date</th>\n",
       "      <th>id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>oral_microbiome</td>\n",
       "      <td>fastqc</td>\n",
       "      <td>8816</td>\n",
       "      <td>8816</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>oral_microbiome</td>\n",
       "      <td>metaphlan</td>\n",
       "      <td>8816</td>\n",
       "      <td>8816</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>oral_microbiome</td>\n",
       "      <td>metaphlan_aggregated</td>\n",
       "      <td>8816</td>\n",
       "      <td>8816</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>oral_microbiome</td>\n",
       "      <td>humann</td>\n",
       "      <td>8816</td>\n",
       "      <td>8816</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>oral_microbiome</td>\n",
       "      <td>humann_aggregated</td>\n",
       "      <td>8816</td>\n",
       "      <td>8816</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            dataset           feature_set  count_col_notna  \\\n",
       "84  oral_microbiome                fastqc             8816   \n",
       "85  oral_microbiome             metaphlan             8816   \n",
       "86  oral_microbiome  metaphlan_aggregated             8816   \n",
       "87  oral_microbiome                humann             8816   \n",
       "88  oral_microbiome     humann_aggregated             8816   \n",
       "\n",
       "    nunique_participants_not_na  first_date   last_date  id  \n",
       "84                         8816  2019-03-11  2022-12-21  70  \n",
       "85                         8816  2019-03-11  2022-12-21  70  \n",
       "86                         8816  2019-03-11  2022-12-21  70  \n",
       "87                         8816  2019-03-11  2022-12-21  70  \n",
       "88                         8816  2019-03-11  2022-12-21  70  "
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset</th>\n",
       "      <th>feature_set</th>\n",
       "      <th>count_col_notna</th>\n",
       "      <th>nunique_participants_not_na</th>\n",
       "      <th>first_date</th>\n",
       "      <th>last_date</th>\n",
       "      <th>id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>population</td>\n",
       "      <td>population</td>\n",
       "      <td>12014</td>\n",
       "      <td>12014</td>\n",
       "      <td>2018-12-05</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>events</td>\n",
       "      <td>events</td>\n",
       "      <td>20723</td>\n",
       "      <td>10434</td>\n",
       "      <td>2018-12-05</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>anthropometrics</td>\n",
       "      <td>anthropometrics</td>\n",
       "      <td>12925</td>\n",
       "      <td>10387</td>\n",
       "      <td>2018-11-22</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>fundus</td>\n",
       "      <td>fundus</td>\n",
       "      <td>7068</td>\n",
       "      <td>7038</td>\n",
       "      <td>2021-02-17</td>\n",
       "      <td>2022-12-04</td>\n",
       "      <td>003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>fundus</td>\n",
       "      <td>microvasculature</td>\n",
       "      <td>7068</td>\n",
       "      <td>7038</td>\n",
       "      <td>2021-02-17</td>\n",
       "      <td>2022-12-04</td>\n",
       "      <td>003</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           dataset       feature_set  count_col_notna  \\\n",
       "0       population        population            12014   \n",
       "1           events            events            20723   \n",
       "2  anthropometrics   anthropometrics            12925   \n",
       "3           fundus            fundus             7068   \n",
       "4           fundus  microvasculature             7068   \n",
       "\n",
       "   nunique_participants_not_na  first_date   last_date   id  \n",
       "0                        12014  2018-12-05  2022-12-29  000  \n",
       "1                        10434  2018-12-05  2022-12-29  001  \n",
       "2                        10387  2018-11-22  2022-12-29  002  \n",
       "3                         7038  2021-02-17  2022-12-04  003  \n",
       "4                         7038  2021-02-17  2022-12-04  003  "
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Function to format the numbers\n",
    "def format_number(x):\n",
    "    return f'{int(x):03d}'\n",
    "\n",
    "# Apply the function to the desired column\n",
    "df['id'] = df['id'].apply(format_number)\n",
    " \n",
    "df.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset</th>\n",
       "      <th>feature_set</th>\n",
       "      <th>count_col_notna</th>\n",
       "      <th>nunique_participants_not_na</th>\n",
       "      <th>first_date</th>\n",
       "      <th>last_date</th>\n",
       "      <th>id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>population</td>\n",
       "      <td>population</td>\n",
       "      <td>12014</td>\n",
       "      <td>12014</td>\n",
       "      <td>2018-12-05</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>events</td>\n",
       "      <td>events</td>\n",
       "      <td>20723</td>\n",
       "      <td>10434</td>\n",
       "      <td>2018-12-05</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>anthropometrics</td>\n",
       "      <td>anthropometrics</td>\n",
       "      <td>12925</td>\n",
       "      <td>10387</td>\n",
       "      <td>2018-11-22</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>fundus</td>\n",
       "      <td>fundus</td>\n",
       "      <td>7068</td>\n",
       "      <td>7038</td>\n",
       "      <td>2021-02-17</td>\n",
       "      <td>2022-12-04</td>\n",
       "      <td>003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>fundus</td>\n",
       "      <td>microvasculature</td>\n",
       "      <td>7068</td>\n",
       "      <td>7038</td>\n",
       "      <td>2021-02-17</td>\n",
       "      <td>2022-12-04</td>\n",
       "      <td>003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>fundus</td>\n",
       "      <td>images</td>\n",
       "      <td>7068</td>\n",
       "      <td>7038</td>\n",
       "      <td>2021-02-17</td>\n",
       "      <td>2022-12-04</td>\n",
       "      <td>003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>fundus</td>\n",
       "      <td>segmented_images</td>\n",
       "      <td>7068</td>\n",
       "      <td>7038</td>\n",
       "      <td>2021-02-17</td>\n",
       "      <td>2022-12-04</td>\n",
       "      <td>003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>liver_ultrasound</td>\n",
       "      <td>liver_ultrasound</td>\n",
       "      <td>26684</td>\n",
       "      <td>8229</td>\n",
       "      <td>2020-02-20</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>diet_logging</td>\n",
       "      <td>diet_logging</td>\n",
       "      <td>149912</td>\n",
       "      <td>9531</td>\n",
       "      <td>2019-09-01</td>\n",
       "      <td>2022-12-31</td>\n",
       "      <td>005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>diet_logging</td>\n",
       "      <td>diet_logging_events</td>\n",
       "      <td>2640312</td>\n",
       "      <td>10197</td>\n",
       "      <td>2019-09-01</td>\n",
       "      <td>2023-03-21</td>\n",
       "      <td>005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>diet_logging</td>\n",
       "      <td>raw_diet_logging_events</td>\n",
       "      <td>6483638</td>\n",
       "      <td>10696</td>\n",
       "      <td>2019-09-01</td>\n",
       "      <td>2023-03-21</td>\n",
       "      <td>005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>hand_grip</td>\n",
       "      <td>hand_grip</td>\n",
       "      <td>12894</td>\n",
       "      <td>10371</td>\n",
       "      <td>2018-12-27</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>blood_pressure</td>\n",
       "      <td>blood_pressure</td>\n",
       "      <td>12946</td>\n",
       "      <td>10398</td>\n",
       "      <td>2018-12-27</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>serum_lipidomics</td>\n",
       "      <td>serum_lipidomics</td>\n",
       "      <td>6321</td>\n",
       "      <td>6144</td>\n",
       "      <td>2019-02-19</td>\n",
       "      <td>2021-08-08</td>\n",
       "      <td>008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>sleep</td>\n",
       "      <td>sleep</td>\n",
       "      <td>21038</td>\n",
       "      <td>7009</td>\n",
       "      <td>2020-01-15</td>\n",
       "      <td>2022-12-27</td>\n",
       "      <td>009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>sleep</td>\n",
       "      <td>hrv</td>\n",
       "      <td>20030</td>\n",
       "      <td>6926</td>\n",
       "      <td>2020-01-15</td>\n",
       "      <td>2022-12-27</td>\n",
       "      <td>009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>sleep</td>\n",
       "      <td>channels_time_series</td>\n",
       "      <td>21072</td>\n",
       "      <td>6973</td>\n",
       "      <td>2020-01-15</td>\n",
       "      <td>2022-12-27</td>\n",
       "      <td>009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>sleep</td>\n",
       "      <td>channels_edf_file</td>\n",
       "      <td>21069</td>\n",
       "      <td>6973</td>\n",
       "      <td>2020-01-15</td>\n",
       "      <td>2022-12-27</td>\n",
       "      <td>009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>sleep</td>\n",
       "      <td>events_time_series</td>\n",
       "      <td>21372</td>\n",
       "      <td>7062</td>\n",
       "      <td>2020-01-15</td>\n",
       "      <td>2022-12-27</td>\n",
       "      <td>009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>vascular_health</td>\n",
       "      <td>vascular_health</td>\n",
       "      <td>11387</td>\n",
       "      <td>9729</td>\n",
       "      <td>2018-12-27</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>body_composition</td>\n",
       "      <td>body_composition</td>\n",
       "      <td>8990</td>\n",
       "      <td>8411</td>\n",
       "      <td>2020-06-09</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>carotid_ultrasound</td>\n",
       "      <td>carotid_ultrasound</td>\n",
       "      <td>8518</td>\n",
       "      <td>7988</td>\n",
       "      <td>2020-07-23</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>gut_microbiome</td>\n",
       "      <td>gut_microbiome</td>\n",
       "      <td>10694</td>\n",
       "      <td>8838</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>2022-12-28</td>\n",
       "      <td>013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>gut_microbiome</td>\n",
       "      <td>raw_fastq</td>\n",
       "      <td>10694</td>\n",
       "      <td>8838</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>2022-12-28</td>\n",
       "      <td>013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>gut_microbiome</td>\n",
       "      <td>trimmed_fastq</td>\n",
       "      <td>10694</td>\n",
       "      <td>8838</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>2022-12-28</td>\n",
       "      <td>013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>gut_microbiome</td>\n",
       "      <td>nonhuman_fastq</td>\n",
       "      <td>10694</td>\n",
       "      <td>8838</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>2022-12-28</td>\n",
       "      <td>013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>gut_microbiome</td>\n",
       "      <td>fastqc</td>\n",
       "      <td>10694</td>\n",
       "      <td>8838</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>2022-12-28</td>\n",
       "      <td>013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>gut_microbiome</td>\n",
       "      <td>human_plink</td>\n",
       "      <td>10694</td>\n",
       "      <td>8838</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>2022-12-28</td>\n",
       "      <td>013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>gut_microbiome</td>\n",
       "      <td>urs_abundances_single</td>\n",
       "      <td>10694</td>\n",
       "      <td>8838</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>2022-12-28</td>\n",
       "      <td>013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>gut_microbiome</td>\n",
       "      <td>urs_abundances_aggregated</td>\n",
       "      <td>10694</td>\n",
       "      <td>8838</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>2022-12-28</td>\n",
       "      <td>013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>gut_microbiome</td>\n",
       "      <td>metaphlan_abundances_single</td>\n",
       "      <td>10694</td>\n",
       "      <td>8838</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>2022-12-28</td>\n",
       "      <td>013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>gut_microbiome</td>\n",
       "      <td>metaphlan_abundances_aggregated</td>\n",
       "      <td>10694</td>\n",
       "      <td>8838</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>2022-12-28</td>\n",
       "      <td>013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>human_genetics</td>\n",
       "      <td>human_genetics</td>\n",
       "      <td>8821</td>\n",
       "      <td>8821</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "      <td>014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>human_genetics</td>\n",
       "      <td>pca</td>\n",
       "      <td>8820</td>\n",
       "      <td>8820</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "      <td>014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>human_genetics</td>\n",
       "      <td>variants_qc</td>\n",
       "      <td>8821</td>\n",
       "      <td>8821</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "      <td>014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>human_genetics</td>\n",
       "      <td>fastq_bam</td>\n",
       "      <td>8821</td>\n",
       "      <td>8821</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "      <td>014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>human_genetics</td>\n",
       "      <td>genotypes_raw</td>\n",
       "      <td>8821</td>\n",
       "      <td>8821</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "      <td>014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>human_genetics</td>\n",
       "      <td>genotypes_postqc</td>\n",
       "      <td>8820</td>\n",
       "      <td>8820</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "      <td>014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>human_genetics</td>\n",
       "      <td>relationship</td>\n",
       "      <td>8821</td>\n",
       "      <td>8821</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "      <td>014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>ecg</td>\n",
       "      <td>ecg</td>\n",
       "      <td>9702</td>\n",
       "      <td>8085</td>\n",
       "      <td>2019-08-13</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>ecg</td>\n",
       "      <td>raw_timeseries_wide</td>\n",
       "      <td>9441</td>\n",
       "      <td>7888</td>\n",
       "      <td>2019-08-13</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>ecg</td>\n",
       "      <td>processed_timeseries_wide</td>\n",
       "      <td>9172</td>\n",
       "      <td>7711</td>\n",
       "      <td>2019-08-13</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>ecg</td>\n",
       "      <td>raw_timeseries_long</td>\n",
       "      <td>9441</td>\n",
       "      <td>7888</td>\n",
       "      <td>2019-08-13</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>ecg</td>\n",
       "      <td>processed_timeseries_long</td>\n",
       "      <td>9172</td>\n",
       "      <td>7711</td>\n",
       "      <td>2019-08-13</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>blood_tests</td>\n",
       "      <td>blood_tests</td>\n",
       "      <td>29074</td>\n",
       "      <td>10252</td>\n",
       "      <td>2010-01-06</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>cgm</td>\n",
       "      <td>cgm</td>\n",
       "      <td>9497</td>\n",
       "      <td>9431</td>\n",
       "      <td>2019-01-07</td>\n",
       "      <td>2022-12-30</td>\n",
       "      <td>017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>cgm</td>\n",
       "      <td>timeseries</td>\n",
       "      <td>9497</td>\n",
       "      <td>9431</td>\n",
       "      <td>2019-01-07</td>\n",
       "      <td>2022-12-30</td>\n",
       "      <td>017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>cgm</td>\n",
       "      <td>iglu</td>\n",
       "      <td>9497</td>\n",
       "      <td>9431</td>\n",
       "      <td>2019-01-07</td>\n",
       "      <td>2022-12-31</td>\n",
       "      <td>017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>cgm</td>\n",
       "      <td>iglu_daily</td>\n",
       "      <td>104845</td>\n",
       "      <td>9431</td>\n",
       "      <td>2019-01-07</td>\n",
       "      <td>2022-12-31</td>\n",
       "      <td>017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>bone_density</td>\n",
       "      <td>bone_density</td>\n",
       "      <td>9032</td>\n",
       "      <td>8459</td>\n",
       "      <td>2020-06-09</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>022</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               dataset                      feature_set  count_col_notna  \\\n",
       "0           population                       population            12014   \n",
       "1               events                           events            20723   \n",
       "2      anthropometrics                  anthropometrics            12925   \n",
       "3               fundus                           fundus             7068   \n",
       "4               fundus                 microvasculature             7068   \n",
       "5               fundus                           images             7068   \n",
       "6               fundus                 segmented_images             7068   \n",
       "7     liver_ultrasound                 liver_ultrasound            26684   \n",
       "8         diet_logging                     diet_logging           149912   \n",
       "9         diet_logging              diet_logging_events          2640312   \n",
       "10        diet_logging          raw_diet_logging_events          6483638   \n",
       "11           hand_grip                        hand_grip            12894   \n",
       "12      blood_pressure                   blood_pressure            12946   \n",
       "13    serum_lipidomics                 serum_lipidomics             6321   \n",
       "14               sleep                            sleep            21038   \n",
       "15               sleep                              hrv            20030   \n",
       "16               sleep             channels_time_series            21072   \n",
       "17               sleep                channels_edf_file            21069   \n",
       "18               sleep               events_time_series            21372   \n",
       "19     vascular_health                  vascular_health            11387   \n",
       "20    body_composition                 body_composition             8990   \n",
       "21  carotid_ultrasound               carotid_ultrasound             8518   \n",
       "22      gut_microbiome                   gut_microbiome            10694   \n",
       "23      gut_microbiome                        raw_fastq            10694   \n",
       "24      gut_microbiome                    trimmed_fastq            10694   \n",
       "25      gut_microbiome                   nonhuman_fastq            10694   \n",
       "26      gut_microbiome                           fastqc            10694   \n",
       "27      gut_microbiome                      human_plink            10694   \n",
       "28      gut_microbiome            urs_abundances_single            10694   \n",
       "29      gut_microbiome        urs_abundances_aggregated            10694   \n",
       "30      gut_microbiome      metaphlan_abundances_single            10694   \n",
       "31      gut_microbiome  metaphlan_abundances_aggregated            10694   \n",
       "32      human_genetics                   human_genetics             8821   \n",
       "33      human_genetics                              pca             8820   \n",
       "34      human_genetics                      variants_qc             8821   \n",
       "35      human_genetics                        fastq_bam             8821   \n",
       "36      human_genetics                    genotypes_raw             8821   \n",
       "37      human_genetics                 genotypes_postqc             8820   \n",
       "38      human_genetics                     relationship             8821   \n",
       "39                 ecg                              ecg             9702   \n",
       "40                 ecg              raw_timeseries_wide             9441   \n",
       "41                 ecg        processed_timeseries_wide             9172   \n",
       "42                 ecg              raw_timeseries_long             9441   \n",
       "43                 ecg        processed_timeseries_long             9172   \n",
       "44         blood_tests                      blood_tests            29074   \n",
       "45                 cgm                              cgm             9497   \n",
       "46                 cgm                       timeseries             9497   \n",
       "47                 cgm                             iglu             9497   \n",
       "48                 cgm                       iglu_daily           104845   \n",
       "49        bone_density                     bone_density             9032   \n",
       "\n",
       "    nunique_participants_not_na  first_date   last_date   id  \n",
       "0                         12014  2018-12-05  2022-12-29  000  \n",
       "1                         10434  2018-12-05  2022-12-29  001  \n",
       "2                         10387  2018-11-22  2022-12-29  002  \n",
       "3                          7038  2021-02-17  2022-12-04  003  \n",
       "4                          7038  2021-02-17  2022-12-04  003  \n",
       "5                          7038  2021-02-17  2022-12-04  003  \n",
       "6                          7038  2021-02-17  2022-12-04  003  \n",
       "7                          8229  2020-02-20  2022-12-29  004  \n",
       "8                          9531  2019-09-01  2022-12-31  005  \n",
       "9                         10197  2019-09-01  2023-03-21  005  \n",
       "10                        10696  2019-09-01  2023-03-21  005  \n",
       "11                        10371  2018-12-27  2022-12-29  006  \n",
       "12                        10398  2018-12-27  2022-12-29  007  \n",
       "13                         6144  2019-02-19  2021-08-08  008  \n",
       "14                         7009  2020-01-15  2022-12-27  009  \n",
       "15                         6926  2020-01-15  2022-12-27  009  \n",
       "16                         6973  2020-01-15  2022-12-27  009  \n",
       "17                         6973  2020-01-15  2022-12-27  009  \n",
       "18                         7062  2020-01-15  2022-12-27  009  \n",
       "19                         9729  2018-12-27  2022-12-29  010  \n",
       "20                         8411  2020-06-09  2022-12-29  011  \n",
       "21                         7988  2020-07-23  2022-12-29  012  \n",
       "22                         8838  2019-02-14  2022-12-28  013  \n",
       "23                         8838  2019-02-14  2022-12-28  013  \n",
       "24                         8838  2019-02-14  2022-12-28  013  \n",
       "25                         8838  2019-02-14  2022-12-28  013  \n",
       "26                         8838  2019-02-14  2022-12-28  013  \n",
       "27                         8838  2019-02-14  2022-12-28  013  \n",
       "28                         8838  2019-02-14  2022-12-28  013  \n",
       "29                         8838  2019-02-14  2022-12-28  013  \n",
       "30                         8838  2019-02-14  2022-12-28  013  \n",
       "31                         8838  2019-02-14  2022-12-28  013  \n",
       "32                         8821  2019-03-11  2022-12-21  014  \n",
       "33                         8820  2019-03-11  2022-12-21  014  \n",
       "34                         8821  2019-03-11  2022-12-21  014  \n",
       "35                         8821  2019-03-11  2022-12-21  014  \n",
       "36                         8821  2019-03-11  2022-12-21  014  \n",
       "37                         8820  2019-03-11  2022-12-21  014  \n",
       "38                         8821  2019-03-11  2022-12-21  014  \n",
       "39                         8085  2019-08-13  2022-12-29  015  \n",
       "40                         7888  2019-08-13  2022-12-29  015  \n",
       "41                         7711  2019-08-13  2022-12-29  015  \n",
       "42                         7888  2019-08-13  2022-12-29  015  \n",
       "43                         7711  2019-08-13  2022-12-29  015  \n",
       "44                        10252  2010-01-06  2022-12-29  016  \n",
       "45                         9431  2019-01-07  2022-12-30  017  \n",
       "46                         9431  2019-01-07  2022-12-30  017  \n",
       "47                         9431  2019-01-07  2022-12-31  017  \n",
       "48                         9431  2019-01-07  2022-12-31  017  \n",
       "49                         8459  2020-06-09  2022-12-29  022  "
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset</th>\n",
       "      <th>feature_set</th>\n",
       "      <th>count_col_notna</th>\n",
       "      <th>nunique_participants_not_na</th>\n",
       "      <th>first_date</th>\n",
       "      <th>last_date</th>\n",
       "      <th>id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>psychological_and_social_health</td>\n",
       "      <td>psychological_and_social_health</td>\n",
       "      <td>14861</td>\n",
       "      <td>9262</td>\n",
       "      <td>2019-01-09</td>\n",
       "      <td>2022-12-31</td>\n",
       "      <td>057</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>health_and_medical_history</td>\n",
       "      <td>ukbb</td>\n",
       "      <td>14861</td>\n",
       "      <td>9262</td>\n",
       "      <td>2019-01-09</td>\n",
       "      <td>2022-12-31</td>\n",
       "      <td>058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>health_and_medical_history</td>\n",
       "      <td>initial_medical</td>\n",
       "      <td>10197</td>\n",
       "      <td>10197</td>\n",
       "      <td>2018-11-21</td>\n",
       "      <td>2022-12-31</td>\n",
       "      <td>058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>participant_relation</td>\n",
       "      <td>participant relation</td>\n",
       "      <td>3739</td>\n",
       "      <td>3106</td>\n",
       "      <td>2019-02-05</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>045</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>rna_seq</td>\n",
       "      <td>rna_seq</td>\n",
       "      <td>3958</td>\n",
       "      <td>3863</td>\n",
       "      <td>2021-02-28</td>\n",
       "      <td>2022-06-21</td>\n",
       "      <td>024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>rna_seq</td>\n",
       "      <td>raw_fastq</td>\n",
       "      <td>3958</td>\n",
       "      <td>3863</td>\n",
       "      <td>2021-02-28</td>\n",
       "      <td>2022-06-21</td>\n",
       "      <td>024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>rna_seq</td>\n",
       "      <td>trimmed_fastq</td>\n",
       "      <td>3958</td>\n",
       "      <td>3863</td>\n",
       "      <td>2021-02-28</td>\n",
       "      <td>2022-06-21</td>\n",
       "      <td>024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>rna_seq</td>\n",
       "      <td>aligned_bam</td>\n",
       "      <td>3958</td>\n",
       "      <td>3863</td>\n",
       "      <td>2021-02-28</td>\n",
       "      <td>2022-06-21</td>\n",
       "      <td>024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>rna_seq</td>\n",
       "      <td>deduplicated_bam</td>\n",
       "      <td>3958</td>\n",
       "      <td>3863</td>\n",
       "      <td>2021-02-28</td>\n",
       "      <td>2022-06-21</td>\n",
       "      <td>024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>rna_seq</td>\n",
       "      <td>multiqc</td>\n",
       "      <td>3958</td>\n",
       "      <td>3863</td>\n",
       "      <td>2021-02-28</td>\n",
       "      <td>2022-06-21</td>\n",
       "      <td>024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>nightingale_metabolomics</td>\n",
       "      <td>nightingale_metabolomics</td>\n",
       "      <td>1644</td>\n",
       "      <td>1552</td>\n",
       "      <td>2019-05-07</td>\n",
       "      <td>2022-08-02</td>\n",
       "      <td>029</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>nightingale_metabolomics</td>\n",
       "      <td>bimarker_qc_tags</td>\n",
       "      <td>1644</td>\n",
       "      <td>1552</td>\n",
       "      <td>2019-05-07</td>\n",
       "      <td>2022-08-02</td>\n",
       "      <td>029</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>oral_microbiome</td>\n",
       "      <td>oral_microbiome</td>\n",
       "      <td>8816</td>\n",
       "      <td>8816</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "      <td>070</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>oral_microbiome</td>\n",
       "      <td>raw_fastq</td>\n",
       "      <td>8816</td>\n",
       "      <td>8816</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "      <td>070</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>oral_microbiome</td>\n",
       "      <td>trimmed_fastq</td>\n",
       "      <td>8816</td>\n",
       "      <td>8816</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "      <td>070</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>oral_microbiome</td>\n",
       "      <td>fastqc</td>\n",
       "      <td>8816</td>\n",
       "      <td>8816</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "      <td>070</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>oral_microbiome</td>\n",
       "      <td>metaphlan</td>\n",
       "      <td>8816</td>\n",
       "      <td>8816</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "      <td>070</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>oral_microbiome</td>\n",
       "      <td>metaphlan_aggregated</td>\n",
       "      <td>8816</td>\n",
       "      <td>8816</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "      <td>070</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>oral_microbiome</td>\n",
       "      <td>humann</td>\n",
       "      <td>8816</td>\n",
       "      <td>8816</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "      <td>070</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>oral_microbiome</td>\n",
       "      <td>humann_aggregated</td>\n",
       "      <td>8816</td>\n",
       "      <td>8816</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "      <td>070</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            dataset                      feature_set  \\\n",
       "69  psychological_and_social_health  psychological_and_social_health   \n",
       "70       health_and_medical_history                             ukbb   \n",
       "71       health_and_medical_history                  initial_medical   \n",
       "72             participant_relation             participant relation   \n",
       "73                          rna_seq                          rna_seq   \n",
       "74                          rna_seq                        raw_fastq   \n",
       "75                          rna_seq                    trimmed_fastq   \n",
       "76                          rna_seq                      aligned_bam   \n",
       "77                          rna_seq                 deduplicated_bam   \n",
       "78                          rna_seq                          multiqc   \n",
       "79         nightingale_metabolomics         nightingale_metabolomics   \n",
       "80         nightingale_metabolomics                 bimarker_qc_tags   \n",
       "81                  oral_microbiome                  oral_microbiome   \n",
       "82                  oral_microbiome                        raw_fastq   \n",
       "83                  oral_microbiome                    trimmed_fastq   \n",
       "84                  oral_microbiome                           fastqc   \n",
       "85                  oral_microbiome                        metaphlan   \n",
       "86                  oral_microbiome             metaphlan_aggregated   \n",
       "87                  oral_microbiome                           humann   \n",
       "88                  oral_microbiome                humann_aggregated   \n",
       "\n",
       "    count_col_notna  nunique_participants_not_na  first_date   last_date   id  \n",
       "69            14861                         9262  2019-01-09  2022-12-31  057  \n",
       "70            14861                         9262  2019-01-09  2022-12-31  058  \n",
       "71            10197                        10197  2018-11-21  2022-12-31  058  \n",
       "72             3739                         3106  2019-02-05  2022-12-29  045  \n",
       "73             3958                         3863  2021-02-28  2022-06-21  024  \n",
       "74             3958                         3863  2021-02-28  2022-06-21  024  \n",
       "75             3958                         3863  2021-02-28  2022-06-21  024  \n",
       "76             3958                         3863  2021-02-28  2022-06-21  024  \n",
       "77             3958                         3863  2021-02-28  2022-06-21  024  \n",
       "78             3958                         3863  2021-02-28  2022-06-21  024  \n",
       "79             1644                         1552  2019-05-07  2022-08-02  029  \n",
       "80             1644                         1552  2019-05-07  2022-08-02  029  \n",
       "81             8816                         8816  2019-03-11  2022-12-21  070  \n",
       "82             8816                         8816  2019-03-11  2022-12-21  070  \n",
       "83             8816                         8816  2019-03-11  2022-12-21  070  \n",
       "84             8816                         8816  2019-03-11  2022-12-21  070  \n",
       "85             8816                         8816  2019-03-11  2022-12-21  070  \n",
       "86             8816                         8816  2019-03-11  2022-12-21  070  \n",
       "87             8816                         8816  2019-03-11  2022-12-21  070  \n",
       "88             8816                         8816  2019-03-11  2022-12-21  070  "
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.tail(20)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df.to_csv('s3://ds-users/mariag/pheno_stage_data_collection_v1_2_count_converted_fulls.csv', index=False)\n",
    "# df.to_csv('s3://ds-users/mariag/pheno_stage_data_collection_v1_2_count_converted_before_2023.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "# for each daatset group by dataset and get the min and max date and max count and max participant\n",
    "df_group = df.groupby('dataset').agg({'first_date': 'min', 'last_date': 'max', 'count_col_notna': 'max', 'nunique_participants_not_na': 'max', 'id':'first'}).reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dataset</th>\n",
       "      <th>first_date</th>\n",
       "      <th>last_date</th>\n",
       "      <th>count_col_notna</th>\n",
       "      <th>nunique_participants_not_na</th>\n",
       "      <th>id</th>\n",
       "      <th>int_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>population</td>\n",
       "      <td>2018-12-05</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>12014</td>\n",
       "      <td>12014</td>\n",
       "      <td>000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>events</td>\n",
       "      <td>2018-12-05</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>20723</td>\n",
       "      <td>10434</td>\n",
       "      <td>001</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>anthropometrics</td>\n",
       "      <td>2018-11-22</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>12925</td>\n",
       "      <td>10387</td>\n",
       "      <td>002</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>fundus</td>\n",
       "      <td>2021-02-17</td>\n",
       "      <td>2022-12-04</td>\n",
       "      <td>7068</td>\n",
       "      <td>7038</td>\n",
       "      <td>003</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>liver_ultrasound</td>\n",
       "      <td>2020-02-20</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>26684</td>\n",
       "      <td>8229</td>\n",
       "      <td>004</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>diet_logging</td>\n",
       "      <td>2019-09-01</td>\n",
       "      <td>2023-03-21</td>\n",
       "      <td>6483638</td>\n",
       "      <td>10696</td>\n",
       "      <td>005</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>hand_grip</td>\n",
       "      <td>2018-12-27</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>12894</td>\n",
       "      <td>10371</td>\n",
       "      <td>006</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>blood_pressure</td>\n",
       "      <td>2018-12-27</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>12946</td>\n",
       "      <td>10398</td>\n",
       "      <td>007</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>serum_lipidomics</td>\n",
       "      <td>2019-02-19</td>\n",
       "      <td>2021-08-08</td>\n",
       "      <td>6321</td>\n",
       "      <td>6144</td>\n",
       "      <td>008</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>sleep</td>\n",
       "      <td>2020-01-15</td>\n",
       "      <td>2022-12-27</td>\n",
       "      <td>21372</td>\n",
       "      <td>7062</td>\n",
       "      <td>009</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>vascular_health</td>\n",
       "      <td>2018-12-27</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>11387</td>\n",
       "      <td>9729</td>\n",
       "      <td>010</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>body_composition</td>\n",
       "      <td>2020-06-09</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>8990</td>\n",
       "      <td>8411</td>\n",
       "      <td>011</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>carotid_ultrasound</td>\n",
       "      <td>2020-07-23</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>8518</td>\n",
       "      <td>7988</td>\n",
       "      <td>012</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>gut_microbiome</td>\n",
       "      <td>2019-02-14</td>\n",
       "      <td>2022-12-28</td>\n",
       "      <td>10694</td>\n",
       "      <td>8838</td>\n",
       "      <td>013</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>human_genetics</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "      <td>8821</td>\n",
       "      <td>8821</td>\n",
       "      <td>014</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ecg</td>\n",
       "      <td>2019-08-13</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>9702</td>\n",
       "      <td>8085</td>\n",
       "      <td>015</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>blood_tests</td>\n",
       "      <td>2010-01-06</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>29074</td>\n",
       "      <td>10252</td>\n",
       "      <td>016</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>cgm</td>\n",
       "      <td>2019-01-07</td>\n",
       "      <td>2022-12-31</td>\n",
       "      <td>104845</td>\n",
       "      <td>9431</td>\n",
       "      <td>017</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>medications</td>\n",
       "      <td>2018-10-18</td>\n",
       "      <td>2022-12-31</td>\n",
       "      <td>43417</td>\n",
       "      <td>10787</td>\n",
       "      <td>018</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>medical_conditions</td>\n",
       "      <td>2018-11-21</td>\n",
       "      <td>2022-12-31</td>\n",
       "      <td>22413</td>\n",
       "      <td>10792</td>\n",
       "      <td>021</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>bone_density</td>\n",
       "      <td>2020-06-09</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>9032</td>\n",
       "      <td>8459</td>\n",
       "      <td>022</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>rna_seq</td>\n",
       "      <td>2021-02-28</td>\n",
       "      <td>2022-06-21</td>\n",
       "      <td>3958</td>\n",
       "      <td>3863</td>\n",
       "      <td>024</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>nightingale_metabolomics</td>\n",
       "      <td>2019-05-07</td>\n",
       "      <td>2022-08-02</td>\n",
       "      <td>1644</td>\n",
       "      <td>1552</td>\n",
       "      <td>029</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>covid</td>\n",
       "      <td>2021-04-22</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>14988</td>\n",
       "      <td>9120</td>\n",
       "      <td>043</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>sex_specific_factors</td>\n",
       "      <td>2019-01-09</td>\n",
       "      <td>2022-12-31</td>\n",
       "      <td>9151</td>\n",
       "      <td>8759</td>\n",
       "      <td>044</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>participant_relation</td>\n",
       "      <td>2019-02-05</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>3739</td>\n",
       "      <td>3106</td>\n",
       "      <td>045</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>medical_symptoms</td>\n",
       "      <td>2019-01-09</td>\n",
       "      <td>2022-12-31</td>\n",
       "      <td>9172</td>\n",
       "      <td>8719</td>\n",
       "      <td>049</td>\n",
       "      <td>49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>medical_procedures</td>\n",
       "      <td>2020-02-26</td>\n",
       "      <td>2022-12-31</td>\n",
       "      <td>7572</td>\n",
       "      <td>7383</td>\n",
       "      <td>050</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>children</td>\n",
       "      <td>2020-02-26</td>\n",
       "      <td>2022-12-31</td>\n",
       "      <td>7416</td>\n",
       "      <td>7383</td>\n",
       "      <td>051</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>family_history</td>\n",
       "      <td>2019-01-09</td>\n",
       "      <td>2022-12-31</td>\n",
       "      <td>14861</td>\n",
       "      <td>9262</td>\n",
       "      <td>052</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>sociodemographics</td>\n",
       "      <td>2019-01-09</td>\n",
       "      <td>2022-12-31</td>\n",
       "      <td>14861</td>\n",
       "      <td>9262</td>\n",
       "      <td>053</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>lifestyle_and_environment</td>\n",
       "      <td>2019-01-09</td>\n",
       "      <td>2022-12-31</td>\n",
       "      <td>14861</td>\n",
       "      <td>9262</td>\n",
       "      <td>055</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>psychological_and_social_health</td>\n",
       "      <td>2019-01-09</td>\n",
       "      <td>2022-12-31</td>\n",
       "      <td>14861</td>\n",
       "      <td>9262</td>\n",
       "      <td>057</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>health_and_medical_history</td>\n",
       "      <td>2018-11-21</td>\n",
       "      <td>2022-12-31</td>\n",
       "      <td>14861</td>\n",
       "      <td>10197</td>\n",
       "      <td>058</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>oral_microbiome</td>\n",
       "      <td>2019-03-11</td>\n",
       "      <td>2022-12-21</td>\n",
       "      <td>8816</td>\n",
       "      <td>8816</td>\n",
       "      <td>070</td>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            dataset  first_date   last_date  count_col_notna  \\\n",
       "27                       population  2018-12-05  2022-12-29            12014   \n",
       "11                           events  2018-12-05  2022-12-29            20723   \n",
       "0                   anthropometrics  2018-11-22  2022-12-29            12925   \n",
       "13                           fundus  2021-02-17  2022-12-04             7068   \n",
       "19                 liver_ultrasound  2020-02-20  2022-12-29            26684   \n",
       "9                      diet_logging  2019-09-01  2023-03-21          6483638   \n",
       "15                        hand_grip  2018-12-27  2022-12-29            12894   \n",
       "1                    blood_pressure  2018-12-27  2022-12-29            12946   \n",
       "30                 serum_lipidomics  2019-02-19  2021-08-08             6321   \n",
       "32                            sleep  2020-01-15  2022-12-27            21372   \n",
       "34                  vascular_health  2018-12-27  2022-12-29            11387   \n",
       "3                  body_composition  2020-06-09  2022-12-29             8990   \n",
       "5                carotid_ultrasound  2020-07-23  2022-12-29             8518   \n",
       "14                   gut_microbiome  2019-02-14  2022-12-28            10694   \n",
       "17                   human_genetics  2019-03-11  2022-12-21             8821   \n",
       "10                              ecg  2019-08-13  2022-12-29             9702   \n",
       "2                       blood_tests  2010-01-06  2022-12-29            29074   \n",
       "6                               cgm  2019-01-07  2022-12-31           104845   \n",
       "23                      medications  2018-10-18  2022-12-31            43417   \n",
       "20               medical_conditions  2018-11-21  2022-12-31            22413   \n",
       "4                      bone_density  2020-06-09  2022-12-29             9032   \n",
       "29                          rna_seq  2021-02-28  2022-06-21             3958   \n",
       "24         nightingale_metabolomics  2019-05-07  2022-08-02             1644   \n",
       "8                             covid  2021-04-22  2022-12-29            14988   \n",
       "31             sex_specific_factors  2019-01-09  2022-12-31             9151   \n",
       "26             participant_relation  2019-02-05  2022-12-29             3739   \n",
       "22                 medical_symptoms  2019-01-09  2022-12-31             9172   \n",
       "21               medical_procedures  2020-02-26  2022-12-31             7572   \n",
       "7                          children  2020-02-26  2022-12-31             7416   \n",
       "12                   family_history  2019-01-09  2022-12-31            14861   \n",
       "33                sociodemographics  2019-01-09  2022-12-31            14861   \n",
       "18        lifestyle_and_environment  2019-01-09  2022-12-31            14861   \n",
       "28  psychological_and_social_health  2019-01-09  2022-12-31            14861   \n",
       "16       health_and_medical_history  2018-11-21  2022-12-31            14861   \n",
       "25                  oral_microbiome  2019-03-11  2022-12-21             8816   \n",
       "\n",
       "    nunique_participants_not_na   id  int_id  \n",
       "27                        12014  000       0  \n",
       "11                        10434  001       1  \n",
       "0                         10387  002       2  \n",
       "13                         7038  003       3  \n",
       "19                         8229  004       4  \n",
       "9                         10696  005       5  \n",
       "15                        10371  006       6  \n",
       "1                         10398  007       7  \n",
       "30                         6144  008       8  \n",
       "32                         7062  009       9  \n",
       "34                         9729  010      10  \n",
       "3                          8411  011      11  \n",
       "5                          7988  012      12  \n",
       "14                         8838  013      13  \n",
       "17                         8821  014      14  \n",
       "10                         8085  015      15  \n",
       "2                         10252  016      16  \n",
       "6                          9431  017      17  \n",
       "23                        10787  018      18  \n",
       "20                        10792  021      21  \n",
       "4                          8459  022      22  \n",
       "29                         3863  024      24  \n",
       "24                         1552  029      29  \n",
       "8                          9120  043      43  \n",
       "31                         8759  044      44  \n",
       "26                         3106  045      45  \n",
       "22                         8719  049      49  \n",
       "21                         7383  050      50  \n",
       "7                          7383  051      51  \n",
       "12                         9262  052      52  \n",
       "33                         9262  053      53  \n",
       "18                         9262  055      55  \n",
       "28                         9262  057      57  \n",
       "16                        10197  058      58  \n",
       "25                         8816  070      70  "
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_group['int_id'] = df_group['id'].astype(int)\n",
    "df_group.sort_values('int_id', inplace=True)\n",
    "df_group\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df_group1 = df_group.copy()\n",
    "# df_group2 = df_group.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "# count_bucket_name = \"pheno-master-data-collection-v1.2-prod-eu-west-1\"\n",
    "# output_path = f'{count_bucket_name}_{ver}_before_2023_{BEFORE_2023}_show_feature_{SHOW_FEATURE_SETS}.csv'\n",
    "# df_group1.to_csv(output_path, index=False)\n",
    "\n",
    "# count_bucket_name = \"pheno-academic-collection-v1.2-prod-eu-west-1\"\n",
    "# output_path = f'{count_bucket_name}_{ver}_before_2023_{BEFORE_2023}_show_feature_{SHOW_FEATURE_SETS}.csv'\n",
    "# df_group2.to_csv(output_path, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "ename": "PermissionError",
     "evalue": "Access Denied",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mClientError\u001b[0m                               Traceback (most recent call last)",
      "File \u001b[0;32m~/miniconda3/lib/python3.11/site-packages/s3fs/core.py:113\u001b[0m, in \u001b[0;36m_error_wrapper\u001b[0;34m(func, args, kwargs, retries)\u001b[0m\n\u001b[1;32m    112\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 113\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;01mawait\u001b[39;00m func(\u001b[38;5;241m*\u001b[39margs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n\u001b[1;32m    114\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m S3_RETRYABLE_ERRORS \u001b[38;5;28;01mas\u001b[39;00m e:\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.11/site-packages/aiobotocore/client.py:408\u001b[0m, in \u001b[0;36mAioBaseClient._make_api_call\u001b[0;34m(self, operation_name, api_params)\u001b[0m\n\u001b[1;32m    407\u001b[0m     error_class \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mexceptions\u001b[38;5;241m.\u001b[39mfrom_code(error_code)\n\u001b[0;32m--> 408\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m error_class(parsed_response, operation_name)\n\u001b[1;32m    409\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "\u001b[0;31mClientError\u001b[0m: An error occurred (AccessDenied) when calling the PutObject operation: Access Denied",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mPermissionError\u001b[0m                           Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[72], line 3\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;66;03m# df_group.to_csv('s3://ds-users/mariag/pheno_stage_data_collection_v1_2_count_converted_full_grouped.csv', index=False)\u001b[39;00m\n\u001b[1;32m      2\u001b[0m \u001b[38;5;66;03m# df_group.to_csv('s3://ds-users/mariag/pheno_stage_data_collection_v1_2_count_converted_before2023_grouped.csv', index=False)\u001b[39;00m\n\u001b[0;32m----> 3\u001b[0m \u001b[43mdf_group\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mto_csv\u001b[49m\u001b[43m(\u001b[49m\u001b[43moutput_path\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mindex\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.11/site-packages/pandas/core/generic.py:3902\u001b[0m, in \u001b[0;36mNDFrame.to_csv\u001b[0;34m(self, path_or_buf, sep, na_rep, float_format, columns, header, index, index_label, mode, encoding, compression, quoting, quotechar, lineterminator, chunksize, date_format, doublequote, escapechar, decimal, errors, storage_options)\u001b[0m\n\u001b[1;32m   3891\u001b[0m df \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(\u001b[38;5;28mself\u001b[39m, ABCDataFrame) \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mto_frame()\n\u001b[1;32m   3893\u001b[0m formatter \u001b[38;5;241m=\u001b[39m DataFrameFormatter(\n\u001b[1;32m   3894\u001b[0m     frame\u001b[38;5;241m=\u001b[39mdf,\n\u001b[1;32m   3895\u001b[0m     header\u001b[38;5;241m=\u001b[39mheader,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m   3899\u001b[0m     decimal\u001b[38;5;241m=\u001b[39mdecimal,\n\u001b[1;32m   3900\u001b[0m )\n\u001b[0;32m-> 3902\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mDataFrameRenderer\u001b[49m\u001b[43m(\u001b[49m\u001b[43mformatter\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mto_csv\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m   3903\u001b[0m \u001b[43m    \u001b[49m\u001b[43mpath_or_buf\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3904\u001b[0m \u001b[43m    \u001b[49m\u001b[43mlineterminator\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mlineterminator\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3905\u001b[0m \u001b[43m    \u001b[49m\u001b[43msep\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msep\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3906\u001b[0m \u001b[43m    \u001b[49m\u001b[43mencoding\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mencoding\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3907\u001b[0m \u001b[43m    \u001b[49m\u001b[43merrors\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43merrors\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3908\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcompression\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcompression\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3909\u001b[0m \u001b[43m    \u001b[49m\u001b[43mquoting\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mquoting\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3910\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcolumns\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcolumns\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3911\u001b[0m \u001b[43m    \u001b[49m\u001b[43mindex_label\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mindex_label\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3912\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmode\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3913\u001b[0m \u001b[43m    \u001b[49m\u001b[43mchunksize\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchunksize\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3914\u001b[0m \u001b[43m    \u001b[49m\u001b[43mquotechar\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mquotechar\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3915\u001b[0m \u001b[43m    \u001b[49m\u001b[43mdate_format\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdate_format\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3916\u001b[0m \u001b[43m    \u001b[49m\u001b[43mdoublequote\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdoublequote\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3917\u001b[0m \u001b[43m    \u001b[49m\u001b[43mescapechar\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mescapechar\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3918\u001b[0m \u001b[43m    \u001b[49m\u001b[43mstorage_options\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mstorage_options\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3919\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.11/site-packages/pandas/io/formats/format.py:1152\u001b[0m, in \u001b[0;36mDataFrameRenderer.to_csv\u001b[0;34m(self, path_or_buf, encoding, sep, columns, index_label, mode, compression, quoting, quotechar, lineterminator, chunksize, date_format, doublequote, escapechar, errors, storage_options)\u001b[0m\n\u001b[1;32m   1131\u001b[0m     created_buffer \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mFalse\u001b[39;00m\n\u001b[1;32m   1133\u001b[0m csv_formatter \u001b[38;5;241m=\u001b[39m CSVFormatter(\n\u001b[1;32m   1134\u001b[0m     path_or_buf\u001b[38;5;241m=\u001b[39mpath_or_buf,\n\u001b[1;32m   1135\u001b[0m     lineterminator\u001b[38;5;241m=\u001b[39mlineterminator,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m   1150\u001b[0m     formatter\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mfmt,\n\u001b[1;32m   1151\u001b[0m )\n\u001b[0;32m-> 1152\u001b[0m \u001b[43mcsv_formatter\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msave\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1154\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m created_buffer:\n\u001b[1;32m   1155\u001b[0m     \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(path_or_buf, StringIO)\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.11/site-packages/pandas/io/formats/csvs.py:247\u001b[0m, in \u001b[0;36mCSVFormatter.save\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    243\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m    244\u001b[0m \u001b[38;5;124;03mCreate the writer & save.\u001b[39;00m\n\u001b[1;32m    245\u001b[0m \u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m    246\u001b[0m \u001b[38;5;66;03m# apply compression and byte/text conversion\u001b[39;00m\n\u001b[0;32m--> 247\u001b[0m \u001b[43m\u001b[49m\u001b[38;5;28;43;01mwith\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mget_handle\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    248\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mfilepath_or_buffer\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    249\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    250\u001b[0m \u001b[43m    \u001b[49m\u001b[43mencoding\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mencoding\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    251\u001b[0m \u001b[43m    \u001b[49m\u001b[43merrors\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43merrors\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    252\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcompression\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcompression\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    253\u001b[0m \u001b[43m    \u001b[49m\u001b[43mstorage_options\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mstorage_options\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    254\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mas\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mhandles\u001b[49m\u001b[43m:\u001b[49m\n\u001b[1;32m    255\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;66;43;03m# Note: self.encoding is irrelevant here\u001b[39;49;00m\n\u001b[1;32m    256\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwriter\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[43mcsvlib\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwriter\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    257\u001b[0m \u001b[43m        \u001b[49m\u001b[43mhandles\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mhandle\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    258\u001b[0m \u001b[43m        \u001b[49m\u001b[43mlineterminator\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mlineterminator\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    263\u001b[0m \u001b[43m        \u001b[49m\u001b[43mquotechar\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mquotechar\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    264\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    266\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_save\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.11/site-packages/pandas/io/common.py:142\u001b[0m, in \u001b[0;36mIOHandles.__exit__\u001b[0;34m(self, *args)\u001b[0m\n\u001b[1;32m    141\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m__exit__\u001b[39m(\u001b[38;5;28mself\u001b[39m, \u001b[38;5;241m*\u001b[39margs: Any) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[0;32m--> 142\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mclose\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.11/site-packages/pandas/io/common.py:134\u001b[0m, in \u001b[0;36mIOHandles.close\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    132\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcreated_handles\u001b[38;5;241m.\u001b[39mremove(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandle)\n\u001b[1;32m    133\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m handle \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcreated_handles:\n\u001b[0;32m--> 134\u001b[0m     handle\u001b[38;5;241m.\u001b[39mclose()\n\u001b[1;32m    135\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcreated_handles \u001b[38;5;241m=\u001b[39m []\n\u001b[1;32m    136\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mis_wrapped \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mFalse\u001b[39;00m\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.11/site-packages/fsspec/spec.py:1932\u001b[0m, in \u001b[0;36mAbstractBufferedFile.close\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1930\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m   1931\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mforced:\n\u001b[0;32m-> 1932\u001b[0m         \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mflush\u001b[49m\u001b[43m(\u001b[49m\u001b[43mforce\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m)\u001b[49m\n\u001b[1;32m   1934\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mfs \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m   1935\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mfs\u001b[38;5;241m.\u001b[39minvalidate_cache(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mpath)\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.11/site-packages/fsspec/spec.py:1803\u001b[0m, in \u001b[0;36mAbstractBufferedFile.flush\u001b[0;34m(self, force)\u001b[0m\n\u001b[1;32m   1800\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mclosed \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mTrue\u001b[39;00m\n\u001b[1;32m   1801\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m\n\u001b[0;32m-> 1803\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_upload_chunk\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfinal\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mforce\u001b[49m\u001b[43m)\u001b[49m \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mFalse\u001b[39;00m:\n\u001b[1;32m   1804\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39moffset \u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mbuffer\u001b[38;5;241m.\u001b[39mseek(\u001b[38;5;241m0\u001b[39m, \u001b[38;5;241m2\u001b[39m)\n\u001b[1;32m   1805\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mbuffer \u001b[38;5;241m=\u001b[39m io\u001b[38;5;241m.\u001b[39mBytesIO()\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.11/site-packages/s3fs/core.py:2299\u001b[0m, in \u001b[0;36mS3File._upload_chunk\u001b[0;34m(self, final)\u001b[0m\n\u001b[1;32m   2296\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mparts\u001b[38;5;241m.\u001b[39mappend(part_header)\n\u001b[1;32m   2298\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mautocommit \u001b[38;5;129;01mand\u001b[39;00m final:\n\u001b[0;32m-> 2299\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcommit\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   2300\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m final\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.11/site-packages/s3fs/core.py:2317\u001b[0m, in \u001b[0;36mS3File.commit\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   2315\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39macl:\n\u001b[1;32m   2316\u001b[0m         kw[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mACL\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39macl\n\u001b[0;32m-> 2317\u001b[0m     write_result \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_call_s3\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mput_object\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkw\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   2318\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m   2319\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mRuntimeError\u001b[39;00m\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.11/site-packages/s3fs/core.py:2162\u001b[0m, in \u001b[0;36mS3File._call_s3\u001b[0;34m(self, method, *kwarglist, **kwargs)\u001b[0m\n\u001b[1;32m   2161\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m_call_s3\u001b[39m(\u001b[38;5;28mself\u001b[39m, method, \u001b[38;5;241m*\u001b[39mkwarglist, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs):\n\u001b[0;32m-> 2162\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mfs\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcall_s3\u001b[49m\u001b[43m(\u001b[49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43ms3_additional_kwargs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwarglist\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.11/site-packages/fsspec/asyn.py:118\u001b[0m, in \u001b[0;36msync_wrapper.<locals>.wrapper\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    115\u001b[0m \u001b[38;5;129m@functools\u001b[39m\u001b[38;5;241m.\u001b[39mwraps(func)\n\u001b[1;32m    116\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mwrapper\u001b[39m(\u001b[38;5;241m*\u001b[39margs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs):\n\u001b[1;32m    117\u001b[0m     \u001b[38;5;28mself\u001b[39m \u001b[38;5;241m=\u001b[39m obj \u001b[38;5;129;01mor\u001b[39;00m args[\u001b[38;5;241m0\u001b[39m]\n\u001b[0;32m--> 118\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43msync\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mloop\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfunc\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.11/site-packages/fsspec/asyn.py:103\u001b[0m, in \u001b[0;36msync\u001b[0;34m(loop, func, timeout, *args, **kwargs)\u001b[0m\n\u001b[1;32m    101\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m FSTimeoutError \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mreturn_result\u001b[39;00m\n\u001b[1;32m    102\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(return_result, \u001b[38;5;167;01mBaseException\u001b[39;00m):\n\u001b[0;32m--> 103\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m return_result\n\u001b[1;32m    104\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    105\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m return_result\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.11/site-packages/fsspec/asyn.py:56\u001b[0m, in \u001b[0;36m_runner\u001b[0;34m(event, coro, result, timeout)\u001b[0m\n\u001b[1;32m     54\u001b[0m     coro \u001b[38;5;241m=\u001b[39m asyncio\u001b[38;5;241m.\u001b[39mwait_for(coro, timeout\u001b[38;5;241m=\u001b[39mtimeout)\n\u001b[1;32m     55\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m---> 56\u001b[0m     result[\u001b[38;5;241m0\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mawait\u001b[39;00m coro\n\u001b[1;32m     57\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mException\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m ex:\n\u001b[1;32m     58\u001b[0m     result[\u001b[38;5;241m0\u001b[39m] \u001b[38;5;241m=\u001b[39m ex\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.11/site-packages/s3fs/core.py:348\u001b[0m, in \u001b[0;36mS3FileSystem._call_s3\u001b[0;34m(self, method, *akwarglist, **kwargs)\u001b[0m\n\u001b[1;32m    346\u001b[0m logger\u001b[38;5;241m.\u001b[39mdebug(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mCALL: \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m - \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m - \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, method\u001b[38;5;241m.\u001b[39m\u001b[38;5;18m__name__\u001b[39m, akwarglist, kw2)\n\u001b[1;32m    347\u001b[0m additional_kwargs \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_get_s3_method_kwargs(method, \u001b[38;5;241m*\u001b[39makwarglist, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n\u001b[0;32m--> 348\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;01mawait\u001b[39;00m _error_wrapper(\n\u001b[1;32m    349\u001b[0m     method, kwargs\u001b[38;5;241m=\u001b[39madditional_kwargs, retries\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mretries\n\u001b[1;32m    350\u001b[0m )\n",
      "File \u001b[0;32m~/miniconda3/lib/python3.11/site-packages/s3fs/core.py:140\u001b[0m, in \u001b[0;36m_error_wrapper\u001b[0;34m(func, args, kwargs, retries)\u001b[0m\n\u001b[1;32m    138\u001b[0m         err \u001b[38;5;241m=\u001b[39m e\n\u001b[1;32m    139\u001b[0m err \u001b[38;5;241m=\u001b[39m translate_boto_error(err)\n\u001b[0;32m--> 140\u001b[0m \u001b[38;5;28;01mraise\u001b[39;00m err\n",
      "\u001b[0;31mPermissionError\u001b[0m: Access Denied"
     ]
    }
   ],
   "source": [
    "# df_group.to_csv('s3://ds-users/mariag/pheno_stage_data_collection_v1_2_count_converted_full_grouped.csv', index=False)\n",
    "# df_group.to_csv('s3://ds-users/mariag/pheno_stage_data_collection_v1_2_count_converted_before2023_grouped.csv', index=False)\n",
    "df_group.to_csv(output_path, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>dataset</th>\n",
       "      <th>feature_set</th>\n",
       "      <th>first_date</th>\n",
       "      <th>last_date</th>\n",
       "      <th>data points (participants)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>000</td>\n",
       "      <td>population</td>\n",
       "      <td>population</td>\n",
       "      <td>2018-12-05</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>11179 (11179)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>001</td>\n",
       "      <td>events</td>\n",
       "      <td>events</td>\n",
       "      <td>2018-12-05</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>20773 (10466)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>002</td>\n",
       "      <td>anthropometrics</td>\n",
       "      <td>anthropometrics</td>\n",
       "      <td>2018-11-22</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>12963 (10419)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>003</td>\n",
       "      <td>fundus</td>\n",
       "      <td>fundus</td>\n",
       "      <td>2021-02-17</td>\n",
       "      <td>2022-12-04</td>\n",
       "      <td>7079 (7049)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>003</td>\n",
       "      <td>fundus</td>\n",
       "      <td>microvasculature</td>\n",
       "      <td>2021-02-17</td>\n",
       "      <td>2022-12-04</td>\n",
       "      <td>7079 (7049)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    id          dataset       feature_set  first_date   last_date  \\\n",
       "0  000       population        population  2018-12-05  2022-12-29   \n",
       "1  001           events            events  2018-12-05  2022-12-29   \n",
       "2  002  anthropometrics   anthropometrics  2018-11-22  2022-12-29   \n",
       "3  003           fundus            fundus  2021-02-17  2022-12-04   \n",
       "4  003           fundus  microvasculature  2021-02-17  2022-12-04   \n",
       "\n",
       "  data points (participants)  \n",
       "0              11179 (11179)  \n",
       "1              20773 (10466)  \n",
       "2              12963 (10419)  \n",
       "3                7079 (7049)  \n",
       "4                7079 (7049)  "
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if not SHOW_FEATURE_SETS: \n",
    "    df  = df_group.copy()\n",
    "    # Reorder the columns\n",
    "    df = df[['id', 'dataset',  'count_col_notna', 'nunique_participants_not_na', 'first_date', 'last_date']] #'feature_set',\n",
    "else: \n",
    "    # Reorder the columns\n",
    "    df = df[['id', 'dataset',  'feature_set', 'count_col_notna', 'nunique_participants_not_na', 'first_date', 'last_date']] \n",
    "\n",
    "# Combine 'count_col_notna' and 'nunique_participants_not_na' columns\n",
    "df['data points (participants)'] = df['count_col_notna'].astype(str) + \" (\" + df['nunique_participants_not_na'].astype(str) + \")\"\n",
    "\n",
    "# Drop the original columns\n",
    "df = df.drop(columns=['count_col_notna', 'nunique_participants_not_na'])\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>dataset</th>\n",
       "      <th>feature_set</th>\n",
       "      <th>first_date</th>\n",
       "      <th>last_date</th>\n",
       "      <th>data points (participants)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>017</td>\n",
       "      <td>cgm</td>\n",
       "      <td>timeseries</td>\n",
       "      <td>2019-01-07</td>\n",
       "      <td>2022-12-30</td>\n",
       "      <td>9524 (9458)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>017</td>\n",
       "      <td>cgm</td>\n",
       "      <td>iglu</td>\n",
       "      <td>2019-01-07</td>\n",
       "      <td>2022-12-31</td>\n",
       "      <td>9524 (9458)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>017</td>\n",
       "      <td>cgm</td>\n",
       "      <td>iglu_daily</td>\n",
       "      <td>2019-01-07</td>\n",
       "      <td>2022-12-31</td>\n",
       "      <td>105149 (9458)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>022</td>\n",
       "      <td>bone_density</td>\n",
       "      <td>bone_density</td>\n",
       "      <td>2020-06-09</td>\n",
       "      <td>2022-12-29</td>\n",
       "      <td>9052 (8477)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>018</td>\n",
       "      <td>medications</td>\n",
       "      <td>medications</td>\n",
       "      <td>2018-10-18</td>\n",
       "      <td>2022-12-31</td>\n",
       "      <td>14308 (10159)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     id       dataset   feature_set  first_date   last_date  \\\n",
       "46  017           cgm    timeseries  2019-01-07  2022-12-30   \n",
       "47  017           cgm          iglu  2019-01-07  2022-12-31   \n",
       "48  017           cgm    iglu_daily  2019-01-07  2022-12-31   \n",
       "49  022  bone_density  bone_density  2020-06-09  2022-12-29   \n",
       "50  018   medications   medications  2018-10-18  2022-12-31   \n",
       "\n",
       "   data points (participants)  \n",
       "46                9524 (9458)  \n",
       "47                9524 (9458)  \n",
       "48              105149 (9458)  \n",
       "49                9052 (8477)  \n",
       "50              14308 (10159)  "
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# Reorder the columns\n",
    "# df = df[['id', 'dataset', 'data points (participants)', 'first_date', 'last_date']]#'feature_set',\n",
    "\n",
    "# Remove '_' from column names\n",
    "df.columns = df.columns.str.replace('_', ' ')\n",
    "\n",
    "# Add 'tabular data' column with \"\" for all rows\n",
    "df['tabular data'] = ''\n",
    "\n",
    "# Add 'time series data' column with \"\" for specified rows\n",
    "df['time series data'] = ''\n",
    "df.loc[df['dataset'] == 'diet logging', 'time series data'] = ''\n",
    "df.loc[(df['dataset'].isin(['sleep', 'ecg', 'cgm'])), 'time series data'] = ''\n",
    "\n",
    "# Add 'image data' column with \"\" for specified rows\n",
    "df['image data'] = ''\n",
    "df.loc[(df['dataset'] == 'fundus') , 'image data'] = ''\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Convert each value in the 'dataset' column to a link\n",
    "df['dataset'] = df.apply(lambda row: f'[{row[\"dataset\"]}](datasets/{row[\"id\"]}-{row[\"dataset\"]}.html)', axis=1)\n",
    "\n",
    "# Convert the DataFrame to a markdown table\n",
    "markdown_table = df.to_markdown(index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (3319058519.py, line 1)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  Cell \u001b[0;32mIn[40], line 1\u001b[0;36m\u001b[0m\n\u001b[0;31m    stop!\u001b[0m\n\u001b[0m        ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "stop!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "|   id | dataset                                                                              | first date   | last date   | data points (participants)   | tabular data   | time series data   | image data   |\n",
      "|-----:|:-------------------------------------------------------------------------------------|:-------------|:------------|:-----------------------------|:---------------|:-------------------|:-------------|\n",
      "|  000 | [population](datasets/000-population.html)                                           | 2018-12-05   | 2022-12-29  | 10434 (10434)                |               |                    |              |\n",
      "|  001 | [events](datasets/001-events.html)                                                   | 2018-12-05   | 2022-12-29  | 20723 (10434)                |               |                    |              |\n",
      "|  002 | [anthropometrics](datasets/002-anthropometrics.html)                                 | 2018-11-22   | 2022-12-29  | 12925 (10387)                |               |                    |              |\n",
      "|  003 | [fundus](datasets/003-fundus.html)                                                   | 2021-02-17   | 2022-12-04  | 7068 (7038)                  |               |                    |             |\n",
      "|  004 | [liver_ultrasound](datasets/004-liver_ultrasound.html)                               | 2020-02-20   | 2022-12-29  | 26684 (8229)                 |               |                    |              |\n",
      "|  005 | [diet_logging](datasets/005-diet_logging.html)                                       | 2019-09-01   | 2022-12-31  | 5924216 (10003)              |               |                    |              |\n",
      "|  006 | [hand_grip](datasets/006-hand_grip.html)                                             | 2018-12-27   | 2022-12-29  | 12894 (10371)                |               |                    |              |\n",
      "|  007 | [blood_pressure](datasets/007-blood_pressure.html)                                   | 2018-12-27   | 2022-12-29  | 12946 (10398)                |               |                    |              |\n"
     ]
    }
   ],
   "source": [
    "\n",
    "# Save the markdown table to a .md file\n",
    "with open('datasets.md', 'w') as f:\n",
    "    f.write(markdown_table)\n",
    "    f.write('\\n\\n')\n",
    "    f.write(': Available Datasets {.striped .hover tbl-colwidths=\"[5, 37, 20 ,20, 20, 5, 5, 5]\"}')\n",
    "    \n",
    "# Check the first few lines of the markdown table\n",
    "print('\\n'.join(markdown_table.split('\\n')[:10]))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "|   id | dataset                                                                              | first date   | last date   | data points (participants)   | tabular data   | time series data   | image data   |\n",
      "|-----:|:-------------------------------------------------------------------------------------|:-------------|:------------|:-----------------------------|:---------------|:-------------------|:-------------|\n",
      "|  000 | [population](datasets/000-population.html)                                           | 2018-12-05   | 2022-12-29  | 10434 (10434)                |               |                    |              |\n",
      "|  001 | [events](datasets/001-events.html)                                                   | 2018-12-05   | 2022-12-29  | 20723 (10434)                |               |                    |              |\n",
      "|  002 | [anthropometrics](datasets/002-anthropometrics.html)                                 | 2018-11-22   | 2022-12-29  | 12925 (10387)                |               |                    |              |\n",
      "|  003 | [fundus](datasets/003-fundus.html)                                                   | 2021-02-17   | 2022-12-04  | 7068 (7038)                  |               |                    |             |\n",
      "|  004 | [liver_ultrasound](datasets/004-liver_ultrasound.html)                               | 2020-02-20   | 2022-12-29  | 26684 (8229)                 |               |                    |              |\n",
      "|  005 | [diet_logging](datasets/005-diet_logging.html)                                       | 2019-09-01   | 2022-12-31  | 5924216 (10003)              |               |                    |              |\n",
      "|  006 | [hand_grip](datasets/006-hand_grip.html)                                             | 2018-12-27   | 2022-12-29  | 12894 (10371)                |               |                    |              |\n",
      "|  007 | [blood_pressure](datasets/007-blood_pressure.html)                                   | 2018-12-27   | 2022-12-29  | 12946 (10398)                |               |                    |              |\n",
      "|  008 | [serum_lipidomics](datasets/008-serum_lipidomics.html)                               | 2019-02-19   | 2021-08-08  | 6321 (6144)                  |               |                    |              |\n",
      "|  009 | [sleep](datasets/009-sleep.html)                                                     | 2020-01-15   | 2022-12-27  | 21589 (7132)                 |               |                   |              |\n",
      "|  010 | [vascular_health](datasets/010-vascular_health.html)                                 | 2018-12-27   | 2022-12-29  | 11387 (9729)                 |               |                    |              |\n",
      "|  011 | [body_composition](datasets/011-body_composition.html)                               | 2020-06-09   | 2022-12-29  | 8990 (8411)                  |               |                    |              |\n",
      "|  012 | [carotid_ultrasound](datasets/012-carotid_ultrasound.html)                           | 2020-07-23   | 2022-12-29  | 8518 (7988)                  |               |                    |              |\n",
      "|  013 | [gut_microbiome](datasets/013-gut_microbiome.html)                                   | 2019-02-14   | 2022-12-28  | 10694 (8838)                 |               |                    |              |\n",
      "|  014 | [human_genetics](datasets/014-human_genetics.html)                                   | 2019-03-11   | 2022-12-21  | 8821 (8821)                  |               |                    |              |\n",
      "|  015 | [ecg](datasets/015-ecg.html)                                                         | 2019-08-13   | 2022-12-29  | 9702 (8085)                  |               |                   |              |\n",
      "|  016 | [blood_tests](datasets/016-blood_tests.html)                                         | 2010-01-06   | 2022-12-29  | 29074 (10252)                |               |                    |              |\n",
      "|  017 | [cgm](datasets/017-cgm.html)                                                         | 2019-01-07   | 2022-12-31  | 104845 (9431)                |               |                   |              |\n"
     ]
    }
   ],
   "source": [
    "print('\\n'.join(markdown_table.split('\\n')[:20]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "|  016 | [blood_tests](datasets/016-blood_tests.html)                                         | 2010-01-06   | 2022-12-29  | 29074 (10252)                |               |                    |              |\n",
      "|  017 | [cgm](datasets/017-cgm.html)                                                         | 2019-01-07   | 2022-12-31  | 104845 (9431)                |               |                   |              |\n",
      "|  018 | [medications](datasets/018-medications.html)                                         | 2018-10-18   | 2022-12-31  | 43417 (10787)                |               |                    |              |\n",
      "|  021 | [medical_conditions](datasets/021-medical_conditions.html)                           | 2018-11-21   | 2022-12-31  | 22413 (10792)                |               |                    |              |\n",
      "|  022 | [bone_density](datasets/022-bone_density.html)                                       | 2020-06-09   | 2022-12-29  | 9032 (8459)                  |               |                    |              |\n",
      "|  024 | [rna_seq](datasets/024-rna_seq.html)                                                 | 2021-02-28   | 2022-06-21  | 3958 (3863)                  |               |                    |              |\n",
      "|  029 | [nightingale_metabolomics](datasets/029-nightingale_metabolomics.html)               | 2019-05-07   | 2022-08-02  | 1644 (1552)                  |               |                    |              |\n",
      "|  043 | [covid](datasets/043-covid.html)                                                     | 2021-04-22   | 2022-12-29  | 14988 (9120)                 |               |                    |              |\n",
      "|  044 | [sex_specific_factors](datasets/044-sex_specific_factors.html)                       | 2019-01-09   | 2022-12-31  | 9151 (8759)                  |               |                    |              |\n",
      "|  045 | [participant_relation](datasets/045-participant_relation.html)                       | 2019-02-05   | 2022-12-29  | 3739 (3106)                  |               |                    |              |\n",
      "|  049 | [medical_symptoms](datasets/049-medical_symptoms.html)                               | 2019-01-09   | 2022-12-31  | 9172 (8719)                  |               |                    |              |\n",
      "|  050 | [medical_procedures](datasets/050-medical_procedures.html)                           | 2020-02-26   | 2022-12-31  | 7572 (7383)                  |               |                    |              |\n",
      "|  051 | [children](datasets/051-children.html)                                               | 2020-02-26   | 2022-12-31  | 7416 (7383)                  |               |                    |              |\n",
      "|  052 | [family_history](datasets/052-family_history.html)                                   | 2019-01-09   | 2022-12-31  | 14861 (9262)                 |               |                    |              |\n",
      "|  053 | [sociodemographics](datasets/053-sociodemographics.html)                             | 2019-01-09   | 2022-12-31  | 14861 (9262)                 |               |                    |              |\n",
      "|  055 | [lifestyle_and_environment](datasets/055-lifestyle_and_environment.html)             | 2019-01-09   | 2022-12-31  | 14861 (9262)                 |               |                    |              |\n",
      "|  057 | [psychological_and_social_health](datasets/057-psychological_and_social_health.html) | 2019-01-09   | 2022-12-31  | 14861 (9262)                 |               |                    |              |\n",
      "|  058 | [health_and_medical_history](datasets/058-health_and_medical_history.html)           | 2018-11-21   | 2022-12-31  | 14861 (10197)                |               |                    |              |\n",
      "|  070 | [oral_microbiome](datasets/070-oral_microbiome.html)                                 | 2019-03-11   | 2022-12-21  | 8816 (8816)                  |               |                    |              |\n",
      "|  100 | [curated_phenotypes](datasets/100-curated_phenotypes.html)                           | 2018-12-05   | 2022-12-29  | 20723 (10434)                |               |                    |              |\n"
     ]
    }
   ],
   "source": [
    "print('\\n'.join(markdown_table.split('\\n')[-20:]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
